---
title: "YTHDC1-METTL3-KO-YTHDC-GO-S2S"
author: "George Boateng-Sarfo"
date: "`r Sys.Date()`"
output: word_document
always_allow_html: true
---

    New S2 METTL3-KO METTKO-DC1-GoF

```{r message=FALSE, warning=FALSE, echo=FALSE}
library(maser)
library(plotly)
library(rtracklayer)
library(dplyr)
library(DESeq2)
library(RColorBrewer)
library(pheatmap)
library(tidyverse)
library(ggrepel)
library(org.Dm.eg.db)
library(AnnotationDbi)
library(ComplexHeatmap)
library(circlize)
library(tximport)
library(ensembldb)
library(AnnotationHub)
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
library(ggplot2)
library(reshape2)
library(tidyr)
library(GO.db)
library(DBI)
```

    ## General statistics obtained from mapping the samples to the dm6 reference genome


```{r message=FALSE, warning=FALSE, echo=FALSE, out.width="100%", out.height="100%"}
data <- data.frame(
  SampleID = c("LK10_IGO_13135_CJ_10_S48", "LK11_IGO_13135_CJ_11_S49", "LK12_IGO_13135_CJ_12_S50", "LK13_IGO_13135_CJ_13_S51", 
               "LK14_IGO_13135_CJ_14_S52", "LK15_IGO_13135_CJ_15_S53", "LK1_IGO_13135_CJ_1_S54", "LK2_IGO_13135_CJ_2_S55", 
               "LK3_IGO_13135_CJ_3_S56", "LK4_IGO_13135_CJ_4_S57", "LK5_IGO_13135_CJ_5_S58", "LK6_IGO_13135_CJ_6_S59", 
               "LK7_IGO_13135_CJ_7_S60", "LK8_IGO_13135_CJ_8_S61", "LK9_IGO_13135_CJ_9_S62"),
  Number_of_input_reads = c(103960983, 125736394, 119410167, 120335851, 117884928, 109690571, 134870234, 107432296, 
                            127120985, 97632467, 120759048, 103871354, 105798848, 104178119, 111623622),
  Average_input_read_length = c(198, 197, 196, 197, 195, 196, 197, 191, 197, 192, 187, 195, 194, 191, 198),
  Uniquely_mapped_reads_number = c(29578132, 37772835, 34787725, 38393708, 38466127, 36235766, 29136034, 20719889, 
                                   25481054, 65401186, 79305783, 69663125, 60239591, 60392748, 65725939),
  Number_of_reads_mapped_to_multiple_loci = c(3122564, 3997159, 3929011, 3713408, 3752017, 3280602, 2613374, 2205849, 
                                              2465312, 8667508, 10825828, 8652958, 7336161, 6984791, 6204108),
  Number_of_reads_unmapped_too_short = c(62002083, 72838904, 70080462, 70352794, 67703709, 62788511, 97160854, 79958096, 
                                         93787366, 4472457, 5766089, 4381965, 23212549, 19573832, 21490182))
# Ensure SampleID is a factor with levels in the desired order
data$SampleID <- factor(data$SampleID, levels = data$SampleID)

# Reshape the data
long_data <- data %>%
  pivot_longer(cols = -SampleID, names_to = "Metric", values_to = "Value")

# Create the bar chart
ggplot(long_data, aes(x = SampleID, y = Value, fill = Metric)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(title = "Sequence statstics per Sample", x = "Sample ID", y = "Value") +
  scale_fill_brewer(palette = "Set3") +
  guides(fill = guide_legend(title = NULL))

```

```{r message=FALSE, warning=FALSE, echo=FALSE, out.width="100%", out.height="100%"}
data <- data.frame(
  SampleID = c("LK10_IGO_13135_CJ_10_S48", "LK11_IGO_13135_CJ_11_S49", "LK12_IGO_13135_CJ_12_S50", "LK13_IGO_13135_CJ_13_S51", 
               "LK14_IGO_13135_CJ_14_S52", "LK15_IGO_13135_CJ_15_S53", "LK1_IGO_13135_CJ_1_S54", "LK2_IGO_13135_CJ_2_S55", 
               "LK3_IGO_13135_CJ_3_S56", "LK4_IGO_13135_CJ_4_S57", "LK5_IGO_13135_CJ_5_S58", "LK6_IGO_13135_CJ_6_S59", 
               "LK7_IGO_13135_CJ_7_S60", "LK8_IGO_13135_CJ_8_S61", "LK9_IGO_13135_CJ_9_S62"),
  Number_of_input_reads = c(103960983, 125736394, 119410167, 120335851, 117884928, 109690571, 134870234, 107432296, 
                            127120985, 97632467, 120759048, 103871354, 105798848, 104178119, 111623622),
  Average_input_read_length = c(198, 197, 196, 197, 195, 196, 197, 191, 197, 192, 187, 195, 194, 191, 198),
  Uniquely_mapped_reads_number = c(29578132, 37772835, 34787725, 38393708, 38466127, 36235766, 29136034, 20719889, 
                                   25481054, 65401186, 79305783, 69663125, 60239591, 60392748, 65725939),
  Number_of_reads_mapped_to_multiple_loci = c(3122564, 3997159, 3929011, 3713408, 3752017, 3280602, 2613374, 2205849, 
                                              2465312, 8667508, 10825828, 8652958, 7336161, 6984791, 6204108),
  Number_of_reads_unmapped_too_short = c(62002083, 72838904, 70080462, 70352794, 67703709, 62788511, 97160854, 79958096, 
                                         93787366, 4472457, 5766089, 4381965, 23212549, 19573832, 21490182)
)

# Plotly bar plot
fig <- plot_ly(data, x = ~SampleID) %>%
  add_trace(y = ~Number_of_input_reads, name = 'Number of input reads', type = 'bar', mode = 'lines+markers') %>%
  add_trace(y = ~Uniquely_mapped_reads_number, name = 'Uniquely mapped reads number', type = 'bar', mode = 'lines+markers') %>%
  add_trace(y = ~Number_of_reads_mapped_to_multiple_loci, name = 'Number of reads mapped to multiple loci', type = 'bar', mode = 'lines+markers') %>%
  add_trace(y = ~Number_of_reads_unmapped_too_short, name = 'Number of reads unmapped too short', type = 'bar', mode = 'lines+markers') %>%
  layout(title = 'Sequencing Data Analysis',
         xaxis = list(title = 'Sample ID', tickangle = -45),
         yaxis = list(title = 'Read Counts'),
         legend = list(x = 1.1, y = 1))

# Show the plot
fig
```

    ## General statistics obtained from mapping the samples to the rDNA reference genome

```{r message=FALSE, warning=FALSE, echo=FALSE, out.width="100%", out.height="100%"}
# Create the data frame manually based on the provided table
data <- data.frame(
  SampleID = c("LK10_IGO_13135_CJ_10_S48", "LK11_IGO_13135_CJ_11_S49", "LK12_IGO_13135_CJ_12_S50", "LK13_IGO_13135_CJ_13_S51", 
               "LK14_IGO_13135_CJ_14_S52", "LK15_IGO_13135_CJ_15_S53", "LK1_IGO_13135_CJ_1_S54", "LK2_IGO_13135_CJ_2_S55", 
               "LK3_IGO_13135_CJ_3_S56", "LK4_IGO_13135_CJ_4_S57", "LK5_IGO_13135_CJ_5_S58", "LK6_IGO_13135_CJ_6_S59", 
               "LK7_IGO_13135_CJ_7_S60", "LK8_IGO_13135_CJ_8_S61", "LK9_IGO_13135_CJ_9_S62"),
  Number_of_input_reads = c(103960983, 125736394, 119410167, 120335851, 117884928, 109690571, 134870234, 107432296, 
                            127120985, 97632467, 120759048, 103871354, 105798848, 104178119, 111623622),
  Average_input_read_length = c(198, 197, 196, 197, 195, 196, 197, 191, 197, 192, 187, 195, 194, 191, 198),
  Uniquely_mapped_reads_number = c(1055325, 1393679, 1368458, 947191, 981940, 953247, 1204636, 1032342, 1117051, 3145038, 
                                   4290660, 3347545, 4865447, 3155895, 2411429),
  Number_of_reads_mapped_to_multiple_loci = c(579, 854, 780, 486, 532, 483, 610, 502, 674, 1562, 1902, 1609, 2304, 1499, 1270),
  Number_of_reads_unmapped_too_short = c(68274394, 83045009, 79644263, 78541485, 77777833, 71702331, 81779496, 67281577, 
                                         77295199, 65503306, 82948529, 68596522, 69786911, 71279817, 73436825)
)

# Ensure SampleID is a factor with levels in the desired order
data$SampleID <- factor(data$SampleID, levels = data$SampleID)

# Reshape the data
long_data <- data %>%
  pivot_longer(cols = -SampleID, names_to = "Metric", values_to = "Value")

# Create the bar chart
ggplot(long_data, aes(x = SampleID, y = Value, fill = Metric)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(title = "Sequence statstics per Sample", x = "Sample ID", y = "Value") +
  scale_fill_brewer(palette = "Set3") +
  guides(fill = guide_legend(title = NULL))
# Plotly bar plot
fig <- plot_ly(data, x = ~SampleID) %>%
  add_trace(y = ~Number_of_input_reads, name = 'Number of input reads', type = 'bar', mode = 'lines+markers') %>%
  add_trace(y = ~Uniquely_mapped_reads_number, name = 'Uniquely mapped reads number', type = 'bar', mode = 'lines+markers') %>%
  add_trace(y = ~Number_of_reads_mapped_to_multiple_loci, name = 'Number of reads mapped to multiple loci', type = 'bar', mode = 'lines+markers') %>%
  add_trace(y = ~Number_of_reads_unmapped_too_short, name = 'Number of reads unmapped too short', type = 'bar', mode = 'lines+markers') %>%
  layout(title = 'Sequencing Data Analysis',
         xaxis = list(title = 'Sample ID', tickangle = -45),
         yaxis = list(title = 'Read Counts'),
         legend = list(x = 1.1, y = 1))

# Show the plot
fig
```

    Read counts files and QC for each genes in every samples. Counts files were obtained from Salmon.

```{r message=FALSE, warning=FALSE, echo=FALSE}
gene<-makeTxDbFromGFF("~/Dropbox/all_files/dm6_new.gtf", format=c("gtf"))
k <- keys(gene, keytype = "TXNAME")
tx2gene <- select(gene, k, "GENEID", "TXNAME")
 
#gene <- makeTxDbFromGFF("/storehouse/sarah/references/dmel-all-r6.49.gtf.gz", format=c("gtf"))
#k <- keys(gene, keytype = "TXNAME")
#tx2gene <- select(gene, k, "GENEID", "TXNAME")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Directory containing quant files
quants_dir <- "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/quant"

# List all quant.sf files in the directory
quant_files <- list.files(quants_dir, pattern = "quant.sf$", recursive = TRUE, full.names = TRUE)

# Check the structure of the directory to list directories containing quant.sf files
quant_dirs <- list.files(quants_dir, pattern = "^LK[0-9]+_IGO_[0-9]+_CJ_[0-9]+_S[0-9]+$", full.names = TRUE)

# Extract sample names from directory names
sample_names <- gsub("_quant$", "", basename(quant_dirs))

# Assign names to the quant files
names(quant_files) <- sample_names

# Display quant files
quant_files

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
txi <- tximport(quant_files, type = "salmon", tx2gene = tx2gene,ignoreTxVersion = TRUE)

names(txi)
head(round(txi$counts))
#write.table(round(txi$counts), file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/quant/march_31_2025_Salmon_read_counts.txt", row.names = TRUE, quote = F, sep = "\t", col.names = T)
```

    Selected the reads counts from the Salmon output file for all samples and rounded each read counts.

```{r message=FALSE, warning=FALSE, echo=FALSE}
read_count_IGO_AQ <- read.table("~/Dropbox/all_files/S2_m6A/salmon_out/Salmon_read_counts.txt", header = TRUE, row.names = 1, sep = "\t", check.names=FALSE)
read_count <- read.table("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/quant/Salmon_read_counts.txt", header = TRUE, row.names = 1, sep = "\t", check.names=FALSE)
#read_count <- read.table("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/quant/march_31_2025_Salmon_read_counts.txt", header = TRUE, row.names = 1, sep = "\t", check.names=FALSE)

# Merge the count files by row names (gene IDs)
merged_counts <- merge(read_count_IGO_AQ, read_count, by = "row.names", all = TRUE)

# Rename the first column to "GeneID" (since the row names are now a column)
colnames(merged_counts)[1] <- "GeneID"

# Set the row names to "GeneID"
rownames(merged_counts) <- merged_counts$GeneID

# Remove the "GeneID" column from the data frame
merged_counts <- merged_counts[ , -1]

# Check the merged data frame
#head(merged_counts)
#write.table(merged_counts, file = "~/Dropbox/all_files/S2_m6A/merged_reads.txt", quote = F, row.names = T, sep = "\t")
```

## Filtering: only kept rows that have a sum count between all samples greater than 10. This will remove most of the genes that mostly have "0" counts. There are a total of 12400 genes after filtering out reads that are less than 10.

```{r message=FALSE, warning=FALSE, echo=FALSE}
read_count = merged_counts[which(rowSums(merged_counts) >10 ),]
dim(read_count)
```

**Sanity check**

***Read counts for differential gene expression analyses***

The correlation plot shows the difference between S2S REP1 and S2S REP2. Observation of a small dispersion of data point indicates samples are biological replicates.

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
logCountData = log2(1+read_count)
par(mfrow = c(1, 2), mar=c(8,4,4,1))  # two columns
#hist(logCountData[,1], main="Histogram of Log Read Counts", xlab="Log transformed counts")
#boxplot(logCountData,las=3, main="Boxplot of Log Read Counts")
```

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
x <- logCountData
myColors = rainbow(dim(x)[2])
#plot(density(x[,1]),col = myColors[1], lwd=2,
#     xlab="Expression values", ylab="Density", main= "Distribution of transformed data",
#     ylim=c(0, max(density(x[,1])$y)+.02 ) )
  
#for( i in 2:dim(x)[2] )
#lines(density(x[,i]),col=myColors[i], lwd=2)
#legend("topright", cex=1.1,colnames(x), lty=rep(1,dim(x)[2]), col=myColors )	
```

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
plot(logCountData[,1],logCountData[,2], xlab="S2S_REP1", ylab="S2S_REP2", xlim=c(0, 15), ylim=c(0, 19))
```

However, this correlation plot show a large dispersion of data points between S2S REP1 and Ythdc1 KO REP1. Hence, observation of a large dispersion of data point indicates samples are not biological replicates.

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
plot(logCountData[,1],logCountData[,17], xlab="S2S_REP1", ylab="DC1-KO_REP1",xlim=c(0, 15), ylim=c(0, 19))
```

```{r message=FALSE, warning=FALSE, echo=FALSE}

#sample_ids <- c("LK1_IGO_13135_CJ_1_S54", "LK2_IGO_13135_CJ_2_S55", "LK3_IGO_13135_CJ_3_S56",
#                "LK4_IGO_13135_CJ_4_S57", "LK5_IGO_13135_CJ_5_S58", "LK6_IGO_13135_CJ_6_S59",
#                "LK7_IGO_13135_CJ_7_S60", "LK8_IGO_13135_CJ_8_S61", "LK9_IGO_13135_CJ_9_S62",
#                "LK10_IGO_13135_CJ_10_S48", "LK11_IGO_13135_CJ_11_S49", "LK12_IGO_13135_CJ_12_S50",
#                "LK13_IGO_13135_CJ_13_S51", "LK14_IGO_13135_CJ_14_S52", "LK15_IGO_13135_CJ_15_S53")

# Treatments
#treatments <- c("S2 GFP", "S2 GFP", "S2 GFP", "S2 DC1-GOF", "S2 DC1-GOF", 
#                "S2 DC1-GOF", "mettl3 KO GFP", "mettl3 KO GFP", "mettl3 KO GFP",
#                "mettl3 KO DC1-GOF", "mettl3 KO DC1-GOF", "mettl3 KO DC1-GOF", 
#                "DC1 KO", "DC1 KO", "DC1 KO")

# Create a data frame
#metadata <- data.frame(SampleID = sample_ids, Treatment = treatments)

# Save the metadata table as a .txt file
#write.table(metadata, file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/metadata.txt", 
#            sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)

```

This was input into the DESeq function for differential gene expression analyses.

```{r message=FALSE, warning=FALSE, echo=FALSE}
metadata <- read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/metadata.txt", header = T, row.names = 1, sep = "\t")
metadata_IGO_AQ <- read.delim("~/Dropbox/all_files/S2_m6A/PhonoS2_m6A.txt", header = T, row.names = 1, sep = "\t")

# Rename the column in metadata_IGO_AQ to match the column name in metadata
colnames(metadata_IGO_AQ)[colnames(metadata_IGO_AQ) == "treatment"] <- "Treatment"

merged_metadata <- bind_rows(metadata_IGO_AQ,metadata)

merged_metadata
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#S2S.GFP_mettl3.KO.GFP = read_count[, c(1, 8, 9, 13:15)]logical statement that shows that the columns names in the counts files are the same as the row names in the meta data
#S2.DC1.GOF_mettl3.KO.DC1.GOF = read_count[, c(10:12, 2:4)]
#S2.DC1.GOF_DC1.KO = read_count[, c(10:12, 5:7)]

#DC1_GOF_DC1_KO_metadata = merged_metadata[c(10:12,25:27),]
#DC1_GOF_DC1_KO_read_counts = read_count[c(10:12,17:19)]
#DC1_GOF_DC1_KOcolData = as.data.frame(cbind(colnames(DC1_GOF_DC1_KO_read_counts), DC1_GOF_DC1_KO_metadata)) 
#all(rownames(DC1_GOF_DC1_KOcolData) %in% colnames(DC1_GOF_DC1_KO_read_counts))


All_colData = as.data.frame(cbind(colnames(merged_counts), merged_metadata)) 


S2_S_DC1_KO_metadata = merged_metadata[c(1:3,25:27),]
S2_S_DC1_KO_read_counts = read_count[c(1:3,17:19)]
S2_S_DC1_KOcolData = as.data.frame(cbind(colnames(S2_S_DC1_KO_read_counts), S2_S_DC1_KO_metadata)) 
#all(rownames(S2_S_DC1_KO_metadata) %in% colnames(S2_S_DC1_KO_read_counts))

S2_S_mettl3_KO_metadata = merged_metadata[1:6,]
S2_S_mettl3_KO_read_counts = read_count[1:6]
S2_S_mettl3_KOcolData = as.data.frame(cbind(colnames(S2_S_mettl3_KO_read_counts), S2_S_mettl3_KO_metadata)) 
#all(rownames(S2_S_mettl3_KO_metadata) %in% colnames(S2_S_mettl3_KO_read_counts))


#S2S_DC1_GOF_metadata = merged_metadata[c(1:3,10:12), ]
#S2S_DC1_GOF = read_count[c(1:3,10:12)]
#S2S_DC1_GOF_colData = as.data.frame(cbind(colnames(S2S_DC1_GOF), S2S_DC1_GOF_metadata))
#all(rownames(S2S_DC1_GOF_metadata) %in% colnames(S2S_DC1_GOF))

S2.GFP_S2.DC1.GOF = read_count[, c(13, 20:24)]
S2.DC1.GOF_S2S.GF_metadata = merged_metadata[c(13:18),]
S2.DC1.GOF_S2S.GF_colData = as.data.frame(cbind(colnames(S2.GFP_S2.DC1.GOF), S2.DC1.GOF_S2S.GF_metadata))
#all(rownames(S2.DC1.GOF_S2S.GF_metadata) %in% colnames(S2.GFP_S2.DC1.GOF))

mettl3.KO.GFP_mettl3.KO.DC1.GOF = read_count[, c(25:27,  14:16)]
mettl3.KO.GFP_mettl3.KO.DC1.GOF_metadata = merged_metadata[c(19:24),]
mettl3.KO.GFP_mettl3.KO.DC1_colData = as.data.frame(cbind(colnames(mettl3.KO.GFP_mettl3.KO.DC1.GOF), mettl3.KO.GFP_mettl3.KO.DC1.GOF_metadata))
#all(rownames(mettl3.KO.GFP_mettl3.KO.DC1.GOF_metadata) %in% colnames(mettl3.KO.GFP_mettl3.KO.DC1.GOF))

S2.YTHDC_mettl3.KO.DC1.GOF = read_count[, c(22:24, 14:16)]
S2.YTHDC_mettl3.KO.DC1.GOF_metadata = merged_metadata[c(16:18, 22:24),]
S2.YTHDC_mettl3.KO.DC1.GOF_colData = as.data.frame(cbind(colnames(S2.YTHDC_mettl3.KO.DC1.GOF), S2.YTHDC_mettl3.KO.DC1.GOF_metadata))

#all(rownames(S2.YTHDC_mettl3.KO.DC1.GOF_metadata) %in% colnames(S2.YTHDC_mettl3.KO.DC1.GOF))





S2.GFP_mettl3.KO.DC1.GOF = read_count[, c(13,20,21, 14:16)]
S2.GFP_mettl3.KO.DC1.GOF_metadata = merged_metadata[c(13:15, 22:24),]
S2.GFP_mettl3.KO.DC1.GOF_colData = as.data.frame(cbind(colnames(S2.GFP_mettl3.KO.DC1.GOF), S2.GFP_mettl3.KO.DC1.GOF_metadata))
#all(rownames(S2.GFP_mettl3.KO.DC1.GOF_metadata) %in% colnames(S2.GFP_mettl3.KO.DC1.GOF))


mettl3.KO.GFP_S2.GFP = read_count[, c(13,20,21,  25:27)]
mettl3.KO.GFP_S2.GFP_metadata = merged_metadata[c(13:15, 19:21),]
mettl3.KO.GFP_S2.GFP_colData = as.data.frame(cbind(colnames(mettl3.KO.GFP_S2.GFP), mettl3.KO.GFP_S2.GFP_metadata))

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#1
#dds_DC1_GOF_DC1_KO = DESeqDataSetFromMatrix(countData = DC1_GOF_DC1_KO_read_counts,
#                                  colData = DC1_GOF_DC1_KOcolData,
#                                  design = ~DC1_GOF_DC1_KO_metadata)
#dds_DC1_GOF_DC1_KO$DC1_GOF_DC1_KO_metadata = relevel(dds_DC1_GOF_DC1_KO$DC1_GOF_DC1_KO_metadata, ref = "DC1_GOF")


#2
dds_S2_S_mettl3_KO = DESeqDataSetFromMatrix(countData = S2_S_mettl3_KO_read_counts,
                                  colData = S2_S_mettl3_KOcolData,
                                  design = ~S2_S_mettl3_KO_metadata)
dds_S2_S_mettl3_KO$S2_S_mettl3_KO_metadata = relevel(dds_S2_S_mettl3_KO$S2_S_mettl3_KO_metadata, ref = "S2_S")

#3
#dds_S2S_DC1_GOF = DESeqDataSetFromMatrix(countData = S2S_DC1_GOF,
#                                  colData = S2S_DC1_GOF_colData,
#                                  design = ~S2S_DC1_GOF_metadata)
#dds_S2S_DC1_GOF$S2S_DC1_GOF_metadata = relevel(dds_S2S_DC1_GOF$S2S_DC1_GOF_metadata, ref = "S2_S")


#4
dds_S2.GFP_S2.DC1.GOF = DESeqDataSetFromMatrix(countData = S2.GFP_S2.DC1.GOF,
                                  colData = S2.DC1.GOF_S2S.GF_colData,
                                  design = ~S2.DC1.GOF_S2S.GF_metadata)
dds_S2.GFP_S2.DC1.GOF$S2.DC1.GOF_S2S.GF_metadata = relevel(dds_S2.GFP_S2.DC1.GOF$S2.DC1.GOF_S2S.GF_metadata, ref = "S2 GFP")

#5
dds_mettl3.KO.GFP_mettl3.KO.DC1.GOF = DESeqDataSetFromMatrix(countData = mettl3.KO.GFP_mettl3.KO.DC1.GOF,
                                  colData = mettl3.KO.GFP_mettl3.KO.DC1_colData,
                                  design = ~mettl3.KO.GFP_mettl3.KO.DC1.GOF_metadata)
dds_mettl3.KO.GFP_mettl3.KO.DC1.GOF$mettl3.KO.GFP_mettl3.KO.DC1.GOF_metadata = relevel(dds_mettl3.KO.GFP_mettl3.KO.DC1.GOF$mettl3.KO.GFP_mettl3.KO.DC1.GOF_metadata, ref = "mettl3 KO GFP")

#6
dds_S2_S_DC1_KO = DESeqDataSetFromMatrix(countData = S2_S_DC1_KO_read_counts,
                                  colData = S2_S_DC1_KOcolData,
                                  design = ~S2_S_DC1_KO_metadata)
dds_S2_S_DC1_KO$S2_S_DC1_KO_metadata = relevel(dds_S2_S_DC1_KO$S2_S_DC1_KO_metadata, ref = "S2_S")

#7
dds_S2.YTHDC_mettl3.KO.DC1.GOF = DESeqDataSetFromMatrix(countData = S2.YTHDC_mettl3.KO.DC1.GOF,
                                  colData = S2.YTHDC_mettl3.KO.DC1.GOF_colData,
                                  design = ~S2.YTHDC_mettl3.KO.DC1.GOF_metadata)
dds_S2.YTHDC_mettl3.KO.DC1.GOF$S2.YTHDC_mettl3.KO.DC1.GOF_metadata = relevel(dds_S2.YTHDC_mettl3.KO.DC1.GOF$S2.YTHDC_mettl3.KO.DC1.GOF_metadata, ref = "mettl3 KO DC1-GOF")



#8
dds_S2.GFP_mettl3.KO.DC1.GOF = DESeqDataSetFromMatrix(countData = S2.GFP_mettl3.KO.DC1.GOF,
                                  colData = S2.GFP_mettl3.KO.DC1.GOF_colData,
                                  design = ~S2.GFP_mettl3.KO.DC1.GOF_metadata)
dds_S2.GFP_mettl3.KO.DC1.GOF$S2.GFP_mettl3.KO.DC1.GOF_metadata = relevel(dds_S2.GFP_mettl3.KO.DC1.GOF$S2.GFP_mettl3.KO.DC1.GOF_metadata, ref = "S2 GFP")


#9
dds_mettl3.KO.GFP_S2.GFP = DESeqDataSetFromMatrix(countData = mettl3.KO.GFP_S2.GFP,
                                  colData = mettl3.KO.GFP_S2.GFP_colData,
                                  design = ~mettl3.KO.GFP_S2.GFP_metadata)
dds_mettl3.KO.GFP_S2.GFP$mettl3.KO.GFP_S2.GFP_metadata = relevel(dds_mettl3.KO.GFP_S2.GFP$mettl3.KO.GFP_S2.GFP_metadata, ref = "S2 GFP")

dds_S2.GFP_S2.DC1.GOF = DESeq(dds_S2.GFP_S2.DC1.GOF)
dds_S2.GFP_mettl3.KO.DC1.GOF = DESeq(dds_S2.GFP_mettl3.KO.DC1.GOF)
dds_mettl3.KO.GFP_S2.GFP = DESeq(dds_mettl3.KO.GFP_S2.GFP)
dds_mettl3.KO.GFP_mettl3.KO.DC1.GOF = DESeq(dds_mettl3.KO.GFP_mettl3.KO.DC1.GOF)
dds_S2_S_DC1_KO = DESeq(dds_S2_S_DC1_KO)
dds_S2.YTHDC_mettl3.KO.DC1.GOF = DESeq(dds_S2.YTHDC_mettl3.KO.DC1.GOF)
dds_S2_S_mettl3_KO = DESeq(dds_S2_S_mettl3_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#dds = DESeqDataSetFromMatrix(countData = read_count,
#                             colData = metadata,
#                             design = ~Treatment)
#dds = DESeq(dds)
#nrow(dds)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#dds = dds[rowSums(counts(dds)) > 5,]
#nrow(dds)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#rld_All_counts = rlog(dds)
#plotPCA(rld_All_counts, intgroup = "Treatment")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#nrow(dds_DC1_GOF_DC1_KO)
```

```{r}

#plotDEXSeq( dxr2, "FBgn0010909", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
```

### ***1*** Differential expressed genes in Mettl3-KO-Ythdc1-GoF KO relative to S2-GFP


```{r message=FALSE, warning=FALSE, echo=FALSE}


#res_S2_GFP_mettl3_KO_DC1_GOF <- results(dds_S2.GFP_mettl3.KO.DC1.GOF)


gene_validation = read.delim("~/Dropbox/id_validation_table_67407.txt", header = TRUE, sep = "\t")
#write.table(res_S2_GFP_mettl3_KO_DC1_GOF, file = "res_S2_GFP_mettl3_KO_DC1_GOF.txt", sep = "\t", quote = F)
#write.table(res_mettl3_KO_GFP_S2_GFP, file = "res_mettl3_KO_GFP_S2_GFP.txt", sep = "\t", quote = F)

res_S2_GFP_mettl3_KO_DC1_GOF = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_S2_GFP_mettl3_KO_DC1_GOF.txt", header = T)
res_S2_GFP_mettl3_KO_DC1_GOF$gene_name = rownames(res_S2_GFP_mettl3_KO_DC1_GOF)
res_S2_GFP_mettl3_KO_DC1_GOF = left_join(res_S2_GFP_mettl3_KO_DC1_GOF, gene_validation, by = c("gene_name" = "submitted_item"))

res_S2_GFP_mettl3_KO_DC1_GOF <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  distinct(gene_name, .keep_all = TRUE)

#res_S2_GFP_mettl3_KO_DC1_GOF_clean <- res_S2_GFP_mettl3_KO_DC1_GOF[res_S2_GFP_mettl3_KO_DC1_GOF$baseMean>0 & !is.na(res_S2_GFP_mettl3_KO_DC1_GOF$padj), ]

res_S2_GFP_mettl3_KO_DC1_GOF$delabel <- ifelse(res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol %in% head(res_S2_GFP_mettl3_KO_DC1_GOF[order(res_S2_GFP_mettl3_KO_DC1_GOF$padj), "current_symbol"], 500), res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol, NA)

# Calculate counts for each category
res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed <- "Not Significant"
res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed[res_S2_GFP_mettl3_KO_DC1_GOF$log2FoldChange > 0.5 & res_S2_GFP_mettl3_KO_DC1_GOF$padj <= 0.05] <- "UP"
res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed[res_S2_GFP_mettl3_KO_DC1_GOF$log2FoldChange < -0.5 & res_S2_GFP_mettl3_KO_DC1_GOF$padj <= 0.05] <- "DOWN"
res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed[!(res_S2_GFP_mettl3_KO_DC1_GOF$log2FoldChange > 0.5 & res_S2_GFP_mettl3_KO_DC1_GOF$padj < 0.05) & !(res_S2_GFP_mettl3_KO_DC1_GOF$log2FoldChange < -0.5 & res_S2_GFP_mettl3_KO_DC1_GOF$padj < 0.05)] <- "Not Significant"

count_UP <- sum(res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed == "UP")
count_DOWN <- sum(res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed == "Not Significant")

# Filter for significant genes to label (UP and DOWN regulated)
significant_genes <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")

# Plot with updated legend labels and adjusted p-values
ggplot(res_S2_GFP_mettl3_KO_DC1_GOF, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed, label = delabel)) +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') + 
  geom_vline(xintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_vline(xintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_point(size = 2) + 
  scale_color_manual(values = c("blue", "grey", "red"), 
                     labels = c(paste("Downregulated (", count_DOWN, ")", sep=""), 
                                paste("Not significant (", count_Not_Significant, ")", sep=""),
                                paste("Upregulated (", count_UP, ")", sep=""))) +
  coord_cartesian(ylim = c(0, 300), xlim = c(-10, 10)) +
  labs(color = ' ', x = expression("log"[2]*"FC"), y = expression("-log"[10]*"padj")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  ggtitle('DEG in Mettl3-KO-Ythdc1-GoF/S2-GFP using padj <= 0.05 and Log2FC |0.5|') + 
  geom_text_repel(data = significant_genes, aes(label = delabel), max.overlaps = 10)

#res_S2_GFP_mettl3_KO_DC1_GOF_clean = res_S2_GFP_mettl3_KO_DC1_GOF_clean %>% dplyr::select(-symbol)
#write.table(res_S2_GFP_mettl3_KO_DC1_GOF_clean, file = "/Users/georgeboateng-sarfo/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/quant/res_S2_GFP_mettl3_KO_DC1_GOF_clean.txt", sep = "\t", quote = FALSE)

```

    
    MA-plot: shows the fold change over the average expression level of Ythdc1 KO relative to S2S.

```{r message=FALSE, warning=FALSE, echo=FALSE}
# List of genes to highlight
highlight_genes <- c("fl(2)d", "aqz", "sky", "pum", "prosap", "Mettl3", "Ythdc1", "unc-13", "Hsp70Ab")

# Add a column for highlighting specific genes
res_S2_GFP_mettl3_KO_DC1_GOF$highlight <- ifelse(res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol %in% highlight_genes, "Highlighted", res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed)

# Count numbers for legend
#count_highlighted <- sum(res_S2_GFP_mettl3_KO_DC1_GOF_clean$highlight == "Highlighted")
count_UP <- sum(res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed == "UP")
count_DOWN <- sum(res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2_GFP_mettl3_KO_DC1_GOF$diffexpressed == "Not Significant")

# Create custom colors for highlighting "Highlighted" = "yellow",
custom_colors <- c("Highlighted" = "yellow", "UP" = "red", "DOWN" = "blue", "Not Significant" = "grey")

# Create a data frame for labeling highlight_genes
highlight_label_genes <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(current_symbol %in% highlight_genes)

# Filter top significant genes for labeling
top_genes <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed != "Not Significant") %>%
  arrange(padj) %>%
  slice(1:50)

# Combine highlight_genes and top_genes for labeling
label_genes <- dplyr::bind_rows(top_genes, highlight_label_genes) %>%
  dplyr::distinct(current_symbol, .keep_all = TRUE)

# Create the MA plot
ggplot(res_S2_GFP_mettl3_KO_DC1_GOF, aes(x = log2(baseMean + 1), y = log2FoldChange, color = highlight)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_text_repel(data = label_genes, aes(label = current_symbol), size = 3, max.overlaps = Inf, color = "black") +
  scale_color_manual(
    values = custom_colors,
    # Only include these groups in the legend (order determines the legend order) MA_METTLE_Ythdc_GoF_S2_GFP
    breaks = c("DOWN", "Not Significant", "UP"),
    labels = c(
      paste("Downregulated (", count_DOWN, ")", sep = ""),
      paste("Not Significant (", count_Not_Significant, ")", sep = ""),
      paste("Upregulated (", count_UP, ")", sep = ""))) + xlim(0, 20) + ylim(c(-10, 10)) +
  labs(title = " ", x = "log2(baseMean)", y = "Log2 Fold Change", color = " ") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle('Mettl3-KO-Ythdc1-GoF/S2-GFP: padj <= 0.05 & Log2FC |0.5|')


```


```{r message=FALSE, warning=FALSE, echo=FALSE}

# Load necessary libraries
library(org.Dm.eg.db)  # Assuming you're working with Drosophila
library(GO.db)
library(AnnotationDbi)

# Assuming your data frame has a column called "gene_name" containing gene symbols
if ("current_symbol" %in% colnames(res_S2_GFP_mettl3_KO_DC1_GOF)) {
  
  # Make gene names unique if needed
  res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol <- make.unique(as.character(res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol))
  
  # Set gene names as rownames (optional)
  rownames(res_S2_GFP_mettl3_KO_DC1_GOF) <- res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol
  
  # Map ENTREZ IDs
  res_S2_GFP_mettl3_KO_DC1_GOF$ENTREZID <- mapIds(org.Dm.eg.db,
                               keys = res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "ENTREZID",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO IDs
  res_S2_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "GO",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO term descriptions
  res_S2_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(res_S2_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  # Map GO ontology (BP, MF, CC)
  res_S2_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "ONTOLOGY",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Optional: Map gene descriptions
  res_S2_GFP_mettl3_KO_DC1_GOF$GENENAME <- mapIds(org.Dm.eg.db,
                               keys = res_S2_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "GENENAME",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
} else {
  cat("The column 'current_symbol' does not exist in the data frame.\n")
}


#write.table(res_S2_GFP_mettl3_KO_DC1_GOF, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/res_mettl3_KO_DC1_GOF_clean_S2_GFP.txt", quote = F, sep = "\t", row.names = F)
```


```{r}

library(clusterProfiler)
library(org.Dm.eg.db)
organism <- org.Dm.eg.db

keytypes(organism)

sig_S2_GFP_mettl3_KO_DC1_GOF_clean <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")


gene_scores <- sig_S2_GFP_mettl3_KO_DC1_GOF_clean$log2FoldChange
names(gene_scores) <- sig_S2_GFP_mettl3_KO_DC1_GOF_clean$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    nPermSimple = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

#head(gse@result)


```




##Category Netplot
The cnetplot depicts the linkages of genes and biological concepts (e.g. GO terms or KEGG pathways) as a network (helpful to see which genes are involved in enriched pathways and genes that may belong to multiple annotation categories).
```{r fig.width=18}
#install.packages("ggnewscale")MA_Ythdc1-KO_S2S 
gene_scores <- sig_S2_GFP_mettl3_KO_DC1_GOF_clean$log2FoldChange
names(gene_scores) <- sig_S2_GFP_mettl3_KO_DC1_GOF_clean$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 
cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 30) + ggtitle("Mettl3-KO-Ythdc1-GoF/S2-GFP")
```

    Gene Ontology analyses Ythdc1 KO relative to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}

# Load the packages
library(clusterProfiler)
#library(org.Dm.eg.db)
library(DOSE)
#library(dplyr)
# Assuming `res_S2_GFP_mettl3_KO_DC1_GOF` contains your differential expression results

# Filter for upregulated and downregulated genes
upregulated_genes_S2_GFP_mettl3_KO_DC1_GOF <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(log2FoldChange > 0.5 & padj <= 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

downregulated_genes_S2_GFP_mettl3_KO_DC1_GOF <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(log2FoldChange < -0.5 & padj <= 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

# Convert gene symbols to Entrez IDs
upregulated_entrez <- bitr(upregulated_genes_S2_GFP_mettl3_KO_DC1_GOF, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)
downregulated_entrez <- bitr(downregulated_genes_S2_GFP_mettl3_KO_DC1_GOF, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)


# GO analysis for upregulated genes
go_up <- enrichGO(gene = upregulated_entrez$ENTREZID,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez$ENTREZID,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes Ythdc-KO_S2S_DGE_GO_up_ALL_20
barplot(go_up, showCategory = 25, 
        title = "Mettl3-KO-Ythdc1-GoF/S2-GFP_DGE:GO Enrichment for Upregulated")

# Bar plot for downregulated genes Ythdc_KO_S2S_DGE_GO_down_ALL_20 
barplot(go_down, showCategory = 25, 
        title = "Mettl3-KO-Ythdc1-GoF/S2-GFP_DGE:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 20, 
        title = "Mettl3-KO-Ythdc1-GoF/S2-GFP_DGE:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 20, 
        title = "Mettl3-KO-Ythdc1-GoF/S2-GFP_DGE:GO Enrichment for Downregulated Genes")

# If you want to see the results in a table
go_up_results <- as.data.frame(go_up)
go_down_results <- as.data.frame(go_down)

# View the top results
head(go_up_results)
head(go_down_results)

```


### ***2*** Differential expressed genes in Mettl3-KO-GFP relative to S2-GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
#res_mettl3_KO_GFP_S2_GFP = results(dds_mettl3.KO.GFP_S2.GFP)
#gene_validation = read.delim("~/Downloads/id_validation_table_67407.txt", header = TRUE, sep = "\t")

#write.table(res_mettl3_KO_GFP_S2_GFP, file = "res_mettl3_KO_GFP_S2_GFP.txt", sep = "\t", quote = F)

res_mettl3_KO_GFP_S2_GFP = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_mettl3_KO_GFP_S2_GFP.txt", header = T)
res_mettl3_KO_GFP_S2_GFP$gene_name = rownames(res_mettl3_KO_GFP_S2_GFP)
res_mettl3_KO_GFP_S2_GFP = left_join(res_mettl3_KO_GFP_S2_GFP, gene_validation, by = c("gene_name" = "submitted_item"))

res_mettl3_KO_GFP_S2_GFP <- res_mettl3_KO_GFP_S2_GFP %>%
  distinct(gene_name, .keep_all = TRUE)

#res_mettl3_KO_GFP_S2_GFP_clean <- res_mettl3_KO_GFP_S2_GFP[res_mettl3_KO_GFP_S2_GFP$baseMean>0 & !is.na(res_mettl3_KO_GFP_S2_GFP$padj), ]

res_mettl3_KO_GFP_S2_GFP$delabel <- ifelse(res_mettl3_KO_GFP_S2_GFP$current_symbol %in% head(res_mettl3_KO_GFP_S2_GFP[order(res_mettl3_KO_GFP_S2_GFP$padj), "current_symbol"], 500), res_mettl3_KO_GFP_S2_GFP$current_symbol, NA)

# Calculate counts for each category
res_mettl3_KO_GFP_S2_GFP$diffexpressed <- "Not Significant"
res_mettl3_KO_GFP_S2_GFP$diffexpressed[res_mettl3_KO_GFP_S2_GFP$log2FoldChange > 0.5 & res_mettl3_KO_GFP_S2_GFP$padj <= 0.05] <- "UP"
res_mettl3_KO_GFP_S2_GFP$diffexpressed[res_mettl3_KO_GFP_S2_GFP$log2FoldChange < -0.5 & res_mettl3_KO_GFP_S2_GFP$padj <= 0.05] <- "DOWN"
res_mettl3_KO_GFP_S2_GFP$diffexpressed[!(res_mettl3_KO_GFP_S2_GFP$log2FoldChange > 0.5 & res_mettl3_KO_GFP_S2_GFP$padj < 0.05) & !(res_mettl3_KO_GFP_S2_GFP$log2FoldChange < -0.5 & res_mettl3_KO_GFP_S2_GFP$padj < 0.05)] <- "Not Significant"

count_UP <- sum(res_mettl3_KO_GFP_S2_GFP$diffexpressed == "UP")
count_DOWN <- sum(res_mettl3_KO_GFP_S2_GFP$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_mettl3_KO_GFP_S2_GFP$diffexpressed == "Not Significant")

# Filter for significant genes to label (UP and DOWN regulated)
significant_genes <- res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(diffexpressed != "Not Significant")

# Plot with updated legend labels and adjusted p-values volc_mettl3_KO_GFP_S2_GFP
ggplot(res_mettl3_KO_GFP_S2_GFP, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed, label = delabel)) +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') + 
  geom_vline(xintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_vline(xintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_point(size = 2) + 
  scale_color_manual(values = c("blue", "grey", "red"), 
                     labels = c(paste("Downregulated (", count_DOWN, ")", sep=""), 
                                paste("Not significant (", count_Not_Significant, ")", sep=""),
                                paste("Upregulated (", count_UP, ")", sep=""))) +
  coord_cartesian(ylim = c(0, 300), xlim = c(-10, 10)) +
  labs(color = ' ', x = expression("log"[2]*"FC"), y = expression("-log"[10]*"padj")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  ggtitle('DEG in Mettl3-KO-GFP/S2-GFP using padj <= 0.05 and Log2FC |0.5|') + 
  geom_text_repel(data = significant_genes, aes(label = delabel), max.overlaps = 10)


```

    
    MA-plot: shows the fold change over the average expression level of Ythdc1 KO relative to S2S.

```{r message=FALSE, warning=FALSE, echo=FALSE}
# List of genes to highlight
highlight_genes <- c("fl(2)d", "aqz", "sky", "pum", "prosap", "Mettl3", "Ythdc1", "unc-13", "Hsp70Ab")

# Add a column for highlighting specific genes volc_mettl3_KO_GFP_S2_GFP
res_mettl3_KO_GFP_S2_GFP$highlight <- ifelse(res_mettl3_KO_GFP_S2_GFP$current_symb %in% highlight_genes, "Highlighted", res_mettl3_KO_GFP_S2_GFP$diffexpressed)

# Count numbers for legend
#count_highlighted <- sum(res_mettl3_KO_GFP_S2_GFP_clean$highlight == "Highlighted")
count_UP <- sum(res_mettl3_KO_GFP_S2_GFP$diffexpressed == "UP")
count_DOWN <- sum(res_mettl3_KO_GFP_S2_GFP$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_mettl3_KO_GFP_S2_GFP$diffexpressed == "Not Significant")

# Create custom colors for highlighting "Highlighted" = "yellow",
custom_colors <- c("Highlighted" = "yellow", "UP" = "red", "DOWN" = "blue", "Not Significant" = "grey")

# Create a data frame for labeling highlight_genes
highlight_label_genes <- res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(current_symbol %in% highlight_genes)

# Filter top significant genes for labeling
top_genes <- res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(diffexpressed != "Not Significant") %>%
  arrange(padj) %>%
  slice(1:50)

# Combine highlight_genes and top_genes for labeling
label_genes <- dplyr::bind_rows(top_genes, highlight_label_genes) %>%
  dplyr::distinct(current_symbol, .keep_all = TRUE)

# Create the MA plot volc_mettl3_KO_GFP_S2_GFP
ggplot(res_mettl3_KO_GFP_S2_GFP, aes(x = log2(baseMean + 1), y = log2FoldChange, color = highlight)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_text_repel(data = label_genes, aes(label = current_symbol), size = 3, max.overlaps = Inf, color = "black") +
  scale_color_manual(
    values = custom_colors,
    # Only include these groups in the legend (order determines the legend order)
    breaks = c("DOWN", "Not Significant", "UP"),
    labels = c(
      paste("Downregulated (", count_DOWN, ")", sep = ""),
      paste("Not Significant (", count_Not_Significant, ")", sep = ""),
      paste("Upregulated (", count_UP, ")", sep = ""))) + xlim(0, 20) + ylim(c(-10, 10)) +
  labs(title = " ", x = "log2(baseMean)", y = "Log2 Fold Change", color = " ") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle('Mettl3-KO-GFP/S2-GFP: padj <= 0.05 & Log2FC |0.5|')


```


```{r}
# Assuming your data frame has a column called "gene_name" containing gene symbols
if ("current_symbol" %in% colnames(res_mettl3_KO_GFP_S2_GFP)) {
  
  # Make gene names unique if needed
  res_mettl3_KO_GFP_S2_GFP$current_symbol <- make.unique(as.character(res_mettl3_KO_GFP_S2_GFP$current_symbol))
  
  # Set gene names as rownames (optional)
  rownames(res_mettl3_KO_GFP_S2_GFP) <- res_mettl3_KO_GFP_S2_GFP$current_symbol
  
  # Map ENTREZ IDs
  res_mettl3_KO_GFP_S2_GFP$ENTREZID <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_S2_GFP$current_symbol,
                               column = "ENTREZID",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO IDs
  res_mettl3_KO_GFP_S2_GFP$GO <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_S2_GFP$current_symbol,
                               column = "GO",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO term descriptions
  res_mettl3_KO_GFP_S2_GFP$GODESCR <- sapply(res_mettl3_KO_GFP_S2_GFP$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  # Map GO ontology (BP, MF, CC)
  res_mettl3_KO_GFP_S2_GFP$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_S2_GFP$current_symbol,
                               column = "ONTOLOGY",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Optional: Map gene descriptions
  res_mettl3_KO_GFP_S2_GFP$GENENAME <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_S2_GFP$current_symbol,
                               column = "GENENAME",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
} else {
  cat("The column 'current_symbol' does not exist in the data frame.\n")
}



#write.table(res_mettl3_KO_GFP_S2_GFP_clean, file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_mettl3_KO_GFP_S2_GFP_clean.txt", sep = "\t", quote = FALSE)

```




```{r}

library(clusterProfiler)
library(org.Dm.eg.db)
organism <- org.Dm.eg.db

keytypes(organism)

Sig_res_mettl3_KO_GFP_S2_GFP <- res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(diffexpressed != "Not Significant")


gene_scores <- Sig_res_mettl3_KO_GFP_S2_GFP$log2FoldChange
names(gene_scores) <- Sig_res_mettl3_KO_GFP_S2_GFP$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    nPermSimple = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

#head(gse@result)


```




##Category Netplot
The cnetplot depicts the linkages of genes and biological concepts (e.g. GO terms or KEGG pathways) as a network (helpful to see which genes are involved in enriched pathways and genes that may belong to multiple annotation categories).
```{r fig.width=18}
#install.packages("ggnewscale")MA_mettl3_KO_GFP_S2_GFP
gene_scores <- Sig_res_mettl3_KO_GFP_S2_GFP$log2FoldChange
names(gene_scores) <- Sig_res_mettl3_KO_GFP_S2_GFP$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 
cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 30) + ggtitle("Mettl3-KO-GFP/S2-GFP")
```

  Gene Ontology analyses Ythdc1 KO relative to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}

# Load the packages
library(clusterProfiler)
#library(org.Dm.eg.db)
library(DOSE)
#library(dplyr)
# Assuming `res_mettl3_KO_GFP_S2_GFP` contains your differential expression results

# Filter for upregulated and downregulated genes
upregulated_genes_mettl3_KO_GFP_S2_GFP <- Sig_res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(log2FoldChange > 0.5 & padj <= 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

downregulated_genes_mettl3_KO_GFP_S2_GFP <- Sig_res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(log2FoldChange < -0.5 & padj <= 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

# Convert gene symbols to Entrez IDs
upregulated_entrez <- bitr(upregulated_genes_mettl3_KO_GFP_S2_GFP, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)
downregulated_entrez <- bitr(downregulated_genes_mettl3_KO_GFP_S2_GFP, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)


# GO analysis for upregulated genes
go_up <- enrichGO(gene = upregulated_entrez$ENTREZID,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez$ENTREZID,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes go_up_mettl3_KO_GFP_S2_GFP
barplot(go_up, showCategory = 25, 
        title = "Mettl3-KO-GFP/S2-GFP_DGE:GO Enrichment for Upregulated")

# Bar plot for downregulated genes go_down_mettl3_KO_GFP_S2_GFP
barplot(go_down, showCategory = 25, 
        title = "Mettl3-KO-GFP/S2-GFP_DGE:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 25, 
        title = "Mettl3-KO-GFPF/S2-GFP_DGE:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 25, 
        title = "Mettl3-KO-GFP/S2-GFP_DGE:GO Enrichment for Downregulated Genes")

# If you want to see the results in a table
go_up_results <- as.data.frame(go_up)
go_down_results <- as.data.frame(go_down)

# View the top results
head(go_up_results)
head(go_down_results)

```




### ***3*** Differential expressed genes in Ythdc1 KO relative to S2S


```{r message=FALSE, warning=FALSE, echo=FALSE}
#res_S2_S_DC1_KO <- results(dds_S2_S_DC1_KO)

#write.table(res_S2_S_DC1_KO, file = "res_S2_S_DC1_KO.txt", sep = "\t", quote = F)

#res_S2_S_DC1_KO = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/quant/S2S_DC1_KO.txt", header = T)
res_S2_S_DC1_KO = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_S2_S_DC1_KO.txt", header = T)
res_S2_S_DC1_KO$gene_name = rownames(res_S2_S_DC1_KO)
res_S2_S_DC1_KO = left_join(res_S2_S_DC1_KO, gene_validation, by = c("gene_name" = "submitted_item"))

res_S2_S_DC1_KO <- res_S2_S_DC1_KO %>%
  distinct(gene_name, .keep_all = TRUE)
#res_S2_S_DC1_KO_clean <- res_S2_S_DC1_KO[res_S2_S_DC1_KO$baseMean>0 & !is.na(res_S2_S_DC1_KO$padj), ]

res_S2_S_DC1_KO$delabel <- ifelse(res_S2_S_DC1_KO$current_symbol %in% head(res_S2_S_DC1_KO[order(res_S2_S_DC1_KO$padj), "current_symbol"], 500), res_S2_S_DC1_KO$current_symbol, NA)

# Calculate counts for each category
res_S2_S_DC1_KO$diffexpressed <- "Not Significant"
res_S2_S_DC1_KO$diffexpressed[res_S2_S_DC1_KO$log2FoldChange > 0.5 & res_S2_S_DC1_KO$padj <= 0.05] <- "UP"
res_S2_S_DC1_KO$diffexpressed[res_S2_S_DC1_KO$log2FoldChange < -0.5 & res_S2_S_DC1_KO$padj <= 0.05] <- "DOWN"
res_S2_S_DC1_KO$diffexpressed[!(res_S2_S_DC1_KO$log2FoldChange > 0.5 & res_S2_S_DC1_KO$padj < 0.05) & !(res_S2_S_DC1_KO$log2FoldChange < -0.5 & res_S2_S_DC1_KO$padj < 0.05)] <- "Not Significant"

count_UP <- sum(res_S2_S_DC1_KO$diffexpressed == "UP")
count_DOWN <- sum(res_S2_S_DC1_KO$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2_S_DC1_KO$diffexpressed == "Not Significant")

# Filter for significant genes to label (UP and DOWN regulated) S2_S_DC1_KO
significant_genes <- res_S2_S_DC1_KO %>%
  dplyr::filter(diffexpressed != "Not Significant")

# Plot with updated legend labels and adjusted p-values MA_Ythdc1-KO_S2S
ggplot(res_S2_S_DC1_KO, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed, label = delabel)) +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') + 
  geom_vline(xintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_vline(xintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_point(size = 2) + 
  scale_color_manual(values = c("blue", "grey", "red"), 
                     labels = c(paste("Downregulated (", count_DOWN, ")", sep=""), 
                                paste("Not significant (", count_Not_Significant, ")", sep=""),
                                paste("Upregulated (", count_UP, ")", sep=""))) +
  coord_cartesian(ylim = c(0, 300), xlim = c(-10, 10)) +
  labs(color = ' ', x = expression("log"[2]*"FC"), y = expression("-log"[10]*"padj")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  ggtitle('DEG in Ythdc1-KO vs S2S using padj <= 0.05 and Log2FC |0.5|') + 
  geom_text_repel(data = significant_genes, aes(label = delabel), max.overlaps = 10)


#res_S2_S_DC1_KO_clean <- res_S2_S_DC1_KO_clean %>% dplyr::select(-symbol)



```

    
    MA-plot: shows the fold change over the average expression level of Ythdc1 KO relative to S2S.

```{r message=FALSE, warning=FALSE, echo=FALSE}
# List of genes to highlight
highlight_genes <- c("fl(2)d", "aqz", "sky", "pum", "prosap", "Mettl3", "Ythdc1", "unc-13", "Hsp70Ab")

# Add a column for highlighting specific genes
res_S2_S_DC1_KO$highlight <- ifelse(res_S2_S_DC1_KO$current_symbol %in% highlight_genes, "Highlighted", res_S2_S_DC1_KO$diffexpressed)

# Count numbers for legend
#count_highlighted <- sum(res_S2_S_DC1_KO_clean$highlight == "Highlighted")
count_UP <- sum(res_S2_S_DC1_KO$diffexpressed == "UP")
count_DOWN <- sum(res_S2_S_DC1_KO$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2_S_DC1_KO$diffexpressed == "Not Significant")

# Create custom colors for highlighting "Highlighted" = "yellow",
custom_colors <- c("Highlighted" = "yellow", "UP" = "red", "DOWN" = "blue", "Not Significant" = "grey")

# Create a data frame for labeling highlight_genes
highlight_label_genes <- res_S2_S_DC1_KO %>%
  dplyr::filter(current_symbol %in% highlight_genes)

# Filter top significant genes for labeling
top_genes <- res_S2_S_DC1_KO %>%
  dplyr::filter(diffexpressed != "Not Significant") %>%
  arrange(padj) %>%
  slice(1:50)

# Combine highlight_genes and top_genes for labeling
label_genes <- dplyr::bind_rows(top_genes, highlight_label_genes) %>%
  dplyr::distinct(current_symbol, .keep_all = TRUE)

# Create the MA plot
ggplot(res_S2_S_DC1_KO, aes(x = log2(baseMean + 1), y = log2FoldChange, color = highlight)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -0.5, col = "black", linetype = 'dashed') + 
  geom_text_repel(data = label_genes, aes(label = current_symbol), size = 3, max.overlaps = Inf, color = "black") +
  scale_color_manual(
    values = custom_colors,
    # Only include these groups in the legend (order determines the legend order)
    breaks = c("DOWN", "Not Significant", "UP"),
    labels = c(
      paste("Downregulated (", count_DOWN, ")", sep = ""),
      paste("Not Significant (", count_Not_Significant, ")", sep = ""),
      paste("Upregulated (", count_UP, ")", sep = ""))) + xlim(0, 20) + ylim(c(-10, 10)) +
  labs(title = " ", x = "log2(baseMean)", y = "Log2 Fold Change", color = " ") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle('Ythdc1-KO/S2S: padj <= 0.05 & Log2FC |0.5|')


```



```{r}
# Assuming your data frame has a column called "gene_name" containing gene symbols
if ("current_symbol" %in% colnames(res_S2_S_DC1_KO)) {
  
  # Make gene names unique if needed
  res_S2_S_DC1_KO$current_symbol <- make.unique(as.character(res_S2_S_DC1_KO$current_symbol))
  
  # Set gene names as rownames (optional)
  rownames(res_S2_S_DC1_KO) <- res_S2_S_DC1_KO$current_symbol
  
  # Map ENTREZ IDs
  res_S2_S_DC1_KO$ENTREZID <- mapIds(org.Dm.eg.db,
                               keys = res_S2_S_DC1_KO$current_symbol,
                               column = "ENTREZID",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO IDs
  res_S2_S_DC1_KO$GO <- mapIds(org.Dm.eg.db,
                               keys = res_S2_S_DC1_KO$current_symbol,
                               column = "GO",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO term descriptions
  res_S2_S_DC1_KO$GODESCR <- sapply(res_S2_S_DC1_KO$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  # Map GO ontology (BP, MF, CC)
  res_S2_S_DC1_KO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = res_S2_S_DC1_KO$current_symbol,
                               column = "ONTOLOGY",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Optional: Map gene descriptions
  res_S2_S_DC1_KO$GENENAME <- mapIds(org.Dm.eg.db,
                               keys = res_S2_S_DC1_KO$current_symbol,
                               column = "GENENAME",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
} else {
  cat("The column 'current_symbol' does not exist in the data frame.\n")
}



#write.table(res_S2_S_DC1_KO_clean, file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/2025_0414_res_S2_S_DC1_KO_clean.txt", sep = "\t", quote = FALSE)

```




    Gene Ontology analyses Ythdc1 KO relative to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}

# Load the packages
library(clusterProfiler)
#library(org.Dm.eg.db)
library(DOSE)
#library(dplyr)
# Assuming `res_S2_S_DC1_KO.df` contains your differential expression results

# Filter for upregulated and downregulated genes
upregulated_genes_S2_S_DC1_KO <- res_S2_S_DC1_KO %>%
  dplyr::filter(log2FoldChange > 0.5 & padj <= 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

downregulated_genes_S2_S_DC1_KO <- res_S2_S_DC1_KO %>%
  dplyr::filter(log2FoldChange < -0.5 & padj <= 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

# Convert gene symbols to Entrez IDs
upregulated_entrez <- bitr(upregulated_genes_S2_S_DC1_KO, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)
downregulated_entrez <- bitr(downregulated_genes_S2_S_DC1_KO, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)


# GO analysis for upregulated genes
go_up <- enrichGO(gene = upregulated_entrez$ENTREZID,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez$ENTREZID,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes Ythdc-KO_S2S_DGE_GO_up_ALL_20
barplot(go_up, showCategory = 25, 
        title = "Ythdc-KO_S2S_DGE:GO Enrichment for Upregulated")

# Bar plot for downregulated genes Ythdc-KO_S2S_DGE_GO_down_ALL_20
barplot(go_down, showCategory = 25, 
        title = "Ythdc-KO_S2S_DGE:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 25, 
        title = "Ythdc-KO_S2S_DGE:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 25, 
        title = "Ythdc-KO_S2S_DGE:GO Enrichment for Downregulated Genes")

# If you want to see the results in a table
go_up_results <- as.data.frame(go_up)
go_down_results <- as.data.frame(go_down)

# View the top results
head(go_up_results)
head(go_down_results)

```



making Ridge plot
```{r}

library(clusterProfiler)
library(org.Dm.eg.db)
organism <- org.Dm.eg.db

keytypes(organism)

sig_S2_S_DC1_KO_clean <- res_S2_S_DC1_KO %>%
  dplyr::filter(diffexpressed != "Not Significant")


gene_scores <- sig_S2_S_DC1_KO_clean$log2FoldChange
names(gene_scores) <- sig_S2_S_DC1_KO_clean$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    nPermSimple = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

#head(gse@result)


```



##Dotplot
```{r echo=TRUE, fig.width=15, fig.height=8}
#require(DOSE)
dotplot(gse, showCategory=20, split=".sign") + facet_grid(.~.sign)
```


##Category Netplot
The cnetplot depicts the linkages of genes and biological concepts (e.g. GO terms or KEGG pathways) as a network (helpful to see which genes are involved in enriched pathways and genes that may belong to multiple annotation categories).
```{r fig.width=18}
#install.packages("ggnewscale")MA_Ythdc1-KO_S2S
gene_scores <- sig_S2_S_DC1_KO_clean$log2FoldChange
names(gene_scores) <- sig_S2_S_DC1_KO_clean$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 
cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 30) + ggtitle("Ythdc1-KO/S2S")
```







### ***4*** Differential expressed genes in S2 Ythdc1-GoF relative to S2 GFP. A total of 12268 genes were implicated in this comparison. 4573 of these genes were up-regulated while 3878 of them were down-regulated. 

The scatter plot below shows genes that were over-expressed in S2 YTHDC1-GoF relative to S2 GFP. GstD5, CadN, unc-13, Ythdc1 are differentially expressed in S2 Ythdc1-GoF relative to S2 GFP


```{r message=FALSE, warning=FALSE, echo=FALSE}
#res_S2.GFP_S2.DC1.GOF <- results(dds_S2.GFP_S2.DC1.GOF)

#write.table(res_S2.GFP_S2.DC1.GOF, file = "res_S2.GFP_S2.DC1.GOF.txt", sep = "\t", quote = F)
res_S2.GFP_S2.DC1.GOF = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_S2.GFP_S2.DC1.GOF.txt", header = T)

res_S2.GFP_S2.DC1.GOF$gene_name = rownames(res_S2.GFP_S2.DC1.GOF)
res_S2.GFP_S2.DC1.GOF = left_join(res_S2.GFP_S2.DC1.GOF, gene_validation, by = c("gene_name" = "submitted_item"))


res_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF %>%
  distinct(gene_name, .keep_all = TRUE)

res_S2.GFP_S2.DC1.GOF_clean <- res_S2.GFP_S2.DC1.GOF[res_S2.GFP_S2.DC1.GOF$baseMean>0 & !is.na(res_S2.GFP_S2.DC1.GOF$padj), ]


res_S2.GFP_S2.DC1.GOF$delabel <- ifelse(res_S2.GFP_S2.DC1.GOF$current_symbol %in% head(res_S2.GFP_S2.DC1.GOF[order(res_S2.GFP_S2.DC1.GOF$padj), "current_symbol"], 500), res_S2.GFP_S2.DC1.GOF$current_symbol, NA)


res_S2.GFP_S2.DC1.GOF$diffexpressed <- "Not Significant"
res_S2.GFP_S2.DC1.GOF$diffexpressed[res_S2.GFP_S2.DC1.GOF$log2FoldChange > 0.5 & res_S2.GFP_S2.DC1.GOF$padj < 0.05] <- "UP"
res_S2.GFP_S2.DC1.GOF$diffexpressed[res_S2.GFP_S2.DC1.GOF$log2FoldChange < -0.5 & res_S2.GFP_S2.DC1.GOF$padj < 0.05] <- "DOWN"
res_S2.GFP_S2.DC1.GOF$diffexpressed[!(res_S2.GFP_S2.DC1.GOF$log2FoldChange > 0.5 & res_S2.GFP_S2.DC1.GOF$padj < 0.05) & !(res_S2.GFP_S2.DC1.GOF$log2FoldChange < -0.5 & res_S2.GFP_S2.DC1.GOF$padj < 0.05)] <- "Not Significant"

count_UP <- sum(res_S2.GFP_S2.DC1.GOF$diffexpressed == "UP")
count_DOWN <- sum(res_S2.GFP_S2.DC1.GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2.GFP_S2.DC1.GOF$diffexpressed == "Not Significant")


significant_genes <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")


ggplot(res_S2.GFP_S2.DC1.GOF, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed, label = delabel)) +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') + 
  geom_vline(xintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_vline(xintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_point(size = 2) + 
  scale_color_manual(values = c("blue", "grey", "red"), 
                     labels = c(paste("Downregulated (", count_DOWN, ")", sep=""), 
                                paste("Not significant (", count_Not_Significant, ")", sep=""),
                                paste("Upregulated (", count_UP, ")", sep=""))) +
  coord_cartesian(ylim = c(0, 300), xlim = c(-10, 10)) +
  labs(color = ' ', x = expression("log"[2]*"FC"), y = expression("-log"[10]*"padj")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  ggtitle('DEG in Ythdc1-GoF vs S2 GFP using padj <= 0.05 and Log2FC |0.5|') + 
  geom_text_repel(data = significant_genes, aes(label = delabel), max.overlaps = 10)  

#res_S2.GFP_S2.DC1.GOF_clean = res_S2.GFP_S2.DC1.GOF_clean %>% dplyr::select(-symbol) S2.GFP_S2.DC1.GOF
#res_S2.GFP_S2.DC1.GOF_clean <- res_S2.GFP_S2.DC1.GOF[res_S2.GFP_S2.DC1.GOF$baseMean > 0, ]

```

MA-plot: shows differential expressed genes in S2 Ythdc1-GoF relative to S2 GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}

highlight_genes <- c("fl(2)d", "aqz", "sky", "pum", "prosap", "Mettl3", "Ythdc1", "unc-13", "Hsp70Ab")

# Add a column for highlighting specific genes
res_S2.GFP_S2.DC1.GOF$highlight <- ifelse(res_S2.GFP_S2.DC1.GOF$current_symbol %in% highlight_genes, "Highlighted", res_S2.GFP_S2.DC1.GOF$diffexpressed)

# Count numbers for legend
count_highlighted <- sum(res_S2.GFP_S2.DC1.GOF$highlight == "Highlighted")
count_UP <- sum(res_S2.GFP_S2.DC1.GOF$diffexpressed == "UP")
count_DOWN <- sum(res_S2.GFP_S2.DC1.GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2.GFP_S2.DC1.GOF$diffexpressed == "Not Significant")

# Create custom colors for highlighting
custom_colors <- c("Highlighted" = "yellow", "UP" = "red", "DOWN" = "blue", "Not Significant" = "grey")

# Create a data frame for labeling highlight_genes
highlight_label_genes <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(current_symbol %in% highlight_genes)

# Filter top significant genes for labeling
top_genes <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(diffexpressed != "Not Significant") %>%
  arrange(padj) %>%
  slice(1:50)

# Combine highlight_genes and top_genes for labeling Volc_S2_DC1_GOF_S2_GFP
label_genes <- dplyr::bind_rows(top_genes, highlight_label_genes) %>%
  dplyr::distinct(current_symbol, .keep_all = TRUE)

# Create the MA plot
ggplot(res_S2.GFP_S2.DC1.GOF, aes(x = log2(baseMean + 1), y = log2FoldChange, color = highlight)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_text_repel(data = label_genes, aes(label = current_symbol), size = 3, max.overlaps = Inf, color = "black") +
  scale_color_manual(
    values = custom_colors,
    breaks = c("DOWN", "Not Significant", "UP"),
    labels = c(
      paste("Downregulated (", count_DOWN, ")", sep = ""),
      paste("Not Significant (", count_Not_Significant, ")", sep = ""),
      paste("Upregulated (", count_UP, ")", sep = ""))) +
  xlim(0, 20) + ylim(c(-10, 10)) +
  labs(title = " ",
       x = "log2(baseMean)",
       y = "Log2 Fold Change",
       color = " ") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle('S2S-Ythdc1-GoF/S2-GFP: padj <= 0.05 & Log2FC |0.5|')

```


```{r}
# Assuming your data frame has a column called "gene_name" containing gene symbols
if ("current_symbol" %in% colnames(res_S2.GFP_S2.DC1.GOF)) {
  
  # Make gene names unique if needed
  res_S2.GFP_S2.DC1.GOF$current_symbol <- make.unique(as.character(res_S2.GFP_S2.DC1.GOF$current_symbol))
  
  # Set gene names as rownames (optional)
  rownames(res_S2.GFP_S2.DC1.GOF) <- res_S2.GFP_S2.DC1.GOF$current_symbol
  
  # Map ENTREZ IDs
  res_S2.GFP_S2.DC1.GOF$ENTREZID <- mapIds(org.Dm.eg.db,
                               keys = res_S2.GFP_S2.DC1.GOF$current_symbol,
                               column = "ENTREZID",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO IDs
  res_S2.GFP_S2.DC1.GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = res_S2.GFP_S2.DC1.GOF$current_symbol,
                               column = "GO",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO term descriptions
  res_S2.GFP_S2.DC1.GOF$GODESCR <- sapply(res_S2.GFP_S2.DC1.GOF$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  # Map GO ontology (BP, MF, CC)
  res_S2.GFP_S2.DC1.GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = res_S2.GFP_S2.DC1.GOF$current_symbol,
                               column = "ONTOLOGY",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Optional: Map gene descriptions
  res_S2.GFP_S2.DC1.GOF$GENENAME <- mapIds(org.Dm.eg.db,
                               keys = res_S2.GFP_S2.DC1.GOF$current_symbol,
                               column = "GENENAME",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
} else {
  cat("The column 'current_symbol' does not exist in the data frame.\n")
}


#write.table(res_S2.GFP_S2.DC1.GOF_clean, file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_S2.GFP_S2.DC1.GOF_clean.txt", sep = "\t", quote = FALSE)


```

    Gene Ontology analyses S2 Ythdc1-GoF relative to S2 GFP
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter for upregulated and downregulated genes
upregulated_genes_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(log2FoldChange > 0.5 & padj < 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

downregulated_genes_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(log2FoldChange < -0.5 & padj < 0.05) %>%
  dplyr::select(current_symbol) %>%
  unlist()

# Convert gene symbols to Entrez IDs
upregulated_entrez <- bitr(upregulated_genes_S2.GFP_S2.DC1.GOF, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)
downregulated_entrez <- bitr(downregulated_genes_S2.GFP_S2.DC1.GOF, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)


# GO analysis for upregulated genes YTHDC1-GoF_S2GFP
go_up <- enrichGO(gene = upregulated_entrez$ENTREZID,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez$ENTREZID,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  # You can also specify "BP", "MF", or "CC"
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes
barplot(go_up, showCategory = 20, 
        title = "YTHDC1-GoF/S2-GFP:GO Enrichment for Upregulated Genes")

# Bar plot for downregulated genes
barplot(go_down, showCategory = 20, 
        title = "YTHDC1-GoF/S2-GFP:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 20, 
        title = "YTHDC1-GoF/S2-GFP:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 20, 
        title = "YTHDC1-GoF/S2-GFP:GO Enrichment for Downregulated Genes")

# If you want to see the results in a table
go_up_results <- as.data.frame(go_up)
go_down_results <- as.data.frame(go_down)

# View the top results
head(go_up_results)
head(go_down_results)

```


making Ridge plot Ythdc1-GoF vs S2 GFP
```{r}

#library(clusterProfiler)
#library(org.Dm.eg.db)
#organism <- org.Dm.eg.db


#keytypes(organism)


sig_S2.GFP_S2.DC1.GOF_clean <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")


gene_scores <- sig_S2.GFP_S2.DC1.GOF_clean$log2FoldChange
names(gene_scores) <- sig_S2.GFP_S2.DC1.GOF_clean$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    nPermSimple = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

head(gse@result)


```

## Ridgeplot
Helpful to interpret up/down-regulated pathways.
```{r fig.width=18, fig.height=12}

#ridgeplot(gse) + labs(x = "Enrichment distribution") + ggtitle("YTHDC1-GoF_S2-GFP")
cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 50) + ggtitle("S2-YTHDC1-GoF/S2-GFP")
```




### ***5*** Differential expressed genes in METTL3-KO GFP relative to METTL3-KO_Ythdc1-GoF. A total of 12348 genes were differentaially expressed. 4345 were up-regulated and 4154 were down-regulated.

```{r message=FALSE, warning=FALSE, echo=FALSE}
#res_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- results(dds_mettl3.KO.GFP_mettl3.KO.DC1.GOF)

#write.table(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, file = "res_mettl3_KO_GFP_mettl3_KO_DC1_GOF.txt", sep = "\t", quote = F)

res_mettl3_KO_GFP_mettl3_KO_DC1_GOF = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_mettl3_KO_GFP_mettl3_KO_DC1_GOF.txt", header = T)
res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$gene_name = rownames(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
res_mettl3_KO_GFP_mettl3_KO_DC1_GOF = left_join(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, gene_validation, by = c("gene_name" = "submitted_item"))


res_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  distinct(gene_name, .keep_all = TRUE)

#res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF[res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$baseMean>0 & !is.na(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$padj), ]


res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$delabel <- ifelse(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol %in% head(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF[order(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$padj), "current_symbol"], 500), res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol, NA)

# Calculate counts for each category
res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed <- "Not Significant"
res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed[res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$log2FoldChange > 0.5  & res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$padj < 0.05] <- "UP"
res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed[res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$log2FoldChange < -0.5 & res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$padj < 0.05] <- "DOWN"
res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed[!(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$log2FoldChange > 0.5 & res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$padj < 0.05) & !(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$log2FoldChange < -0.5 & res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$padj < 0.05)] <- "Not Significant"

count_UP <- sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed == "UP")
count_DOWN <- sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed == "Not Significant")

# Filter for significant genes to label (UP and DOWN regulated)
significant_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")

# Plot with updated legend labels and adjusted p-values
ggplot(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed, label = delabel)) +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') + 
  geom_vline(xintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_vline(xintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_point(size = 2) + 
  scale_color_manual(values = c("blue", "grey", "red"), 
                     labels = c(paste("Downregulated (", count_DOWN, ")", sep=""), 
                                paste("Not significant (", count_Not_Significant, ")", sep=""),
                                paste("Upregulated (", count_UP, ")", sep=""))) +
  coord_cartesian(ylim = c(0, 300), xlim = c(-10, 10)) +
  labs(color = ' ', x = expression("log"[2]*"FC"), y = expression("-log"[10]*"p-value")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  ggtitle('METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP: padj <= 0.05 and Log2FC |0.5|') + 
  geom_text_repel(data = significant_genes, aes(label = delabel), max.overlaps = 10)

#res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean = res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean %>% dplyr::select(-symbol) Volc_mettl3KO_DC1_GOF_mettl3_KO_GFP


```

MA-plot: shows differential expressed genes in METTL3-KO Ythdc1-GoF relative to METTL3-KO GFP.

```{r message=FALSE, warning=FALSE, echo=FALSE}
highlight_genes <- c("fl(2)d", "aqz", "sky", "pum", "prosap", "Mettl3", "Ythdc1", "unc-13", "Hsp70Ab")

# Add a column for highlighting specific genes
res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$highlight <- ifelse(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol %in% highlight_genes, "Highlighted", res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed)

# Count numbers for legend
count_highlighted <- sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$highlight == "Highlighted")
count_UP <- sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed == "UP")
count_DOWN <- sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$diffexpressed == "Not Significant")

# Create custom colors for highlighting
custom_colors <- c("Highlighted" = "yellow", "UP" = "red", "DOWN" = "blue", "Not Significant" = "grey")

# Create a data frame for labeling highlight_genes
highlight_label_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(current_symbol %in% highlight_genes)

# Filter top significant genes for labeling
top_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed != "Not Significant") %>%
  arrange(padj) %>%
  slice(1:50)

# Combine highlight_genes and top_genes for labeling
label_genes <- dplyr::bind_rows(top_genes, highlight_label_genes) %>%
  dplyr::distinct(current_symbol, .keep_all = TRUE)

# Create the MA plot
ggplot(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, aes(x = log2(baseMean + 1), y = log2FoldChange, color = highlight)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_text_repel(data = label_genes, aes(label = current_symbol), size = 3, max.overlaps = Inf, color = "black") +
  scale_color_manual(
    values = custom_colors,
    # Only include these groups in the legend (order determines the legend order)
    breaks = c("DOWN", "Not Significant", "UP"),
    labels = c(
      paste("Downregulated (", count_DOWN, ")", sep = ""),
      paste("Not Significant (", count_Not_Significant, ")", sep = ""),
      paste("Upregulated (", count_UP, ")", sep = ""))) +
  xlim(0, 20) + ylim(c(-10, 10)) +
  labs(title = " ",
       x = "log2(baseMean)",
       y = "Log2 Fold Change",
       color = " ") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle('METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP: padj <= 0.05 & Log2FC |0.5|')



```


```{r}
# Assuming your data frame has a column called "gene_name" containing gene symbols
if ("current_symbol" %in% colnames(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF)) {
  
  # Make gene names unique if needed
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol <- make.unique(as.character(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol))
  
  # Set gene names as rownames (optional)
  rownames(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF) <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol
  
  # Map ENTREZ IDs
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$ENTREZID <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "ENTREZID",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO IDs
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "GO",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO term descriptions
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  # Map GO ontology (BP, MF, CC)
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "ONTOLOGY",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Optional: Map gene descriptions
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GENENAME <- mapIds(org.Dm.eg.db,
                               keys = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol,
                               column = "GENENAME",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
} else {
  cat("The column 'current_symbol' does not exist in the data frame.\n")
}


#write.table(res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean, file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean.txt", sep = "\t", quote = FALSE)
```


making Ridge plot
```{r}

#library(clusterProfiler)
#library(org.Dm.eg.db)
#organism <- org.Dm.eg.db


#keytypes(organism)

sig_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")

gene_scores <- sig_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean$log2FoldChange
names(gene_scores) <- sig_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean$current_symbol 

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    nPermSimple = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

#head(gse@result) cnet_METTL3_KO_YTHDC1_GoF_METTL3_KO_GFP


```

## Ridgeplot
Helpful to interpret up/down-regulated pathways. cnet_METTL3-KO-YTHDC1-GoF_METTL3-KO-GFP
```{r fig.width=18, fig.height=12}

#ridgeplot(gse) + labs(x = "Enrichment distribution") + ggtitle("Gene Set Enrichment for METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP")
cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 30) + ggtitle("Gene Set Enrichment for METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP")
```

    Gene Ontology analyses METTL3-KO YTHDC1-GoF relative to METTL3-KO
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter for upregulated and downregulated genes
upregulated_entrez <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed == "UP") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

downregulated_entrez <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(diffexpressed == "DOWN") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

# GO analysis for upregulated genes
go_up <- enrichGO(gene = upregulated_entrez,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes METTL3-KO-YTHDC1-GoF_METTL3-KO-GFP_GO_UP
barplot(go_up, showCategory = 20, 
        title = "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP:GO Enrichment for Upregulated Genes")

# Bar plot for downregulated genes
barplot(go_down, showCategory = 20, title = "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 20, 
        title = "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 20, title = "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP:GO Enrichment for Downregulated Genes")


```

    # dds_S2.YTHDC_mettl3.KO.DC1.GOF S2.YTHDC-GOF LK4_IGO_13135_CJ_4_S57 LK5_IGO_13135_CJ_5_S58 LK6_IGO_13135_CJ_6_S59 mettl3.KO.DC1.GOF LK10_IGO_13135_CJ_10_S48 LK11_IGO_13135_CJ_11_S49 LK12_IGO_13135_CJ_12_S50


# 3A Volcano shows differential expressed genes in S2S-YTHDC1-GoF/METTL3-KO Ythdc1-GoF.
```{r message=FALSE, warning=FALSE, echo=FALSE}
#res_S2.YTHDC_gof_mettl3.KO.DC1.GOF = results(dds_S2.YTHDC_mettl3.KO.DC1.GOF)
#write.table(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF, file = "res_S2.YTHDC_gof_mettl3.KO.DC1.GOF.txt", sep = "\t", quote = F)

res_S2.YTHDC_gof_mettl3.KO.DC1.GOF = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_S2.YTHDC_gof_mettl3.KO.DC1.GOF.txt", header = T)
res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$gene_name = rownames(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF)
res_S2.YTHDC_gof_mettl3.KO.DC1.GOF = left_join(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF, gene_validation, by = c("gene_name" = "submitted_item"))

res_S2.YTHDC_gof_mettl3.KO.DC1.GOF <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  distinct(gene_name, .keep_all = TRUE)
#res_S2.YTHDC_gof_mettl3.KO.DC1.GOF <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF[res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$baseMean>0 & !is.na(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$padj), ]


res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$delabel <- ifelse(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol %in% head(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF[order(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$padj), "current_symbol"], 500), res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol, NA)

# Calculate counts for each category
res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed <- "Not Significant"
res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed[res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$log2FoldChange > 0.5 & res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$padj < 0.05] <- "UP"
res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed[res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$log2FoldChange < -0.5 & res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$padj < 0.05] <- "DOWN"
res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed[!(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$log2FoldChange > 0.5 & res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$padj < 0.05) & !(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$log2FoldChange < -0.5 & res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$padj < 0.05)] <- "Not Significant"

count_UP <- sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed == "UP")
count_DOWN <- sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed == "Not Significant")

# Filter for significant genes to label (UP and DOWN regulated)
significant_genes <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")

# Plot with updated legend labels and adjusted p-values
ggplot(data = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed, label = delabel)) +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') + 
  geom_vline(xintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_vline(xintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_point(size = 2) + 
  scale_color_manual(values = c("blue", "grey", "red"), 
                     labels = c(paste("Downregulated (", count_DOWN, ")", sep=""), 
                                paste("Not significant (", count_Not_Significant, ")", sep=""),
                                paste("Upregulated (", count_UP, ")", sep=""))) +
  coord_cartesian(ylim = c(0, 300), xlim = c(-10, 10)) +
  labs(color = ' ', x = expression("log"[2]*"FC"), y = expression("-log"[10]*"p-value")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  ggtitle('DEG:S2S-YTHDC1-GoF/METTL3-KO-GFP padj <= 0.05 and Log2FC |0.5|') + 
  geom_text_repel(data = significant_genes, aes(label = delabel), max.overlaps = 10)


#res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean %>% dplyr::select(-symbol) Volc_S2_YTHDC_GoF_Mettl3_KO_DC1_GoF


```

MA-plot: shows differential expressed genes in METTL3-KO Ythdc1-GoF relative to METTL3-KO GFP.

```{r message=FALSE, warning=FALSE, echo=FALSE}

highlight_genes <- c("fl(2)d", "aqz", "sky", "pum", "prosap", "Mettl3", "Ythdc1", "unc-13", "Hsp70Ab")

# Add a column for highlighting specific genes
res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$highlight <- ifelse(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol %in% highlight_genes, "Highlighted", res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed)

# Count numbers for legend
count_highlighted <- sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$highlight == "Highlighted")
count_UP <- sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed == "UP")
count_DOWN <- sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$diffexpressed == "Not Significant")

# Create custom colors for highlighting
custom_colors <- c("Highlighted" = "yellow", "UP" = "red", "DOWN" = "blue", "Not Significant" = "grey")

# Create a data frame for labeling highlight_genes
highlight_label_genes <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(current_symbol %in% highlight_genes)

# Filter top significant genes for labeling
top_genes <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(diffexpressed != "Not Significant") %>%
  arrange(padj) %>%
  slice(1:50)

# Combine highlight_genes and top_genes for labeling
label_genes <- dplyr::bind_rows(top_genes, highlight_label_genes) %>%
  dplyr::distinct(current_symbol, .keep_all = TRUE)

# Create the MA plot
ggplot(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF, aes(x = log2(baseMean + 1), y = log2FoldChange, color = highlight)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_text_repel(data = label_genes, aes(label = current_symbol), size = 3, max.overlaps = Inf, color = "black") +
  scale_color_manual(
    values = custom_colors,
    # Only include these groups in the legend (order determines the legend order)MA_S2S_Ythdc1-GoF_S2S_GFP
    breaks = c("DOWN", "Not Significant", "UP"),
    labels = c(
      paste("Downregulated (", count_DOWN, ")", sep = ""),
      paste("Not Significant (", count_Not_Significant, ")", sep = ""),
      paste("Upregulated (", count_UP, ")", sep = ""))) +
  xlim(0, 20) + ylim(c(-10, 10)) +
  labs(title = " ",
       x = "log2(baseMean)",
       y = "Log2 Fold Change",
       color = " ") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle('S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF: padj <= 0.05 and Log2FC |0.5|')



```



```{r}
# Assuming your data frame has a column called "gene_name" containing gene symbols
if ("current_symbol" %in% colnames(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF)) {
  
  # Make gene names unique if needed
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol <- make.unique(as.character(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol))
  
  # Set gene names as rownames (optional)
  rownames(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF) <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol
  
  # Map ENTREZ IDs
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$ENTREZID <- mapIds(org.Dm.eg.db,
                               keys = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol,
                               column = "ENTREZID",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO IDs
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol,
                               column = "GO",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO term descriptions
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$GODESCR <- sapply(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  # Map GO ontology (BP, MF, CC)
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol,
                               column = "ONTOLOGY",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Optional: Map gene descriptions
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$GENENAME <- mapIds(org.Dm.eg.db,
                               keys = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$current_symbol,
                               column = "GENENAME",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
} else {
  cat("The column 'current_symbol' does not exist in the data frame.\n")
}


#write.table(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean, file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean.txt", sep = "\t", quote = FALSE)
```




making Ridge plot
```{r}

library(clusterProfiler)
library(org.Dm.eg.db)
organism <- org.Dm.eg.db


#keytypes(organism)

sig_res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")

gene_scores <- sig_res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$log2FoldChange
names(gene_scores) <- sig_res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$current_symbol 

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    nPermSimple = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

#head(gse@result)


```

## Ridgeplot
Helpful to interpret up/down-regulated pathways.
```{r fig.width=18, fig.height=12}

#ridgeplot(gse) + labs(x = "Enrichment distribution") + ggtitle("Gene Set Enrichment for S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF")
cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 30) + ggtitle("Gene Set Enrichment for S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF")
```

    Gene Ontology analyses METTL3-KO YTHDC1-GoF relative to METTL3-KO
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter for upregulated and downregulated genes
upregulated_entrez <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(diffexpressed == "UP") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

downregulated_entrez <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(diffexpressed == "DOWN") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

# GO analysis for upregulated genes
go_up <- enrichGO(gene = upregulated_entrez,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes S2-YTHDC1-GoF_METTL3-KO-YTHDC1-GoF_GO_UP
barplot(go_up, showCategory = 20, 
        title = "S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF:GO Enrichment for Upregulated Genes")

# Bar plot for downregulated genes
barplot(go_down, showCategory = 20, title = "S2-YTHDC1-GoF vs METTL3-KO YTHDC1-GoF:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 20, 
        title = "S2-YTHDC1-GoF vs METTL3-KO YTHDC1-GoF:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 20, title = "S2-YTHDC1-GoF vs METTL3-KO YTHDC1-GoF:GO Enrichment for Downregulated Genes")


```

### **6** Differential expressed genes in METTL3-KO relative to S2-S (from first data sets)
Selected the reads counts from the Salmon output file for all samples and rounded each read counts.


```{r message=FALSE, warning=FALSE, echo=FALSE}
#res_S2_m6A = results(dds_S2_S_mettl3_KO)

#write.table(res_S2_m6A, file = "res_METTL3_KO_S2.txt", sep = "\t", quote = F)

res_S2_m6A = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/res_METTL3_KO_S2.txt", header = T)
res_S2_m6A$gene_name = rownames(res_S2_m6A)
res_S2_m6A = full_join(res_S2_m6A, gene_validation, by = c("gene_name" = "submitted_item"))


res_S2_m6A <- res_S2_m6A %>%
  distinct(gene_name, .keep_all = TRUE)
#res_S2_m6A.df <- read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/quant/S2-S_vs_mettl3_KO.txt", header = TRUE)


#excluded_genes <- c("His1:CG31617", "His2B:CG17949", "Hsp70Ba")  & !is.na(res_S2_m6A.df$padj)
#res_S2_m6A.df$delabel <- ifelse(res_S2_m6A.df$current_symbol %in% head(res_S2_m6A.df[order(res_S2_m6A.df$padj), "current_symbol"], 500), res_S2_m6A.df$current_symbol, NA)

#res_S2_m6A_clean = res_S2_m6A.df[res_S2_m6A.df$baseMean>0 & !is.na(res_S2_m6A.df$padj),]

res_S2_m6A$delabel <- ifelse(res_S2_m6A$current_symbol %in% head(res_S2_m6A[order(res_S2_m6A$padj), "current_symbol"], 500), res_S2_m6A$current_symbol, NA)

# Calculate counts for each category
res_S2_m6A$diffexpressed <- "Not Significant"
res_S2_m6A$diffexpressed[res_S2_m6A$log2FoldChange > 0.5  & res_S2_m6A$padj < 0.05] <- "UP"
res_S2_m6A$diffexpressed[res_S2_m6A$log2FoldChange < -0.5 & res_S2_m6A$padj < 0.05] <- "DOWN"
res_S2_m6A$diffexpressed[!(res_S2_m6A$log2FoldChange > 0.5 & res_S2_m6A$padj < 0.05) & !(res_S2_m6A$log2FoldChange < -0.5 & res_S2_m6A$padj < 0.05)] <- "Not Significant"

count_UP <- sum(res_S2_m6A$diffexpressed == "UP")
count_DOWN <- sum(res_S2_m6A$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2_m6A$diffexpressed == "Not Significant")

# Filter for significant genes to label (UP and DOWN regulated)
significant_genes <- res_S2_m6A %>%
  dplyr::filter(diffexpressed != "Not Significant")

# Plot with updated legend labels and adjusted p-values
ggplot(res_S2_m6A, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed, label = delabel)) +
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = 'dashed') + 
  geom_vline(xintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_vline(xintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_point(size = 2) + 
  scale_color_manual(values = c("blue", "grey", "red"), 
                     labels = c(paste("Downregulated (", count_DOWN, ")", sep=""), 
                                paste("Not significant (", count_Not_Significant, ")", sep=""),
                                paste("Upregulated (", count_UP, ")", sep=""))) +
  coord_cartesian(ylim = c(0, 300), xlim = c(-10, 10)) +
  labs(color = ' ', x = expression("log"[2]*"FC"), y = expression("-log"[10]*"p-value")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  ggtitle('METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP: padj <= 0.05 and Log2FC |0.5|') + 
  geom_text_repel(data = significant_genes, aes(label = delabel), max.overlaps = 10)


#res_S2_m6A_clean = res_S2_m6A_clean %>% dplyr::select(-symbol) Volc_Mettl3_KO_S2S






```


    MA-plot: shows differential expressed genes in METTL3-KO relative to S2-S.

```{r message=FALSE, warning=FALSE, echo=FALSE}

res_S2_m6A$highlight <- ifelse(res_S2_m6A$current_symbol %in% highlight_genes, "Highlighted", res_S2_m6A$diffexpressed)

# Count numbers for legend
count_highlighted <- sum(res_S2_m6A$highlight == "Highlighted")
count_UP <- sum(res_S2_m6A$diffexpressed == "UP")
count_DOWN <- sum(res_S2_m6A$diffexpressed == "DOWN")
count_Not_Significant <- sum(res_S2_m6A$diffexpressed == "Not Significant")

# Create custom colors for highlighting
custom_colors <- c("Highlighted" = "yellow", "UP" = "red", "DOWN" = "blue", "Not Significant" = "grey")

# Create a data frame for labeling highlight_genes
highlight_label_genes <- res_S2_m6A %>%
  dplyr::filter(current_symbol %in% highlight_genes)

# Filter top significant genes for labeling
top_genes <- res_S2_m6A %>%
  dplyr::filter(diffexpressed != "Not Significant") %>%
  arrange(padj) %>%
  slice(1:50)

# Combine highlight_genes and top_genes for labeling
label_genes <- dplyr::bind_rows(top_genes, highlight_label_genes) %>%
  dplyr::distinct(current_symbol, .keep_all = TRUE)

# Create the MA plot
ggplot(res_S2_m6A, aes(x = log2(baseMean + 1), y = log2FoldChange, color = highlight)) +
  geom_point(alpha = 0.6, size = 2) +
  geom_hline(yintercept = 0.5, col = "black", linetype = 'dashed') +
  geom_hline(yintercept = -0.5, col = "black", linetype = 'dashed') +
  geom_text_repel(data = label_genes, aes(label = current_symbol), size = 3, max.overlaps = Inf, color = "black") +
  scale_color_manual(
    values = custom_colors,
    # Only include these groups in the legend (order determines the legend order)
    breaks = c("DOWN", "Not Significant", "UP"),
    labels = c(
      paste("Downregulated (", count_DOWN, ")", sep = ""),
      paste("Not Significant (", count_Not_Significant, ")", sep = ""),
      paste("Upregulated (", count_UP, ")", sep = ""))) +
  xlim(0, 20) + ylim(c(-10, 10)) +
  labs(title = " ",
       x = "log2(baseMean)",
       y = "Log2 Fold Change",
       color = " ") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggtitle('Mettl3-KO/S2S: padj <= 0.05 & Log2FC |0.5|')
```


```{r}
# Assuming your data frame has a column called "gene_name" containing gene symbols
if ("current_symbol" %in% colnames(res_S2_m6A)) {
  
  # Make gene names unique if needed
  res_S2_m6A$current_symbol <- make.unique(as.character(res_S2_m6A$current_symbol))
  
  # Set gene names as rownames (optional)
  rownames(res_S2_m6A) <- res_S2_m6A$current_symbol
  
  # Map ENTREZ IDs
  res_S2_m6A$ENTREZID <- mapIds(org.Dm.eg.db,
                               keys = res_S2_m6A$current_symbol,
                               column = "ENTREZID",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO IDs
  res_S2_m6A$GO <- mapIds(org.Dm.eg.db,
                               keys = res_S2_m6A$current_symbol,
                               column = "GO",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Map GO term descriptions
  res_S2_m6A$GODESCR <- sapply(res_S2_m6A$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  # Map GO ontology (BP, MF, CC)
  res_S2_m6A$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = res_S2_m6A$current_symbol,
                               column = "ONTOLOGY",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
  # Optional: Map gene descriptions
  res_S2_m6A$GENENAME <- mapIds(org.Dm.eg.db,
                               keys = res_S2_m6A$current_symbol,
                               column = "GENENAME",
                               keytype = "SYMBOL",
                               multiVals = "first")
  
} else {
  cat("The column 'current_symbol' does not exist in the data frame.\n")
}


#write.table(res_S2_m6A_clean, file = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/S2S_vs_mettl3_KO.txt", sep = "\t", quote = FALSE)
```


    Gene Ontology analyses METTL3-KO relative to S2-S
```{r message=FALSE, warning=FALSE, echo=FALSE}
library(clusterProfiler)
# Filter for upregulated and downregulated genes
upregulated_entrez <- res_S2_m6A %>%
  dplyr::filter(diffexpressed == "UP") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

downregulated_entrez <- res_S2_m6A %>%
  dplyr::filter(diffexpressed == "DOWN") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

# GO analysis for upregulated genes
go_up <- enrichGO(gene = upregulated_entrez,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",  # You can specify "BP", "MF", or "CC" if needed
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  # You can specify "BP", "MF", or "CC" if needed
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes
barplot(go_up, showCategory = 20, 
        title = "Mettl3-KO/S2S:GO Enrichment for Upregulated Genes")

# Bar plot for downregulated genes
barplot(go_down, showCategory = 20, 
        title = "Mettl3-KO_vs_S2S:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 20, 
        title = "Mettl3-KO_vs_S2S:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes Mettl3-KO_S2S_GO_UP
dotplot(go_down, showCategory = 20, 
        title = "Mettl3-KO_vs_S2S:GO Enrichment for Downregulated Genes")

# If you want to see the results in a table
go_up_results <- as.data.frame(go_up)
go_down_results <- as.data.frame(go_down)

# View the top results
head(go_up_results)
head(go_down_results)

```


making Ridge plot
```{r}
library(org.Dm.eg.db)
organism <- org.Dm.eg.db

#keytypes(organism)

sig_S2_m6A_clean <- res_S2_m6A %>%
  dplyr::filter(diffexpressed != "Not Significant")

gene_scores <- sig_S2_m6A_clean$log2FoldChange
names(gene_scores) <- sig_S2_m6A_clean$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    #nPerm = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

#head(gse@result)


```

## Ridgeplot
Helpful to interpret up/down-regulated pathways. Mettl3-KO_S2S_GO_down
```{r fig.width=18, fig.height=12}

#ridgeplot(gse) + labs(x = "Enrichment distribution") + ggtitle("Gene Set Enrichment: METTL3-KO/S2S")Volc_Mettl3_KO_S2S
cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 30) + ggtitle("Gene Set Enrichment: METTL3-KO/S2S")
```

merge all datasets for network plot


```{r}
# Load required libraries
library(clusterProfiler)
library(enrichplot)
library(org.Dm.eg.db)
library(ggplot2)
library(dplyr)
library(ggnewscale) 
# Combine all datasets into a single list
datasets <- list(
  res_mettl3_KO_GFP_S2_GFP,
  res_S2_GFP_mettl3_KO_DC1_GOF,
  res_S2_S_DC1_KO,
  res_S2.GFP_S2.DC1.GOF, 
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, 
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF, 
  res_S2_m6A)

# Function to extract significant genes
extract_significant_genes <- function(dataset) {
  dataset %>% dplyr::filter(diffexpressed != "Not Significant") %>%
    dplyr::select(current_symbol, log2FoldChange)
}


# Apply function to all datasets
significant_lists <- lapply(datasets, extract_significant_genes)
names(significant_lists) <- c(
  "Ythdc1-KO/S2S",
  "Mettl3-KO/S2S",
  "Mettl3KO-Ythdc1-GoF/S2-GFP",
  "Mettl3KO-GFP/S2-GFP",
  "S2-Ythdc1-GoF/S2-GFP",
  "Mettl3KO-Ythdc1-GoF/Mettl3-KO-GFP",
  "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF")



logfc_combined <- Reduce(function(x, y) full_join(x, y, by = "current_symbol"), significant_lists)


rownames(logfc_combined) <- logfc_combined$current_symbol
logfc_combined$current_symbol <- NULL


colnames(logfc_combined) <- names(significant_lists)


logfc_combined[is.na(logfc_combined)] <- 0

library(ComplexHeatmap)

pheatmap(
  mat = as.matrix(logfc_combined),
  color = colorRampPalette(c("blue", "white", "red"))(100),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_rownames = TRUE,
  show_colnames = TRUE,
  fontsize_row = 8,
  fontsize_col = 10,
  main = "Differential Expression Heatmap")


# --- FILTERED HEATMAP for selected genes ---
selected_genes <- c(
  "Hsp70Aa", "Hsp70Ab", "Hsp70Ba", "Hsp70Bb", "Hsp70Bc", "fl(2)d", "vir", "nito",
  "Mettl3", "Ythdc1", "baz", "Flacc", "Adh", "Aldh-III", "Aldh7A1", "twi", "Ace",
  "ben", "unc-13", "Sxl", "cv-c", "Smr", "spen", "Bsg", "gsb", "Ten-a", "olf186-M","mfas", "miple2", "prosap", "aqz", "sky","Abd-B","Act88F","Argk1","cv-2", "Hsp83","TotM", "Swim", "Ten-a", "CG8046", "NijA", "Spn31A", "nemy", "slgA")

logfc_combined_filtered <- logfc_combined[rownames(logfc_combined) %in% selected_genes, ]

pheatmap(
  mat = as.matrix(logfc_combined_filtered),
  color = colorRampPalette(c("blue", "white", "red"))(100),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_rownames = TRUE,
  show_colnames = TRUE,
  fontsize_row = 9,
  fontsize_col = 10,
  main = "Differential Expression Heatmap", name = "log2FC" )



```

**Screenshot showing *DOWN* regulation of Ythdc in Ythdc-KO samples S2S GFP and METTL3-KO**

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
knitr::include_graphics("~/Dropbox/2022_signor_lai_m6A/Splicing_Analysis_George/S2S_YTHDC1_GoF_KO_METTL3-KO/Screenshots/Screenshot 2024-06-03 at 4.23.59PM.png")
```

Screenshot showing *DOWN* regulation of twi in METTL3-KO samples

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
knitr::include_graphics("~/Dropbox/2022_signor_lai_m6A/Splicing_Analysis_George/S2S_YTHDC1_GoF_KO_METTL3-KO/Screenshots/Screenshot 2024-06-03 at 4.27.44PM.png")
```

Screenshot showing *DOWN* regulation of Tollo in Ythdc1-KO

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
knitr::include_graphics("~/Dropbox/2022_signor_lai_m6A/Splicing_Analysis_George/S2S_YTHDC1_GoF_KO_METTL3-KO/Screenshots/Screenshot 2024-06-03 at 4.35.56PM.png")
```

Screenshot showing *DOWN* regulation of TotM in Ythdc1-GoF

```{r out.width="100%", out.height="100%", message=FALSE, warning=FALSE, echo=FALSE}
knitr::include_graphics("~/Dropbox/2022_signor_lai_m6A/Splicing_Analysis_George/S2S_YTHDC1_GoF_KO_METTL3-KO/Screenshots/Screenshot 2024-06-03 at 4.38.32PM.png")
```


Below are some commonly shared genes in METTL3-KO and YTHDC1-KO relative to S2S

# Lijuan's criteria

```{r}
# Find genes in res_S2_m6A_clean that are not in res_S2_S_DC1_KO_clean
res_S2_m6A_clean$gene_name = row.names(res_S2_m6A)

exclusive_mettl3 <- res_S2_m6A_clean %>%
  dplyr::filter(!gene_name %in% res_S2_S_DC1_KO_clean$gene_name)


res_S2_S_DC1_KO_clean$gene_name = row.names(res_S2_S_DC1_KO_clean)

exclusive_ythdc1 <- res_S2_S_DC1_KO_clean %>%
  dplyr::filter(!gene_name %in% res_S2_m6A_clean$gene_name)


# Print the number of exclusive genes
cat("Number of genes only in METTL3-KO/S2S:", nrow(exclusive_mettl3), "\n")
cat("Number of genes only in YTHDC1-KO/S2S:", nrow(exclusive_ythdc1), "\n")

#write.table(exclusive_mettl3, file = "mettl3-KO_genes_only.tsv", row.names = FALSE, quote = F, sep = "\t")
#write.table(exclusive_ythdc1, file = "ythdc1-KO_genes_only.tsv", row.names = FALSE, quote = F, sep = "\t")
# Optionally, display the result
print(exclusive_mettl3)
print(exclusive_ythdc1)

# Filter the data frame for "DOWN" entries and count them
exclusive_ythdc1_down <- nrow(exclusive_ythdc1[exclusive_ythdc1$diffexpressed == "DOWN", ])
exclusive_ythdc1_up <- nrow(exclusive_ythdc1[exclusive_ythdc1$diffexpressed == "UP", ])

exclusive_mettl3_down <- nrow(exclusive_mettl3[exclusive_mettl3$diffexpressed == "DOWN", ])
exclusive_mettl31_up <- nrow(exclusive_mettl3[exclusive_mettl3$diffexpressed == "UP", ])

# Print the result
print(paste("Number of exclusively downregulated genes in YTHDC1:", exclusive_ythdc1_down))
print(paste("Number of exclusively upregulated genes in YTHDC1:", exclusive_ythdc1_up))


print(paste("Number of exclusively downregulated genes in METTL3-KO:", exclusive_mettl3_down))
print(paste("Number of exclusively upregulated genes in METTL3-KO:", exclusive_mettl31_up))

library(clusterProfiler)
# FilterclusterProfiler# Filter for upregulated and downregulated genes
upregulated_entrez <- exclusive_ythdc1 %>%
  dplyr::filter(diffexpressed == "UP") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

downregulated_entrez <- exclusive_ythdc1 %>%
  dplyr::filter(diffexpressed == "DOWN") %>%
  dplyr::select(ENTREZID) %>%
  unlist()

# GO analysis for upregulated genes
go_up <- enrichGO(gene = upregulated_entrez,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = downregulated_entrez,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes
#barplot(go_up, title = "S2-YTHDC1-GoF vs METTL3-KO YTHDC1-GoF:GO Enrichment for Upregulated Genes")

# Bar plot for downregulated genes
#barplot(go_down, showCategory = 20, title = "S2-YTHDC1-GoF vs METTL3-KO YTHDC1-GoF:GO Enrichment for Downregulated Genes")

```



    
     make a correlation plot and a Venn diagram Using genes that exist in both mettl3 and YTHDC1 KO ( filter out genes only have one reading in one genotype)
```{r}
# Extract validated IDs from each dataset
ids_1 <- res_S2_m6A$validated_id
ids_2 <- res_S2_S_DC1_KO$validated_id

# Find common validated IDs between datasets
shared_ids <- intersect(ids_1, ids_2)
num_shared_gene <- length(shared_ids)

# Filter shared genes in each dataset based on logFC and padj
upregulated_1 <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% shared_ids & log2FoldChange > 0.5 & padj < 0.05)

downregulated_1 <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% shared_ids & log2FoldChange < -0.5 & padj < 0.05)

upregulated_2 <- res_S2_S_DC1_KO %>%
  dplyr::filter(validated_id %in% shared_ids & log2FoldChange > 0.5 & padj < 0.05)

downregulated_2 <- res_S2_S_DC1_KO %>%
  dplyr::filter(validated_id %in% shared_ids & log2FoldChange < -0.5 & padj < 0.05)


shared_gene_up = intersect(upregulated_1, upregulated_2)
num_shared_gene_up = length(shared_gene_up)
shared_gene_down = intersect(downregulated_1, downregulated_2)
num_shared_gene_down = length(shared_gene_down)


num_up_1 <- nrow(upregulated_1)
num_down_1 <- nrow(downregulated_1)

num_up_2 <- nrow(upregulated_2)
num_down_2 <- nrow(downregulated_2)

# Print the results
cat("Number of shared genes between both datasets:", num_shared_gene, "\n")

cat("Number of shared UP genes:", num_shared_gene_up, "\n")

cat("Number of shared DOWN genes:", num_shared_gene_down, "\n")


cat("Dataset 1 (Mettl3-KO/S2S):\n")
cat("Upregulated:", num_up_1, "\n")
cat("Downregulated:", num_down_1, "\n\n")

cat("Dataset 2 (Ythdc1-KO/S2S):\n")
cat("Upregulated:", num_up_2, "\n")
cat("Downregulated:", num_down_2, "\n\n")

```


```{r}
# ==== Set thresholds ====
padj_threshold <- 0.05
logFC_threshold <- 0.5

# ==== Step 1: Filter DE genes in each dataset ====

# Upregulated in METTL3
up_m6A <- res_S2_m6A %>%
  dplyr::filter(log2FoldChange > logFC_threshold, padj < padj_threshold)

# Downregulated in METTL3
down_m6A <- res_S2_m6A %>%
  dplyr::filter(log2FoldChange < -logFC_threshold, padj < padj_threshold)

# Upregulated in YTHDC1
up_YTHDC1 <- res_S2_S_DC1_KO %>%
  dplyr::filter(log2FoldChange > logFC_threshold, padj < padj_threshold)

# Downregulated in YTHDC1
down_YTHDC1 <- res_S2_S_DC1_KO %>%
  dplyr::filter(log2FoldChange < -logFC_threshold, padj < padj_threshold)

# ==== Step 2: Find shared DE genes ====

# Shared Upregulated
shared_up <- intersect(up_m6A$validated_id, up_YTHDC1$validated_id)

# Shared Downregulated
shared_down <- intersect(down_m6A$validated_id, down_YTHDC1$validated_id)

# Total shared DE genes (up or down)
shared_all <- union(shared_up, shared_down)

# ==== Step 3: Print results ====
cat("===== Shared DE Genes Between METTL3-KO and YTHDC1-KO =====\n")
cat("Shared Upregulated Genes:", length(shared_up), "\n")
cat("Shared Downregulated Genes:", length(shared_down), "\n")
cat("Total Shared DE Genes (up or down):", length(shared_all), "\n")

```


```{r}
library(ggvenn)

# Set thresholds
padj_threshold <- 0.05
logFC_threshold <- 0.5

# Filter up/downregulated genes
up_m6A <- res_S2_m6A %>% 
  dplyr::filter(log2FoldChange > logFC_threshold, padj < padj_threshold)
down_m6A <- res_S2_m6A %>% 
  dplyr::filter(log2FoldChange < -logFC_threshold, padj < padj_threshold)
up_YTHDC1 <- res_S2_S_DC1_KO %>% 
  dplyr::filter(log2FoldChange > logFC_threshold, padj < padj_threshold)
down_YTHDC1 <- res_S2_S_DC1_KO %>% 
  dplyr::filter(log2FoldChange < -logFC_threshold, padj < padj_threshold)

# Find shared up/down genes
shared_up_ids <- intersect(up_m6A$validated_id, up_YTHDC1$validated_id)
shared_down_ids <- intersect(down_m6A$validated_id, down_YTHDC1$validated_id)

# Prepare merged data for scatter plot
shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% c(shared_up_ids, shared_down_ids)) %>%
  dplyr::select(validated_id, gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange) %>%
  inner_join(
    res_S2_S_DC1_KO %>%
      dplyr::select(validated_id, symbol_YTHDC1 = current_symbol, log2FoldChange_YTHDC1 = log2FoldChange),
    by = "validated_id")

# Optional: label a few genes manually (you can change these)
highlight_genes <- c("Mettl3", "fl(2)d", "mfas", "Ten-a", "sky")

highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_YTHDC1 %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_YTHDC1))

# Correlation
cor_pearson <- cor(shared_genes_with_metrics$log2FoldChange_m6A,
                   shared_genes_with_metrics$log2FoldChange_YTHDC1,
                   method = "pearson")

# Plot
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_YTHDC1)) +
  geom_point(alpha = 0.6, size = 1.5, color = "steelblue") +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  size = 4, color = "black", box.padding = 0.3) +
  geom_smooth(method = "lm", linetype = "dashed", color = "black") +
  labs(
    title = paste0("Shared DEGs Between METTL3-KO and YTHDC1-KO (Pearson r = ", round(cor_pearson, 2), ")"),
    x = "Log2 Fold Change (METTL3-KO/S2S)",
    y = "Log2 Fold Change (YTHDC1-KO/S2S)") +
  theme_minimal(base_size = 14)



```
     
```{r}
library(dplyr)
library(pheatmap)

# Set thresholds
padj_threshold <- 0.05
logFC_threshold <- 0.5

# Filter DEGs from both datasets
up_m6A <- res_S2_m6A %>%
  dplyr::filter(log2FoldChange > logFC_threshold, padj < padj_threshold)
down_m6A <- res_S2_m6A %>% 
  dplyr::filter(log2FoldChange < -logFC_threshold, padj < padj_threshold)

up_YTHDC1 <- res_S2_S_DC1_KO %>% 
  dplyr::filter(log2FoldChange > logFC_threshold, padj < padj_threshold)
down_YTHDC1 <- res_S2_S_DC1_KO %>% 
  dplyr::filter(log2FoldChange < -logFC_threshold, padj < padj_threshold)

# Combine shared up and downregulated genes
shared_up <- intersect(up_m6A$validated_id, up_YTHDC1$validated_id)
shared_down <- intersect(down_m6A$validated_id, down_YTHDC1$validated_id)
shared_all <- union(shared_up, shared_down)

# Extract log2FoldChange for shared genes from both datasets
m6A_fc <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% shared_all) %>%
  dplyr::select(validated_id, gene_symbol = current_symbol, log2FoldChange_m6A = log2FoldChange)

YTHDC1_fc <- res_S2_S_DC1_KO %>%
  dplyr::filter(validated_id %in% shared_all) %>%
  dplyr::select(validated_id, log2FoldChange_YTHDC1 = log2FoldChange)

# Merge into a matrix
heatmap_data <- m6A_fc %>%
  inner_join(YTHDC1_fc, by = "validated_id") %>%
  dplyr::select(gene_symbol, log2FoldChange_m6A, log2FoldChange_YTHDC1) %>%
  distinct(gene_symbol, .keep_all = TRUE) %>%  # avoid duplicated gene symbols
  tibble::column_to_rownames("gene_symbol")

# Replace NA (if any) with 0 or leave as NA to allow clustering
heatmap_data[is.na(heatmap_data)] <- 0  # optional


# Plot heatmap
pheatmap(heatmap_data,
         cluster_rows = F,
         cluster_cols = F,
         #color = colorRampPalette(c("blue", "white", "red"))(100),
         main = "Log2 Fold Change of Shared DEGs (METTL3-KO/S2S vs YTHDC1-KO/S2S)",
         fontsize_row = 8, 
         name = "logFC")


```
     

```{r}
# ==== Load libraries ====
library(dplyr)
library(ggplot2)
library(ggrepel)

# ==== Step 1: Set thresholds ====
padj_threshold <- 0.05
logFC_threshold <- 0.5

# ==== Step 2: Filter shared genes ====
shared_genes <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% res_S2_S_DC1_KO$validated_id,
                padj <= padj_threshold,
                abs(log2FoldChange) >= logFC_threshold)

# ==== Step 3: Merge datasets and extract logFC ====
shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% shared_genes$validated_id,
                padj <= padj_threshold,
                abs(log2FoldChange) >= logFC_threshold) %>%
  dplyr::select(validated_id, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange) %>%
  inner_join(
    res_S2_S_DC1_KO %>%
      dplyr::filter(validated_id %in% shared_genes$validated_id,
                    padj <= padj_threshold,
                    abs(log2FoldChange) >= logFC_threshold) %>%
      dplyr::select(validated_id, symbol_YTHDC1 = current_symbol, log2FoldChange_YTHDC1 = log2FoldChange),
    by = "validated_id")

# ==== Step 4: Categorize genes ====
shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
    log2FoldChange_m6A > logFC_threshold & log2FoldChange_YTHDC1 > logFC_threshold ~ "Shared Upregulated",
    log2FoldChange_m6A < -logFC_threshold & log2FoldChange_YTHDC1 < -logFC_threshold ~ "Shared Downregulated",
    log2FoldChange_m6A > logFC_threshold & log2FoldChange_YTHDC1 < -logFC_threshold ~ "Up in METTL3, Down in YTHDC1",
    log2FoldChange_m6A < -logFC_threshold & log2FoldChange_YTHDC1 > logFC_threshold ~ "Down in METTL3, Up in YTHDC1",
    TRUE ~ "Other"))

# ==== Step 5: Create category labels for legend ====
category_counts <- as.data.frame(table(shared_genes_with_metrics$Category))
colnames(category_counts) <- c("Category", "n")
category_counts$label <- paste0(category_counts$Category, " (n=", category_counts$n, ")")
category_labels <- setNames(category_counts$label, category_counts$Category)

# ==== Step 6: Calculate correlation ONLY for genes changing in the same direction ====
same_direction_genes <- shared_genes_with_metrics %>%
  dplyr::filter(Category %in% c("Shared Upregulated", "Shared Downregulated", "Up in METTL3, Down in YTHDC1", "Down in METTL3, Up in YTHDC1"))

cor_pearson <- cor(same_direction_genes$log2FoldChange_m6A,
                   same_direction_genes$log2FoldChange_YTHDC1, method = "pearson")

# ==== Step 7: Highlight genes ====
highlight_genes <- c("Mettl3", "fl(2)d", "mfas", "miple2", "Ten-a", "Aldh7A1", "prosap", "aqz", "sky")
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_YTHDC1 %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_YTHDC1))

# ==== Step 8: Create plot ====
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_YTHDC1, color = Category)) +
  geom_point(size = 0.8, alpha = 0.8) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 20, size = 4.5, box.padding = 0.5, force = 1.2, segment.color = "gray70") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  scale_color_manual(
    values = c("Shared Upregulated" = "red",
               "Shared Downregulated" = "blue",
               "Up in METTL3, Down in YTHDC1" = "darkorange",
               "Down in METTL3, Up in YTHDC1" = "darkgreen",
               "Other" = "gray70"),
    labels = category_labels) +
  labs(title = paste0("Shared DE Genes METTL3-KO and YTHDC1-KOgenes at padj <=0.05 &log2FC|0.2|\n",
                      "Pearson r = ", round(cor_pearson, 2)),
       x = "Log2 Fold Change (METTL3-KO/S2S)",
       y = "Log2 Fold Change (YTHDC1-KO/S2S)",
       color = "") +
  xlim(-15, 15) +
  ylim(-15, 15) +
  theme_minimal(base_size = 14) +
  theme(plot.margin = margin(10, 10, 10, 10))


#write.table(shared_genes_with_metrics, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/novel/shared_gene_Ythdc1_Mettl3_KO.txt", quote = F, sep = "\t", row.names = F)
```

```{r}
library(ggvenn)

# Create sets based on regulation direction
up_in_m6A <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > logFC_threshold) %>%
  pull(validated_id)

down_in_m6A <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -logFC_threshold) %>%
  pull(validated_id)

up_in_YTHDC1 <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_YTHDC1 > logFC_threshold) %>%
  pull(validated_id)

down_in_YTHDC1 <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_YTHDC1 < -logFC_threshold) %>%
  pull(validated_id)


# Venn diagram for upregulated genes Venn_Mettl3_Ythdc1_UP_05
ggvenn(show_percentage = F, auto_scale = T,
  list("Up in METTL3-KO/S2S" = up_in_m6A,
    "Up in YTHDC1-KO/S2S" = up_in_YTHDC1),
  fill_color = c("#4DAF4A", "#984EA3"),
  stroke_size = 0.5,
  set_name_size = 4)


# Venn diagram for downregulated genes
ggvenn(show_percentage = F, auto_scale = T,
       list("Down in METTL3-KO/S2S" = down_in_m6A,
    "Down in YTHDC1-KO/S2S" = down_in_YTHDC1),
  fill_color = c("#4DAF4A", "#984EA3"),
  stroke_size = 0.5,
  set_name_size = 4)

```

# Soller comparism
```{r message=FALSE, warning=FALSE, echo=FALSE}
# ==== Load and clean Soller data ====
Soller_diff <- read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/Roignant_dif_exp.txt")
Soller_diff$log2.fold_change. <- as.numeric(Soller_diff$log2.fold_change.)

# ==== Step 1: Merge by FlyBase Gene ID (validated_id == gene_name) ====
shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% Soller_diff$gene_name,
         abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
   dplyr::select(validated_id, symbol_m6A = current_symbol,
         log2FoldChange_m6A = log2FoldChange, padj_m6A = padj) %>%
  inner_join(
    Soller_diff %>%
      dplyr::filter(gene_name %in% res_S2_m6A$validated_id,
             abs(log2.fold_change.) >= 0.5, q_value <= 0.01) %>%
       dplyr::select(validated_id = gene_name,
             symbol_Soller = gene_id, log2FoldChange_Soller = log2.fold_change.,
             padj_Soller = q_value), by = "validated_id")

# ==== Step 2: Count shared up/downregulated ====
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6A <= 0.05 &
         log2FoldChange_Soller > 0.5 & padj_Soller <= 0.01)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6A <= 0.05 &
         log2FoldChange_Soller < -0.5 & padj_Soller <= 0.01)

# ==== Step 3: Unique genes ====
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(validated_id %in% Soller_diff$gene_name),
         log2FoldChange > 0.5,
         padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(validated_id %in% Soller_diff$gene_name),
         log2FoldChange < -0.5,
         padj <= 0.05)

unique_up_Soller <- Soller_diff %>%
  dplyr::filter(!(gene_name %in% res_S2_m6A$validated_id),
         log2.fold_change. > 0.5,
         q_value <= 0.01)

unique_down_Soller <- Soller_diff %>%
  dplyr::filter(!(gene_name %in% res_S2_m6A$validated_id),
         log2.fold_change. < -0.5,
         q_value <= 0.01)

# ==== Step 4: Summary stats ====
num_total_m6A <- nrow(res_S2_m6A)
num_total_Soller <- nrow(Soller_diff)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_Soller <- nrow(unique_up_Soller)
num_unique_down_Soller <- nrow(unique_down_Soller)

num_mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6A <= 0.05) %>%
  nrow()

num_mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6A <= 0.05) %>%
  nrow()

num_Soller_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Soller > 0.5 & padj_Soller <= 0.01) %>%
  nrow()

num_Soller_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Soller < -0.5 & padj_Soller <= 0.01) %>%
  nrow()

# ==== Step 5: Print ====
cat("==== METTL3 vs Soller DE Comparison ====\n")
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in Soller:", num_total_Soller, "\n")
cat("Shared genes:", num_shared, "\n\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n\n")

cat("Upregulated genes in METTL3 (shared):", num_mettl3_up, "\n")
cat("Downregulated genes in METTL3 (shared):", num_mettl3_down, "\n")
cat("Upregulated genes in Soller (shared):", num_Soller_up, "\n")
cat("Downregulated genes in Soller (shared):", num_Soller_down, "\n\n")

cat("Unique upregulated in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated in METTL3:", num_unique_down_m6A, "\n")
cat("Unique upregulated in Soller:", num_unique_up_Soller, "\n")
cat("Unique downregulated in Soller:", num_unique_down_Soller, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_Soller > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_Soller < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_Soller < 0 ~ "Up in METTL3, Down in Soller",
      log2FoldChange_m6A < 0 & log2FoldChange_Soller > 0 ~ "Down in METTL3, Up in Soller",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Soller <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Soller")
Down_METTL3_Up_Soller <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Soller")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Hsp83", "GstD10", "AttB", "GluProRS","Hsc70-2", "Lime")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_Soller %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_Soller))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_Soller, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Soller (", Up_METTL3_Down_Soller, ")", sep=""), 
                                paste("Down in METTL3, Up in Soller (", Down_METTL3_Up_Soller, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Soller_METTL3-KO", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_Soller), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Soller_METTL3-KO)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40)) 


# Step 1: Filter shared genes and get metrics for both datasets

upregulated_m6A <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5) %>%
  pull(validated_id)

downregulated_m6A <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5) %>%
  pull(validated_id)

upregulated_Soller_diff <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Soller > 0.5) %>%
  pull(validated_id)

downregulated_Soller_diff <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Soller < -0.5) %>%
  pull(validated_id)


# Step 2: Create named lists for upregulated and downregulated genes
venn_data_up <- list(
  METTL3 = upregulated_m6A,
  YTHDC1 = upregulated_Soller_diff)

venn_data_down <- list(
  METTL3 = downregulated_m6A,
  YTHDC1 = downregulated_Soller_diff)

# Step 3: Generate Venn diagrams
# Venn diagram for upregulated genes
ggvenn(venn_data_up, auto_scale = T, show_elements = FALSE, show_percentage = FALSE) +
  ggtitle("Venn Diagram of Upregulated Genes: padj<=0.05 and log2foldchange|0.05|") +
  theme_void() + # Remove axes and background
  theme(plot.title = element_text(size = 16, face = "bold", hjust = 0.5))

# Venn diagram for downregulated genes
ggvenn(venn_data_down, auto_scale = T, show_elements = FALSE, show_percentage = FALSE) +
  ggtitle("Venn Diagram of Downregulated Genes:padj<=0.05 and log2foldchange|0.05|") +  
  theme_void() + 
  theme(plot.title = element_text(size = 16, face = "bold", hjust = 0.5))

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_Soller.txt", sep = "\t", quote = F)
```




# Roignant: m6A modulates neuronal functions and sex determination in Drosophila Ime4_KD_versus_LacZKD_S2R
```{r message=FALSE, warning=FALSE, echo=FALSE}
# ==== Load and read Ime4 KD dataset ====
Ime4_KD_versus_LacZKD_S2R <- read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/Ime4_KD_versus_LacZKD_S2R.txt")

# ==== Step 1: Identify shared genes (by gene_name) ====
G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(validated_id %in% Ime4_KD_versus_LacZKD_S2R$gene_id)

# ==== Step 2: Merge datasets and filter shared DE genes ====
shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name,
         abs(log2FoldChange) >= 0.5,
         padj <= 0.05) %>%
  dplyr::select(gene_name,
         symbol_m6A = current_symbol,
         log2FoldChange_m6A = log2FoldChange,
         padj_m6A = padj) %>%
  inner_join(
    Ime4_KD_versus_LacZKD_S2R %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name,
             #abs(log2.fold.change..MAP...groupIme4.vs.groupControl) >= 0.5,
             BH.adjusted.p.values <= 0.05) %>%
      dplyr::select(gene_name,
             symbol_Ime4 = current_symbol,
             log2FoldChange_Ime4 = log2.fold.change..MAP...groupIme4.vs.groupControl,
             padj_Ime4 = BH.adjusted.p.values),
    by = "gene_name")

# ==== Step 3: Shared DE Genes (Up/Down in both datasets) ====
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6A <= 0.05 &
         log2FoldChange_Ime4 > 0.5 & padj_Ime4 <= 0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6A <= 0.05 &
         log2FoldChange_Ime4 < -0.5 & padj_Ime4 <= 0.05)

# ==== Step 4: Unique DE genes in each dataset ====
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name),
         log2FoldChange > 0.5,
         padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name),
         log2FoldChange < -0.5,
         padj <= 0.05)

unique_up_Ime4 <- Ime4_KD_versus_LacZKD_S2R %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name),
         log2.fold.change..MAP...groupIme4.vs.groupControl > 0.5,
         BH.adjusted.p.values <= 0.05)

unique_down_Ime4 <- Ime4_KD_versus_LacZKD_S2R %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name),
         log2.fold.change..MAP...groupIme4.vs.groupControl < -0.5,
         BH.adjusted.p.values <= 0.05)

# ==== Step 5: Direction-specific DE stats from shared genes ====
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6A <= 0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6A <= 0.05)

Ime4_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Ime4 > 0.5 & padj_Ime4 <= 0.05)

Ime4_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Ime4 < -0.5 & padj_Ime4 <= 0.05)

# ==== Step 6: Count Stats ====
num_total_m6A <- nrow(res_S2_m6A)
num_total_Ime4 <- nrow(Ime4_KD_versus_LacZKD_S2R)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_Ime4 <- nrow(unique_up_Ime4)
num_unique_down_Ime4 <- nrow(unique_down_Ime4)

num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_Ime4_up <- nrow(Ime4_up)
num_Ime4_down <- nrow(Ime4_down)

# ==== Step 7: Print Summary ====
cat("===== METTL3 vs Lence (Roignant lab) Ime4-KD DE Comparison =====\n")
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in Ime4-KD:", num_total_Ime4, "\n")
cat("Shared genes analyzed:", num_shared, "\n\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n\n")

cat("Upregulated in METTL3 (shared):", num_mettl3_up, "\n")
cat("Downregulated in METTL3 (shared):", num_mettl3_down, "\n")
cat("Upregulated in Ime4-KD (shared):", num_Ime4_up, "\n")
cat("Downregulated in Ime4-KD (shared):", num_Ime4_down, "\n\n")

cat("Unique upregulated in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated in METTL3:", num_unique_down_m6A, "\n")
cat("Unique upregulated in Ime4-KD:", num_unique_up_Ime4, "\n")
cat("Unique downregulated in Ime4-KD:", num_unique_down_Ime4, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_Ime4 > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_Ime4 < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_Ime4 < 0 ~ "Up in METTL3, Down in Roignant",
      log2FoldChange_m6A < 0 & log2FoldChange_Ime4 > 0 ~ "Down in METTL3, Up in Roignant",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Roignant <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Roignant")
Down_METTL3_Up_Roignant <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Roignant")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Hsp83", "GstD10", "AttB", "GluProRS","Hsc70-2", "Lime")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | log2FoldChange_Ime4 %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, log2FoldChange_Ime4))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_Ime4, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Roignant (", Up_METTL3_Down_Roignant, ")", sep=""), 
                                paste("Down in METTL3, Up in Roignant (", Down_METTL3_Up_Roignant, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Roignant_Ime4_KD_versus_LacZKD_S2R", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_Ime4), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Roignant_Ime4_KD_versus_LacZKD_S2R)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40)) 

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_Ime4_KD_versus_LacZKD_Roignant.txt", sep = "\t", quote = F)
```

# Rioganant: m6A modulates neuronal functions and sex determination in Drosophila Ime4_dMettl14_KD_versus_LacZKD_S2R
```{r message=FALSE, warning=FALSE, echo=FALSE}
Ime4_dMettl14_KD_versus_LacZKD = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/Ime4_dMettl14_KD_versus_LacZKD_S2R.txt")
#Ime4_dMettl14_KD_versus_LacZKD

G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% Ime4_dMettl14_KD_versus_LacZKD$gene_name)

shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(Ime4_dMettl14_KD_versus_LacZKD %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2.fold.change..MAP...groupIme4_CG7818.vs.groupControl) >= 0.5, BH.adjusted.p.values <= 0.05) %>%
      dplyr::select(gene_name, symbol_Ime4_dMettl14_KD_versus_LacZKD = current_symbol, log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD = log2.fold.change..MAP...groupIme4_CG7818.vs.groupControl, padj_Ime4_dMettl14_KD_versus_LacZKD = BH.adjusted.p.values),
    by = "gene_name")

# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD > 0.5, padj_Ime4_dMettl14_KD_versus_LacZKD <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD < -0.5, padj_Ime4_dMettl14_KD_versus_LacZKD <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_Ime4_dMettl14_KD_versus_LacZKD <- Ime4_dMettl14_KD_versus_LacZKD %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.fold.change..MAP...groupIme4_CG7818.vs.groupControl > 0.5 & BH.adjusted.p.values <= 0.05)

unique_down_Ime4_dMettl14_KD_versus_LacZKD <- Ime4_dMettl14_KD_versus_LacZKD %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.fold.change..MAP...groupIme4_CG7818.vs.groupControl < -0.5 & BH.adjusted.p.values <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_Ime4_dMettl14_KD_versus_LacZKD <- nrow(Ime4_dMettl14_KD_versus_LacZKD)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_Ime4_dMettl14_KD_versus_LacZKD <- nrow(unique_up_Ime4_dMettl14_KD_versus_LacZKD)
num_unique_down_Ime4_dMettl14_KD_versus_LacZKD <- nrow(unique_down_Ime4_dMettl14_KD_versus_LacZKD)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
Ime4_dMettl14_KD_versus_LacZKD_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD > 0 & padj_m6a<=0.05)

Ime4_dMettl14_KD_versus_LacZKD_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD < 0 & padj_m6a<=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
Ime4_dMettl14_KD_versus_LacZKD_up <- nrow(Ime4_dMettl14_KD_versus_LacZKD_up)
Ime4_dMettl14_KD_versus_LacZKD_down <- nrow(Ime4_dMettl14_KD_versus_LacZKD_down)

# Print results
cat("===== METTL3 vs Lence (Roignant lab) Ime4_dMettl14_KD_versus_LacZKD_S2R DE Comparison =====\n")
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in Ime4_dMettl14_KD_versus_LacZKD (from shared genes):", Ime4_dMettl14_KD_versus_LacZKD_up, "\n")
cat("Number of downregulated genes in Ime4_dMettl14_KD_versus_LacZKD (from shared genes):", Ime4_dMettl14_KD_versus_LacZKD_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in Ime4_dMettl14_KD_versus_LacZKD:", num_total_Ime4_dMettl14_KD_versus_LacZKD, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in Ime4_dMettl14_KD_versus_LacZKD:", num_unique_up_Ime4_dMettl14_KD_versus_LacZKD, "\n")
cat("Unique downregulated genes in Ime4_dMettl14_KD_versus_LacZKD:", num_unique_down_Ime4_dMettl14_KD_versus_LacZKD, "\n") 
```


```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD < 0 ~ "Up in METTL3, Down in Roignant",
      log2FoldChange_m6A < 0 & log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD > 0 ~ "Down in METTL3, Up in Roignant",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Roignant <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Roignant")
Down_METTL3_Up_Roignant <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Roignant")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_Ime4_dMettl14_KD_versus_LacZKD %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_Ime4_dMettl14_KD_versus_LacZKD))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Roignant (", Up_METTL3_Down_Roignant, ")", sep=""), 
                                paste("Down in METTL3, Up in Roignant (", Down_METTL3_Up_Roignant, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Roignant_Ime4_KD_versus_LacZKD_S2R", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_Ime4_dMettl14_KD_versus_LacZKD), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Roignant_Ime4_KD_versus_LacZKD_S2R)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40)) 

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_Ime4_dMettl14_KD_versus_LacZKD_Roignant.txt", sep = "\t", quote = F)
```

# Rioganant: m6A modulates neuronal functions and sex determination in Drosophila dMettl14KD_versus_LacZKD_in_S2R
```{r message=FALSE, warning=FALSE, echo=FALSE}
dMettl14KD_versus_LacZKD_in_S2R = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/dMettl14KD_versus_LacZKD_in_S2R.txt")
#dMettl14KD_versus_LacZKD_in_S2R

G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% dMettl14KD_versus_LacZKD_in_S2R$gene_name)

shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(dMettl14KD_versus_LacZKD_in_S2R %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2.fold.change..MAP...groupCG7818.vs.groupControl) >= 0.5, BH.adjusted.p.values <= 0.05) %>%
      dplyr::select(gene_name, symbol_dMettl14KD_versus_LacZKD_in_S2R = current_symbol, log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R = log2.fold.change..MAP...groupCG7818.vs.groupControl, padj_dMettl14KD_versus_LacZKD_in_S2R = BH.adjusted.p.values),
    by = "gene_name")

# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R > 0.5, padj_dMettl14KD_versus_LacZKD_in_S2R <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R < -0.5, padj_dMettl14KD_versus_LacZKD_in_S2R <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_dMettl14KD_versus_LacZKD_in_S2R <- dMettl14KD_versus_LacZKD_in_S2R %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.fold.change..MAP...groupCG7818.vs.groupControl > 0.5 & BH.adjusted.p.values <= 0.05)

unique_down_dMettl14KD_versus_LacZKD_in_S2R <- dMettl14KD_versus_LacZKD_in_S2R %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.fold.change..MAP...groupCG7818.vs.groupControl < -0.5 & BH.adjusted.p.values <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_dMettl14KD_versus_LacZKD_in_S2R <- nrow(dMettl14KD_versus_LacZKD_in_S2R)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_dMettl14KD_versus_LacZKD_in_S2R <- nrow(unique_up_dMettl14KD_versus_LacZKD_in_S2R)
num_unique_down_dMettl14KD_versus_LacZKD_in_S2R <- nrow(unique_down_Ime4_dMettl14_KD_versus_LacZKD)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
dMettl14KD_versus_LacZKD_in_S2R_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R > 0 & padj_m6a<=0.05)

dMettl14KD_versus_LacZKD_in_S2R_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R < 0 & padj_m6a<=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_dMettl14KD_versus_LacZKD_in_S2R_up <- nrow(dMettl14KD_versus_LacZKD_in_S2R_up)
num_dMettl14KD_versus_LacZKD_in_S2R_down <- nrow(dMettl14KD_versus_LacZKD_in_S2R_down)

# Print results
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in dMettl14KD_versus_LacZKD_in_S2R (from shared genes):", num_dMettl14KD_versus_LacZKD_in_S2R_up, "\n")
cat("Number of downregulated genes in dMettl14KD_versus_LacZKD_in_S2R (from shared genes):", num_dMettl14KD_versus_LacZKD_in_S2R_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in dMettl14KD_versus_LacZKD_in_S2R:", num_total_dMettl14KD_versus_LacZKD_in_S2R, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in dMettl14KD_versus_LacZKD_in_S2R:", num_unique_up_dMettl14KD_versus_LacZKD_in_S2R, "\n")
cat("Unique downregulated genes in dMettl14KD_versus_LacZKD_in_S2R:", num_unique_down_dMettl14KD_versus_LacZKD_in_S2R, "\n")

```


```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R < 0 ~ "Up in METTL3, Down in Roignant",
      log2FoldChange_m6A < 0 & log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R > 0 ~ "Down in METTL3, Up in Roignant",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Roignant <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Roignant")
Down_METTL3_Up_Roignant <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Roignant")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_dMettl14KD_versus_LacZKD_in_S2R %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_dMettl14KD_versus_LacZKD_in_S2R))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Roignant (", Up_METTL3_Down_Roignant, ")", sep=""), 
                                paste("Down in METTL3, Up in Roignant (", Down_METTL3_Up_Roignant, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Roignant_Ime4_KD_versus_LacZKD_S2R", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_dMettl14KD_versus_LacZKD_in_S2R), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Roignant_Ime4_KD_versus_LacZKD_S2R)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40))

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_dMettl14KD_versus_LacZKD_Roignant.txt", sep = "\t", quote = F)
```



# Rioganant: m6A modulates neuronal functions and sex determination in Drosophila Fl2d_KD_versus_LacZKD_in_S2R
```{r message=FALSE, warning=FALSE, echo=FALSE}
Fl2d_KD_versus_LacZKD_in_S2R = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/Fl2d_KD_versus_LacZKD_in_S2R.txt")
#Fl2d_KD_versus_LacZKD_in_S2R

G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% Fl2d_KD_versus_LacZKD_in_S2R$gene_name)

shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(Fl2d_KD_versus_LacZKD_in_S2R %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2.fold.change..MAP...groupFl2d.vs.groupControl) >= 0.5, BH.adjusted.p.values <= 0.05) %>%
      dplyr::select(gene_name, symbol_Fl2d_KD_versus_LacZKD_in_S2R = current_symbol, log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R = log2.fold.change..MAP...groupFl2d.vs.groupControl, padj_Fl2d_KD_versus_LacZKD_in_S2R = BH.adjusted.p.values),
    by = "gene_name")

# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R > 0.5, padj_Fl2d_KD_versus_LacZKD_in_S2R <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R < -0.5, padj_Fl2d_KD_versus_LacZKD_in_S2R <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_Fl2d_KD_versus_LacZKD_in_S2R <- Fl2d_KD_versus_LacZKD_in_S2R %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.fold.change..MAP...groupFl2d.vs.groupControl > 0.5 & BH.adjusted.p.values <= 0.05)

unique_down_Fl2d_KD_versus_LacZKD_in_S2R <- Fl2d_KD_versus_LacZKD_in_S2R %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.fold.change..MAP...groupFl2d.vs.groupControl < -0.5 & BH.adjusted.p.values <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_Fl2d_KD_versus_LacZKD_in_S2R <- nrow(Fl2d_KD_versus_LacZKD_in_S2R)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_Fl2d_KD_versus_LacZKD_in_S2R <- nrow(unique_up_Fl2d_KD_versus_LacZKD_in_S2R)
num_unique_down_Fl2d_KD_versus_LacZKD_in_S2R <- nrow(unique_down_Fl2d_KD_versus_LacZKD_in_S2R)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
Fl2d_KD_versus_LacZKD_in_S2R_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R > 0 & padj_m6a<=0.05)

Fl2d_KD_versus_LacZKD_in_S2R_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R < 0 & padj_m6a<=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_Fl2d_KD_versus_LacZKD_in_S2R_up <- nrow(Fl2d_KD_versus_LacZKD_in_S2R_up)
num_Fl2d_KD_versus_LacZKD_in_S2R_down <- nrow(Fl2d_KD_versus_LacZKD_in_S2R_down)

# Print results
cat("===== METTL3 vs Lence (Roignant lab) Fl2d_KD_versus_LacZKD_in_S2R DE Comparison =====\n")
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in Fl2d_KD_versus_LacZKD_in_S2R (from shared genes):", num_Fl2d_KD_versus_LacZKD_in_S2R_up, "\n")
cat("Number of downregulated genes in Fl2d_KD_versus_LacZKD_in_S2R (from shared genes):", num_Fl2d_KD_versus_LacZKD_in_S2R_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in Fl2d_KD_versus_LacZKD_in_S2R:", num_total_Fl2d_KD_versus_LacZKD_in_S2R, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in Fl2d_KD_versus_LacZKD_in_S2R:", num_unique_up_Fl2d_KD_versus_LacZKD_in_S2R, "\n")
cat("Unique downregulated genes in Fl2d_KD_versus_LacZKD_in_S2R:", num_unique_down_Fl2d_KD_versus_LacZKD_in_S2R, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R < 0 ~ "Up in METTL3, Down in Roignant",
      log2FoldChange_m6A < 0 & log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R > 0 ~ "Down in METTL3, Up in Roignant",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Roignant <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Roignant")
Down_METTL3_Up_Roignant <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Roignant")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_Fl2d_KD_versus_LacZKD_in_S2R %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_Fl2d_KD_versus_LacZKD_in_S2R))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
#shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Roignant (", Up_METTL3_Down_Roignant, ")", sep=""), 
                                paste("Down in METTL3, Up in Roignant (", Down_METTL3_Up_Roignant, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Roignant_Ime4_KD_versus_LacZKD_S2R", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_Fl2d_KD_versus_LacZKD_in_S2R), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Roignant_Ime4_KD_versus_LacZKD_S2R)", 
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40)) 


#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_Fl2d_KD_versus_LacZKD_Roignant.txt", sep = "\t", quote = F)
```



Overlap Analysis with Bonini
```{r}

Ythdc_RNAi_basal_vs_HS_2 = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/Ythdc_RNAi_basal_vs_HS_2.txt", header = T)
Ythdc_RNAi_basal_vs_HS_2 = Ythdc_RNAi_basal_vs_HS_2[Ythdc_RNAi_basal_vs_HS_2$baseMean>0 & !is.na(Ythdc_RNAi_basal_vs_HS_2$padj), ]


Mettl3_RNAi_basal_vs_HS = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/Mettl3_RNAi_basal_vs_HS.txt", header = T)
Mettl3_RNAi_basal_vs_HS = Mettl3_RNAi_basal_vs_HS[Mettl3_RNAi_basal_vs_HS$baseMean>0 & !is.na(Mettl3_RNAi_basal_vs_HS$padj), ]


Control_RNAi_vs_HS = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/Control_RNAi_vs_HS.txt", header = T)
Control_RNAi_vs_HS = Control_RNAi_vs_HS[Control_RNAi_vs_HS$baseMean>0 & !is.na(Control_RNAi_vs_HS$padj), ]
```

# Bonini: Mettl3_RNAi_basal_vs_HS
```{r message=FALSE, warning=FALSE, echo=FALSE}

G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% Mettl3_RNAi_basal_vs_HS$gene_name)

shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(Mettl3_RNAi_basal_vs_HS %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2.hs.vs.ctl.mettl3.) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_Mettl3_RNAi_basal_vs_HS = current_symbol, log2FoldChange_Mettl3_RNAi_basal_vs_HS = log2.hs.vs.ctl.mettl3., padj_Mettl3_RNAi_basal_vs_HS = padj),
    by = "gene_name")

# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_Mettl3_RNAi_basal_vs_HS > 0.5, padj_Mettl3_RNAi_basal_vs_HS <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_Mettl3_RNAi_basal_vs_HS < -0.5, padj_Mettl3_RNAi_basal_vs_HS <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_Mettl3_RNAi_basal_vs_HS <- Mettl3_RNAi_basal_vs_HS %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.hs.vs.ctl.mettl3. > 0.5 & padj <= 0.05)

unique_down_Mettl3_RNAi_basal_vs_HS <- Mettl3_RNAi_basal_vs_HS %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2.hs.vs.ctl.mettl3. < -0.5 & padj <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_Mettl3_RNAi_basal_vs_HS <- nrow(Mettl3_RNAi_basal_vs_HS)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_Mettl3_RNAi_basal_vs_HS <- nrow(unique_up_Mettl3_RNAi_basal_vs_HS)
num_unique_down_Mettl3_RNAi_basal_vs_HS <- nrow(unique_down_Mettl3_RNAi_basal_vs_HS)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
Mettl3_RNAi_basal_vs_HS_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Mettl3_RNAi_basal_vs_HS > 0.5 & padj_Mettl3_RNAi_basal_vs_HS <=0.05)

Mettl3_RNAi_basal_vs_HS_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Mettl3_RNAi_basal_vs_HS < 0.5 & padj_Mettl3_RNAi_basal_vs_HS <=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_Mettl3_RNAi_basal_vs_HS_up <- nrow(Mettl3_RNAi_basal_vs_HS_up)
num_Mettl3_RNAi_basal_vs_HS_down <- nrow(Mettl3_RNAi_basal_vs_HS_down)

# Print results
cat("===== METTL3 vs Perlegoset et al (Bonini lab)  Mettl3_RNAi_basal_vs_HS DE Comparison =====\n")
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in Mettl3_RNAi_basal_vs_HS (from shared genes):", num_Mettl3_RNAi_basal_vs_HS_up, "\n")
cat("Number of downregulated genes in Mettl3_RNAi_basal_vs_HS (from shared genes):", num_Mettl3_RNAi_basal_vs_HS_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in Mettl3_RNAi_basal_vs_HS:", num_total_Mettl3_RNAi_basal_vs_HS, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in Mettl3_RNAi_basal_vs_HS:", num_unique_up_Mettl3_RNAi_basal_vs_HS, "\n")
cat("Unique downregulated genes in Mettl3_RNAi_basal_vs_HS:", num_unique_down_Mettl3_RNAi_basal_vs_HS, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_Mettl3_RNAi_basal_vs_HS > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_Mettl3_RNAi_basal_vs_HS < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_Mettl3_RNAi_basal_vs_HS < 0 ~ "Up in METTL3-KO, Down in Bonini",
      log2FoldChange_m6A < 0 & log2FoldChange_Mettl3_RNAi_basal_vs_HS > 0 ~ "Down in METTL3-KO, Up in Bonini",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Bonini <- sum(shared_genes_with_metrics$Category == "Up in METTL3-KO, Down in Bonini")
Down_METTL3_Up_Bonini <- sum(shared_genes_with_metrics$Category == "Down in METTL3-KO, Up in Bonini")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_Mettl3_RNAi_basal_vs_HS %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_Mettl3_RNAi_basal_vs_HS))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_Mettl3_RNAi_basal_vs_HS, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3-KO, Down in Roignant (", Up_METTL3_Down_Bonini, ")", sep=""), 
                                paste("Down in METTL3-KO, Up in Roignant (", Down_METTL3_Up_Bonini, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Bonini_Mettl3_RNAi_basal_vs_HS", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_Mettl3_RNAi_basal_vs_HS), 2), ")"),
       x = "Log2 Fold Change (Lai_METTL3-KO)",
       y = "Log2 Fold Change (Bonini_Mettl3_RNAi_basal_vs_HS)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40))

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_Mettl3_RNAi_basal_vs_HS_Bonini.txt", sep = "\t", quote = F)
```

# Bonini: res_S2_S_DC1_KO_clean Ythdc_RNAi_basal_vs_HS_2
```{r message=FALSE, warning=FALSE, echo=FALSE}

G_R_shared_gene <- res_S2_S_DC1_KO %>%
  dplyr::filter(gene_name %in% Ythdc_RNAi_basal_vs_HS_2$gene_name)

shared_genes_with_metrics <- res_S2_S_DC1_KO %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_YTHDC1_KO = current_symbol, log2FoldChange_YTHDC1_KO = log2FoldChange, padj_YTHDC1_KO = padj) %>%
  inner_join(Ythdc_RNAi_basal_vs_HS_2 %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(logs.YthdcRNAi.HSvcntrl.) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_Ythdc_RNAi_basal_vs_HS_2 = current_symbol, log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 = logs.YthdcRNAi.HSvcntrl., padj_Ythdc_RNAi_basal_vs_HS_2 = padj),
    by = "gene_name")

# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_YTHDC1_KO > 0.5, padj_YTHDC1_KO <=0.05 & log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 > 0.5, padj_Ythdc_RNAi_basal_vs_HS_2 <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_YTHDC1_KO < -0.5, padj_YTHDC1_KO <=0.05 & log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 < -0.5, padj_Ythdc_RNAi_basal_vs_HS_2 <=0.05)

# Unique genes
unique_up_YTHDC1_KO <- res_S2_S_DC1_KO %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_YTHDC1_KO <- res_S2_S_DC1_KO %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_Ythdc_RNAi_basal_vs_HS_2 <- Ythdc_RNAi_basal_vs_HS_2 %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & logs.YthdcRNAi.HSvcntrl. > 0.5 & padj <= 0.05)

unique_down_Ythdc_RNAi_basal_vs_HS_2 <- Ythdc_RNAi_basal_vs_HS_2 %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & logs.YthdcRNAi.HSvcntrl. < -0.5 & padj <= 0.05)

# Stats
num_total_YTHDC1_KO <- nrow(res_S2_S_DC1_KO)
num_total_Ythdc_RNAi_basal_vs_HS_2 <- nrow(Ythdc_RNAi_basal_vs_HS_2)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_YTHDC1_KO <- nrow(unique_up_YTHDC1_KO)
num_unique_down_YTHDC1_KO <- nrow(unique_down_YTHDC1_KO)

num_unique_up_Ythdc_RNAi_basal_vs_HS_2 <- nrow(unique_up_Ythdc_RNAi_basal_vs_HS_2)
num_unique_down_Ythdc_RNAi_basal_vs_HS_2 <- nrow(unique_down_Ythdc_RNAi_basal_vs_HS_2)

# Upregulated and Downregulated in METTL3
YTHDC1_KO_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_YTHDC1_KO > 0.5 & padj_YTHDC1_KO<=0.05)

YTHDC1_KO_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_YTHDC1_KO < -0.5 & padj_YTHDC1_KO<=0.05)

# Upregulated and Downregulated in YTHDC1
Ythdc_RNAi_basal_vs_HS_2_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 > 0.5 & padj_Ythdc_RNAi_basal_vs_HS_2 <=0.05)

Ythdc_RNAi_basal_vs_HS_2_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 < -0.5 & padj_Ythdc_RNAi_basal_vs_HS_2 <=0.05)

# Count numbers
num_YTHDC1_KO_up <- nrow(YTHDC1_KO_up)
num_YTHDC1_KO_down <- nrow(YTHDC1_KO_down)
num_Ythdc_RNAi_basal_vs_HS_2_up <- nrow(Ythdc_RNAi_basal_vs_HS_2_up)
num_Ythdc_RNAi_basal_vs_HS_2_down <- nrow(Ythdc_RNAi_basal_vs_HS_2_down)

# Print results
cat("===== Ythdc1-KO vs Perlegoset et al (Bonini lab) Ythdc_RNAi_basal_vs_HS_2 DE Comparison =====\n")
cat("Number of upregulated genes in Ythdc1-KO (from shared genes):", num_YTHDC1_KO_up, "\n")
cat("Number of downregulated genes in Ythdc1-KO (from shared genes):", num_YTHDC1_KO_down, "\n")
cat("Number of upregulated genes in Ythdc_RNAi_basal_vs_HS_2 (from shared genes):", num_Ythdc_RNAi_basal_vs_HS_2_up, "\n")
cat("Number of downregulated genes in Ythdc_RNAi_basal_vs_HS_2 (from shared genes):", num_Ythdc_RNAi_basal_vs_HS_2_down, "\n")
# Print statistics
cat("Total genes in Ythdc1-KO:", num_total_YTHDC1_KO, "\n")
cat("Total genes in Ythdc_RNAi_basal_vs_HS_2:", num_total_Ythdc_RNAi_basal_vs_HS_2, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in Ythdc1-KO:", num_unique_up_YTHDC1_KO, "\n")
cat("Unique downregulated genes in Ythdc1-KO:", num_unique_down_YTHDC1_KO, "\n")

cat("Unique upregulated genes in Ythdc_RNAi_basal_vs_HS_2:", num_unique_up_Ythdc_RNAi_basal_vs_HS_2, "\n")
cat("Unique downregulated genes in Ythdc_RNAi_basal_vs_HS_2:", num_unique_down_Ythdc_RNAi_basal_vs_HS_2, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_YTHDC1_KO > 0 & log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 > 0 ~ "Shared Upregulated",
      log2FoldChange_YTHDC1_KO < 0 & log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 < 0 ~ "Shared Downregulated",
      log2FoldChange_YTHDC1_KO > 0 & log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 < 0 ~ "Up in Ythdc1-KO, Down in Bonini",
      log2FoldChange_YTHDC1_KO < 0 & log2FoldChange_Ythdc_RNAi_basal_vs_HS_2 > 0 ~ "Down in Ythdc1-KO, Up in Bonini",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_Ythdc1_KO_Down_Bonini <- sum(shared_genes_with_metrics$Category == "Up in Ythdc1-KO, Down in Bonini")
Down_Ythdc1_KO_Up_Bonini <- sum(shared_genes_with_metrics$Category == "Down in Ythdc1-KO, Up in Bonini")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_YTHDC1_KO %in% highlight_genes | symbol_Ythdc_RNAi_basal_vs_HS_2 %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_YTHDC1_KO, symbol_Ythdc_RNAi_basal_vs_HS_2))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_YTHDC1_KO, y = log2FoldChange_Ythdc_RNAi_basal_vs_HS_2, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in Ythdc1-KO, Down in Bonini (", Up_Ythdc1_KO_Down_Bonini, ")", sep=""), 
                                paste("Down in Ythdc1-KO, Up in Bonini (", Down_Ythdc1_KO_Up_Bonini, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between Ythdc1-KO/S2S and Bonini_Mettl3_RNAi_basal_vs_HS", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_YTHDC1_KO,
                                                        shared_genes_with_metrics$log2FoldChange_Ythdc_RNAi_basal_vs_HS_2), 2), ")"),
       x = "Log2 Fold Change (Lai_Yhdc1-KO)",
       y = "Log2 Fold Change (Bonini_Ythdc_RNAi_basal_vs_HS_2)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40)) 

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_Ythdc_RNAi_basal_vs_HS_2_Bonini.txt", sep = "\t", quote = F)
```



Overlap Analysis with Kan et al
```{r}

one_week_mettl3_vs_mettl3 = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/one_week_mettl3_vs_mettl3.txt", header = T)
one_week_mettl3_vs_mettl3 = one_week_mettl3_vs_mettl3[one_week_mettl3_vs_mettl3$baseMean>0 & !is.na(one_week_mettl3_vs_mettl3$padj), ]


three_week_mettl3_vs_mettl3 = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/three_week_mettl3_vs_mettl3.txt", header = T)
three_week_mettl3_vs_mettl3 = three_week_mettl3_vs_mettl3[three_week_mettl3_vs_mettl3$baseMean>0 & !is.na(three_week_mettl3_vs_mettl3$padj), ]


one_week_ythdf_vs_ythdf = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/one_week_ythdf_vs_ythdf.txt", header = T)
one_week_ythdf_vs_ythdf = one_week_ythdf_vs_ythdf[one_week_ythdf_vs_ythdf$baseMean>0 & !is.na(one_week_ythdf_vs_ythdf$padj), ]


three_week_ythdf_vs_ythdf = read.delim("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/three_week_ythdf_vs_ythdf.txt", header = T)
three_week_ythdf_vs_ythdf = three_week_ythdf_vs_ythdf[three_week_ythdf_vs_ythdf$baseMean>0 & !is.na(three_week_ythdf_vs_ythdf$padj), ]
```


# Kan et al: one_week_mettl3_vs_mettl3
```{r message=FALSE, warning=FALSE, echo=FALSE}

G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% one_week_mettl3_vs_mettl3$gene_name)

shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(one_week_mettl3_vs_mettl3 %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_one_week_mettl3_vs_mettl3 = current_symbol, log2FoldChange_one_week_mettl3_vs_mettl3 = log2FoldChange, padj_one_week_mettl3_vs_mettl3 = padj),
    by = "gene_name")

# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_one_week_mettl3_vs_mettl3 > 0.5, padj_one_week_mettl3_vs_mettl3 <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_one_week_mettl3_vs_mettl3 < -0.5, padj_one_week_mettl3_vs_mettl3 <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_one_week_mettl3_vs_mettl3 <- one_week_mettl3_vs_mettl3 %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_one_week_mettl3_vs_mettl3 <- one_week_mettl3_vs_mettl3 %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_one_week_mettl3_vs_mettl3 <- nrow(one_week_mettl3_vs_mettl3)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_one_week_mettl3_vs_mettl3 <- nrow(unique_up_one_week_mettl3_vs_mettl3)
num_unique_down_one_week_mettl3_vs_mettl3 <- nrow(unique_down_one_week_mettl3_vs_mettl3)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
one_week_mettl3_vs_mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_one_week_mettl3_vs_mettl3 > 0.5 & padj_one_week_mettl3_vs_mettl3 <=0.05)

one_week_mettl3_vs_mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_one_week_mettl3_vs_mettl3 < 0.5 & padj_one_week_mettl3_vs_mettl3 <=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_one_week_mettl3_vs_mettl3_up <- nrow(one_week_mettl3_vs_mettl3_up)
num_one_week_mettl3_vs_mettl3_down <- nrow(one_week_mettl3_vs_mettl3_down)

# Print results
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in one_week_mettl3_vs_mettl3 (from shared genes):", num_Mettl3_RNAi_basal_vs_HS_up, "\n")
cat("Number of downregulated genes in one_week_mettl3_vs_mettl3 (from shared genes):", num_Mettl3_RNAi_basal_vs_HS_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in one_week_mettl3_vs_mettl3:", num_total_one_week_mettl3_vs_mettl3, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in one_week_mettl3_vs_mettl3:", num_unique_up_one_week_mettl3_vs_mettl3, "\n")
cat("Unique downregulated genes in one_week_mettl3_vs_mettl3:", num_unique_down_one_week_mettl3_vs_mettl3, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_one_week_mettl3_vs_mettl3 > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_one_week_mettl3_vs_mettl3 < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_one_week_mettl3_vs_mettl3 < 0 ~ "Up in METTL3, Down in Kan_one_week_mettl3",
      log2FoldChange_m6A < 0 & log2FoldChange_one_week_mettl3_vs_mettl3 > 0 ~ "Down in METTL3, Up in Kan_one_week_mettl3",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Kan_one_week_mettl3 <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Kan_one_week_mettl3")
Down_METTL3_Up_Kan_one_week_mettl3 <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Kan_one_week_mettl3")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_one_week_mettl3_vs_mettl3 %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_one_week_mettl3_vs_mettl3))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
#shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_one_week_mettl3_vs_mettl3, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Kan_one_week_mettl3 (", Up_METTL3_Down_Kan_one_week_mettl3, ")", sep=""), 
                                paste("Down in METTL3, Up in Kan_one_week_mettl3 (", Down_METTL3_Up_Kan_one_week_mettl3, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Kan_one_week_mettl3", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_one_week_mettl3_vs_mettl3), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Kan_one_week_mettl3)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40)) 

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_one_week_mettl3_vs_mettl3_Kan.txt", sep = "\t", quote = F)
```

# Kan et al: three_week_mettl3_vs_mettl3
```{r message=FALSE, warning=FALSE, echo=FALSE}

G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% three_week_mettl3_vs_mettl3$gene_name)

shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(three_week_mettl3_vs_mettl3 %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_three_week_mettl3_vs_mettl3 = current_symbol, log2FoldChange_three_week_mettl3_vs_mettl3 = log2FoldChange, padj_three_week_mettl3_vs_mettl3 = padj),
    by = "gene_name")

# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_three_week_mettl3_vs_mettl3 > 0.5, padj_three_week_mettl3_vs_mettl3 <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_three_week_mettl3_vs_mettl3 < -0.5, padj_three_week_mettl3_vs_mettl3 <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_three_week_mettl3_vs_mettl3 <- three_week_mettl3_vs_mettl3 %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_three_week_mettl3_vs_mettl3 <- three_week_mettl3_vs_mettl3 %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_three_week_mettl3_vs_mettl3 <- nrow(three_week_mettl3_vs_mettl3)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_three_week_mettl3_vs_mettl3 <- nrow(unique_up_three_week_mettl3_vs_mettl3)
num_unique_down_three_week_mettl3_vs_mettl3 <- nrow(unique_down_three_week_mettl3_vs_mettl3)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
three_week_mettl3_vs_mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_three_week_mettl3_vs_mettl3 > 0.5 & padj_three_week_mettl3_vs_mettl3 <=0.05)

three_week_mettl3_vs_mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_three_week_mettl3_vs_mettl3 < 0.5 & padj_three_week_mettl3_vs_mettl3 <=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_three_week_mettl3_vs_mettl3_up <- nrow(three_week_mettl3_vs_mettl3_up)
num_three_week_mettl3_vs_mettl3_down <- nrow(three_week_mettl3_vs_mettl3_down)

# Print results
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in three_week_mettl3_vs_mettl3 (from shared genes):", num_three_week_mettl3_vs_mettl3_up, "\n")
cat("Number of downregulated genes in three_week_mettl3_vs_mettl3 (from shared genes):", num_three_week_mettl3_vs_mettl3_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in three_week_mettl3_vs_mettl3:", num_total_three_week_mettl3_vs_mettl3, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in three_week_mettl3_vs_mettl3:", num_unique_up_three_week_mettl3_vs_mettl3, "\n")
cat("Unique downregulated genes in three_week_mettl3_vs_mettl3:", num_unique_down_three_week_mettl3_vs_mettl3, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_three_week_mettl3_vs_mettl3 > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_three_week_mettl3_vs_mettl3 < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_three_week_mettl3_vs_mettl3 < 0 ~ "Up in METTL3, Down in Kan_three_week_mettl3",
      log2FoldChange_m6A < 0 & log2FoldChange_three_week_mettl3_vs_mettl3 > 0 ~ "Down in METTL3, Up in Kan_three_week_mettl3",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_Kan_three_week_mettl3 <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Kan_three_week_mettl3")
Down_METTL3_Up_Kan_three_week_mettl3 <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Kan_three_week_mettl3")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_three_week_mettl3_vs_mettl3 %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_three_week_mettl3_vs_mettl3))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
#shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_three_week_mettl3_vs_mettl3  , color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Kan_one_week_mettl3 (", Up_METTL3_Down_Kan_three_week_mettl3, ")", sep=""), 
                                paste("Down in METTL3, Up in Kan_one_week_mettl3 (", Down_METTL3_Up_Kan_three_week_mettl3, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Kan_one_week_mettl3", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_three_week_mettl3_vs_mettl3), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Kan_three_week_mettl3)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40))

#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_three_week_mettl3_vs_mettl3_Kan.txt", sep = "\t", quote = F)
```

# Kan et al: one_week_ythdf_vs_ythdf
```{r message=FALSE, warning=FALSE, echo=FALSE}

# Define the shared genes based on `one_week_ythdf_vs_ythdf`
G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% one_week_ythdf_vs_ythdf$gene_name)

# Select genes with |log2FoldChange| >= 0.5 and padj <= 0.05
shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(one_week_ythdf_vs_ythdf %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_one_week_ythdf_vs_ythdf = current_symbol, log2FoldChange_one_week_ythdf_vs_ythdf = log2FoldChange, padj_one_week_ythdf_vs_ythdf = padj),
    by = "gene_name")


# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_one_week_ythdf_vs_ythdf > 0.5, padj_one_week_ythdf_vs_ythdf <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_one_week_ythdf_vs_ythdf < -0.5, padj_one_week_ythdf_vs_ythdf <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_one_week_ythdf_vs_ythdf <- one_week_ythdf_vs_ythdf %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_one_week_ythdf_vs_ythdf <- one_week_ythdf_vs_ythdf %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_one_week_ythdf_vs_ythdf <- nrow(one_week_ythdf_vs_ythdf)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_one_week_ythdf_vs_ythdf <- nrow(unique_up_one_week_ythdf_vs_ythdf)
num_unique_down_one_week_ythdf_vs_ythdf <- nrow(unique_down_one_week_ythdf_vs_ythdf)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
one_week_ythdf_vs_ythdf_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_one_week_ythdf_vs_ythdf > 0.5 & padj_one_week_ythdf_vs_ythdf <=0.05)

one_week_ythdf_vs_ythdf_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_one_week_ythdf_vs_ythdf < 0.5 & padj_one_week_ythdf_vs_ythdf <=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_one_week_ythdf_vs_ythdf_up <- nrow(one_week_ythdf_vs_ythdf_up)
num_one_week_ythdf_vs_ythdf_down <- nrow(one_week_ythdf_vs_ythdf_down)

# Print results
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in one_week_ythdf_vs_ythdf (from shared genes):", num_one_week_ythdf_vs_ythdf_up, "\n")
cat("Number of downregulated genes in one_week_ythdf_vs_ythdf (from shared genes):", num_one_week_ythdf_vs_ythdf_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in one_week_ythdf_vs_ythdf:", num_total_three_week_mettl3_vs_mettl3, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in one_week_ythdf_vs_ythdf:", num_unique_up_one_week_ythdf_vs_ythdf, "\n")
cat("Unique downregulated genes in one_week_ythdf_vs_ythdf:", num_unique_down_one_week_ythdf_vs_ythdf, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_one_week_ythdf_vs_ythdf > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_one_week_ythdf_vs_ythdf < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_one_week_ythdf_vs_ythdf < 0 ~ "Up in METTL3, Down in Kan_one_week_ythdf_vs_ythdf",
      log2FoldChange_m6A < 0 & log2FoldChange_one_week_ythdf_vs_ythdf > 0 ~ "Down in METTL3, Up in Kan_one_week_ythdf_vs_ythdf",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_one_week_ythdf_vs_ythdf <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Kan_one_week_ythdf_vs_ythdf")
Down_METTL3_Up_one_week_ythdf_vs_ythdf <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Kan_one_week_ythdf_vs_ythdf")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_one_week_ythdf_vs_ythdf %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_one_week_ythdf_vs_ythdf))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
#shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = log2FoldChange_one_week_ythdf_vs_ythdf  , color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Kan_one_week_mettl3 (", Up_METTL3_Down_one_week_ythdf_vs_ythdf, ")", sep=""), 
                                paste("Down in METTL3, Up in Kan_one_week_mettl3 (", Down_METTL3_Up_one_week_ythdf_vs_ythdf, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Kan_one_week_ythdf_vs_ythdf", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$log2FoldChange_one_week_ythdf_vs_ythdf), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Kan_one_week_ythdf_vs_ythdf)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40))
#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_one_week_ythdf_vs_ythdf_Kan.txt", sep = "\t", quote = F)
```


# Kan et al: three_week_ythdf_vs_ythdf
```{r message=FALSE, warning=FALSE, echo=FALSE}

# Define the shared genes based on `one_week_ythdf_vs_ythdf`
G_R_shared_gene <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% three_week_ythdf_vs_ythdf$gene_name)

# Select genes with |log2FoldChange| >= 0.5 and padj <= 0.05
shared_genes_with_metrics <- res_S2_m6A %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_m6A = current_symbol, log2FoldChange_m6A = log2FoldChange, padj_m6a = padj) %>%
  inner_join(three_week_ythdf_vs_ythdf %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_three_week_ythdf_vs_ythdf = current_symbol, log2FoldChange_three_week_ythdf_vs_ythdf = log2FoldChange, padj_three_week_ythdf_vs_ythdf = padj),
    by = "gene_name")


# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5, padj_m6a <=0.05 & log2FoldChange_three_week_ythdf_vs_ythdf > 0.5, padj_three_week_ythdf_vs_ythdf <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < -0.5, padj_m6a <=0.05 & log2FoldChange_three_week_ythdf_vs_ythdf < -0.5, padj_three_week_ythdf_vs_ythdf <=0.05)

# Unique genes
unique_up_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_m6A <- res_S2_m6A %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_three_week_ythdf_vs_ythdf <- three_week_ythdf_vs_ythdf %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_three_week_ythdf_vs_ythdf <- three_week_ythdf_vs_ythdf %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

# Stats
num_total_m6A <- nrow(res_S2_m6A)
num_total_three_week_ythdf_vs_ythdf <- nrow(three_week_ythdf_vs_ythdf)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_m6A <- nrow(unique_up_m6A)
num_unique_down_m6A <- nrow(unique_down_m6A)

num_unique_up_three_week_ythdf_vs_ythdf <- nrow(unique_up_three_week_ythdf_vs_ythdf)
num_unique_down_three_week_ythdf_vs_ythdf <- nrow(unique_down_three_week_ythdf_vs_ythdf)

# Upregulated and Downregulated in METTL3
mettl3_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A > 0.5 & padj_m6a<=0.05)

mettl3_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_m6A < 0.5 & padj_m6a<=0.05)

# Upregulated and Downregulated in YTHDC1
three_week_ythdf_vs_ythdf_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_three_week_ythdf_vs_ythdf > 0.5 & padj_three_week_ythdf_vs_ythdf <=0.05)

three_week_ythdf_vs_ythdf_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_three_week_ythdf_vs_ythdf < 0.5 & padj_three_week_ythdf_vs_ythdf <=0.05)

# Count numbers
num_mettl3_up <- nrow(mettl3_up)
num_mettl3_down <- nrow(mettl3_down)
num_three_week_ythdf_vs_ythdf_up <- nrow(three_week_ythdf_vs_ythdf_up)
num_three_week_ythdf_vs_ythdf_down <- nrow(three_week_ythdf_vs_ythdf_down)

# Print results
cat("Number of upregulated genes in METTL3 (from shared genes):", num_mettl3_up, "\n")
cat("Number of downregulated genes in METTL3 (from shared genes):", num_mettl3_down, "\n")
cat("Number of upregulated genes in three_week_ythdf_vs_ythdf (from shared genes):", num_three_week_ythdf_vs_ythdf_up, "\n")
cat("Number of downregulated genes in three_week_ythdf_vs_ythdf (from shared genes):", num_three_week_ythdf_vs_ythdf_down, "\n")
# Print statistics
cat("Total genes in METTL3:", num_total_m6A, "\n")
cat("Total genes in three_week_ythdf_vs_ythdf:", num_total_three_week_mettl3_vs_mettl3, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in METTL3:", num_unique_up_m6A, "\n")
cat("Unique downregulated genes in METTL3:", num_unique_down_m6A, "\n")

cat("Unique upregulated genes in three_week_ythdf_vs_ythdf:", num_unique_up_three_week_ythdf_vs_ythdf, "\n")
cat("Unique downregulated genes in three_week_ythdf_vs_ythdf:", num_unique_down_three_week_ythdf_vs_ythdf, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_m6A > 0 & log2FoldChange_three_week_ythdf_vs_ythdf > 0 ~ "Shared Upregulated",
      log2FoldChange_m6A < 0 & log2FoldChange_three_week_ythdf_vs_ythdf < 0 ~ "Shared Downregulated",
      log2FoldChange_m6A > 0 & log2FoldChange_three_week_ythdf_vs_ythdf < 0 ~ "Up in METTL3, Down in Kan_three_week_ythdf_vs_ythdf",
      log2FoldChange_m6A < 0 & log2FoldChange_three_week_ythdf_vs_ythdf > 0 ~ "Down in METTL3, Up in Kan_three_week_ythdf_vs_ythdf",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_METTL3_Down_three_week_ythdf_vs_ythdf <- sum(shared_genes_with_metrics$Category == "Up in METTL3, Down in Kan_three_week_ythdf_vs_ythdf")
Down_METTL3_Up_three_week_ythdf_vs_ythdf <- sum(shared_genes_with_metrics$Category == "Down in METTL3, Up in Kan_three_week_ythdf_vs_ythdf")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Myo61F", "Sxl", "rgn", "Hsp26", "Prosap", "fl(2)d", "Rdh1", "mfas", "puml")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_m6A %in% highlight_genes | symbol_three_week_ythdf_vs_ythdf %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_m6A, symbol_three_week_ythdf_vs_ythdf))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
#shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_m6A, y = padj_three_week_ythdf_vs_ythdf  , color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in METTL3, Down in Kan_one_week_mettl3 (", Up_METTL3_Down_three_week_ythdf_vs_ythdf, ")", sep=""), 
                                paste("Down in METTL3, Up in Kan_one_week_mettl3 (", Down_METTL3_Up_three_week_ythdf_vs_ythdf, ")", sep=""))) +
  xlim(-7, 7) +
  ylim(-7, 7) +
  labs(title = paste( "shared genes between METTL3-KO/S2S and Kan_one_week_ythdf_vs_ythdf", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_m6A,
                                                        shared_genes_with_metrics$padj_three_week_ythdf_vs_ythdf), 2), ")"),
       x = "Log2 Fold Change (George_METTL3-KO)",
       y = "Log2 Fold Change (Kan_one_week_ythdf_vs_ythdf)",
       color = "") +
  theme(text = element_text(size = 14), plot.margin = margin(10, 10, 10, 40))
#write.table(shared_genes_with_metrics, file = "METTL3_KO_shared_genes_three_week_ythdf_vs_ythdf_Kan.txt", sep = "\t", quote = F)
```

    
    
    
    res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean 
    res_S2.GFP_S2.DC1.GOF_clean 
    res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean 
    
    
# Kan et al: res_S2.GFP_S2.DC1.GOF_clean
```{r message=FALSE, warning=FALSE, echo=FALSE}
#res_S2.GFP_S2.DC1.GOF_clean$gene_name = rownames(res_S2.GFP_S2.DC1.GOF_clean)
#res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean$gene_name = rownames(res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean)
#res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$gene_name = rownames(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean)

G_R_shared_gene <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(gene_name %in% res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$gene_name, padj <= 0.05, abs(log2FoldChange) >= 0.5) %>%
  dplyr::filter(gene_name %in% res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$gene_name, padj <= 0.05, abs(log2FoldChange) >= 0.5)

# Select genes with |log2FoldChange| >= 0.5 and padj <= 0.05
shared_genes_with_metrics <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_S2.GFP_S2 = current_symbol, log2FoldChange_S2.GFP_S2 = log2FoldChange, padj_S2.GFP_S2 = padj) %>%
  inner_join(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_mettl3.KO.GFP = current_symbol, log2FoldChange_mettl3.KO.GFP = log2FoldChange, padj_mettl3.KO.GFP = padj),
    by = "gene_name")


# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 > 0.5, padj_S2.GFP_S2 <=0.05 & log2FoldChange_mettl3.KO.GFP > 0.5, padj_mettl3.KO.GFP <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 < -0.5, padj_S2.GFP_S2 <=0.05 & log2FoldChange_mettl3.KO.GFP < -0.5, padj_mettl3.KO.GFP <=0.05)

# Unique genes
unique_up_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF_clean %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF_clean %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_mettl3.KO.GFP_mettl3.KO.DC1.GOF <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_mettl3.KO.GFP_mettl3.KO.DC1.GOF <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

# Stats
num_total_S2.GFP_S2.DC1.GOF <- nrow(res_S2.GFP_S2.DC1.GOF)
num_total_mettl3.KO.GFP_mettl3.KO.DC1.GOF <- nrow(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_S2.GFP_S2.DC1.GOF <- nrow(unique_up_S2.GFP_S2.DC1.GOF)
num_unique_down_S2.GFP_S2.DC1.GOF <- nrow(unique_down_S2.GFP_S2.DC1.GOF)

num_unique_up_mettl3.KO.GFP_mettl3.KO.DC1.GOF <- nrow(unique_up_mettl3.KO.GFP_mettl3.KO.DC1.GOF)
num_unique_down_mettl3.KO.GFP_mettl3.KO.DC1.GOF <- nrow(unique_down_mettl3.KO.GFP_mettl3.KO.DC1.GOF)

# Upregulated and Downregulated in METTL3
S2.GFP_S2_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 > 0.5 & padj_S2.GFP_S2<=0.05)

S2.GFP_S2_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 < -0.5 & padj_S2.GFP_S2<=0.05)

# Upregulated and Downregulated in YTHDC1

mettl3.KO.GFP_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_mettl3.KO.GFP > 0.5 & padj_mettl3.KO.GFP <=0.05)

mettl3.KO.GFP_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_mettl3.KO.GFP < 0.5 & padj_mettl3.KO.GFP <=0.05)

# Count numbers
num_S2.GFP_S2_up <- nrow(S2.GFP_S2_up)
num_S2.GFP_S2_down <- nrow(S2.GFP_S2_down)
num_mettl3.KO.GFP_up <- nrow(mettl3.KO.GFP_up)
num_mettl3.KO.GFP_down <- nrow(mettl3.KO.GFP_down)

# Print results
cat("Number of upregulated genes in S2S-YTHDC1-GoF/S2S-GFP (from shared genes):", num_S2.GFP_S2_up, "\n")
cat("Number of downregulated genes in S2S-YTHDC1-GoF/S2S-GFP (from shared genes):", num_S2.GFP_S2_down, "\n")
cat("Number of upregulated genes in METTL3-KO-YTHDC1/METTL3-KO-GFP-GoF (from shared genes):", num_mettl3.KO.GFP_up, "\n")
cat("Number of downregulated genes in METTL3-KO-YTHDC1/METTL3-KO-GFP-GoF (from shared genes):", num_mettl3.KO.GFP_down, "\n")
# Print statistics
cat("Total genes in S2S-YTHDC1-GoF/S2S-GFP:", num_total_S2.GFP_S2.DC1.GOF, "\n")
cat("Total genes in METTL3-KO-YTHDC1/METTL3-KO-GFP-GoF:", num_total_mettl3.KO.GFP_mettl3.KO.DC1.GOF, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in S2S-YTHDC1-GoF/S2S-GFP:", num_unique_up_S2.GFP_S2.DC1.GOF, "\n")
cat("Unique downregulated genes in S2S-YTHDC1-GoF/S2S-GFP:", num_unique_down_S2.GFP_S2.DC1.GOF, "\n")

cat("Unique upregulated genes in METTL3-KO-YTHDC1/METTL3-KO-GFP-GoF:", num_unique_up_mettl3.KO.GFP_mettl3.KO.DC1.GOF, "\n")
cat("Unique downregulated genes in METTL3-KO-YTHDC1/METTL3-KO-GFP-GoF:", num_unique_down_mettl3.KO.GFP_mettl3.KO.DC1.GOF, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_S2.GFP_S2 > 0 & log2FoldChange_mettl3.KO.GFP > 0 ~ "Shared Upregulated",
      log2FoldChange_S2.GFP_S2 < 0 & log2FoldChange_mettl3.KO.GFP < 0 ~ "Shared Downregulated",
      log2FoldChange_S2.GFP_S2 > 0 & log2FoldChange_mettl3.KO.GFP < 0 ~ "Up in S2S-YTHDC1-GoF/S2S-GFP, Down in METTL3-KO-YTHDC1/METTL3-KO-GFP",
      log2FoldChange_S2.GFP_S2 < 0 & log2FoldChange_mettl3.KO.GFP > 0 ~ "Down in S2S-YTHDC1-GoF/S2S-GFP, Up in METTL3-KO-YTHDC1/METTL3-KO-GFP",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_S2S_YTHDC1_GoF_S2S_GFP_Down_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF <- sum(shared_genes_with_metrics$Category == "Up in S2S-YTHDC1-GoF/S2S-GFP, Down in METTL3-KO-YTHDC1/METTL3-KO-GFP")
Down_S2S_YTHDC1_GoF_S2S_GFP_Up_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF <- sum(shared_genes_with_metrics$Category == "Down in S2S-YTHDC1-GoF/S2S-GFP, Up in METTL3-KO-YTHDC1/METTL3-KO-GFP")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Ythdc1", "Arc1", "AOX1", "nvy", "bun", "brp", "Gr5a", "Tsp", "Zip89B")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_S2.GFP_S2 %in% highlight_genes | symbol_mettl3.KO.GFP %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_S2.GFP_S2, symbol_mettl3.KO.GFP))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
#shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_S2.GFP_S2, y = log2FoldChange_mettl3.KO.GFP  , color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in S2S-YTHDC1-GoF/S2S-GFP, Down in METTL3-KO-YTHDC1/METTL3-KO-GFP (", Up_S2S_YTHDC1_GoF_S2S_GFP_Down_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF, ")", sep=""), 
                                paste("Down in S2S-YTHDC1-GoF/S2S-GFP, Up in METTL3-KO-YTHDC1/METTL3-KO-GFP (", Down_S2S_YTHDC1_GoF_S2S_GFP_Up_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF, ")", sep=""))) +
  labs(title = paste( "shared genes between S2S-YTHDC1-GoF/S2S-GFP and METTL3-KO-YTHDC1/METTL3-KO-GFP", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_S2.GFP_S2, shared_genes_with_metrics$log2FoldChange_mettl3.KO.GFP), 2), ")"),
       x = "Log2 Fold Change (S2S-YTHDC1-GoF/S2S-GFP)",
       y = "Log2 Fold Change (METTL3-KO-YTHDC1/METTL3-KO-GFP)",
       color = "")
#write.table(shared_genes_with_metrics, file = "S2.GFP_S2_shared_genes_METTL3-KO-YTHDC1_Kan.txt", sep = "\t", quote = F)
```

# Kan et al: res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean
```{r message=FALSE, warning=FALSE, echo=FALSE}

G_R_shared_gene <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(gene_name %in% res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$gene_name)

# Select genes with |log2FoldChange| >= 0.5 and padj <= 0.05
shared_genes_with_metrics <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
  dplyr::select(gene_name, symbol_S2.GFP_S2 = current_symbol, log2FoldChange_S2.GFP_S2 = log2FoldChange, padj_S2.GFP_S2 = padj) %>%
  inner_join(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
      dplyr::filter(gene_name %in% G_R_shared_gene$gene_name, abs(log2FoldChange) >= 0.5, padj <= 0.05) %>%
      dplyr::select(gene_name, symbol_S2.YTHDC_gof = current_symbol, log2FoldChange_S2.YTHDC_gof = log2FoldChange, padj_S2.YTHDC_gof = padj),
    by = "gene_name")


# Count shared upregulated and downregulated genes
shared_upregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 > 0.5, padj_S2.GFP_S2 <=0.05 & log2FoldChange_S2.YTHDC_gof > 0.5, padj_S2.YTHDC_gof <=0.05)

shared_downregulated <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 < -0.5, padj_S2.GFP_S2 <=0.05 & log2FoldChange_S2.YTHDC_gof < -0.5, padj_S2.YTHDC_gof <=0.05)

# Unique genes
unique_up_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

unique_up_S2.YTHDC_gof <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange > 0.5 & padj <= 0.05)

unique_down_S2.YTHDC_gof <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(!(gene_name %in% G_R_shared_gene$gene_name) & log2FoldChange < -0.5 & padj <= 0.05)

# Stats
num_total_S2.GFP_S2.DC1.GOF <- nrow(res_S2.GFP_S2.DC1.GOF)
num_total_S2.YTHDC_gof <- nrow(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF)
num_shared <- nrow(shared_genes_with_metrics)

num_shared_up <- nrow(shared_upregulated)
num_shared_down <- nrow(shared_downregulated)

num_unique_up_S2.GFP_S2.DC1.GOF <- nrow(unique_up_S2.GFP_S2.DC1.GOF)
num_unique_down_S2.GFP_S2.DC1.GOF <- nrow(unique_down_S2.GFP_S2.DC1.GOF)

num_unique_up_S2.YTHDC_gof <- nrow(unique_up_S2.YTHDC_gof)
num_unique_down_S2.YTHDC_gof <- nrow(unique_down_S2.YTHDC_gof)

# Upregulated and Downregulated in METTL3
S2.GFP_S2_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 > 0.5 & padj_S2.GFP_S2<=0.05)

S2.GFP_S2_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.GFP_S2 < 0.5 & padj_S2.GFP_S2<=0.05)

# Upregulated and Downregulated in YTHDC1

S2.YTHDC_gof_up <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.YTHDC_gof > 0.5 & padj_S2.YTHDC_gof <=0.05)

S2.YTHDC_gof_down <- shared_genes_with_metrics %>%
  dplyr::filter(log2FoldChange_S2.YTHDC_gof < 0.5 & padj_S2.YTHDC_gof <=0.05)

# Count numbers
num_S2.GFP_S2_up <- nrow(S2.GFP_S2_up)
num_S2.GFP_S2_down <- nrow(S2.GFP_S2_down)
num_S2.YTHDC_gof_up <- nrow(S2.YTHDC_gof_up)
num_S2.YTHDC_gof_down <- nrow(S2.YTHDC_gof_down)

# Print results
cat("Number of upregulated genes in S2S-YTHDC1-GoF/S2S-GFP (from shared genes):", num_S2.GFP_S2_up, "\n")
cat("Number of downregulated genes in S2S-YTHDC1-GoF/S2S-GFP (from shared genes):", num_S2.GFP_S2_down, "\n")
cat("Number of upregulated genes in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1 (from shared genes):", num_S2.YTHDC_gof_up, "\n")
cat("Number of downregulated genes in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1 (from shared genes):", num_S2.YTHDC_gof_down, "\n")
# Print statistics
cat("Total genes in S2S-YTHDC1-GoF/S2S-GFP:", num_total_S2.GFP_S2.DC1.GOF, "\n")
cat("Total genes in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1:", num_total_S2.YTHDC_gof, "\n")
cat("Shared genes:", num_shared, "\n")

cat("Shared upregulated genes:", num_shared_up, "\n")
cat("Shared downregulated genes:", num_shared_down, "\n")

cat("Unique upregulated genes in S2S-YTHDC1-GoF/S2S-GFP:", num_unique_up_S2.GFP_S2.DC1.GOF, "\n")
cat("Unique downregulated genes in S2S-YTHDC1-GoF/S2S-GFP:", num_unique_down_S2.GFP_S2.DC1.GOF, "\n")

cat("Unique upregulated genes in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1:", num_unique_up_S2.YTHDC_gof, "\n")
cat("Unique downregulated genes in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1:", num_unique_down_S2.YTHDC_gof, "\n")

```

```{r}

shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
      log2FoldChange_S2.GFP_S2 > 0 & log2FoldChange_S2.YTHDC_gof > 0 ~ "Shared Upregulated",
      log2FoldChange_S2.GFP_S2 < 0 & log2FoldChange_S2.YTHDC_gof < 0 ~ "Shared Downregulated",
      log2FoldChange_S2.GFP_S2 > 0 & log2FoldChange_S2.YTHDC_gof < 0 ~ "Up in S2S-YTHDC1-GoF/S2S-GFP, Down in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF",
      log2FoldChange_S2.GFP_S2 < 0 & log2FoldChange_S2.YTHDC_gof > 0 ~ "Down in S2S-YTHDC1-GoF/S2S-GFP, Up in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF",
      TRUE ~ "Other"))



Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_S2S_YTHDC1_GoF_S2S_GFP_Down_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF <- sum(shared_genes_with_metrics$Category == "Up in S2S-YTHDC1-GoF/S2S-GFP, Down in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF")
Down_S2S_YTHDC1_GoF_S2S_GFP_Up_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF <- sum(shared_genes_with_metrics$Category == "Down in S2S-YTHDC1-GoF/S2S-GFP, Up in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF")

# Step 1: Define the highlighted genes
highlight_genes <- c("Mettl3", "Ythdc1", "Arc1", "AOX1", "nvy", "bun", "brp", "Gr5a", "Tsp", "Zip89B")

# Step 2: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(symbol_S2.GFP_S2 %in% highlight_genes | symbol_S2.YTHDC_gof %in% highlight_genes) %>%
  mutate(label = coalesce(symbol_S2.GFP_S2, symbol_S2.YTHDC_gof))  # Use the non-NA symbol for labeling

# Step 3: Create the plot with highlighted gene labels
#shared_genes_with_metrics = na.omit(shared_genes_with_metrics)
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_S2.GFP_S2, y = log2FoldChange_S2.YTHDC_gof, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +
  theme_minimal() +
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in S2S-YTHDC1-GoF/S2S-GFP, Down in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF (", Up_S2S_YTHDC1_GoF_S2S_GFP_Down_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF, ")", sep=""), 
                                paste("Down in S2S-YTHDC1-GoF/S2S-GFP, Up in S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF (", Down_S2S_YTHDC1_GoF_S2S_GFP_Up_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF, ")", sep=""))) +
  labs(title = paste( "shared genes between S2S-YTHDC1-GoF/S2S-GFP and S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF", 
                      "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_S2.GFP_S2, shared_genes_with_metrics$log2FoldChange_S2.YTHDC_gof), 2), ")"),
       x = "Log2 Fold Change (S2S-YTHDC1-GoF/S2S-GFP)",
       y = "Log2 Fold Change (S2S-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF)",
       color = "")

#write.table(shared_genes_with_metrics, file = "S2S-YTHDC1-GoF_shared_genes_METTL3-KO-YTHDC1-GoF_Kan.txt", sep = "\t", quote = F)
```


    


    Venn diagram METTL3-KO-YTHDC1-GoF vs S2-YTHDC-GoF, S2-YTHDC1-GoF vs S2-GFP and METTL3-KO-YTHDC1-GoF vs METTL3-KO-GFP
```{r}
library(ggvenn)

# Add gene names to the clean data frames
#res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$gene_name <- row.names(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF)
#res_S2.GFP_S2.DC1.GOF$gene_name <- row.names(res_S2.GFP_S2.DC1.GOF)
#res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean$gene_name <- row.names(res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean)

# Extract upregulated and downregulated genes for each dataset
S2_YTHDC_up <- res_S2.GFP_S2.DC1.GOF$validated_id[res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$log2FoldChange > 0]
S2_YTHDC_down <- res_S2.GFP_S2.DC1.GOF$validated_id[res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$log2FoldChange < 0]

S2_GFP_up <- res_S2.GFP_S2.DC1.GOF$validated_id[res_S2.GFP_S2.DC1.GOF$log2FoldChange > 0]
S2_GFP_down <- res_S2.GFP_S2.DC1.GOF$validated_id[res_S2.GFP_S2.DC1.GOF$log2FoldChange < 0]

mettl3_KO_up <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$validated_id[res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$log2FoldChange > 0]
mettl3_KO_down <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$validated_id[res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$log2FoldChange < 0]

# Create gene sets for upregulated and downregulated genes
gene_sets_up <- list(
  "METTL3-KO-YTHDC1-GOF vs S2-YTHDC-GoF (Upregulated)" = S2_YTHDC_up,
  "S2-YTHDC1-GoF vs S2-GFP (Upregulated)" = S2_GFP_up,
  "METTL3-KO-YTHDC1-GOF vs METTL3-KO-GFP (Upregulated)" = mettl3_KO_up)

gene_sets_down <- list(
  "METTL3-KO-YTHDC1-GOF vs S2-YTHDC-GoF (Downregulated)" = S2_YTHDC_down,
  "S2-YTHDC1-GoF vs S2-GFP (Downregulated)" = S2_GFP_down,
  "METTL3-KO-YTHDC1-GOF vs METTL3-KO-GFP (Downregulated)" = mettl3_KO_down)

# Identify shared and unique genes for upregulated
shared_genes_up <- Reduce(intersect, gene_sets_up)
unique_genes_up <- lapply(gene_sets_up, function(x) setdiff(x, shared_genes_up))

# Identify shared and unique genes for downregulated
shared_genes_down <- Reduce(intersect, gene_sets_down)
unique_genes_down <- lapply(gene_sets_down, function(x) setdiff(x, shared_genes_down))

# Calculate counts and percentages for upregulated genes
total_up_genes <- length(unlist(gene_sets_up))
shared_up_count <- length(shared_genes_up)
unique_up_counts <- sapply(unique_genes_up, length)
shared_up_percentage <- (shared_up_count / total_up_genes) * 100
unique_up_percentages <- (unique_up_counts / total_up_genes) * 100

# Calculate counts and percentages for downregulated genes
total_down_genes <- length(unlist(gene_sets_down))
shared_down_count <- length(shared_genes_down)
unique_down_counts <- sapply(unique_genes_down, length)
shared_down_percentage <- (shared_down_count / total_down_genes) * 100
unique_down_percentages <- (unique_down_counts / total_down_genes) * 100

# Print results for upregulated genes
cat("Upregulated Genes:\n")
cat("Total genes:", total_up_genes, "\n")
cat("Shared genes:", shared_up_count, "(", round(shared_up_percentage, 2), "%)\n")
cat("Unique genes per group:\n")
for (i in seq_along(unique_up_counts)) {
  cat(names(unique_up_counts)[i], ":", unique_up_counts[i], "(", round(unique_up_percentages[i], 2), "%)\n")
}

# Print results for downregulated genes
cat("\nDownregulated Genes:\n")
cat("Total genes:", total_down_genes, "\n")
cat("Shared genes:", shared_down_count, "(", round(shared_down_percentage, 2), "%)\n")
cat("Unique genes per group:\n")
for (i in seq_along(unique_down_counts)) {
  cat(names(unique_down_counts)[i], ":", unique_down_counts[i], "(", round(unique_down_percentages[i], 2), "%)\n")
}

# Optional: Generate Venn diagrams
ggvenn(gene_sets_up, 
       fill_color = c("#E69F00", "#56B4E9", "red"), 
       stroke_size = 0.5, 
       text_size = 5, 
       show_percentage = FALSE, 
       auto_scale = FALSE) + 
  labs(title = "Venn Diagram for Upregulated Genes") + 
  theme(plot.title = element_text(hjust = 0.5, size = 16))

ggvenn(gene_sets_down, 
       fill_color = c("#E69F00", "#56B4E9", "red"), 
       stroke_size = 0.5, 
       text_size = 5, 
       show_percentage = FALSE, 
       auto_scale = FALSE) + 
  labs(title = "Venn Diagram for Downregulated Genes") + 
  theme(plot.title = element_text(hjust = 0.5, size = 16))

# Print shared and unique genes
#cat("Shared upregulated genes:\n", shared_genes_up, "\n")
#cat("Shared downregulated genes:\n", shared_genes_down, "\n")

```




```{r}
# Extract gene names from each dataset
genes_1 <- res_S2.GFP_S2.DC1.GOF$validated_id
genes_2 <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$validated_id
genes_3 <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$validated_id

# Find common genes between all three lists
shared_genes <- Reduce(intersect, list(genes_1, genes_2, genes_3))
num_shared_gene <- length(shared_genes)

# Define thresholds
padj_threshold <- 0.05
logFC_threshold <- 1


# Filter and count up/downregulated genes per dataset using validated_id filtering
upregulated_1 <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(validated_id %in% shared_genes, log2FoldChange > logFC_threshold, padj < padj_threshold)

downregulated_1 <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(validated_id %in% shared_genes, log2FoldChange < -logFC_threshold, padj < padj_threshold)

upregulated_2 <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(validated_id %in% shared_genes, log2FoldChange > logFC_threshold, padj < padj_threshold)

downregulated_2 <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(validated_id %in% shared_genes, log2FoldChange < -logFC_threshold, padj < padj_threshold)

upregulated_3 <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(validated_id %in% shared_genes, log2FoldChange > logFC_threshold, padj < padj_threshold)

downregulated_3 <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(validated_id %in% shared_genes, log2FoldChange < -logFC_threshold, padj < padj_threshold)

# Count the number of upregulated and downregulated shared genes in each dataset
num_up_1 <- nrow(upregulated_1)
num_down_1 <- nrow(downregulated_1)

num_up_2 <- nrow(upregulated_2)
num_down_2 <- nrow(downregulated_2)

num_up_3 <- nrow(upregulated_3)
num_down_3 <- nrow(downregulated_3)

# Print the results
cat("Number of shared genes between all three data sets:", num_shared_gene, "\n\n")

cat("Dataset 1 (res_S2.GFP_S2.DC1.GOF):\n")
cat("  Upregulated:", num_up_1, "\n")
cat("  Downregulated:", num_down_1, "\n\n")

cat("Dataset 2 (res_mettl3_KO_GFP_mettl3_KO_DC1_GOF):\n")
cat("  Upregulated:", num_up_2, "\n")
cat("  Downregulated:", num_down_2, "\n\n")

cat("Dataset 3 (res_S2.YTHDC_gof_mettl3.KO.DC1.GOF):\n")
cat("  Upregulated:", num_up_3, "\n")
cat("  Downregulated:", num_down_3, "\n")


```

```{r}
# Define threshold values
padj_threshold <- 0.05
logFC_threshold <- 1

# Get upregulated genes (log2FoldChange > 0.5 and padj < 0.05)
upregulated_list <- list(
  "S2-YTHDC1-GoF/S2-GFP" = res_S2.GFP_S2.DC1.GOF %>%
    dplyr::filter(log2FoldChange > logFC_threshold & padj < padj_threshold) %>%
    pull(validated_id),

  "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP" = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
    dplyr::filter(log2FoldChange > logFC_threshold & padj < padj_threshold) %>%
    pull(validated_id),

  "S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF" = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
    dplyr::filter(log2FoldChange > logFC_threshold & padj < padj_threshold) %>%
    pull(validated_id))

# Get downregulated genes (log2FoldChange < -0.5 and padj < 0.05)
downregulated_list <- list(
  "S2-YTHDC1-GoF/S2-GFP" = res_S2.GFP_S2.DC1.GOF %>%
    dplyr::filter(log2FoldChange < -logFC_threshold & padj < padj_threshold) %>%
    pull(validated_id),

  "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP" = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
    dplyr::filter(log2FoldChange < -logFC_threshold & padj < padj_threshold) %>%
    pull(validated_id),

  "S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF" = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
    dplyr::filter(log2FoldChange < -logFC_threshold & padj < padj_threshold) %>%
    pull(validated_id))


ggvenn(upregulated_list, show_percentage = F,
       fill_color = c("#FF9999", "#99CCFF", "#99FF99"),
       stroke_size = 0.5,
       set_name_size = 5) +
  ggtitle("Upregulated Genes (padj < 0.05, log2FC > 2)")


ggvenn(downregulated_list, show_percentage = F,
       fill_color = c("#FF9999", "#99CCFF", "#99FF99"),
       stroke_size = 0.5,
       set_name_size = 5) +
  ggtitle("Downregulated Genes (padj < 0.05, log2FC < 2)")

```


```{r}
library(dplyr)

# Define threshold values
padj_threshold <- 0.05
logFC_threshold <- 0.5

# Get upregulated genes (log2FoldChange > 0.5 and padj < 0.05)
upregulated_list <- list(
  "S2-YTHDC1-GoF/S2-GFP" = res_S2.GFP_S2.DC1.GOF_clean %>%
    dplyr::filter(log2FoldChange > logFC_threshold & padj < padj_threshold) %>%
    pull(gene_name),

  "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP" = res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean %>%
    dplyr::filter(log2FoldChange > logFC_threshold & padj < padj_threshold) %>%
    pull(gene_name),

  "S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF" = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean %>%
    dplyr::filter(log2FoldChange > logFC_threshold & padj < padj_threshold) %>%
    pull(gene_name))

# Get downregulated genes (log2FoldChange < -0.5 and padj < 0.05)
downregulated_list <- list(
  "S2-YTHDC1-GoF/S2-GFP" = res_S2.GFP_S2.DC1.GOF_clean %>%
    dplyr::filter(log2FoldChange < -logFC_threshold & padj < padj_threshold) %>%
    pull(gene_name),

  "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP" = res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean %>%
    dplyr::filter(log2FoldChange < -logFC_threshold & padj < padj_threshold) %>%
    pull(gene_name),

  "S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF" = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean %>%
    dplyr::filter(log2FoldChange < -logFC_threshold & padj < padj_threshold) %>%
    pull(gene_name))

# Function to identify shared and exclusive genes
identify_shared_exclusive_genes <- function(gene_list) {
  # Extract gene sets
  genes_1 <- gene_list[[1]]
  genes_2 <- gene_list[[2]]
  genes_3 <- gene_list[[3]]

  # Find shared and exclusive genes
  shared_all <- Reduce(intersect, gene_list)  # Genes shared by all three datasets
  shared_12 <- setdiff(intersect(genes_1, genes_2), shared_all)  # Shared by 1 and 2 only
  shared_13 <- setdiff(intersect(genes_1, genes_3), shared_all)  # Shared by 1 and 3 only
  shared_23 <- setdiff(intersect(genes_2, genes_3), shared_all)  # Shared by 2 and 3 only
  exclusive_1 <- setdiff(genes_1, union(genes_2, genes_3))  # Exclusive to 1
  exclusive_2 <- setdiff(genes_2, union(genes_1, genes_3))  # Exclusive to 2
  exclusive_3 <- setdiff(genes_3, union(genes_1, genes_2))  # Exclusive to 3

  # Combine results into a data frame
  result <- data.frame(
    gene_name = c(shared_all, shared_12, shared_13, shared_23, exclusive_1, exclusive_2, exclusive_3),
    category = c(
      rep("Shared by all", length(shared_all)),
      rep("Shared by S2-YTHDC1-GoF/S2-GFP and METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP", length(shared_12)),
      rep("Shared by S2-YTHDC1-GoF/S2-GFP and S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF", length(shared_13)),
      rep("Shared by METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP and S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF", length(shared_23)),
      rep("Exclusive to S2-YTHDC1-GoF/S2-GFP", length(exclusive_1)),
      rep("Exclusive to METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP", length(exclusive_2)),
      rep("Exclusive to S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF", length(exclusive_3))))

  return(result)
}

# Identify shared and exclusive genes for upregulated and downregulated lists
upregulated_shared_exclusive <- identify_shared_exclusive_genes(upregulated_list)
downregulated_shared_exclusive <- identify_shared_exclusive_genes(downregulated_list)

# Save the results to files
#write.table(upregulated_shared_exclusive, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/novel/upregulated_shared_exclusive_genes_all_3.txt", sep = "\t", quote = FALSE, row.names = FALSE)
#write.table(downregulated_shared_exclusive, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/novel/downregulated_shared_exclusive_genes_all_3.txt", sep = "\t", quote = FALSE, row.names = FALSE)

# Print the results
cat("Upregulated shared and exclusive genes:\n")
print(upregulated_shared_exclusive)

cat("\nDownregulated shared and exclusive genes:\n")
print(downregulated_shared_exclusive)
```

```{r}
library(ggplot2)
library(ggrepel)
library(dplyr)

# Step 1: Extract shared genes with metrics from the datasets
shared_genes_with_metrics <- data.frame(
  gene_name = shared_genes,
  log2FoldChange_S2.GFP_S2 = res_S2.GFP_S2.DC1.GOF_clean[shared_genes, "log2FoldChange"],
  padj_S2.GFP_S2 = res_S2.GFP_S2.DC1.GOF_clean[shared_genes, "padj"],
  log2FoldChange_mettl3.KO.GFP = res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean[shared_genes, "log2FoldChange"],
  padj_mettl3.KO.GFP = res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean[shared_genes, "padj"])

# Step 2: Categorize genes based on log2 fold changes
shared_genes_with_metrics <- shared_genes_with_metrics %>%
  mutate(Category = case_when(
    log2FoldChange_S2.GFP_S2 > 0 & log2FoldChange_mettl3.KO.GFP > 0 ~ "Shared Upregulated",
    log2FoldChange_S2.GFP_S2 < 0 & log2FoldChange_mettl3.KO.GFP < 0 ~ "Shared Downregulated",
    log2FoldChange_S2.GFP_S2 > 0 & log2FoldChange_mettl3.KO.GFP < 0 ~ "Up in S2S-YTHDC1-GoF/S2S-GFP, Down in METTL3-KO-YTHDC1/METTL3-KO-GFP",
    log2FoldChange_S2.GFP_S2 < 0 & log2FoldChange_mettl3.KO.GFP > 0 ~ "Down in S2S-YTHDC1-GoF/S2S-GFP, Up in METTL3-KO-YTHDC1/METTL3-KO-GFP",
    TRUE ~ "Other"))

# Step 3: Count genes in each category
Shared_Upregulated <- sum(shared_genes_with_metrics$Category == "Shared Upregulated")
Shared_Downregulated <- sum(shared_genes_with_metrics$Category == "Shared Downregulated")
Up_S2S_YTHDC1_GoF_S2S_GFP_Down_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF <- sum(shared_genes_with_metrics$Category == "Up in S2S-YTHDC1-GoF/S2S-GFP, Down in METTL3-KO-YTHDC1/METTL3-KO-GFP")
Down_S2S_YTHDC1_GoF_S2S_GFP_Up_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF <- sum(shared_genes_with_metrics$Category == "Down in S2S-YTHDC1-GoF/S2S-GFP, Up in METTL3-KO-YTHDC1/METTL3-KO-GFP")

# Step 4: Define the highlighted genes
highlight_genes <- c("Mettl3", "Ythdc1", "Arc1", "AOX1", "nvy", "bun", "brp", "Gr5a", "Tsp", "Zip89B")

# Step 5: Filter the shared_genes_with_metrics data frame for the highlighted genes
highlighted_data <- shared_genes_with_metrics %>%
  dplyr::filter(gene_name %in% highlight_genes) %>%
  mutate(label = gene_name)  # Use the gene name for labeling

# Step 6: Create the plot with highlighted gene labels
ggplot(shared_genes_with_metrics, aes(x = log2FoldChange_S2.GFP_S2, y = log2FoldChange_mettl3.KO.GFP, color = Category)) +
  geom_point(size = 2, alpha = 0.7) +  # Scatter plot points
  geom_text_repel(data = highlighted_data, aes(label = label),
                  max.overlaps = 10, size = 5, color = "black") +  # Labels for highlighted genes
  geom_smooth(method = "lm", color = "black", linetype = "dashed") +  # Linear regression line
  theme_minimal() +  # Minimal theme
  scale_color_manual(values = c("red", "blue", "darkorange", "darkgreen", "gray"),
                     labels = c(paste("Shared Upregulated (", Shared_Upregulated, ")", sep=""), 
                                paste("Shared Downregulated (", Shared_Downregulated, ")", sep=""),
                                paste("Up in S2S-YTHDC1-GoF/S2S-GFP, Down in METTL3-KO-YTHDC1/METTL3-KO-GFP (", Up_S2S_YTHDC1_GoF_S2S_GFP_Down_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF, ")", sep=""), 
                                paste("Down in S2S-YTHDC1-GoF/S2S-GFP, Up in METTL3-KO-YTHDC1/METTL3-KO-GFP (", Down_S2S_YTHDC1_GoF_S2S_GFP_Up_METTL3_KO_YTHDC1_METTL3_KO_GFP_GoF, ")", sep=""))) +
  labs(title = paste("Shared Genes Between S2S-YTHDC1-GoF/S2S-GFP and METTL3-KO-YTHDC1/METTL3-KO-GFP", 
                     "(r =", round(cor(shared_genes_with_metrics$log2FoldChange_S2.GFP_S2, shared_genes_with_metrics$log2FoldChange_mettl3.KO.GFP), 2), ")"),
       x = "Log2 Fold Change (S2S-YTHDC1-GoF/S2S-GFP)",
       y = "Log2 Fold Change (METTL3-KO-YTHDC1/METTL3-KO-GFP)",
       color = "Category") +
  theme(legend.position = "right")  # Move legend to the bottom
```




      
      500 significant change gene from S2-YTHDC-GOF vs  S2-GFP 
      
      how many of them do not significantly change in S2-YTHDC-GoF vs mettl3 KO-YTHDC-GoF?(as mettl3 rescued)
      
      how many of them still significantly change in S2-YTHDC-GoF vs mettl3 KO-YTHDC-GoF?(as mettl3 independent)
Identify significantly changed genes in S2-YTHDC1-GoF vs S2-GFP by filtering the genes that are significantly changed in res_S2.GFP_S2.DC1.GOF_clean.


```{r message=FALSE, warning=FALSE, echo=FALSE}


# Filter significant genes in S2-YTHDC1-GoF vs S2S-GFP
significant_genes_S2_GoF_vs_GFP <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(padj < 0.05 & abs(log2FoldChange) > 0.5)


#Now, you want to see how many of these significantly changed genes are rescued or remain changed in the METTL3-KO background.

# Genes that do not significantly change in S2-YTHDC1-GoF vs METTL3-KO-YTHDC1-GoF (METTL3 rescued)



rescued_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(validated_id %in% significant_genes_S2_GoF_vs_GFP$validated_id) %>%
  dplyr::filter(padj >= 0.05 | abs(log2FoldChange) <= 0.5)

num_rescued_genes <- nrow(rescued_genes)

# Genes that still significantly change in S2-YTHDC1-GoF vs METTL3-KO-YTHDC1-GoF (METTL3 independent)

independent_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(validated_id %in% significant_genes_S2_GoF_vs_GFP$validated_id) %>%
  dplyr::filter(padj < 0.05 & abs(log2FoldChange) > 0.5)

num_independent_genes <- nrow(independent_genes)


cat("Number of genes rescued by METTL3 loss (no longer significant in METTL3-KO-YTHDC1-GoF vs METTL3-KO-GFP):", num_rescued_genes, "\n")
cat("Number of genes independent of METTL3 (remain significant in METTL3-KO-YTHDC1-GoF vs METTL3-KO-GFP):", num_independent_genes, "\n")



print(rescued_genes)

# Independent genes
cat("\nIndependent genes (still significantly changed in METTL3-KO-YTHDC1-GoF vs METTL3-KO-GFP):\n")
print(independent_genes)

#write.table(rescued_genes, file = "rescued_genes_METTL3_loss.txt", row.names = FALSE, quote = FALSE, sep = "\t")


#write.table(independent_genes, file = "independent_genes_METTL3_loss.txt", row.names = FALSE, quote = FALSE, sep = "\t")


```

```{r}
# Merge the datasets and determine if genes are rescued or independent
rescued_DEGs <- significant_genes_S2_GoF_vs_GFP %>%
  dplyr::left_join(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, by = "gene_name", suffix = c("_YTHDC1_GoF", "_Mettl3KO_YTHDC1GoF")) %>%
  dplyr::mutate(
    rescued = ifelse(padj_Mettl3KO_YTHDC1GoF >= 0.05 | abs(log2FoldChange_Mettl3KO_YTHDC1GoF) <= 0.5, "Yes", "No"),
    current_symbol = coalesce(current_symbol_YTHDC1_GoF, current_symbol_Mettl3KO_YTHDC1GoF))

# Count the number of rescued and independent genes for the legend
rescued_count <- sum(rescued_DEGs$rescued == "Yes")
independent_count <- sum(rescued_DEGs$rescued == "No")

# Select genes to label (e.g., top 5 most rescued + top 5 most independent)
top_rescued <- rescued_DEGs %>% 
  dplyr::filter(rescued == "Yes") %>% 
  arrange(padj_Mettl3KO_YTHDC1GoF) %>% head(15)
top_independent <- rescued_DEGs %>% 
  dplyr::filter(rescued == "No") %>% 
  arrange(padj_Mettl3KO_YTHDC1GoF) %>% head(15)

genes_to_label <- rbind(top_rescued, top_independent)

#rescued_DEGs = na.omit(rescued_DEGs)



```



```{r}
res_upregulated_entrez <- rescued_genes %>%
  dplyr::filter(log2FoldChange > 0) %>%
  dplyr::select(ENTREZID) %>%
  unlist()

res_downregulated_entrez <- rescued_genes %>%
  dplyr::filter(log2FoldChange < 0) %>%
  dplyr::select(ENTREZID) %>%
  unlist()

# GO analysis for upregulated genes
go_up <- enrichGO(gene = res_upregulated_entrez,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",  # You can specify "BP", "MF", or "CC" if needed
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = res_downregulated_entrez,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  # You can specify "BP", "MF", or "CC" if needed
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes
barplot(go_up, showCategory = 20, 
        title = "Mettl3-KO Rescue:GO Enrichment for Upregulated Genes")

# Bar plot for downregulated genes
barplot(go_down, showCategory = 20, 
        title = "Mettl3-KO Rescue:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 20, 
        title = "Mettl3-KO Rescue:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 20, 
        title = "Mettl3-KO Rescue:GO Enrichment for Downregulated Genes")

# If you want to see the results in a table
go_up_results <- as.data.frame(go_up)
go_down_results <- as.data.frame(go_down)

# View the top results
head(go_up_results)
head(go_down_results)

```
all rescued genes

```{r}
# Load necessary libraries
library(clusterProfiler)
library(org.Dm.eg.db)  # Use the correct organism database
library(enrichplot)

# Get all genes (without up/down regulation filtering)
all_entrez_genes <- rescued_genes %>%
  dplyr::select(ENTREZID) %>%
  unlist()

# Perform GO enrichment analysis
go_all <- enrichGO(gene = all_entrez_genes,
                   OrgDb = org.Dm.eg.db,
                   keyType = "ENTREZID",
                   ont = "ALL",  # Includes BP, MF, and CC
                   pAdjustMethod = "BH",
                   pvalueCutoff = 0.05,
                   qvalueCutoff = 0.05)

# Convert results to data frames
go_all_results <- as.data.frame(go_all)


# Bar plot for GO terms
barplot(go_all, showCategory = 20, 
        title = "Mettl3-KO Rescue: GO Enrichment for All Genes")

# Dot plot for GO terms
dotplot(go_all, showCategory = 20, 
        title = "Mettl3-KO Rescue: GO Enrichment for All Genes")


# Load necessary libraries
library(clusterProfiler)
library(org.Dm.eg.db)

# Step 1: Convert gene SYMBOL to FlyBase ID (FBgn)
flybase_ids <- bitr(
  rescued_genes$current_symbol,  # Replace with your gene list
  fromType = "SYMBOL",
  toType = "FLYBASE",
  OrgDb = org.Dm.eg.db)

# Step 2: Convert FlyBase ID (FBgn) to KEGG-compatible CG ID (Dmel_CGXXXX)
kegg_ids <- bitr(
  flybase_ids$FLYBASE,
  fromType = "FLYBASE",
  toType = "FLYBASECG",  # KEGG uses "Dmel_CGXXXX"
  OrgDb = org.Dm.eg.db)

# Step 3: Remove NA values (unmapped genes)
kegg_ids <- na.omit(kegg_ids)

# Step 4: Format KEGG IDs by adding "Dmel_" prefix
kegg_ids$KEGG_ID <- paste0("Dmel_", kegg_ids$FLYBASECG)

# Step 5: Run KEGG enrichment analysis with formatted KEGG IDs
kegg_results <- enrichKEGG(
  gene = kegg_ids$KEGG_ID,  # Use KEGG-compatible IDs
  organism = "dme",  # KEGG code for *Drosophila melanogaster*
  pAdjustMethod = "BH",
  pvalueCutoff = 0.05)

# Bar plot for KEGG pathways
barplot(kegg_results, showCategory = 20, 
        title = "Mettl3-KO Rescue: KEGG Enrichment for All Genes")



```


```{r}
ind_upregulated_entrez <- independent_genes %>%
  dplyr::filter(log2FoldChange > 0) %>%
  dplyr::select(ENTREZID) %>%
  unlist()

ind_downregulated_entrez <- independent_genes %>%
  dplyr::filter(log2FoldChange < 0) %>%
  dplyr::select(ENTREZID) %>%
  unlist()

# GO analysis for upregulated genes
go_up <- enrichGO(gene = ind_upregulated_entrez,
                  OrgDb = org.Dm.eg.db,
                  keyType = "ENTREZID",
                  ont = "ALL",  # You can specify "BP", "MF", or "CC" if needed
                  pAdjustMethod = "BH",
                  pvalueCutoff = 0.05,
                  qvalueCutoff = 0.05)

# GO analysis for downregulated genes
go_down <- enrichGO(gene = ind_downregulated_entrez,
                    OrgDb = org.Dm.eg.db,
                    keyType = "ENTREZID",
                    ont = "ALL",  # You can specify "BP", "MF", or "CC" if needed
                    pAdjustMethod = "BH",
                    pvalueCutoff = 0.05,
                    qvalueCutoff = 0.05)



# Bar plot for upregulated genes
barplot(go_up, showCategory = 20, 
        title = "Mettl3-KO independent:GO Enrichment for Upregulated Genes")

# Bar plot for downregulated genes
barplot(go_down, showCategory = 20, 
        title = "Mettl3-KO independent:GO Enrichment for Downregulated Genes")

# Dot plot for upregulated genes
dotplot(go_up, showCategory = 20, 
        title = "Mettl3-KO independent:GO Enrichment for Upregulated Genes")

# Dot plot for downregulated genes
dotplot(go_down, showCategory = 20, 
        title = "Mettl3-KO independent:GO Enrichment for Downregulated Genes")

# If you want to see the results in a table
go_up_results <- as.data.frame(go_up)
go_down_results <- as.data.frame(go_down)

# View the top results
head(go_up_results)
head(go_down_results)
```





```{r}

# Define the genes you want to label additionally
highlight_genes <- c("fl(2)d", "aqz", "sky", "pum", "Mettl3", "Ythdc1", "unc-13", "Hsp70Ab", "GstD5","HERC2","CadN","nvy","AOX1","Desat1","Set2","mrva","form3","Swim","drongo","clumsy","Prosap")

padj_threshold <- 0.05
lfc_threshold <- 0.5

# Compute Rescued Genes (lost significance in METTL3-KO-YTHDC1-GoF vs METTL3-KO-GFP)
rescued_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(validated_id %in% significant_genes_S2_GoF_vs_GFP$validated_id) %>%
  dplyr::filter(padj >= padj_threshold | abs(log2FoldChange) <= lfc_threshold)

num_rescued_genes <- nrow(rescued_genes)

# Compute Independent Genes (still significant in METTL3-KO-YTHDC1-GoF vs METTL3-KO-GFP)
independent_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(validated_id %in% significant_genes_S2_GoF_vs_GFP$validated_id) %>%
  dplyr::filter(padj < padj_threshold & abs(log2FoldChange) > lfc_threshold)

num_independent_genes <- nrow(independent_genes)

# Compute METTL3-Dependent Genes (rescued + significantly changed in METTL3-KO vs S2S)
res_S2_m6A$validated_id = rownames(res_S2_m6A)
mettl3_dependent_genes <- rescued_genes %>%
  dplyr::filter(validated_id %in% res_S2_m6A$validated_id)

num_mettl3_dependent_genes <- nrow(mettl3_dependent_genes)

# Merge classification into one dataset
rescued_DEGs <- significant_genes_S2_GoF_vs_GFP %>%
  left_join(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, by = "validated_id", 
            suffix = c("_YTHDC1_GoF", "_Mettl3KO_YTHDC1GoF")) %>%
  mutate(
    classification = case_when(
      validated_id %in% rescued_genes$validated_id ~ "Rescued",
      validated_id %in% independent_genes$validated_id ~ "Independent",
      validated_id %in% mettl3_dependent_genes$validated_id ~ "METTL3-Dependent",
      TRUE ~ "Other"),
    current_symbol = coalesce(current_symbol_YTHDC1_GoF, current_symbol_Mettl3KO_YTHDC1GoF))

# Count the number of each classification for the legend
classification_counts <- table(rescued_DEGs$classification)

# Update classification factor levels (for consistent legend order)
rescued_DEGs <- rescued_DEGs %>%
  mutate(classification = factor(classification, 
                                 levels = c("METTL3-Dependent", "Rescued", "Independent", "Other")))

legend_labels <- c(
  paste0("METTL3-Dependent (", num_mettl3_dependent_genes, ")"),
  paste0("Rescued (", num_rescued_genes, ")"),
  paste0("Independent (", num_independent_genes, ")"),
  "Other")

ggplot(rescued_DEGs, 
       aes(x = log2FoldChange_YTHDC1_GoF, y = log2FoldChange_Mettl3KO_YTHDC1GoF, color = classification)) +
  geom_point(alpha = 0.7) +
  # Label METTL3-Dependent genes as before
  geom_text_repel(data = rescued_DEGs %>% filter(classification == "METTL3-Dependent"), 
                  aes(label = current_symbol), 
                  size = 4, 
                  box.padding = 0.5) +
  # Additional labeling for the specific highlight_genes
  geom_text_repel(data = rescued_DEGs %>% filter(current_symbol %in% highlight_genes),
                  aes(label = current_symbol),
                  size = 4,
                  box.padding = 0.5,
                  color = "black") +
  theme_minimal() +
  labs(title = "METTL3-Dependent and Independent genes",
    x = "log2FC (S2-YTHDC1-GoF/S2S-GFP)",
    y = "log2FC (METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP)",
    color = " ") +
  scale_color_manual(
    values = c("METTL3-Dependent" = "purple", "Rescued" = "blue", "Independent" = "red", "Other" = "gray"),
    labels = legend_labels)


#write.table(rescued_DEGs, file = "2025_0412_METTL3_Rescued_DEGs_S2-YTHDC1-GoF_vs_METTL3-KO-YTHDC1-GoF.txt", row.names = FALSE, quote = FALSE, sep = "\t")
```

```{r}
gene_scores <- independent_genes$log2FoldChange
names(gene_scores) <- independent_genes$current_symbol

geneList <- sort(gene_scores, decreasing = TRUE) 

gse <- gseGO(geneList = geneList,
    ont = "ALL",
    keyType = "SYMBOL",
    #nPerm = 10000, 
    minGSSize = 3,
    maxGSSize = 800,
    pvalueCutoff = 0.05, 
    verbose = TRUE, 
    OrgDb = organism,
    pAdjustMethod = "none")

cnetplot(gse, categorySize="pvalue", foldChange=geneList, showCategory = 30) + ggtitle("Gene Set Enrichment: Mettl3 independent genes")
```

```{r}

# Create named list of your data frames
data_frames <- list(
  res_S2_m6A,
  res_S2_S_DC1_KO,
  res_S2.GFP_S2.DC1.GOF,
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF,
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF,
  res_S2_GFP_mettl3_KO_DC1_GOF,
  res_mettl3_KO_GFP_S2_GFP)

names(data_frames) <- c(
  "METTL3-KO/S2S",
  "Ythdc1-KO/S2S",
  "S2S-YTHDC1-GOF/S2S-GFP",
  "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP",
  "S2S-YTHDC-GoF/METTL3-KO-YTHDC1-GoF",
  "Mettl3-KO-Ythdc1-GoF/S2-GFP",
  "Mettl3-KO-GFP/S2-GFP")

# Process each dataframe to count significant genes
results <- lapply(names(data_frames), function(name) {
  df <- data_frames[[name]]
  
  # Ensure diffexpressed column exists and is properly formatted
  if(!"diffexpressed" %in% colnames(df)) {
    stop(paste("Dataframe", name, "does not contain 'diffexpressed' column"))
  }
  
  # Convert to data frame if it's not already (common with DESeq2 results)
  if(!is.data.frame(df)) {
    df <- as.data.frame(df)
  }
  
  # Count UP and DOWN regulated genes
  counts <- df %>%
    dplyr::filter(diffexpressed %in% c("UP", "DOWN")) %>%
    dplyr::count(diffexpressed) %>%
    dplyr::mutate(Dataset = name)
  
  return(counts)
}) %>% dplyr::bind_rows()

ggplot(results, aes(x = Dataset, y = n, fill = diffexpressed)) +
  geom_col(width = 0.7) +
  geom_text(aes(label = n),
            position = position_stack(vjust = 0.5),
            size = 3.5, color = "white", fontface = "bold") +
  scale_fill_manual(values = c("UP" = "#e41a1c", "DOWN" = "#377eb8")) +
  theme_minimal(base_size = 14) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    plot.title = element_text(hjust = 0.5, face = "bold"),
    legend.position = "right",
    panel.grid.major.x = element_blank(),
    plot.margin = margin(10, 10, 10, 10)) +
  labs(
    title = "Significant DE Genes (UP/DOWN) per Dataset",
    x = "Comparison",
    y = "Number of Genes",
    fill = " ") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.05)))
```

# DGE lollipop

```{r}
# Load libraries
library(ggplot2)
library(tidyr)
library(dplyr)



res_mettl3_KO_GFP_S2_GFP_clean = res_mettl3_KO_GFP_S2_GFP[res_mettl3_KO_GFP_S2_GFP$baseMean>0 & !is.na(res_mettl3_KO_GFP_S2_GFP$padj),]

res_S2_GFP_mettl3_KO_DC1_GOF_clean = res_S2_GFP_mettl3_KO_DC1_GOF[res_S2_GFP_mettl3_KO_DC1_GOF$baseMean>0 & !is.na(res_S2_GFP_mettl3_KO_DC1_GOF$padj),]

res_S2_S_DC1_KO_clean = res_S2_S_DC1_KO[res_S2_S_DC1_KO$baseMean>0 & !is.na(res_S2_S_DC1_KO$padj),]

res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_clean = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF[res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$baseMean>0 & !is.na(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$padj),]

res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF[res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$baseMean>0 & !is.na(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF$padj),]

res_S2_m6A_clean = res_S2_m6A[res_S2_m6A$baseMean>0 & !is.na(res_S2_m6A$padj),]

res_S2.GFP_S2.DC1.GOF_clean = res_S2.GFP_S2.DC1.GOF[res_S2.GFP_S2.DC1.GOF$baseMean>0 & !is.na(res_S2.GFP_S2.DC1.GOF$padj),]

# Count upregulated and downregulated genes per dataset
deg_counts <- data.frame(
  Dataset = c("Ythdc1-KO/S2S", "S2-YTHDC1-GOF/S2-GFP", "Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP", 
              "S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF", "Mettl3-KO/S2S", "Mettl3-KO-Ythdc1-GoF/S2-GFP","Mettl3-KO-GFP/S2-GFP"),
  Upregulated = c(
    sum(res_S2_S_DC1_KO_clean$log2FoldChange > 0.5 & res_S2_S_DC1_KO_clean$padj <= 0.05),
    sum(res_S2.GFP_S2.DC1.GOF_clean$log2FoldChange > 0.5 & res_S2.GFP_S2.DC1.GOF_clean$padj <= 0.05),
    sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_clean$log2FoldChange > 0.5 & res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_clean$padj <= 0.05),
    sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$log2FoldChange > 0.5 & res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$padj <= 0.05),
    sum(res_S2_m6A_clean$log2FoldChange > 0.5 & res_S2_m6A_clean$padj <= 0.05), sum(res_S2_GFP_mettl3_KO_DC1_GOF_clean$log2FoldChange > 0.5 & res_S2_GFP_mettl3_KO_DC1_GOF_clean$padj <= 0.05), sum(res_mettl3_KO_GFP_S2_GFP_clean$log2FoldChange > 0.5 & res_mettl3_KO_GFP_S2_GFP_clean$padj <= 0.05)),
  Downregulated = c(
    sum(res_S2_S_DC1_KO_clean$log2FoldChange < -0.5 & res_S2_S_DC1_KO_clean$padj <= 0.05),
    sum(res_S2.GFP_S2.DC1.GOF_clean$log2FoldChange < -0.5 & res_S2.GFP_S2.DC1.GOF_clean$padj <= 0.05),
    sum(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_clean$log2FoldChange < -0.5 & res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_clean$padj <= 0.05),
    sum(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$log2FoldChange < -0.5 & res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$padj <= 0.05),
    sum(res_S2_m6A_clean$log2FoldChange < -0.5 & res_S2_m6A_clean$padj <= 0.05), 
    sum(res_S2_GFP_mettl3_KO_DC1_GOF_clean$log2FoldChange < -0.5 & res_S2_GFP_mettl3_KO_DC1_GOF_clean$padj <= 0.05),
    sum(res_mettl3_KO_GFP_S2_GFP_clean$log2FoldChange < -0.5 & res_mettl3_KO_GFP_S2_GFP_clean$padj <= 0.05)))


# Reshape data for ggplot
deg_counts_long <- pivot_longer(deg_counts, cols = c("Upregulated", "Downregulated"), 
                                names_to = "Direction", values_to = "Count")
deg_counts_long$Dataset <- factor(deg_counts_long$Dataset, 
  levels = c("Mettl3-KO/S2S", 
             "Ythdc1-KO/S2S",
             "Mettl3-KO-Ythdc1-GoF/S2-GFP", 
             "Mettl3-KO-GFP/S2-GFP",
             "Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP",
             "S2-YTHDC1-GOF/S2-GFP", 
             "S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF"))

# Create Lollipop Plot
ggplot(deg_counts_long, aes(x = Dataset, y = Count, color = Direction)) +
  geom_segment(aes(x = Dataset, xend = Dataset, y = 0, yend = Count), size = 1.2, color = "gray") +  # Lollipop stems
  geom_point(size = 6, aes(fill = Direction), shape = 21, stroke = 1.5) +  # Lollipop heads
  geom_text(aes(label = Count), vjust = -1.2, size = 5, fontface = "bold") +  # Add count labels
  scale_color_manual(values = c("Upregulated" = "red", "Downregulated" = "blue")) +
  scale_fill_manual(values = c("Upregulated" = "red", "Downregulated" = "blue")) +
  labs(title = " ",
       x = "Dataset",
       y = "Number of Significant Genes") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12, face = "bold"),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5),
        legend.position = "right")


```



# Differential Expression Heatmap Across Labs
```{r}
library(dplyr)
library(pheatmap)

# --- 1. Dataset List (in your order) ---
datasets <- list(
  res_mettl3_KO_GFP_S2_GFP,
  res_S2_GFP_mettl3_KO_DC1_GOF,
  res_S2_S_DC1_KO,
  res_S2.GFP_S2.DC1.GOF, 
  res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, 
  res_S2.YTHDC_gof_mettl3.KO.DC1.GOF, 
  res_S2_m6A, 
  Soller_diff,
  Ime4_dMettl14_KD_versus_LacZKD,
  Fl2d_KD_versus_LacZKD_in_S2R,
  Ythdc_RNAi_basal_vs_HS_2,
  Mettl3_RNAi_basal_vs_HS,
  Control_RNAi_vs_HS,
  one_week_mettl3_vs_mettl3,
  three_week_mettl3_vs_mettl3,
  one_week_ythdf_vs_ythdf
)

# --- 2. Metadata about each dataset ---
dataset_meta <- data.frame(
  name = c(
    "Ythdc1-KO/S2S",
    "Mettl3-KO/S2S",
    "Mettl3KO-Ythdc1-GoF/S2-GFP",
    "Mettl3KO-GFP/S2-GFP",
    "Mettl3KO-Ythdc1-GoF/Mettl3-KO-GFP",
    "S2-Ythdc1-GoF/S2-GFP",
    "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF",
    "Matthias et al", 
    "Roignant lab dMettl3-KO",
    "Roignant lab Fl2d_KD_versus_LacZKD_in_S2R",
    "Bonini lab Ythdc_RNAi_basal_vs_HS_2",
    "Bonini lab Mettl3_RNAi_basal_vs_HS",
    "Bonini lab Control_RNAi_vs_HS",
    "Lai lab one_week_mettl3_vs_mettl3",
    "Lai lab three_week_mettl3_vs_mettl3",
    "Lai lab one_week_ythdf_vs_ythdf"),
  gene_col = c(
    rep("current_symbol", 7),
    "gene_id",
    rep("current_symbol", 8)),
  logfc_col = c(
    rep("log2FoldChange", 7),
    "log2.fold_change.",
    "log2.fold.change..MAP...groupIme4_CG7818.vs.groupControl",
    "log2.fold.change..MAP...groupFl2d.vs.groupControl",
    "logs.YthdcRNAi.HSvcntrl.",
    "log2.hs.vs.ctl.mettl3.",
    "log2.hs.vs.ctl.",
    "log2FoldChange",
    "log2FoldChange",
    "log2FoldChange"),
  filter_sig = c(rep(TRUE, 7), rep(FALSE, 9)),
  stringsAsFactors = FALSE)

# --- 3. Generalized cleaner function ---
standardize_dataset <- function(dataset, symbol_col, fc_col, filter_sig = FALSE) {
  if (!(symbol_col %in% colnames(dataset)) || !(fc_col %in% colnames(dataset))) {
    warning(paste("Missing expected columns in dataset. Skipping:", symbol_col, fc_col))
    return(data.frame(current_symbol = character(), log2FoldChange = numeric()))
  }

  data <- dataset %>%
    dplyr::rename(current_symbol = all_of(symbol_col), log2FoldChange = all_of(fc_col))

  if (filter_sig && "diffexpressed" %in% colnames(data)) {
    data <- data %>% dplyr::filter(diffexpressed != "Not Significant")
  }

  data %>%
    dplyr::select(current_symbol, log2FoldChange) %>%
    dplyr::filter(!is.na(current_symbol), !is.na(log2FoldChange))
}

# --- 4. Apply to all datasets ---
significant_lists <- mapply(function(data, sym, lfc, sig) {
  standardize_dataset(data, symbol_col = sym, fc_col = lfc, filter_sig = sig)
}, datasets, dataset_meta$gene_col, dataset_meta$logfc_col, dataset_meta$filter_sig, SIMPLIFY = FALSE)

names(significant_lists) <- dataset_meta$name

# --- 5. Combine ---
logfc_combined <- Reduce(function(x, y) full_join(x, y, by = "current_symbol"), significant_lists)
logfc_combined <- logfc_combined[!duplicated(logfc_combined$current_symbol), ]

rownames(logfc_combined) <- logfc_combined$current_symbol
logfc_combined$current_symbol <- NULL
colnames(logfc_combined) <- dataset_meta$name
logfc_combined[is.na(logfc_combined)] <- 0

# --- 6. Filter to selected genes ---
selected_genes <- c(
  "Hsp70Aa", "Hsp70Ab", "Hsp70Ba", "Hsp70Bb", "Hsp70Bc", "fl(2)d", "vir", "nito",
  "Mettl3", "Ythdc1", "baz", "Flacc", "Adh", "Aldh-III", "Aldh7A1", "twi", "Ace",
  "ben", "unc-13", "Sxl", "cv-c", "Smr", "spen", "Bsg", "gsb", "Ten-a", "olf186-M",
  "mfas", "miple2", "prosap", "aqz", "sky", "Abd-B", "Act88F", "Argk1", "cv-2", 
  "Hsp83", "TotM", "Swim", "CG8046", "NijA", "Spn31A", "nemy", "slgA", "nkd", "Dso1", "CG4404", "P58IPK", "eIF4E4", "Invadolysin", "corn", "blo", "magu", "tim", "CG17646", "slow", "CG9664", "trpl", "Pde6", "Tpst", "Dpp10", "TP53INP", "Dop2R", "CG44247", "Sirup", "LysB", "CG33303", "Atg1", "Tollo", "CG31760", "CG8034", "spir", "CG33144", "Snx27", "Clc", "CG12502", "msi", "Grp170", "Eip63E", "raw", "Drat","Liprin-", "side", "fru", "bnl", "dnc", "Hr39", "pyd", "Wdr62", "CG45088",  "T48", "rdx", "shep", "CG9674", "sgg", "CG44085", "Tab2", "Eip75B", "tai")

logfc_combined_filtered <- logfc_combined[rownames(logfc_combined) %in% selected_genes, ]

# --- 7. Plot heatmap ---
pheatmap(
  mat = as.matrix(logfc_combined_filtered),
  color = colorRampPalette(c("blue", "white", "red"))(100),
  cluster_rows = F,
  cluster_cols = F,
  show_rownames = TRUE,
  show_colnames = T,
  fontsize_row = 9,
  fontsize_col = 10,
  main = "Differential Expression Heatmap Across Labs", 
  name = "log2FC")

```


```{r}
# Extract significant DEGs from all datasets
#res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean$Gene_ID = rownames(res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean)


#res_S2.GFP_S2.DC1.GOF_clean$Gene_ID = rownames(res_S2.GFP_S2.DC1.GOF_clean)

#res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$Gene_ID = rownames(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean)

sig_mettl3_KO <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

sig_other <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

# Find common genes (Mettl3-independent)
mettl3_independent_genes <- intersect(sig_mettl3_KO$validated_id, sig_other$validated_id)

# View the list
#print(mettl3_independent_genes)

# Extract DEGs from Mettl3-KO vs. Control
sig_mettl3_KO <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

# Extract DEGs from Mettl3-KO + YTHDC1-GOF
sig_mettl3_YTHDC1 <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(padj > 0.05) %>%  # Not significantly different anymore
  dplyr::select(validated_id)

# Find rescued genes (DE in Mettl3-KO, not DE in Mettl3-KO + YTHDC1-GOF)
rescued_genes <- intersect(sig_mettl3_KO$validated_id, sig_mettl3_YTHDC1$validated_id)

# View rescued genes
#print(rescued_genes)

```



# Combine GO terms 


```{r}
library(clusterProfiler)
library(org.Dm.eg.db)
library(dplyr)
library(ggplot2)
library(tidyr)

# Define function for GO enrichment
run_GO_enrichment <- function(gene_list) {
  enrichGO(
    gene = gene_list, 
    OrgDb = org.Dm.eg.db, 
    keyType = "FLYBASE", 
    ont = "BP", 
    pAdjustMethod = "BH", 
    qvalueCutoff = 0.05) %>%
  as.data.frame()
}

#res_S2_S_DC1_KO_clean$gene_name = rownames(res_S2_S_DC1_KO_clean)

#res_S2.GFP_S2.DC1.GOF_clean$gene_name = rownames(res_S2.GFP_S2.DC1.GOF_clean)

#res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean$gene_name = rownames(res_mettl3.KO.GFP_mettl3.KO.DC1.GOF_clean)

#res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean$gene_name = rownames(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean)

#res_S2_m6A_clean$gene_name = rownames(res_S2_m6A_clean)

# Perform GO enrichment for each dataset

sig_Ythdc1_KO_S2S <- res_S2_S_DC1_KO %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

sig_S2.GFP_S2.DC1.GOF <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

sig_Mettl3_KO_Ythdc1_Mettl3_KO_GFP <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

sig_S2.YTHDC_gof_mettl3.KO.DC1.GOF <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

sig_Mettl3_KO_S2S <- res_S2_m6A %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

sig_Mettl3_KO_Ythdc1_GoF_S2_GFP <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

sig_Mettl3_KO_GFP_S2_GFP <- res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(validated_id)

go_Ythdc1_KO_S2S = run_GO_enrichment(sig_Ythdc1_KO_S2S$validated_id)
go_S2.GFP_S2.DC1.GOF = run_GO_enrichment(sig_S2.GFP_S2.DC1.GOF$validated_id)
go_Mettl3_KO_Ythdc1_Mettl3_KO_GFP = run_GO_enrichment(sig_Mettl3_KO_Ythdc1_Mettl3_KO_GFP$validated_id)
go_S2.YTHDC_gof_mettl3.KO.DC1.GOF= run_GO_enrichment(sig_S2.YTHDC_gof_mettl3.KO.DC1.GOF$validated_id)
go_Mettl3_KO_S2S <- run_GO_enrichment(sig_Mettl3_KO_S2S$validated_id)
go_Mettl3_KO_Ythdc1_GoF_S2_GFP = run_GO_enrichment(sig_Mettl3_KO_Ythdc1_GoF_S2_GFP$validated_id)
go_Mettl3_KO_GFP_S2_GFP= run_GO_enrichment(sig_Mettl3_KO_GFP_S2_GFP$validated_id)

# Add dataset column
go_Ythdc1_KO_S2S$Dataset = "Ythdc1-KO/S2S"
go_Mettl3_KO_S2S$Dataset <- "Mettl3-KO/S2S"
go_S2.GFP_S2.DC1.GOF$Dataset = "S2-YTHDC1-GOF/S2-GFP"
go_Mettl3_KO_Ythdc1_Mettl3_KO_GFP$Dataset = "Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP"
go_S2.YTHDC_gof_mettl3.KO.DC1.GOF$Dataset = "S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF"
go_Mettl3_KO_Ythdc1_GoF_S2_GFP$Dataset = "Mettl3-KO-Ythdc1-GoF/S2-GFP"
go_Mettl3_KO_GFP_S2_GFP$Dataset = "Mettl3-KO-GFP/S2-GFP"

# Select relevant columns and bind all results
go_results <- bind_rows(
  go_Ythdc1_KO_S2S %>% select(Description, p.adjust, Count, Dataset),
  go_S2.GFP_S2.DC1.GOF %>% select(Description, p.adjust, Count, Dataset),
  go_Mettl3_KO_Ythdc1_Mettl3_KO_GFP %>% select(Description, p.adjust, Count, Dataset),
  go_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>% select(Description, p.adjust, Count, Dataset),
  go_Mettl3_KO_S2S %>% select(Description, p.adjust, Count, Dataset),
  go_Mettl3_KO_Ythdc1_GoF_S2_GFP %>% select(Description, p.adjust, Count, Dataset),
  go_Mettl3_KO_GFP_S2_GFP %>% select(Description, p.adjust, Count, Dataset))

# Convert dataset to factor (for proper ordering in the plot)
go_results$Dataset <- factor(go_results$Dataset, 
  levels = c("Mettl3-KO/S2S", 
             "Ythdc1-KO/S2S",
             "Mettl3-KO-Ythdc1-GoF/S2-GFP", 
             "Mettl3-KO-GFP/S2-GFP",
             "Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP",
             "S2-YTHDC1-GOF/S2-GFP", 
             "S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF"))

# Select **top 10 enriched GO terms** across datasets
top_terms <- go_results %>%
  group_by(Dataset) %>%
  slice_min(order_by = p.adjust, n = 20)




# Plot GO terms across datasets
ggplot(top_terms, aes(x = Dataset, y = reorder(Description, p.adjust), size = Count, color = p.adjust)) +
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +  # Red = more significant
  labs(title = "Top Enriched GO Terms Across Datasets",
       x = "Dataset",
       y = "GO Biological Process",
       size = "Gene Count",
       color = "p.adjust") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```


```{r}
library(clusterProfiler)
library(org.Dm.eg.db)
library(ggplot2)
library(dplyr)
library(tidyr)


# Filter significant genes (padj  0.05 and log2FoldChange > 0.5 or < -0.5)
sig_genes_S2_S_DC1_KO <- res_S2_S_DC1_KO %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(current_symbol)

sig_genes_S2_GFP_S2_DC1_GOF <- res_S2.GFP_S2.DC1.GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(current_symbol)

sig_genes_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(current_symbol)

sig_genes_S2_YTHDC_gof_mettl3_KO_DC1_GOF <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(current_symbol)

sig_genes_S2_m6A <- res_S2_m6A %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(current_symbol)


sig_genes_S2_GFP_mettl3_KO_DC1_GOF <- res_S2_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(current_symbol)

sig_mettl3_KO_GFP_S2_GFP <- res_mettl3_KO_GFP_S2_GFP %>%
  dplyr::filter(padj <= 0.05 & abs(log2FoldChange) > 0.5) %>%
  dplyr::select(current_symbol)


# Convert SYMBOL to FlyBase Gene ID (FBgn) for each dataset
flybase_ids_S2_S_DC1_KO <- bitr(sig_genes_S2_S_DC1_KO$current_symbol, 
                                fromType = "SYMBOL", toType = "FLYBASE", OrgDb = org.Dm.eg.db)
flybase_ids_S2_GFP_S2_DC1_GOF <- bitr(sig_genes_S2_GFP_S2_DC1_GOF$current_symbol, fromType = "SYMBOL", toType = "FLYBASE", OrgDb = org.Dm.eg.db)
flybase_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- bitr(sig_genes_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol, fromType = "SYMBOL", toType = "FLYBASE", OrgDb = org.Dm.eg.db)
flybase_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF <- bitr(sig_genes_S2_YTHDC_gof_mettl3_KO_DC1_GOF$current_symbol, fromType = "SYMBOL", toType = "FLYBASE", OrgDb = org.Dm.eg.db)
flybase_ids_S2_m6A <- bitr(sig_genes_S2_m6A$current_symbol, fromType = "SYMBOL", toType = "FLYBASE", OrgDb = org.Dm.eg.db)

flybase_ids_S2_GFP_mettl3_KO_DC1_GOF <- bitr(sig_genes_S2_GFP_mettl3_KO_DC1_GOF$current_symbol, fromType = "SYMBOL", toType = "FLYBASE", OrgDb = org.Dm.eg.db)
flybase_ids_mettl3_KO_GFP_S2_GFP <- bitr(sig_mettl3_KO_GFP_S2_GFP$current_symbol, fromType = "SYMBOL", toType = "FLYBASE", OrgDb = org.Dm.eg.db)

# Convert FlyBase Gene IDs to KEGG-compatible Dmel_CGXXXX format
kegg_ids_S2_S_DC1_KO <- bitr(flybase_ids_S2_S_DC1_KO$FLYBASE, fromType = "FLYBASE", toType = "FLYBASECG", OrgDb = org.Dm.eg.db)
kegg_ids_S2_GFP_S2_DC1_GOF <- bitr(flybase_ids_S2_GFP_S2_DC1_GOF$FLYBASE, fromType = "FLYBASE", toType = "FLYBASECG", OrgDb = org.Dm.eg.db)
kegg_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- bitr(flybase_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FLYBASE, fromType = "FLYBASE", toType = "FLYBASECG", OrgDb = org.Dm.eg.db)
kegg_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF <- bitr(flybase_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF$FLYBASE, fromType = "FLYBASE", toType = "FLYBASECG", OrgDb = org.Dm.eg.db)
kegg_ids_S2_m6A <- bitr(flybase_ids_S2_m6A$FLYBASE, fromType = "FLYBASE", toType = "FLYBASECG", OrgDb = org.Dm.eg.db)

kegg_ids_S2_GFP_mettl3_KO_DC1_GOF <- bitr(flybase_ids_S2_GFP_mettl3_KO_DC1_GOF$FLYBASE, fromType = "FLYBASE", toType = "FLYBASECG", OrgDb = org.Dm.eg.db)
kegg_ids_mettl3_KO_GFP_S2_GFP <- bitr(flybase_ids_mettl3_KO_GFP_S2_GFP$FLYBASE, fromType = "FLYBASE", toType = "FLYBASECG", OrgDb = org.Dm.eg.db)
# Remove NA values
kegg_ids_S2_S_DC1_KO <- na.omit(kegg_ids_S2_S_DC1_KO)
kegg_ids_S2_GFP_S2_DC1_GOF <- na.omit(kegg_ids_S2_GFP_S2_DC1_GOF)
kegg_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- na.omit(kegg_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
kegg_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF <- na.omit(kegg_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF)
kegg_ids_S2_m6A <- na.omit(kegg_ids_S2_m6A)

kegg_ids_S2_GFP_mettl3_KO_DC1_GOF <- na.omit(kegg_ids_S2_GFP_mettl3_KO_DC1_GOF)
kegg_ids_mettl3_KO_GFP_S2_GFP <- na.omit(kegg_ids_mettl3_KO_GFP_S2_GFP)


kegg_ids_S2_S_DC1_KO$KEGG_ID <- paste0("Dmel_", kegg_ids_S2_S_DC1_KO$FLYBASECG)
kegg_ids_S2_GFP_S2_DC1_GOF$KEGG_ID <- paste0("Dmel_", kegg_ids_S2_GFP_S2_DC1_GOF$FLYBASECG)
kegg_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF$KEGG_ID <- paste0("Dmel_", kegg_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FLYBASECG)
kegg_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF$KEGG_ID <- paste0("Dmel_", kegg_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF$FLYBASECG)
kegg_ids_S2_m6A$KEGG_ID <- paste0("Dmel_", kegg_ids_S2_m6A$FLYBASECG)


kegg_ids_S2_GFP_mettl3_KO_DC1_GOF$KEGG_ID <- paste0("Dmel_", kegg_ids_S2_GFP_mettl3_KO_DC1_GOF$FLYBASECG)
kegg_ids_mettl3_KO_GFP_S2_GFP$KEGG_ID <- paste0("Dmel_", kegg_ids_mettl3_KO_GFP_S2_GFP$FLYBASECG)
```

```{r}
# Run KEGG enrichment with **Dmel_ prefixed IDs**
kegg_S2_S_DC1_KO <- enrichKEGG(gene = kegg_ids_S2_S_DC1_KO$KEGG_ID, 
                               organism = 'dme', pAdjustMethod = "BH", 
                               pvalueCutoff = 0.05)

kegg_S2_GFP_S2_DC1_GOF <- enrichKEGG(gene = kegg_ids_S2_GFP_S2_DC1_GOF$KEGG_ID, 
                                     organism = 'dme', pAdjustMethod = "BH", 
                                     pvalueCutoff = 0.05)

kegg_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- enrichKEGG(gene = kegg_ids_mettl3_KO_GFP_mettl3_KO_DC1_GOF$KEGG_ID, organism = 'dme', 
                                                   pAdjustMethod = "BH", pvalueCutoff = 0.05)

kegg_S2_YTHDC_gof_mettl3_KO_DC1_GOF <- enrichKEGG(gene = kegg_ids_S2_YTHDC_gof_mettl3_KO_DC1_GOF$KEGG_ID, organism = 'dme', pAdjustMethod = "BH", pvalueCutoff = 0.05)
kegg_S2_m6A <- enrichKEGG(gene = kegg_ids_S2_m6A$KEGG_ID, 
                          organism = 'dme', pAdjustMethod = "BH", pvalueCutoff = 0.05)


kegg_S2_GFP_mettl3_KO_DC1_GOF <- enrichKEGG(gene = kegg_ids_S2_GFP_mettl3_KO_DC1_GOF$KEGG_ID, 
                          organism = 'dme', pAdjustMethod = "BH", pvalueCutoff = 0.05)

kegg_mettl3_KO_GFP_S2_GFP <- enrichKEGG(gene = kegg_ids_mettl3_KO_GFP_S2_GFP$KEGG_ID, 
                          organism = 'dme', pAdjustMethod = "BH", pvalueCutoff = 0.05)
```

```{r}
# Convert KEGG results to data frames
kegg_S2_S_DC1_KO <- as.data.frame(kegg_S2_S_DC1_KO)
kegg_S2_GFP_S2_DC1_GOF <- as.data.frame(kegg_S2_GFP_S2_DC1_GOF)
kegg_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- as.data.frame(kegg_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
kegg_S2_YTHDC_gof_mettl3_KO_DC1_GOF <- as.data.frame(kegg_S2_YTHDC_gof_mettl3_KO_DC1_GOF)
kegg_S2_m6A <- as.data.frame(kegg_S2_m6A)

kegg_S2_GFP_mettl3_KO_DC1_GOF <- as.data.frame(kegg_S2_GFP_mettl3_KO_DC1_GOF)
kegg_mettl3_KO_GFP_S2_GFP <- as.data.frame(kegg_mettl3_KO_GFP_S2_GFP)
# Add dataset column
kegg_S2_S_DC1_KO$Dataset = "Ythdc1-KO/S2S"
kegg_S2_GFP_S2_DC1_GOF$Dataset = "S2-YTHDC1-GOF/S2-GFP"
kegg_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Dataset = "Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP"
kegg_S2_YTHDC_gof_mettl3_KO_DC1_GOF$Dataset = "S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF"
kegg_S2_m6A$Dataset <- "Mettl3-KO/S2S"
kegg_S2_GFP_mettl3_KO_DC1_GOF = "Mettl3-KO-Ythdc1-GoF/S2-GFP"
kegg_mettl3_KO_GFP_S2_GFP = "Mettl3-KO-GFP/S2-GFP"

# Merge all KEGG results
kegg_results <- bind_rows(
  kegg_S2_S_DC1_KO %>% 
    dplyr::select(Description, p.adjust, Count, Dataset),
  kegg_S2_GFP_S2_DC1_GOF %>% 
    dplyr::select(Description, p.adjust, Count, Dataset),
  kegg_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>% 
    dplyr::select(Description, p.adjust, Count, Dataset),
  kegg_S2_YTHDC_gof_mettl3_KO_DC1_GOF %>% 
    dplyr::select(Description, p.adjust, Count, Dataset),
  kegg_S2_m6A %>% 
    dplyr::select(Description, p.adjust, Count, Dataset))

kegg_results$Dataset <- factor(kegg_results$Dataset, 
  levels = c("Mettl3-KO/S2S", 
             "Ythdc1-KO/S2S",
             "Mettl3-KO-Ythdc1-GoF/S2-GFP", 
             "Mettl3-KO-GFP/S2-GFP",
             "Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP",
             "S2-YTHDC1-GOF/S2-GFP", 
             "S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF"))


ggplot(kegg_results, aes(x = Dataset, y = reorder(Description, p.adjust), size = Count, color = p.adjust)) +
  geom_point() +
  scale_color_gradient(low = "red", high = "blue") +  # Red = more significant
  labs(title = "Top Enriched KEGG Pathways Across Datasets",
       x = "Dataset",
       y = "KEGG Pathway",
       size = "Gene Count",
       color = "p.adjust") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))


```

# Merge the cnetplot
```{r}
samples <- list(
  S1 = res_mettl3_KO_GFP_S2_GFP,
  S2 = res_S2_GFP_mettl3_KO_DC1_GOF,
  S3 = res_S2_S_DC1_KO,
  S4 = res_S2.GFP_S2.DC1.GOF,
  S5 = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF,
  S6 = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF,
  S7 = res_S2_m6A)








# --- Run GSEA for all datasets ---
gsea_results <- lapply(samples, function(sig_df) {
  gene_scores <- sig_df %>%
    filter(diffexpressed != "Not Significant") %>%
    { setNames(.$log2FoldChange, .$current_symbol) } %>%
    sort(decreasing = TRUE)
  
  gseGO(geneList = gene_scores,
        ont = "ALL",
        keyType = "SYMBOL",
        #nPermSimple = 10000,
        minGSSize = 3,
        maxGSSize = 800,
        pvalueCutoff = 0.05,
        verbose = FALSE,
        OrgDb = organism,
        pAdjustMethod = "none")
})




# Count how often each GO term appears
all_terms <- unlist(lapply(gsea_results, function(x) x@result$ID))
term_freq <- table(all_terms)

# Keep only terms found in at least 3 datasets
frequent_terms <- names(term_freq[term_freq >= 2])
cat("Terms found in >=3 datasets:", length(frequent_terms), "\n")

# --- Identify shared GO terms ---
shared_terms <- Reduce(intersect, lapply(gsea_results, function(gse) gse@result$ID))

# --- Combine all gene fold changes ---
all_gene_fc <- lapply(samples, function(sig_df) {
  sig_df %>%
    filter(diffexpressed != "Not Significant") %>%
    { setNames(.$log2FoldChange, .$current_symbol) }
})
merged_fc <- Reduce(function(x, y) {
  x[names(y)] <- y[names(y)]
  x
}, all_gene_fc)

# --- Use a representative GSEA result for shared terms ---
gse_shared <- gsea_results[[1]]
gse_shared@result <- gse_shared@result %>% filter(ID %in% shared_terms)

# --- Optionally cap the number of categories shown ---
show_n <- 50
gse_shared@result <- gse_shared@result %>% arrange(pvalue) %>% head(show_n)


# Count how often each GO term appears
all_terms <- unlist(lapply(gsea_results, function(x) x@result$ID))
term_freq <- table(all_terms)

# Keep only terms found in at least 3 datasets
frequent_terms <- names(term_freq[term_freq >= 3])
cat("Terms found in >=3 datasets:", length(frequent_terms), "\n")


# Use dataset 1 as a base
gse_shared <- gsea_results[[1]]

# Filter to the frequent terms and take top 10
gse_shared@result <- gse_shared@result %>%
  dplyr::filter(ID %in% frequent_terms) %>%
  dplyr::arrange(pvalue) %>%
  head(10)

cat("GO terms used for plot:", gse_shared@result$Description, "\n")


# All gene FCs across samples, merged
merged_fc <- Reduce(function(x, y) {
  x[names(y)] <- y[names(y)]
  x
}, lapply(samples, function(sig_df) {
  sig_df %>%
    filter(diffexpressed != "Not Significant") %>%
    { setNames(.$log2FoldChange, .$current_symbol) }
}))

library(enrichplot)

cnetplot(
  gse_shared,
  categorySize = "geneNum",
  foldChange = merged_fc,
  showCategory = 10,
  node_label = "all",       # Show both terms and genes
  circular = FALSE,
  colorEdge = TRUE
) + 
  ggtitle("Top 10 Frequently Shared GO Terms (3 datasets)")


```

```{r}
# Load libraries
library(clusterProfiler)
library(enrichplot)
library(org.Dm.eg.db)
library(dplyr)
library(ggplot2)
library(tibble)
library(tidygraph)
library(ggraph)

# ---- STEP 1: Select datasets ----
selected_samples <- c("S1", "S3", "S5")  # Choose any subset

# ---- STEP 2: Subset GSEA and DE results ----
selected_gsea <- gsea_results[selected_samples]
selected_sig_lists <- samples[selected_samples]

# ---- STEP 3: Combine gene fold changes & dataset origins ----
gene_info <- bind_rows(lapply(seq_along(selected_sig_lists), function(i) {
  df <- selected_sig_lists[[i]]
  df <- df %>%
    dplyr::filter(diffexpressed != "Not Significant") %>%
    dplyr::select(current_symbol, log2FoldChange) %>%
    mutate(dataset = selected_samples[i])
})) %>%
  distinct(current_symbol, .keep_all = TRUE)

# Fix rownames
rownames(gene_info) <- NULL
rownames(gene_info) <- gene_info$current_symbol

# ---- STEP 4: Prepare fold change vector ----
merged_fc <- gene_info$log2FoldChange
names(merged_fc) <- rownames(gene_info)

# ---- STEP 5: Get frequently enriched GO terms ----
all_terms <- unlist(lapply(selected_gsea, function(x) x@result$ID))
term_freq <- table(all_terms)
frequent_terms <- names(term_freq[term_freq >= 2])  # Shared in 2 datasets

# ---- STEP 6: Use one GSEA object to base plot on ----
gse_shared <- selected_gsea[[1]]
gse_shared@result <- gse_shared@result %>%
  dplyr::filter(ID %in% frequent_terms) %>%
  arrange(pvalue) %>%
  head(10)

# ---- STEP 7: Dataset color clusters ----
cluster_colors <- setNames(c("red", "blue", "green"), selected_samples)
gene_info$color <- cluster_colors[gene_info$dataset]

# Generate cnetplot with access to data
p_data <- cnetplot(
  gse_shared,
  categorySize = "geneNum",
  foldChange = merged_fc,
  showCategory = 10,
  node_label = "all",
  colorEdge = TRUE,
  plot.data = TRUE  # returns a list with graph and ggplot
)


```



## PCA
```{r}
count_data <- read.delim2("~/Dropbox/all_files/read_counts_from_Lijuan.txt", header = T, check.names = FALSE)
count_data <- count_data[, -1]
count_data = na.omit(count_data)
count_data <- count_data[, !grepl("NOT USED", colnames(count_data))]
count_data <- count_data[, !grepl("DC1_GoF*", colnames(count_data))]
sample_info <- data.frame(
  sample = colnames(count_data),
  condition = c(
    rep("S2-GFP", 3),
    rep("S2-Ythdc1-GoF", 3),
    rep("Mettl3-KO-GFP", 3),
    rep("Mettl3-KO-Ythdc1-GoF", 3),
    rep("Ythdc1-KO", 3),
    rep("S2S", 3),
    rep("Mettl3-KO", 3)))#,
    #rep("DC1_KO_UNUSED", 3),
    #rep("Ythdc1-GoF", 3)))

rownames(sample_info) <- sample_info$sample


# Construct DESeqDataSet
dds <- DESeqDataSetFromMatrix(countData = count_data,
                              colData = sample_info,
                              design = ~ condition)

# Filter out low counts (optional but recommended)
dds <- dds[rowSums(counts(dds)) > 10, ]

# Variance Stabilizing Transformation
vsd <- vst(dds, blind = TRUE)

# Generate PCA plot
pca_data <- plotPCA(vsd, intgroup = "condition", returnData = TRUE)
percentVar <- round(100 * attr(pca_data, "percentVar"))


custom_colors <- c(
  "S2S" = "red",
  "S2-GFP" = "brown",
  "Mettl3-KO-GFP" = "orange",
  "Mettl3-KO" = "blue",
  "Ythdc1-KO" = "green",
  "S2-Ythdc1-GoF" = "purple",
  "Mettl3-KO-Ythdc1-GoF" = "pink")



ggplot(pca_data, aes(PC1, PC2, color = condition, fill = condition)) +
  stat_ellipse(type = "euclid", alpha = 0.1) +
  geom_point(size = 5, shape = 21, color = "black", stroke = 0.5, aes(fill = condition)) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  coord_fixed() +
  ggtitle("Principal Component Analysis of Gene Expression") +
  scale_fill_manual(values = custom_colors) +
  scale_color_manual(values = custom_colors) +
  theme_minimal(base_size = 12) +
  theme(
    panel.grid.major = element_line(color = "grey90", linewidth = 0.2),
    panel.grid.minor = element_blank(),
    panel.border = element_rect(fill = NA, color = "grey70"),
    plot.title = element_text(hjust = 0.5, face = "bold"),
    legend.position = "right",
    legend.title = element_text(face = "bold")) +
  guides(
    color = guide_legend(title = "Condition"),
    fill = guide_legend(title = "Condition"))

```



# Alternative splicing

Significant genes were selected based on the following criteria:

*1. an adjusted p-value (FDR) less than 0.05  and*
*2. an inclusion level difference (IncLevelDifference) greater than 0.2.*


```{r message=FALSE, warning=FALSE, echo=FALSE}
make_single_PSI_box_METTL3_KO_S2S <- function(event_IDs, Splicing_type) {
    # Define the path to your splicing files
    file_paths <- list(
        SE = ~"~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt")
    
    # Check if the specified splicing type is available
    if (!Splicing_type %in% names(file_paths)) {
        stop("Splicing type not found in the list of files.")
    }

    # Read the data file for the specified splicing type
    file_path <- file_paths[[Splicing_type]]
    data <- read.table(file_path, header = TRUE)

    # Filter for specified event IDs
    event_data <- data[data$ID %in% event_IDs, ]

    # Reshape the data
    psi_data <- data.frame(ID = event_data$ID, IncLevel1 = event_data$IncLevel1, IncLevel2 = event_data$IncLevel2)
    psi_data_long <- melt(psi_data, id.vars = "ID", variable.name = "Sample", value.name = "PSI")
    psi_data_long$Condition <- ifelse(psi_data_long$Sample == "IncLevel1", "METTL3-KO", "S2S")

    # Reorder the factor levels to put S2S as the first level
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("METTL3-KO", "S2S"))

    # Split the concatenated PSI values and convert them to numeric
    psi_data_long$PSI <- as.character(psi_data_long$PSI)
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(PSI, ",")) %>%
        unnest(PSI) %>%
        mutate(PSI = as.numeric(PSI))

    # Create the box plot with statistical comparisons
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6) +
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
        labs(fill = "Condition") + 
        scale_fill_manual(values = c("S2S" = "red", "METTL3" = "blue")) +
        stat_compare_means(aes(label = ..p.signif..), method = "t.test", comparisons = list(c("YTHDC1-KO", "YTHDC1-GoF")))

    print(plot)
}

# Example usage
#make_single_PSI_box_DC1_GoF(event_IDs = c("3972", "3992"), Splicing_type = "DSE")

```

    
    
    
    METTL3-KO relative to S2S 
 
 Total number of Significant splicing events in **METTL3-KO relative to S2S**: METTL3-KO promotes exon exclusion relative to S2S. it also promotes more exon skipping and A5SS events than any other splicing type/event.

```{r message=FALSE, warning=FALSE, echo=FALSE}

prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)
  
  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  return(data)
}

# Function to count splicing events after filtering
count_splicing_events <- function(filename) {
  data <- prepare_and_filter_data(filename)
  
  # Filter for significant genes
  significant_genes <- subset(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  n_exclusion <- sum(significant_genes$IncLevelDifference < 0)
  n_inclusion <- sum(significant_genes$IncLevelDifference > 0)
  
  sample_id <- basename(filename)
  sample_id <- gsub(".MATS.JCEC.txt", "", sample_id)
  
  return(data.frame(SampleID = sample_id, Exclusion = n_exclusion, Inclusion = n_inclusion))
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt")

# Apply function to each file and combine results
results <- do.call(rbind, lapply(files, count_splicing_events))
# Calculate totals
total_exclusion <- sum(results$Exclusion)
total_inclusion <- sum(results$Inclusion)

# Melt the results data for ggplot
results_melted <- tidyr::pivot_longer(results, cols = c(Exclusion, Inclusion), names_to = "EventType", values_to = "Count")
# Perform a paired t-test
t_test_result <- t.test(results$Inclusion, results$Exclusion, paired = TRUE)

# Extract p-value from the t-test result
p_value <- t_test_result$p.value
formatted_p_value <- sprintf("P-value: %.3g", p_value)

# Generate the enhanced ggplot with the p-value in the legend
ggplot(results_melted, aes(x = SampleID, y = Count, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 4, color = "black") +
  labs(title = sprintf("Mettl3-KO/S2S: Total Exclusion : %d, Total Inclusion: %d", total_exclusion, total_inclusion),
       x = "Splicing Event Type", 
       y = "Number of Significant Splicing Events",
       fill = paste("Splicing Event (", formatted_p_value, ")", sep = "")) +
  scale_fill_manual(values = c("Exclusion" = "skyblue", "Inclusion" = "#ff7f0e")) +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "top",
        legend.title = element_text(face = "bold"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', color = 'gray')) +
  guides(fill = guide_legend(title = paste("Splicing Event (", formatted_p_value, ")", sep = "")))


# Print the t-test results
print(t_test_result)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Define vector of gene names
genes_of_interest <- c("Srrm234", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "kuz", "C1GalTA", "Sema1b", "Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega",   "Vinc", "Bsg", "Ptp4E", "dlg1", "Ank", "CG2225", "Nrg", "sd", "AP-1gamma", "Galphaq", "anne", "lilli", "CG9399", "Gbeta13F", "Pif1A", "Cals", "CG43110", "CG43998", "sau", "Ac3", "Ankle2", "Arp2", "Atg18a", "C3G", "spi", "CaMKII", "Acsl", "wdb", "LanB2", "CG31522", "Pvf2", "gce", "Tep2", "Adar", "Hsc70-4", "ABCD", "Tnpo")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0) %>%
    dplyr::filter(FDR < 0.05) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}
# Function to count splicing events per gene and splicing type
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extracting the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  return(gene_event_counts)
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt")

# Initialize a list to hold the results from each file
results_list <- lapply(files, count_splicing_events, genes = genes_of_interest)

# Combine results from all files into a single data frame
final_results <- bind_rows(results_list)

# Reshape the data to a wide format for inclusion and exclusion counts
final_results_wide <- final_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final table
print(final_results_wide)

#write.table(final_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_mettl3_KO_Sig_0_read_gene_of_intereset.txt", row.names = FALSE, quote = F, sep = "\t")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Define vector of gene names
genes_of_interest <- c("Srrm234", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b","Gbeta13F", "Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega", "CG16952", "wdb", "NK7.1")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and extract high-frequency exons
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extract the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  # Calculate exon frequencies and extract the highest frequency exons
  exon_frequency <- significant_genes %>%
    group_by(geneSymbol) %>%
    summarise(Frequency = n()) %>%
    arrange(desc(Frequency)) %>%
    slice(1:10) %>%  # Extract top 10 highest frequency exons
    mutate(EventType = event_type) # Add event type
  
  return(list(gene_counts = gene_event_counts, high_freq_exons = exon_frequency))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt", "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt")

# Initialize lists to hold results
gene_results_list <- list()
exon_results_list <- list()

for (file in files) {
  results <- count_splicing_events(file, genes_of_interest)
  gene_results_list <- append(gene_results_list, list(results$gene_counts))
  exon_results_list <- append(exon_results_list, list(results$high_freq_exons))
}

# Combine results from all files into single data frames
final_gene_results <- bind_rows(gene_results_list)
final_exon_results <- bind_rows(exon_results_list)

# Reshape the gene data to a wide format for inclusion and exclusion counts
final_gene_results_wide <- final_gene_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final tables
print(final_gene_results_wide)
print(final_exon_results)


```
        
## Exon skipping Analyses for METTL3-KO vs S2S

There are a total of 24762 events in SE. 21922 events after filtering reads that were less than 20 from each sample, and 229 events were recorded for when psi = 0.1 and FDR = 0.05

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))

#nrow(SE_mettl3_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO <- SE_mettl3_KO %>%
  #Split the replicate read counts that are separated by commas into different columns
  tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = T, convert = T) %>%
  tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = T, convert = T) %>%
  tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = T, convert = T) %>%
  tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = T, convert = T)

#nrow(SE_mettl3_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO_filt <- SE_mettl3_KO %>%
  mutate(., S1R1counts = IJC_S1R1 + SJC_S1R1) %>%
  mutate(., S1R2counts = IJC_S1R2 + SJC_S1R2) %>%
  mutate(., S1R3counts = IJC_S1R3 + SJC_S1R3) %>%
  mutate(., S2R1counts = IJC_S2R1 + SJC_S2R1) %>%
  mutate(., S2R2counts = IJC_S2R2 + SJC_S2R2) %>%
  mutate(., S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
  #Filter on read counts
  dplyr::filter(., S1R1counts > 0 & S1R2counts > 0 & S1R3counts > 0 &
           S2R1counts > 0 & S2R2counts > 0 & S2R3counts > 0)

nrow(SE_mettl3_KO_filt)
```

An important point we need to consider for downstream analysis is that exons with unregulated inclusion typically exhibit PSI (Percent Spliced In) values that are near 0 or close to 1, indicating that they are almost invariably included or excluded. On the other hand, exons with regulated inclusion often have PSI values that are more uniformly distributed across the spectrum between 0 and 1.

***Plot distribution of PSI values***

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO_filt_psi <-SE_mettl3_KO_filt %>%
  #Separate psi value replicates into individual columns
   tidyr::separate(., col = IncLevel1, into = c('PSI_S1R1', 'PSI_S1R2', 'PSI_S1R3'), sep = ',', remove = T, convert = T) %>%
   tidyr::separate(., col = IncLevel2, into = c('PSI_S2R1', 'PSI_S2R2', 'PSI_S2R3'), sep = ',', remove = T, convert = T) %>%
  #Select the columns we want
  dplyr::select(., c(contains('PSI'), FDR))

#Turn data from wide format into long format for plotting purposes

SE_mettl3_KO_filt_psi_long <- gather(SE_mettl3_KO_filt_psi, key = sample, value = psi, PSI_S1R1:PSI_S2R3) %>%
  # can tell that by asking if the substring 'S1' is somewhere in 'sample'
  mutate(., condition = ifelse(grepl('S1', sample), 'mettl3-KO', 'S2S')) %>%
  #Make a column indicating whether the FDR in this row is significant
  mutate(., sig = ifelse(FDR < 0.05, 'yes', 'no'))

#Plot
colors <- c('DarkOrange', 'DarkViolet')

#ggplot(SE_mettl3_KO_filt_psi_long, aes(x = psi, linetype = condition, color = sig)) +
#  geom_density() + 
#  facet_wrap(~sig, scales = "free_y", nrow = 2) +
#  scale_color_manual(values = colors)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#Make a matrix of psi values
S2S_mko.matrix <- dplyr::select(SE_mettl3_KO_filt_psi, -FDR) %>% na.omit(.)

#Use prcomp() to derive principle component coordinants of PSI values
psi.pca <- prcomp(t(S2S_mko.matrix))

#Add annotations of the conditions to the samples
psi.pca.pc <- data.frame(psi.pca$x, sample = colnames(S2S_mko.matrix)) %>%
  mutate(., condition = ifelse(grepl('S1', sample), 'mettl3-KO', 'S2S'))

#Get the amount of variances contained within PC1 and PC2
psi.pca.summary <- summary(psi.pca)$importance
pc1var = round(psi.pca.summary[2,1] * 100, 1)
pc2var = round(psi.pca.summary[2,2] * 100, 1)

#Plot PCA data
#ggplot(psi.pca.pc, aes(x = PC1, y = PC2, shape = condition, color = condition)) +
#  geom_point(size = 5)  +
#  scale_color_manual(values = colors) +
#  xlab(paste('PC1,', pc1var, '% explained var.')) + 
#  ylab(paste('PC2,', pc2var, '% explained var.'))
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter only significant events
psi_sig <- SE_mettl3_KO_filt_psi %>% 
  dplyr::filter(FDR < 0.05) %>%
  dplyr::select(-FDR)

# row scaled heatmap
#pheatmap(mat = psi_sig,
#         clustering_method = "ward.D2",
#         scale = "row",
#         show_rownames = F)
```


```{r message=FALSE, warning=FALSE, echo=FALSE} 
S2S_mettl3ko = c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/")

S2S_mettl3ko_data = maser(S2S_mettl3ko,c("METTLE-KO","S2S") , ftype = "JCEC")
#S2S_mettl3ko_data
```



There are a total of 1575 splicing events. Below is a break down of how many are in each Splicing type
```{r message=FALSE, warning=FALSE, echo=FALSE}
#S2S_mettl3ko_data_filt = filterByCoverage(S2S_mettl3ko_data, avg_reads = 50)
S2S_mettl3ko_data_top = topEvents(S2S_mettl3ko_data, fdr = 0.05, deltaPSI = 0.2)
#S2S_mettl3ko_data_top
```

     
     function to make PSI box for mettl3-ko and Ythdc1-ko relative to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}

# File paths configuration for METTL3-KO and Ythdc1-KO
file_paths_config <- list(
    METTL3_KO = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt"),
    Ythdc1_KO = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt"))

# Function to create box plots for PSI values
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO <- function(event_IDs, Splicing_type) {
    # Retrieve file paths from the configuration
    file_path_METTL3_KO <- file_paths_config$METTL3_KO[[Splicing_type]]
    file_path_Ythdc1_KO <- file_paths_config$Ythdc1_KO[[Splicing_type]]

    # Process METTL3-KO data
    data_METTL3_KO <- read.delim(file_path_METTL3_KO, header = TRUE)
    data_METTL3_KO_filtered <- data_METTL3_KO[data_METTL3_KO$ID == event_IDs[1], ]
    data_METTL3_KO_filtered$Dataset <- "METTL3-KO"

    # Process Ythdc1-KO data
    data_Ythdc1_KO <- read.delim(file_path_Ythdc1_KO, header = TRUE)
    data_Ythdc1_KO_filtered <- data_Ythdc1_KO[data_Ythdc1_KO$ID == event_IDs[2], ]
    data_Ythdc1_KO_filtered$Dataset <- "Ythdc1-KO"

    # Combine and reshape data
    combined_data <- rbind(data_METTL3_KO_filtered, data_Ythdc1_KO_filtered)
    psi_data <- data.frame(
        ID = combined_data$ID,
        IncLevel1 = combined_data$IncLevel1,
        IncLevel2 = combined_data$IncLevel2,
        Dataset = combined_data$Dataset)
    psi_data_long <- melt(psi_data, id.vars = c("ID", "Dataset"), variable.name = "Sample", value.name = "PSI")

    # Add Condition based on Dataset and Sample
    psi_data_long$Condition <- ifelse(
        psi_data_long$Dataset == "METTL3-KO",
        ifelse(psi_data_long$Sample == "IncLevel1", "METTL3-KO", "S2S"),
        ifelse(psi_data_long$Sample == "IncLevel1", "Ythdc1-KO", "S2S"))

    # Reorder the factor levels to put S2S as the first level
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("S2S", "METTL3-KO", "Ythdc1-KO"))

    # Split the concatenated PSI values
    psi_data_long$PSI <- as.character(psi_data_long$PSI)
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(PSI, ",")) %>%
        unnest(PSI)
    psi_data_long$PSI <- as.numeric(psi_data_long$PSI)

    # Create the box plot with statistical comparisons
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) +
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID + Dataset, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
            axis.text.y = element_text(size = 12),
            axis.title.x = element_text(face = "plain", colour = "black", size = 12),
            axis.title.y = element_text(face = "plain", colour = "black", size = 12),
            legend.title = element_blank(),
            legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
        scale_fill_manual(values = c("S2S" = "red", "METTL3-KO" = "blue", "Ythdc1-KO" = "green")) +
        stat_compare_means(aes(label = ..p.signif..),
            method = "t.test",
            comparisons = list(c("S2S", "METTL3-KO"), c("S2S", "Ythdc1-KO")))
    print(plot)
}
```

   
   
    function to process Ythdc1-GoF relative to S2 GFP and METTL3-KO 
```{r message=FALSE, warning=FALSE, echo=FALSE}

library(ggpubr)  # For stat_compare_means

# File paths configuration for METTL3_KO_GFP and Ythdc1_GoF
file_paths_config <- list(
    Ythdc1_GoF = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt"),
    METTL3_KO_GFP = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt"))

# Function to create box plots for PSI values
function_to_make_PSI_box_METTL3_KO_GFP_Ythdc1_GoF <- function(event_IDs, Splicing_type) {
    # Retrieve file paths from the configuration
    file_path_METTL3_KO_GFP <- file_paths_config$METTL3_KO_GFP[[Splicing_type]]
    file_path_Ythdc1_GoF <- file_paths_config$Ythdc1_GoF[[Splicing_type]]

    # Process METTL3-KO_GFP data
    data_KO_GFP <- read.table(file_path_METTL3_KO_GFP, header = TRUE)
    data_KO_GFP_filtered <- data_KO_GFP[data_KO_GFP$ID == event_IDs[1], ]
    data_KO_GFP_filtered$Dataset <- "METTL3-KO-GFP"

    # Process Ythdc1-GoF data
    data_GoF <- read.table(file_path_Ythdc1_GoF, header = TRUE)
    data_GoF_filtered <- data_GoF[data_GoF$ID == event_IDs[2], ]
    data_GoF_filtered$Dataset <- "Ythdc1-GoF"

    # Combine and reshape data
    combined_data <- rbind(data_KO_GFP_filtered, data_GoF_filtered)
    psi_data <- data.frame(
        ID = combined_data$ID,
        IncLevel1 = combined_data$IncLevel1,
        IncLevel2 = combined_data$IncLevel2,
        Dataset = combined_data$Dataset)
    psi_data_long <- melt(psi_data, id.vars = c("ID", "Dataset"), variable.name = "Sample", value.name = "PSI")

    # Add Condition based on Dataset and Sample
    psi_data_long$Condition <- ifelse(
        psi_data_long$Dataset == "METTL3-KO-GFP",
        ifelse(psi_data_long$Sample == "IncLevel1", "METTL3-YTHDC1-GoF", "METTL3-KO-GFP"),
        ifelse(psi_data_long$Sample == "IncLevel1", "Ythdc1-GoF", "S2S"))

    # Reorder the factor levels to put S2S as the first level
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("S2S", "METTL3-KO-GFP", "Ythdc1-GoF", ))

    # Split the concatenated PSI values
    psi_data_long$PSI <- as.character(psi_data_long$PSI)
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(PSI, ",")) %>%
        unnest(PSI)
    psi_data_long$PSI <- as.numeric(psi_data_long$PSI)

    # Create the box plot with statistical comparisons
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) +
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +  
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID + Dataset, scales = "free") +
        theme_bw() +
        theme(
            axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
            axis.text.y = element_text(size = 12),
            axis.title.x = element_text(face = "plain", colour = "black", size = 12),
            axis.title.y = element_text(face = "plain", colour = "black", size = 12),
            legend.title = element_blank(),
            legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
        scale_fill_manual(values = c("S2S" = "red", "METTL3-KO-GFP" = "blue", "Ythdc1-GoF" = "green")) +
        stat_compare_means(
            aes(label = ..p.signif..),
            method = "t.test",
            comparisons = list(c("S2S", "METTL3-KO-GFP"), c("S2S", "Ythdc1-GoF")))

    print(plot)
}

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
make_single_PSI_box_mettl3_ko <- function(event_IDs, Splicing_type) {
    # Load necessary libraries
    library(ggplot2)
    library(reshape2)
    library(dplyr)
    library(tidyr)
    library(ggpubr)  # For stat_compare_means
    
    # Define the path to your splicing files
    file_paths <- list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt")
    
    # Check if the specified splicing type is available
    if (!Splicing_type %in% names(file_paths)) {
        stop("Splicing type not found in the list of files.")
    }

    # Read the data file for the specified splicing type
    file_path <- file_paths[[Splicing_type]]
    data <- read.table(file_path, header = TRUE)

    # Filter for specified event IDs
    event_data <- data[data$ID %in% event_IDs, ]

    # Reshape the data
    psi_data <- data.frame(ID = event_data$ID, IncLevel1 = event_data$IncLevel1, IncLevel2 = event_data$IncLevel2)
    psi_data_long <- melt(psi_data, id.vars = "ID", variable.name = "Sample", value.name = "PSI")
    psi_data_long$Condition <- ifelse(psi_data_long$Sample == "IncLevel1", "METTL3-KO", "S2S")

    # Reorder the factor levels to put S2S as the first level
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("S2S", "METTL3-KO"))

    # Split the concatenated PSI values and convert them to numeric
    psi_data_long$PSI <- as.character(psi_data_long$PSI)
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(PSI, ",")) %>%
        unnest(PSI) %>%
        mutate(PSI = as.numeric(PSI))

    # Create the box plot with jitter and statistical comparisons
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) +  # Add boxplot without outliers
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +  # Add jitter
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
        labs(fill = "Condition") + 
        scale_fill_manual(values = c("S2S" = "red", "METTL3-KO" = "blue")) +
        stat_compare_means(aes(label = ..p.signif..), method = "t.test", comparisons = list(c("S2S", "METTL3-KO")))

    print(plot)
}

# Example usage
#make_single_PSI_box_m(event_IDs = c("3972", "3992"), Splicing_type = "SE")

```

    ## SE METTL3-KO relative to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}
### SE for S2S vs METTL3-KO: Fig shows significant alternatively spliced genes in METTL3-KO relative to S2S. Transformed the FDR values using a -1 \* log10 to get the Log10 Adj. P-value. 
SE_mettl3_KO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt", header = T)


SE_mettl3_KO <- SE_mettl3_KO %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(SE_mettl3_KO)
```

## Volcano plot shows significantly spliced *SE* in METTL3-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO$logPValue <- -1 * log10(SE_mettl3_KO$FDR)
# Create a binary inclusion/exclusion column
SE_mettl3_KO$Splicing <- ifelse(SE_mettl3_KO$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(SE_mettl3_KO, FDR < 0.05 & abs(IncLevelDifference) > 0.1)

# Update colors for significant and non-significant genes in the main dataframe
SE_mettl3_KO$Color <- ifelse(SE_mettl3_KO$FDR < 0.05 & abs(SE_mettl3_KO$IncLevelDifference) > 0.1, SE_mettl3_KO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(SE_mettl3_KO$Color)

# Update the scale_color_manual function with new labels including counts
ggplot(SE_mettl3_KO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "blue"),
                     labels = c(paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.1, 0.1), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Exon skipping events in METTL3-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 
```

```{r message=FALSE, warning=FALSE, echo=FALSE} 

if ("GeneID" %in% colnames(SE_mettl3_KO)) {
  
  # Make GeneID unique
  SE_mettl3_KO$GeneID <- make.unique(as.character(SE_mettl3_KO$GeneID))
  
  # Set GeneID as rownames
  rownames(SE_mettl3_KO) <- SE_mettl3_KO$GeneID
  # SE_mettl3_KO$GeneID <- NULL
  
  # Now, you can use the mapIds function
  SE_mettl3_KO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_mettl3_KO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
  SE_mettl3_KO$GO <- mapIds(org.Dm.eg.db,
                             keys = rownames(SE_mettl3_KO),
                             column = "GO",
                             keytype = "ENSEMBL",
                             multiVals = "first")

  
  # Retrieve GO term descriptions
  SE_mettl3_KO$GODESCR <- sapply(SE_mettl3_KO$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  SE_mettl3_KO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                                   keys = rownames(SE_mettl3_KO),
                                   column = "ONTOLOGY",
                                   keytype = "ENSEMBL",
                                   multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}
#write.table(SE_mettl3_KO, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_mettl3_KO_S2S.txt', quote = F, sep = "\t", row.names = F)
```



***Sxl:chrX:7,087,672-7,091,875*** PCR validated loci. There are a total of 17 splicing events for Sxl. Most of the Splicing event was recorded as Exon Skipping
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Sxl = geneEvents(S2S_mettl3ko_data_top, geneS = "Sxl", fdr = 0.05, deltaPSI = 0.2)
print(S2S_mettl3ko_Sxl)
```

*Box plot of PSI values  for Sxl Exon skipping event shows that these exon are more included in S2S than METTL3-KO*
```{r message=FALSE, warning=FALSE, echo=FALSE}
#plotGenePSI(S2S_mettl3ko_Sxl, type = "SE", show_replicates = TRUE)METTL3_KO_S2S_vol_SE
S2S_mettl3ko_Sxl@SE_stats
make_single_PSI_box_mettl3_ko(event_ID = "3973", Splicing_type = "SE")
make_single_PSI_box_mettl3_ko(event_ID = "17946", Splicing_type = "SE")
make_single_PSI_box_mettl3_ko(event_ID = "17948", Splicing_type = "SE")
#make_single_PSI_box_mettl3_ko(event_ID = "17946", Splicing_type = "SE")
#make_single_PSI_box_mettl3_ko(event_ID = "17948", Splicing_type = "SE")
#make_single_PSI_box_mettl3_ko(event_ID = "17953", Splicing_type = "SE")
```






## A3SS splicng event and mechanism: Altenative 3'(A3SS) is caused by change in the acceptor sites

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3SS_mko<- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt", header = T) #%>%

A3SS_mko <- A3SS_mko %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A3SS_mko)
```

## Volcano plot shows significantly spliced *A3SS* in METTL3-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3SS_mko$logPValue <- -1 * log10(A3SS_mko$FDR)
# Create a binary inclusion/exclusion column
A3SS_mko$Splicing <- ifelse(A3SS_mko$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A3SS_mko, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A3SS_mko$Color <- ifelse(A3SS_mko$FDR < 0.05 & abs(A3SS_mko$IncLevelDifference) > 0.2, A3SS_mko$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A3SS_mko$Color)

# Update the scale_color_manual function with new labels including counts
ggplot(A3SS_mko, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1,1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "blue"),
                     labels = c(paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "A3SS events in METTL3-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 
```

```{r message=FALSE, warning=FALSE, echo=FALSE} 
if ("GeneID" %in% colnames(A3SS_mko)) {
  
  # Make GeneID unique
  A3SS_mko$GeneID <- make.unique(as.character(A3SS_mko$GeneID))
  
  # Set GeneID as rownames
  rownames(A3SS_mko) <- A3SS_mko$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  # A3SS_mko$GeneID <- NULL
  
  # Now, you can use the mapIds function
  A3SS_mko$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3SS_mko),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
  A3SS_mko$GO <- mapIds(org.Dm.eg.db,
                             keys = rownames(A3SS_mko),
                             column = "GO",
                             keytype = "ENSEMBL",
                             multiVals = "first")

  
  # Retrieve GO term descriptions
  A3SS_mko$GODESCR <- sapply(A3SS_mko$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  A3SS_mko$ONTOLOGY <- mapIds(org.Dm.eg.db,
                                   keys = rownames(A3SS_mko),
                                   column = "ONTOLOGY",
                                   keytype = "ENSEMBL",
                                   multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}

#write.table(A3SS_mko, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS_S2S_METTL3.txt", quote = F, sep = "\t", row.names = F)

```


## A5SS splicng event and mechanism: METTL3-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5SS_mko<- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt", header = T) #%>%

  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(A5SS_mko)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5SS_mko <- A5SS_mko %>%
 tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A5SS_mko)
```



## Volcano plot shows significantly spliced *A5*  in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5SS_mko$logPValue <- -1 * log10(A5SS_mko$FDR)
# Create a binary inclusion/exclusion column
A5SS_mko$Splicing <- ifelse(A5SS_mko$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A5SS_mko, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A5SS_mko$Color <- ifelse(A5SS_mko$FDR < 0.05 & abs(A5SS_mko$IncLevelDifference) > 0.2, A5SS_mko$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label volc_SE_METTL3-KO_S2S
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A5SS_mko$Color)

# Update the scale_color_manual function with new labels including counts
ggplot(A5SS_mko, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1,1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "blue"),
                     labels = c(paste("Increased -", counts["Increased"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced - ", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "A5SS event transcripts/genes in METTL3-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}

if ("GeneID" %in% colnames(A5SS_mko)) {
  
# Make GeneID unique
  A5SS_mko$GeneID <- make.unique(as.character(A5SS_mko$GeneID))
  
  # Set GeneID as rownames
  rownames(A5SS_mko) <- A5SS_mko$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A5SS_mko$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A5SS_mko$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5SS_mko),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A5SS_mko$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5SS_mko),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A5SS_mko$GODESCR <- sapply(A5SS_mko$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A5SS_mko$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5SS_mko),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}

#write.table(A5SS_mko, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS_mko.txt", quote = F, sep = "\t", row.names = F)
```


Differenrial expressed genes in A5 METTL3-KO S2S
```{r}
sig_METTL3_KO_S2S = res_S2_m6A %>%
  dplyr::filter(diffexpressed != "Not Significant")

mettl3_ko_genes <- unique(sig_METTL3_KO_S2S$validated_id) 

A5_mettl3_KO_genes = unique(A5SS_mko$GeneID)
# Filter out transcripts by removing suffixes
A5_mettl3_KO_genes_clean <- unique(sub("\\..*", "", A5_mettl3_KO_genes))


# Update gene sets for Venn diagram
gene_sets <- list(
  "DEG METTL3-KO vs S2S" = mettl3_ko_genes,
  "A5 METTL3-KO vs S2S" = A5_mettl3_KO_genes_clean)

# Generate the Venn diagram
ggvenn(gene_sets, fill_color = c("#E69F00", "#56B4E9"), stroke_size = 0.5, text_size = 5,  show_percentage = FALSE, auto_scale = TRUE) +
  labs(title = "Number of genes shared between METTL3-KO/S2S and A5'SS") +
  theme(plot.title = element_text(hjust = 0.5, size = 16))


```

## MXE METLL3-KO relavtive to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_MKO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_S2_MKO)
``` 

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_MKO <- MXE_S2_MKO %>%
 tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(MXE_S2_MKO)
```



## Volcano plot shows significantly spliced *SE*  in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_MKO$logPValue <- -1 * log10(MXE_S2_MKO$FDR)
# Create a binary inclusion/exclusion column
MXE_S2_MKO$Splicing <- ifelse(MXE_S2_MKO$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(MXE_S2_MKO, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
MXE_S2_MKO$Color <- ifelse(MXE_S2_MKO$FDR < 0.05 & abs(MXE_S2_MKO$IncLevelDifference) > 0.2, MXE_S2_MKO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(MXE_S2_MKO$Color)

# Update the scale_color_manual function with new labels including counts 
ggplot(MXE_S2_MKO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1,1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "blue"),
                     labels = c(paste("Increased -", counts["Increased"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced - ", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "MXE event transcripts/genes in METTL3-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}

if ("GeneID" %in% colnames(MXE_S2_MKO)) {
  
# Make GeneID unique
  MXE_S2_MKO$GeneID <- make.unique(as.character(MXE_S2_MKO$GeneID))
  
  # Set GeneID as rownames
  rownames(MXE_S2_MKO) <- MXE_S2_MKO$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #MXE_S2_MKO$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  MXE_S2_MKO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_MKO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   MXE_S2_MKO$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_MKO),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
MXE_S2_MKO$GODESCR <- sapply(MXE_S2_MKO$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  MXE_S2_MKO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_MKO),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}

#write.table(MXE_S2_MKO, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE_S2_MKO.txt", quote = F, sep = "\t", row.names = F)
```


## RI METLL3-KO relavtive to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_MKO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(RI_S2_MKO)
``` 

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_MKO <- RI_S2_MKO %>%
 tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(RI_S2_MKO)
```

## Volcano plot shows significantly spliced *RI_S2_MKO*  in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_MKO$logPValue <- -1 * log10(RI_S2_MKO$FDR)
# Create a binary inclusion/exclusion column
RI_S2_MKO$Splicing <- ifelse(RI_S2_MKO$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(RI_S2_MKO, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
RI_S2_MKO$Color <- ifelse(RI_S2_MKO$FDR < 0.05 & abs(RI_S2_MKO$IncLevelDifference) > 0.2, RI_S2_MKO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(RI_S2_MKO$Color)

# Update the scale_color_manual function with new labels including counts volc_SE_METTL3-KO_S2S
ggplot(RI_S2_MKO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "blue"),
                     labels = c(paste("Increased -", counts["Increased"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced - ", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "RI event transcripts/genes in Mettl3-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}

if ("GeneID" %in% colnames(RI_S2_MKO)) {
  
 # Make GeneID unique
  RI_S2_MKO$GeneID <- make.unique(as.character(RI_S2_MKO$GeneID))
  
  # Set GeneID as rownames
  rownames(RI_S2_MKO) <- RI_S2_MKO$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #RI_S2_MKO$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  RI_S2_MKO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_MKO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   RI_S2_MKO$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_MKO),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
RI_S2_MKO$GODESCR <- sapply(RI_S2_MKO$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  RI_S2_MKO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_MKO),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}


#write.table(RI_S2_MKO, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI_S2_MKO.txt", quote = F, sep = "\t", row.names = F)
```




***Differentially splice genes in METTL3-KO relative to S2S*** 


```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to prepare data with read filtering steps
prepare_data <- function(file_path, event_type) {
  # Read the data
  data <- read.table(file_path, header = TRUE, sep = "\t")
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)

  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  # Add necessary columns
  data <- data %>%
    mutate(delta_PSI = IncLevelDifference,
           Significant = ifelse(FDR < 0.05 & abs(delta_PSI) > 0.2, "Significant", "Not Significant"),
           EventType = event_type,
           Event_Types = ifelse(Significant == "Significant", event_type, "Not Significant"))
  
  # Select necessary columns
  data <- data[, c("geneSymbol", "delta_PSI", "FDR", "Significant", "EventType", "Event_Types")]
  
  return(data)
}


file_paths <- list(
  SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
  A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
  A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt",
  MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
  RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt"
)

event_types <- names(file_paths)

# Extract significant genes for each event type
significant_genes_list <- mapply(prepare_data, file_paths, event_types, SIMPLIFY = FALSE)

# Combine all splicing events into a single dataframe
all_genes <- do.call(rbind, significant_genes_list)

# Calculate counts for each event type
event_counts <- as.data.frame(table(all_genes$Event_Types))
colnames(event_counts) <- c("Event_Types", "Count")

# Define colors for splicing event types
type_colors <- c("SE" = "blue", "A3SS" = "black", "A5SS" = "darkgreen", "MXE" = "red", "RI" = "#984EA3", "Not Significant" = "grey")

# Identify top genes (top 100 based on FDR and delta_PSI)
top_genes <- all_genes[order(all_genes$FDR, -abs(all_genes$delta_PSI)),][1:1000,]

# Prepare custom labels with counts for the legend
custom_labels <- with(event_counts, setNames(paste(Event_Types, " (", Count, ")", sep=""), Event_Types))


ggplot(all_genes, aes(x = delta_PSI, y = -log10(FDR), color = Event_Types)) +
  geom_point(alpha = 0.5) +
  scale_color_manual(values = type_colors, labels = custom_labels) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype = "dashed", color = "black") +
  theme_minimal() + xlim(-1.0, 1.0) +
  labs(x = "", y = "Log10 Adj. P-value", title = "Significant spliced transcripts in METTL3-KO relative to S2S cells") +
  theme(legend.position = "right", plot.title = element_text(hjust = 0.5)) +
  # Use geom_text_repel to remove boxes and filter for significant genes only
  geom_text_repel(data = top_genes %>% dplyr::filter(Significant == "Significant"), 
                  aes(label = geneSymbol), size = 3, 
                  box.padding = 0.35, point.padding = 0.5, max.overlaps = 10)
```



Below are the genes implicated in METTL3-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}

# Define file paths for splicing data
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt")

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_S2_m6A$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

# Add differential expression information
splicing_diff <- splicing_diff %>%
  left_join(res_S2_m6A %>% dplyr::select(current_symbol, log2FoldChange, padj), by = c("geneSymbol" = "current_symbol")) %>%
  mutate(Regulation_DE = ifelse(log2FoldChange > 2, "Upregulated", ifelse(log2FoldChange < -0.5, "Downregulated", "Not Significant")))

# Exclude "Not Significant" genes
splicing_diff <- splicing_diff %>% dplyr::filter(Regulation_DE != "Not Significant")

# Create the dot plot
ggplot(splicing_diff, aes(x = log2FoldChange, y = IncLevelDifference, color = Regulation_DE)) +
  geom_point(aes(size = -log10(FDR)), alpha = 0.8) +
  scale_color_manual(values = c("Upregulated" = "red", "Downregulated" = "blue", "Not Significant" = "grey")) +
  labs(x = "log2 Fold Change", y = "Inclusion Level Difference ()", color = "Differential Expression", size = "-log10(FDR)") +
  theme_minimal() +
  geom_text_repel(aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  ggtitle("Significantly Spliced and Differentially Expressed Genes in METTL3-KO relative to S2S") +
    labs(x = "", y = "Log10 Adj. P-value", title = "Significantly Spliced and Differentially Expressed Genes in METTL3-KO relative to S2S") +
    theme(legend.position = "right", plot.title = element_text(hjust = 0.5))
 #+
 # theme(
 #   text = element_text(size = 20), # Change font size of all text
  #  axis.text = element_text(size = 20), # Change font size of axis text
 #   axis.title = element_text(size = 20), # Change font size of axis titles
 #   plot.title = element_text(size = 20, hjust = 0.5), # Change font size and center plot title
 #   legend.text = element_text(size = 20), # Change font size of legend text
  #  legend.title = element_text(size = 20) # Change font size of legend title
  #)
```


What proportion of METTL3-KO differential expressed genes are implicated in splicing 
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter significant and non-significant genes
significant_genes <- res_S2_m6A %>% 
  dplyr::filter(diffexpressed != "Not Significant")
non_significant_genes <- res_S2_m6A %>% 
  dplyr::filter(diffexpressed == "Not Significant")

# Function to count the number of genes involved in each splicing event type
count_genes_by_event <- function(data, gene_list) {
  data %>%
    dplyr::filter(geneSymbol %in% gene_list$current_symbol) %>%
    group_by(EventType) %>%
    summarise(
      Count = n(),
      GeneNames = paste(geneSymbol, collapse = ", ")  # Collect gene names
    )
}

# Count for significant DEGs
sig_event_counts <- count_genes_by_event(all_significant_genes, significant_genes)
sig_event_counts <- sig_event_counts %>% mutate(Significance = "Differentially Expressed")

# Count for non-significant DEGs
non_sig_event_counts <- count_genes_by_event(all_significant_genes, non_significant_genes)
non_sig_event_counts <- non_sig_event_counts %>% mutate(Significance = "Not Differentially Expressed")

# Combine the counts into one dataframe
combined_event_counts <- bind_rows(sig_event_counts, non_sig_event_counts)

# Reorder the Significance factor to ensure "Significant" is on top
combined_event_counts$Significance <- factor(combined_event_counts$Significance, 
                                             levels = c("Not Differentially Expressed", "Differentially Expressed"))

# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])

# Extract the p-value and format it
p_value <- chi_square_result$p.value
p_value_formatted <- sprintf("Chi-Square Test p-value: %.2e", p_value)


plot <- plot_ly(combined_event_counts, 
                x = ~EventType, 
                y = ~Count, 
                color = ~Significance, 
                type = 'bar', 
                text = ~Count,  # Display the count number on each bar
                textposition = 'auto',  # Position the text within the bars
                textfont = list(color = "black"),  # Set the color of the text to black
                hoverinfo = 'text') %>%
  layout(
    title = "Number of DEGs for Each Splicing Event Type in METTL3-KO relative to S2S",
    xaxis = list(title = "Splicing Event Type"),
    yaxis = list(title = "Number of Genes"),
    barmode = 'stack',
    legend = list(
      title = list(text = p_value_formatted),  # Add the p-value to the legend title
      traceorder = "reversed"))

# Display the plot
plot



# Define custom colors
custom_colors <- c("Not Differentially Expressed" = "#a6bddb", "Differentially Expressed" = "#de2d26")

# Create the ggplot with p-value in legend
ggplot(combined_event_counts, aes(x = EventType, y = Count, fill = Significance)) +
  geom_bar(stat = "identity", position = "stack", width = 0.7) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 5, fontface = "bold", color = "black") +
  scale_fill_manual(values = custom_colors) +
  labs(
    title = "DEGs for each splicing event in METTL3-KO/S2S",
    x = "Splicing Event Type",
    y = "Number of Genes",
    fill = sprintf("Gene Expression Status\n(p = %.2e)", p_value)) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank())
```
  
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)


```

```{r}
# Calculate the total number of splicing genes
total_splicing_genes <- length(unique(all_significant_genes$geneSymbol))

# Calculate the number and percentage of DEGs implicated in splicing
num_diff_splicing_genes <- length(unique(splicing_diff$geneSymbol))
percent_diff_splicing_genes <- (num_diff_splicing_genes / total_splicing_genes) * 100

# Print the summary
cat("Total number of genes implicated in splicing:", total_splicing_genes, "\n")
cat("Number of DEGs implicated in splicing:", num_diff_splicing_genes, "\n")
cat("Percentage of DEGs implicated in splicing:", round(percent_diff_splicing_genes, 2), "%\n")

# Optional: Print the number of splicing events per event type
splicing_event_counts <- all_significant_genes %>%
  group_by(EventType) %>%
  summarise(Count = n())

cat("\nNumber of significant splicing events by type:\n")
print(splicing_event_counts)

```

    
    
    Ythdc1-KO RELATIVE TO S2S (from previous data sets) 


```{r message=FALSE, warning=FALSE, echo=FALSE}
make_single_PSI_box_S2S_YHTDC1_KO <- function(event_IDs, Splicing_type) {
    # Define the path to your splicing files
    file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt")
    
    # Check if the specified splicing type is available
    if (!Splicing_type %in% names(file_paths)) {
        stop("Splicing type not found in the list of files.")
    }

    # Read the data file for the specified splicing type
    file_path <- file_paths[[Splicing_type]]
    data <- read.table(file_path, header = TRUE)

    # Filter for specified event IDs
    event_data <- data[data$ID %in% event_IDs, ]

    # Reshape the data
    psi_data <- data.frame(ID = event_data$ID, IncLevel1 = event_data$IncLevel1, IncLevel2 = event_data$IncLevel2)
    psi_data_long <- melt(psi_data, id.vars = "ID", variable.name = "Sample", value.name = "PSI")
    psi_data_long$Condition <- ifelse(psi_data_long$Sample == "IncLevel1", "Ythdc1-KO", "S2S")

    # Reorder the factor levels to put S2S as the first level
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("Ythdc1-KO", "S2S"))

    # Split the concatenated PSI values and convert them to numeric 
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(PSI, ",")) %>%
        unnest(PSI) %>%
        mutate(PSI = as.numeric(PSI))

    # Create the box plot with statistical comparisons
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
      geom_boxplot(alpha = 0.6, outlier.shape = NA) +  
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) + 
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
       labs(fill = "Condition") + 
        scale_fill_manual(values = c("S2S" = "red", "Ythdc1-KO" = "blue")) +
        stat_compare_means(aes(label = ..p.signif..), method = "t.test", comparisons = list(c("Ythdc1-KO", "S2S")))

    print(plot)
}

# Example usage
#make_single_PSI_box_m(event_IDs = c("3972", "3992"), Splicing_type = "SE")

```

  
  
    Bar chart below show the number of significant splicing difference in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}

prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)
  
  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  return(data)
}

# Function to count splicing events after filtering
count_splicing_events <- function(filename) {
  data <- prepare_and_filter_data(filename)
  
  # Filter for significant genes
  significant_genes <- subset(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  n_exclusion <- sum(significant_genes$IncLevelDifference < 0)
  n_inclusion <- sum(significant_genes$IncLevelDifference > 0)
  
  sample_id <- basename(filename)
  sample_id <- gsub(".MATS.JCEC.txt", "", sample_id)
  
  return(data.frame(SampleID = sample_id, Exclusion = n_exclusion, Inclusion = n_inclusion))
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt")

# Apply function to each file and combine results
results <- do.call(rbind, lapply(files, count_splicing_events))

# Calculate totals
total_exclusion <- sum(results$Exclusion)
total_inclusion <- sum(results$Inclusion)

# Melt the results data for ggplot
results_melted <- tidyr::pivot_longer(results, cols = c(Exclusion, Inclusion), names_to = "EventType", values_to = "Count")
# Perform a paired t-test
t_test_result <- t.test(results$Inclusion, results$Exclusion, paired = TRUE)

# Extract p-value from the t-test result
p_value <- t_test_result$p.value
formatted_p_value <- sprintf("P-value: %.3g", p_value)

# Generate the enhanced ggplot with the p-value in the legend
ggplot(results_melted, aes(x = SampleID, y = Count, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 4, color = "black") +
  labs(title = sprintf("Ythdc1-KO relative to S2S: Total Exclusion : %d, Total Inclusion: %d", total_exclusion, total_inclusion),
       x = "Splicing Event Type", 
       y = "Number of Significant Splicing Events",
       fill = paste("Splicing Event (", formatted_p_value, ")", sep = "")) +
  scale_fill_manual(values = c("Exclusion" = "#377EB8", "Inclusion" = "#ff7f0e")) +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "top",
        legend.title = element_text(face = "bold"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', color = 'gray')) +
  guides(fill = guide_legend(title = paste("Splicing Event (", formatted_p_value, ")", sep = "")))


# Print the t-test results
print(t_test_result)


```



    Below are the number of spliced events in Ythdc1-KO relative to S2S that might be of interest to us and validate some of the genes from PCR
```{r message=FALSE, warning=FALSE, echo=FALSE}


genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega",   "Bsg", "NK7.1", "alpha-Man-Ia", "Vinc", "Ptp4E", "shi", "RpL15", "scrib", "dlg1", "PlexA", "CG13124", "Pka-R2", "Tep2", "mfas", "CG13366", "CG45050", "Fer1HCH", "CG11357", "CG2767", "CaMKII", "ND-B12", "ATPCL", "AnxB11", "Best1", "CDC50", "CG10616", "CG32772", "Cals", "sd", "CG4822", "Dscam1", "l(2)k01209", 
"par-1", "CkIIalpha")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(
      S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and splicing type
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extracting the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  return(gene_event_counts)
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt")

# Initialize a list to hold the results from each file
results_list <- lapply(files, count_splicing_events, genes = genes_of_interest)

# Combine results from all files into a single data frame
final_results <- bind_rows(results_list)

# Reshape the data to a wide format for inclusion and exclusion counts
final_results_wide <- final_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final table
print(final_results_wide)

#Write the final results to a file
#write.table(final_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/S2S_DC1_KO_0_Sig_gene_of_interest.txt", row.names = FALSE, quote = FALSE, sep = "\t")

```

frequency of gene

```{r}
# Define vector of gene names
genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, ID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and extract high-frequency exons
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extract the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  # Calculate exon frequencies and extract the highest frequency exons
  exon_frequency <- significant_genes %>%
    group_by(geneSymbol) %>%
    summarise(Frequency = n()) %>%
    arrange(desc(Frequency)) %>%
    slice(1:10) %>%  # Extract top 10 highest frequency exons
    mutate(EventType = event_type) # Add event type
  
  return(list(gene_counts = gene_event_counts, high_freq_exons = exon_frequency))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt")

# Initialize lists to hold results
gene_results_list <- list()
exon_results_list <- list()

for (file in files) {
  results <- count_splicing_events(file, genes_of_interest)
  gene_results_list <- append(gene_results_list, list(results$gene_counts))
  exon_results_list <- append(exon_results_list, list(results$high_freq_exons))
}

# Combine results from all files into single data frames
final_gene_results <- bind_rows(gene_results_list)
final_exon_results <- bind_rows(exon_results_list)

# Reshape the gene data to a wide format for inclusion and exclusion counts
final_gene_results_wide <- final_gene_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final tables
print(final_gene_results_wide)
print(final_exon_results)

# Write the final tables to files
#write.table(final_gene_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/Sig_gene_of_interest.txt", row.names = FALSE, quote = F, sep = "\t")
#write.table(final_exon_results, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/High_frequency_exons.txt", row.names = FALSE, quote = F, sep = "\t")

```



```         
### SE Ythdc1-KO relative to S2S
```

There are a total of 31466 A5SS genes or transcripts and 18330 after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2S_DC1_KO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(SE_S2S_DC1_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2S_DC1_KO <- SE_S2S_DC1_KO %>%
 tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(SE_S2S_DC1_KO)
```

## Volcano plot shows significantly spliced *SE*  in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2S_DC1_KO$logPValue <- -1 * log10(SE_S2S_DC1_KO$FDR)
# Create a binary inclusion/exclusion column
SE_S2S_DC1_KO$Splicing <- ifelse(SE_S2S_DC1_KO$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(SE_S2S_DC1_KO, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
SE_S2S_DC1_KO$Color <- ifelse(SE_S2S_DC1_KO$FDR < 0.05 & abs(SE_S2S_DC1_KO$IncLevelDifference) > 0.2, SE_S2S_DC1_KO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(SE_S2S_DC1_KO$Color)

# Update the scale_color_manual function with new labels including counts
ggplot(SE_S2S_DC1_KO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) +  xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "navy"),
                     labels = c(paste("Increased -", counts["Increased"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced - ", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "SE event transcripts/genes in Ythdc1-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5))+
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(SE_S2S_DC1_KO)) {
  
  # Make GeneID unique
  SE_S2S_DC1_KO$GeneID <- make.unique(as.character(SE_S2S_DC1_KO$GeneID))
  
  # Set GeneID as rownames
  rownames(SE_S2S_DC1_KO) <- SE_S2S_DC1_KO$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  # SE_S2S_DC1_KO$GeneID <- NULL
  
  # Now, you can use the mapIds function
  SE_S2S_DC1_KO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2S_DC1_KO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
  SE_S2S_DC1_KO$GO <- mapIds(org.Dm.eg.db,
                             keys = rownames(SE_S2S_DC1_KO),
                             column = "GO",
                             keytype = "ENSEMBL",
                             multiVals = "first")

  
  # Retrieve GO term descriptions
  SE_S2S_DC1_KO$GODESCR <- sapply(SE_S2S_DC1_KO$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  SE_S2S_DC1_KO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                                   keys = rownames(SE_S2S_DC1_KO),
                                   column = "ONTOLOGY",
                                   keytype = "ENSEMBL",
                                   multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}

#write.table(SE_S2S_DC1_KO, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE_S2S_DC1_KO.txt", quote = F, sep = "\t", row.names = F)
```

```         
### A3 Ythdc1-KO relative to S2S
```

There are a total of 15559 A3SS genes or transcripts and 10611 after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2S_DC1_KO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
nrow(A3_S2S_DC1_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2S_DC1_KO <- A3_S2S_DC1_KO %>%
 tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A3_S2S_DC1_KO)

```

## Volcano plot shows significantly spliced *A3SS*  in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2S_DC1_KO$logPValue <- -1 * log10(A3_S2S_DC1_KO$FDR)
# Create a binary inclusion/exclusion column
A3_S2S_DC1_KO$Splicing <- ifelse(A3_S2S_DC1_KO$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A3_S2S_DC1_KO, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A3_S2S_DC1_KO$Color <- ifelse(A3_S2S_DC1_KO$FDR < 0.05 & abs(A3_S2S_DC1_KO$IncLevelDifference) > 0.2, A3_S2S_DC1_KO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A3_S2S_DC1_KO$Color)

# Update the scale_color_manual function with new labels including counts volc_SE_ythdc1-KO_S2S
ggplot(A3_S2S_DC1_KO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "navy"),
                     labels = c(paste("Increased - ", counts["Increased"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced - ", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "A3SS event transcripts/genes in Ythdc1-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A3_S2S_DC1_KO)) {
  # Make GeneID unique
  A3_S2S_DC1_KO$GeneID <- make.unique(as.character(A3_S2S_DC1_KO$GeneID))
  
  # Set GeneID as rownames
  rownames(A3_S2S_DC1_KO) <- A3_S2S_DC1_KO$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  # A3_S2S_DC1_KO$GeneID <- NULL
  
  # Now, you can use the mapIds function
  A3_S2S_DC1_KO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_S2S_DC1_KO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
  A3_S2S_DC1_KO$GO <- mapIds(org.Dm.eg.db,
                             keys = rownames(A3_S2S_DC1_KO),
                             column = "GO",
                             keytype = "ENSEMBL",
                             multiVals = "first")

  
  # Retrieve GO term descriptions
  A3_S2S_DC1_KO$GODESCR <- sapply(A3_S2S_DC1_KO$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  A3_S2S_DC1_KO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                                   keys = rownames(A3_S2S_DC1_KO),
                                   column = "ONTOLOGY",
                                   keytype = "ENSEMBL",
                                   multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}
 

#write.table(A3_S2S_DC1_KO, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3_S2S_DC1_KO.txt", quote = F, sep = "\t", row.names = F)
```


```         
### A5 Ythdc1-KO relative to S2S
```

There are a total of 31466 A5SS genes or transcripts and 18330 after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2S_DC1_KO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(A5_S2S_DC1_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2S_DC1_KO <- A5_S2S_DC1_KO %>%
 tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A5_S2S_DC1_KO)
```

## Volcano plot shows significantly spliced *A5SS*  in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2S_DC1_KO$logPValue <- -1 * log10(A5_S2S_DC1_KO$FDR)
# Create a binary inclusion/exclusion column
A5_S2S_DC1_KO$Splicing <- ifelse(A5_S2S_DC1_KO$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A5_S2S_DC1_KO, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A5_S2S_DC1_KO$Color <- ifelse(A5_S2S_DC1_KO$FDR < 0.05 & abs(A5_S2S_DC1_KO$IncLevelDifference) > 0.2, A5_S2S_DC1_KO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A5_S2S_DC1_KO$Color)

# Update the scale_color_manual function with new labels including counts volc_A3SS_ythdc1-KO_S2S
ggplot(A5_S2S_DC1_KO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1,1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "navy"),
                     labels = c(paste("Increased - ", counts["Increased"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced - ", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "A5SS event transcripts/genes in Ythdc1-KO relative to S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A5_S2S_DC1_KO)) {
  
A5_S2S_DC1_KO$GeneID <- make.unique(as.character(A5_S2S_DC1_KO$GeneID))
rownames(A5_S2S_DC1_KO) <- A5_S2S_DC1_KO$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  # A5_S2S_DC1_KO$GeneID <- NULL
  
  # Now, you can use the mapIds function
  A5_S2S_DC1_KO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_S2S_DC1_KO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first") 
  
  A5_S2S_DC1_KO$GO <- mapIds(org.Dm.eg.db,
                             keys = rownames(A5_S2S_DC1_KO),
                             column = "GO",
                             keytype = "ENSEMBL",
                             multiVals = "first")

  
  # Retrieve GO term descriptions
  A5_S2S_DC1_KO$GODESCR <- sapply(A5_S2S_DC1_KO$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  A5_S2S_DC1_KO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                                   keys = rownames(A5_S2S_DC1_KO),
                                   column = "ONTOLOGY",
                                   keytype = "ENSEMBL",
                                   multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}

#write.table(A5_S2S_DC1_KO, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO//A5_S2S_DC1_KO.txt", quote = F, sep = "\t", row.names = F)
```

```         
### RI Ythdc1-KO relative to S2S
```

There are a total of 6600 RI genes after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2S_DC1_KO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
nrow(RI_S2S_DC1_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2S_DC1_KO <- RI_S2S_DC1_KO %>%
 tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(RI_S2S_DC1_KO)
```

## Volcano plot shows significantly spliced *RI*  in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2S_DC1_KO$logPValue <- -1 * log10(RI_S2S_DC1_KO$FDR)
# Create a binary inclusion/exclusion column
RI_S2S_DC1_KO$Splicing <- ifelse(RI_S2S_DC1_KO$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(RI_S2S_DC1_KO, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
RI_S2S_DC1_KO$Color <- ifelse(RI_S2S_DC1_KO$FDR < 0.05 & abs(RI_S2S_DC1_KO$IncLevelDifference) > 0.2, RI_S2S_DC1_KO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(RI_S2S_DC1_KO$Color)

# Update the scale_color_manual function with new labels including counts volc_A3SS_ythdc1-KO_S2S
ggplot(RI_S2S_DC1_KO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "red",  "Reduced" = "navy"),
                     labels = c(paste("Increased", counts["Increased"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "RI event transcripts/genes in Ythdc1-KO/S2S") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(RI_S2S_DC1_KO)) {

RI_S2S_DC1_KO$GeneID <- make.unique(as.character(RI_S2S_DC1_KO$GeneID))
rownames(RI_S2S_DC1_KO) <- RI_S2S_DC1_KO$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  # RI_S2S_DC1_KO$GeneID <- NULL
  
  # Now, you can use the mapIds function
  RI_S2S_DC1_KO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2S_DC1_KO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
  RI_S2S_DC1_KO$GO <- mapIds(org.Dm.eg.db,
                             keys = rownames(RI_S2S_DC1_KO),
                             column = "GO",
                             keytype = "ENSEMBL",
                             multiVals = "first")

  
  # Retrieve GO term descriptions
  RI_S2S_DC1_KO$GODESCR <- sapply(RI_S2S_DC1_KO$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  RI_S2S_DC1_KO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                                   keys = rownames(RI_S2S_DC1_KO),
                                   column = "ONTOLOGY",
                                   keytype = "ENSEMBL",
                                   multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}
#write.table(RI_S2S_DC1_KO, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI_S2S_DC1_KO.txt", quote = F, sep = "\t", row.names = F)
```

       
    MXE in Ythdc1-KO relative to S2S


There are a total of 26475 MXE genes or transcripts and 17835 after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2S_DC1_KO <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
nrow(MXE_S2S_DC1_KO)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2S_DC1_KO <- MXE_S2S_DC1_KO %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(MXE_S2S_DC1_KO)
```

## Volcano plot shows significantly spliced *Mutually exclusive events* in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2S_DC1_KO$logPValue <- -1 * log10(MXE_S2S_DC1_KO$FDR)
# Create a binary inclusion/exclusion column
MXE_S2S_DC1_KO$Splicing <- ifelse(MXE_S2S_DC1_KO$IncLevelDifference < 0,  "Reduced in S2S", "Increased in S2S")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(MXE_S2S_DC1_KO, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
MXE_S2S_DC1_KO$Color <- ifelse(MXE_S2S_DC1_KO$FDR < 0.05 & abs(MXE_S2S_DC1_KO$IncLevelDifference) > 0.2, MXE_S2S_DC1_KO$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(MXE_S2S_DC1_KO$Color)

# Update the scale_color_manual function with new labels including counts volc_A3SS_ythdc1-KO_S2S
ggplot(MXE_S2S_DC1_KO, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1,1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased in S2S" = "red",  "Reduced in S2S" = "navy"),
                     labels = c(paste("Increased in S2S -", counts["Increased in S2S"]),paste("Not significant -", counts["Not significant"]),
                                paste("Reduced in S2S -", counts["Reduced in S2S"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Ythdc1-KO/S2S (MXE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(MXE_S2S_DC1_KO)) {
 MXE_S2S_DC1_KO$GeneID <- make.unique(as.character(MXE_S2S_DC1_KO$GeneID))
rownames(MXE_S2S_DC1_KO) <- MXE_S2S_DC1_KO$GeneID
  

  MXE_S2S_DC1_KO$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2S_DC1_KO),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
  MXE_S2S_DC1_KO$GO <- mapIds(org.Dm.eg.db,
                             keys = rownames(MXE_S2S_DC1_KO),
                             column = "GO",
                             keytype = "ENSEMBL",
                             multiVals = "first")

  
  MXE_S2S_DC1_KO$GODESCR <- sapply(MXE_S2S_DC1_KO$GO, function(go_id) {
    if (!is.na(go_id)) {
      Term(go_id)
    } else {
      NA
    }
  })
  
  MXE_S2S_DC1_KO$ONTOLOGY <- mapIds(org.Dm.eg.db,
                                   keys = rownames(MXE_S2S_DC1_KO),
                                   column = "ONTOLOGY",
                                   keytype = "ENSEMBL",
                                   multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}
#write.table(MXE_S2S_DC1_KO, file = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO//MXE_S2S_DC1_KO.txt", quote = F, sep = "\t", row.names = F)
```

    
    The scatter plot below shows significant spliced events in Ythdc1-KO relative to S2S
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to prepare data with read filtering steps
prepare_data <- function(file_path, event_type) {
  # Read the data
  data <- read.table(file_path, header = TRUE, sep = "\t")
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)

  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  # Add necessary columns
  data <- data %>%
    mutate(delta_PSI = IncLevelDifference,
           Significant = ifelse(FDR < 0.05 & abs(delta_PSI) > 0.2, "Significant", "Not Significant"),
           EventType = event_type,
           Event_Types = ifelse(Significant == "Significant", event_type, "Not Significant"))
  
  # Select necessary columns
  data <- data[, c("geneSymbol", "delta_PSI", "FDR", "Significant", "EventType", "Event_Types")]
  
  return(data)
}


file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt"
    )

event_types <- names(file_paths)

# Extract significant genes for each event type
significant_genes_list <- mapply(prepare_data, file_paths, event_types, SIMPLIFY = FALSE)

# Combine all splicing events into a single dataframe
all_genes <- do.call(rbind, significant_genes_list)

# Calculate counts for each event type
event_counts <- as.data.frame(table(all_genes$Event_Types))
colnames(event_counts) <- c("Event_Types", "Count")

# Define colors for splicing event types
type_colors <- c("SE" = "blue", "A3SS" = "black", "A5SS" = "darkgreen", "MXE" = "red", "RI" = "brown", "Not Significant" = "grey")

# Identify top genes (top 100 based on FDR and delta_PSI) volc_MXE_ythdc1-KO_S2S
top_genes <- all_genes[order(all_genes$FDR, -abs(all_genes$delta_PSI)),][1:1000,]

# Prepare custom labels with counts for the legend
custom_labels <- with(event_counts, setNames(paste(Event_Types, " (", Count, ")", sep=""), Event_Types))

# Create the plot
ggplot(all_genes, aes(x = delta_PSI, y = -log10(FDR), color = Event_Types)) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = type_colors, labels = custom_labels) +
    geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
    geom_vline(xintercept = c(-0.2, 0.2), linetype = "dashed", color = "black") +
    theme_minimal() + xlim(-1.0, 1.0) +
    labs(x = "", y = "Log10 Adj. P-value", title = "Significant splicing events in Ythdc1-KO relative to S2S") +
    theme(legend.position = "right", plot.title = element_text(hjust = 0.5)) +
    # Use geom_text_repel to remove boxes and filter for significant genes only
    geom_text_repel(data = top_genes %>% dplyr::filter(Significant == "Significant"), 
                    aes(label = geneSymbol), size = 3, 
                    box.padding = 0.35, point.padding = 0.5, max.overlaps = 10)
```

    
    Below are differential genes that are implicated in YTHDC1-KO relative in S2S splicing
    
```{r echo=FALSE, message=FALSE, warning=FALSE}

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}

# Define file paths for splicing data
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt")

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_S2_S_DC1_KO$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

# Add differential expression information
splicing_diff <- splicing_diff %>%
  left_join(res_S2_S_DC1_KO %>% dplyr::select(current_symbol, log2FoldChange, padj), by = c("geneSymbol" = "current_symbol")) %>%
  mutate(Regulation_DE = ifelse(log2FoldChange > 2, "Upregulated", ifelse(log2FoldChange < -2, "Downregulated", "Not Significant")))

# Print the results
print(splicing_diff)

```

    
    
```{r message=FALSE, warning=FALSE, echo=FALSE}

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}

# Define file paths for splicing data
files <- c(
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt",
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt"
)

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_S2_S_DC1_KO$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

# Add differential expression information
splicing_diff <- splicing_diff %>%
  left_join(res_S2_S_DC1_KO %>% dplyr::select(current_symbol, log2FoldChange, padj), by = c("geneSymbol" = "current_symbol")) %>%
  mutate(Regulation_DE = ifelse(log2FoldChange > 0.5, "Upregulated", ifelse(log2FoldChange < -0.5, "Downregulated", "Not Significant")))

# Exclude "Not Significant" genes
splicing_diff <- splicing_diff %>% dplyr::filter(Regulation_DE != "Not Significant")

# Create the dot plot
ggplot(splicing_diff, aes(x = log2FoldChange, y = IncLevelDifference, color = Regulation_DE)) +
  geom_point(aes(size = -log10(FDR)), alpha = 0.8) +
  scale_color_manual(values = c("Upregulated" = "red", "Downregulated" = "blue", "Not Significant" = "grey")) +
  labs(x = "log2 Fold Change", y = "Inclusion Level Difference ()", color = "Differential Expression", size = "-log10(FDR)") +
  theme_minimal() +
  geom_text_repel(aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  ggtitle("Significantly Spliced and Differentially Expressed Genes in YTHDC1-KO relative to S2S")

 #+
 # theme(
 #   text = element_text(size = 20), # Change font size of all text
  #  axis.text = element_text(size = 20), # Change font size of axis text
 #   axis.title = element_text(size = 20), # Change font size of axis titles
 #   plot.title = element_text(size = 20, hjust = 0.5), # Change font size and center plot title
 #   legend.text = element_text(size = 20), # Change font size of legend text
  #  legend.title = element_text(size = 20) # Change font size of legend title
  #)
```
  
```{r message=FALSE, warning=FALSE, echo=FALSE}

significant_genes <- res_S2_S_DC1_KO %>% 
  dplyr::filter(diffexpressed != "Not Significant")

non_significant_genes <- res_S2_S_DC1_KO %>% 
  dplyr::filter(diffexpressed == "Not Significant")


count_genes_by_event <- function(data, gene_list) {
  data %>%
    dplyr::filter(geneSymbol %in% gene_list$current_symbol) %>%
    group_by(EventType) %>%
    summarise(Count = n(),
      GeneNames = paste(geneSymbol, collapse = ", "))
}


non_sig_event_counts <- count_genes_by_event(all_significant_genes, non_significant_genes)
non_sig_event_counts <- non_sig_event_counts %>% mutate(Significance = "Not Differentially Expressed")


sig_event_counts <- count_genes_by_event(all_significant_genes, significant_genes)
sig_event_counts <- sig_event_counts %>% mutate(Significance = "Differentially Expressed")


combined_event_counts <- bind_rows(sig_event_counts, non_sig_event_counts)


combined_event_counts$Significance <- factor(combined_event_counts$Significance, 
                                             levels = c("Not Differentially Expressed", "Differentially Expressed"))

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# Extract the p-value and format it
p_value <- chi_square_result$p.value
p_value_formatted <- sprintf("Chi-Square Test p-value: %.2e", p_value)  # Scientific notation
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column bar_plot_DEG_METTL3-KO_S2S

# View the results
print(chi_square_result)

plot_ly(combined_event_counts, 
                x = ~EventType, 
                y = ~Count, 
                color = ~Significance, 
                type = 'bar', 
                text = ~Count,
                textposition = 'auto',
                textfont = list(color = "black"),
                hoverinfo = 'text') %>%
  layout(title = "Number of DEGs for Each Splicing Event Type in YTHDC1-KO relative to S2S",
    xaxis = list(title = "Splicing Event Type"),
    yaxis = list(title = "Number of Genes"),
    barmode = 'stack',
    legend = list(
      title = list(text = p_value_formatted),
      traceorder = "reversed"))



custom_colors <- c("Not Differentially Expressed" = "#a6bddb", "Differentially Expressed" = "#de2d26")

# Create the ggplot
ggplot(combined_event_counts, aes(x = EventType, y = Count, fill = Significance)) +
  geom_bar(stat = "identity", position = "stack", width = 0.7) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 5, fontface = "bold", color = "black") +
  scale_fill_manual(values = custom_colors) +
  labs(
    title = "DEGs for each splicing event type in Ythdc1-KO/S2S",
    x = "Splicing Event Type",
    y = "Number of Genes",
    fill = sprintf("Gene Expression Status\n(p = %.2e)", p_value)) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank())
```
  
    
```{r}
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)


```

```{r}
# Calculate the total number of splicing genes
total_splicing_genes <- length(unique(all_significant_genes$geneSymbol))

# Calculate the number and percentage of DEGs implicated in splicing
num_diff_splicing_genes <- length(unique(splicing_diff$geneSymbol))
percent_diff_splicing_genes <- (num_diff_splicing_genes / total_splicing_genes) * 100

# Print the summary
cat("Total number of genes implicated in splicing:", total_splicing_genes, "\n")
cat("Number of DEGs implicated in splicing:", num_diff_splicing_genes, "\n")
cat("Percentage of DEGs implicated in splicing:", round(percent_diff_splicing_genes, 2), "%\n")

# Optional: Print the number of splicing events per event type
splicing_event_counts <- all_significant_genes %>%
  group_by(EventType) %>%
  summarise(Count = n())

cat("\nNumber of significant splicing events by type:\n")
print(splicing_event_counts)

```
   
   
   
    S2_DC1_GOF relative to S2_GFP
    

```{r message=FALSE, warning=FALSE, echo=FALSE}
make_single_PSI_box_S2_GFP_S2_DC1_GOF <- function(event_IDs, Splicing_type) {
    # Define the path to your splicing files
    file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt"
    )
    
    # Check if the specified splicing type is available
    if (!Splicing_type %in% names(file_paths)) {
        stop("Splicing type not found in the list of files.")
    }

    # Read the data file for the specified splicing type
    file_path <- file_paths[[Splicing_type]]
    data <- read.table(file_path, header = TRUE)

    # Filter for specified event IDs
    event_data <- data[data$ID %in% event_IDs, ]

    # Reshape the data
    psi_data <- data.frame(ID = event_data$ID, IncLevel1 = event_data$IncLevel1, IncLevel2 = event_data$IncLevel2)
    psi_data_long <- melt(psi_data, id.vars = "ID", variable.name = "Sample", value.name = "PSI")
    psi_data_long$Condition <- ifelse(psi_data_long$Sample == "IncLevel1", "S2 Ythdc1-GoF", "S2 GFP")

    # Reorder the factor levels to put S2S as the first level
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("S2 Ythdc1-GoF", "S2 GFP"))

    # Split the concatenated PSI values and convert them to numeric 
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(PSI, ",")) %>%
        unnest(PSI) %>%
        mutate(PSI = as.numeric(PSI))

    # Create the box plot with statistical comparisons
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) +
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
       labs(fill = "Condition") + 
        scale_fill_manual(values = c("S2 GFP" = "red", "S2 Ythdc1-GoF" = "purple")) +
        stat_compare_means(aes(label = ..p.signif..), method = "t.test", comparisons = list(c("S2 Ythdc1-GoF", "S2 GFP")))

    print(plot)
}

# Example usage
#make_single_PSI_box_m(event_IDs = c("3972", "3992"), Splicing_type = "SE")

```

    
    
    Bar chart below show the number of significant splicing difference in S2_GFP relative to S2_DC1_GOF

```{r message=FALSE, warning=FALSE, echo=FALSE}
prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)
  
  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  return(data)
}

# Function to count splicing events after filtering
count_splicing_events <- function(filename) {
  data <- prepare_and_filter_data(filename)
  
  # Filter for significant genes
  significant_genes <- subset(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  n_exclusion <- sum(significant_genes$IncLevelDifference < 0)
  n_inclusion <- sum(significant_genes$IncLevelDifference > 0)
  
  sample_id <- basename(filename)
  sample_id <- gsub(".MATS.JCEC.txt", "", sample_id)
  
  return(data.frame(SampleID = sample_id, Exclusion = n_exclusion, Inclusion = n_inclusion))
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt")

# Apply function to each file and combine results
results <- do.call(rbind, lapply(files, count_splicing_events))

# Calculate totals
total_exclusion <- sum(results$Exclusion)
total_inclusion <- sum(results$Inclusion)

# Melt the results data for ggplot
results_melted <- tidyr::pivot_longer(results, cols = c(Exclusion, Inclusion), names_to = "EventType", values_to = "Count")

t_test_result <- t.test(results$Inclusion, results$Exclusion, paired = TRUE)

# Extract p-value from the t-test result
p_value <- t_test_result$p.value
formatted_p_value <- sprintf("P-value: %.3g", p_value)

# Generate the enhanced ggplot with the p-value in the legend
ggplot(results_melted, aes(x = SampleID, y = Count, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 4, color = "white") +
  labs(title = sprintf("S2 Ythdc1-GOF relative to S2 GFP\nTotal Exclusion: %d, Total Inclusion: %d", total_exclusion, total_inclusion),
       x = "Splicing Event Type", 
       y = "Number of Significant Splicing Events",
       fill = paste("Splicing Event (", formatted_p_value, ")", sep = "")) +
  scale_fill_manual(values = c("Exclusion" = "#1f77b4", "Inclusion" = "#ff7f0e")) +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "top",
        legend.title = element_text(face = "bold"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', color = 'gray')) +
  guides(fill = guide_legend(title = paste("Splicing Event (", formatted_p_value, ")", sep = "")))

# Print the t-test results S2_Ythdc1_GOF_S2_GFP_AS_Inc_Exc
print(t_test_result)
```

```         
   Total number of splicing events in S2_GFP relative to S2_DC1_GOF
```



There are a total of 18516 Exon Skipping transcripts in S2S YthDC1-GoF relative to S2S GFP after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(SE_S2_GFP_S2_DC1_GOF)
```


    Below are the number of spliced events in S2S YthDC1-GoF relative to S2S GFP that might be of interest to us and validate some of the genes from PCR
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Load required libraries
library(dplyr)
library(tidyr)
# Define vector of gene names
genes_of_interest <- c(
  "Srrm234", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas",
  "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", 
  "Gbeta13F", "Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega", 
  "NK7.1", "gw", "zip", "Ptp10D", "CG16952", "CG2310", "CG42668", "CG31635", 
  "CG32280", "Naprt", "B4", "CG32521", "CG45050", "aru", "eIF3j", "vap", "ABCD", 
  "Arf102F", "Unc-115b", "CG7611", "pyd", "Ars2", "CG32772", "MTA1-like", 
  "ATPsynF", "Ac3", "Adf1", "ArgRS-m", "Dscam1", "sd", "mth", "Acsl", "Tep2", 
  "aqz", "l(2)gl", "unc-13", "Tet", "Moe", "hdly", "mod(mdg4)", "CG2246", "CG1146")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(
      S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and splicing type
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extracting the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  return(gene_event_counts)
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize a list to hold the results from each file
results_list <- lapply(files, count_splicing_events, genes = genes_of_interest)

# Combine results from all files into a single data frame
final_results <- bind_rows(results_list)

# Reshape the data to a wide format for inclusion and exclusion counts
final_results_wide <- final_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final table
print(final_results_wide)

# Write the final results to a file
#write.table(final_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/S2_GFP_S2_DC1_GOF_0_Sig_gene_of_interest.txt", row.names = FALSE, quote = FALSE, sep = "\t")

```

    Gene frequency 
```{r}
genes_of_interest <- c(
  "Srrm234", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas",
  "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", 
  "Gbeta13F", "Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega", 
  "NK7.1", "gw", "zip", "Ptp10D", "CG16952", "CG2310", "CG42668", "CG31635", 
  "CG32280", "Naprt", "B4", "CG32521", "CG45050", "aru", "eIF3j", "vap", "ABCD", 
  "Arf102F", "Unc-115b", "CG7611", "pyd", "Ars2", "CG32772", "MTA1-like", 
  "ATPsynF", "Ac3", "Adf1", "ArgRS-m", "Dscam1", "sd", "mth", "Acsl", "Tep2", 
  "aqz", "l(2)gl", "unc-13", "Tet", "Moe", "hdly", "mod(mdg4)", "CG2246", "CG1146")


# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, ID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and extract high-frequency exons
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extract the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  # Calculate exon frequencies and extract the highest frequency exons
  exon_frequency <- significant_genes %>%
    group_by(geneSymbol) %>%
    summarise(Frequency = n()) %>%
    arrange(desc(Frequency)) %>%
    slice(1:10) %>%  # Extract top 10 highest frequency exons
    mutate(EventType = event_type) # Add event type
  
  return(list(gene_counts = gene_event_counts, high_freq_exons = exon_frequency))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize lists to hold results
gene_results_list <- list()
exon_results_list <- list()

for (file in files) {
  results <- count_splicing_events(file, genes_of_interest)
  gene_results_list <- append(gene_results_list, list(results$gene_counts))
  exon_results_list <- append(exon_results_list, list(results$high_freq_exons))
}

# Combine results from all files into single data frames
final_gene_results <- bind_rows(gene_results_list)
final_exon_results <- bind_rows(exon_results_list)

# Reshape the gene data to a wide format for inclusion and exclusion counts
final_gene_results_wide <- final_gene_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final tables
print(final_gene_results_wide)
print(final_exon_results)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_S2_DC1_GOF <- SE_S2_GFP_S2_DC1_GOF %>%
  #Split the replicate read counts that are separated by commas into different columns
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(SE_S2_GFP_S2_DC1_GOF)
```


    ## Volcano plot shows significantly spliced *SE* in S2S YthDC1-GoF relative to S2S

```{r echo=FALSE, message=FALSE, warning=FALSE}
SE_S2_GFP_S2_DC1_GOF$logPValue <- -1 * log10(SE_S2_GFP_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
SE_S2_GFP_S2_DC1_GOF$Splicing <- ifelse(SE_S2_GFP_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Over-expressed")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(SE_S2_GFP_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
SE_S2_GFP_S2_DC1_GOF$Color <- ifelse(SE_S2_GFP_S2_DC1_GOF$FDR < 0.05 & abs(SE_S2_GFP_S2_DC1_GOF$IncLevelDifference) > 0.2, SE_S2_GFP_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(SE_S2_GFP_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts S2_Ythdc1_GOF_S2_GFP_AS_Inc_Exc volc_RI_S2_Ythdc1_GOF_S2_GFP
ggplot(SE_S2_GFP_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Over-expressed" = "#ff7f0e",  "Reduced" = "#1f77b4"),
                     labels = c(paste("Not significant -", counts["Not significant"]),
                                paste("Over-expressed -", counts["Over-expressed"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Exon Skipping events in S2 Ythdc1-GOF relative to S2 GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(SE_S2_GFP_S2_DC1_GOF)) {
  
  # Make GeneID unique
  SE_S2_GFP_S2_DC1_GOF$GeneID <- make.unique(as.character(SE_S2_GFP_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(SE_S2_GFP_S2_DC1_GOF) <- SE_S2_GFP_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #SE_S2_GFP_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  SE_S2_GFP_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   SE_S2_GFP_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
 # Retrieve GO term descriptions
SE_S2_GFP_S2_DC1_GOF$GODESCR <- sapply(SE_S2_GFP_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  SE_S2_GFP_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}
#write.table(SE_S2_GFP_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE_S2_GFP_S2_DC1_GOF.txt', quote = F, sep = "\t", row.names = F)
```

```         
 A3SS S2 Ythdc1-GOF relative to S2 GFP
```

There are a total of 9662 A3SS transcripts in S2S YthDC1-GoF relative to S2S GFP after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_SS2_GFP_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(A3_SS2_GFP_S2_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_SS2_GFP_S2_DC1_GOF <- A3_SS2_GFP_S2_DC1_GOF %>%
  #Split the replicate read counts that are separated by commas into different columns
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A3_SS2_GFP_S2_DC1_GOF)
```

### Volcano plot shows significantly spliced *A3SS* in S2S YthDC1-GoF relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_SS2_GFP_S2_DC1_GOF$logPValue <- -1 * log10(A3_SS2_GFP_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A3_SS2_GFP_S2_DC1_GOF$Splicing <- ifelse(A3_SS2_GFP_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Over-expressed")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A3_SS2_GFP_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A3_SS2_GFP_S2_DC1_GOF$Color <- ifelse(A3_SS2_GFP_S2_DC1_GOF$FDR < 0.05 & abs(A3_SS2_GFP_S2_DC1_GOF$IncLevelDifference) > 0.2, A3_SS2_GFP_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A3_SS2_GFP_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts volc_RI_S2_Ythdc1_GOF_S2_GFP
ggplot(A3_SS2_GFP_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1,1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Over-expressed" = "#ff7f0e",  "Reduced" = "#1f77b4"),
                     labels = c(paste("Not significant -", counts["Not significant"]),
                                paste("Over-expressed -", counts["Over-expressed"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "A3SS transcripts/genes in S2 Ythdc1-GOF relative to S2 GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A3_SS2_GFP_S2_DC1_GOF)) {
# Make GeneID unique
  A3_SS2_GFP_S2_DC1_GOF$GeneID <- make.unique(as.character(A3_SS2_GFP_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A3_SS2_GFP_S2_DC1_GOF) <- A3_SS2_GFP_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A3_SS2_GFP_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A3_SS2_GFP_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_SS2_GFP_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A3_SS2_GFP_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_SS2_GFP_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A3_SS2_GFP_S2_DC1_GOF$GODESCR <- sapply(A3_SS2_GFP_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A3_SS2_GFP_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_SS2_GFP_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}
#write.table(A3_SS2_GFP_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3_SS2_GFP_S2_DC1_GOF.txt', quote = F, sep = "\t", row.names = F)
```

```         
### A5SS S2 Ythdc1-GOF relative to S2 GFP
```

There are a total of 17034 A5SS transcripts in S2S YthDC1-GoF relative to S2S GFP after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5SS_S2_GFP_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(A5SS_S2_GFP_S2_DC1_GOF) volc_A3SS_S2_Ythdc1_GOF_S2_GFP
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5SS_S2_GFP_S2_DC1_GOF <- A5SS_S2_GFP_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A5SS_S2_GFP_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *A5SS* in S2 Ythdc1-GOF relative to S2 GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5SS_S2_GFP_S2_DC1_GOF$logPValue <- -1 * log10(A5SS_S2_GFP_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A5SS_S2_GFP_S2_DC1_GOF$Splicing <- ifelse(A5SS_S2_GFP_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Over-expressed")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A5SS_S2_GFP_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A5SS_S2_GFP_S2_DC1_GOF$Color <- ifelse(A5SS_S2_GFP_S2_DC1_GOF$FDR < 0.05 & abs(A5SS_S2_GFP_S2_DC1_GOF$IncLevelDifference) > 0.2, A5SS_S2_GFP_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A5SS_S2_GFP_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts volc_A3SS_S2_Ythdc1_GOF_S2_GFP
ggplot(A5SS_S2_GFP_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Over-expressed" = "#ff7f0e",  "Reduced" = "#1f77b4"),
                     labels = c(paste("Not significant -", counts["Not significant"]),
                                paste("Over-expressed -", counts["Over-expressed"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "A5SS event in S2 Ythdc1-GOF relative to S2 GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A5SS_S2_GFP_S2_DC1_GOF)) {
# Make GeneID unique
  A5SS_S2_GFP_S2_DC1_GOF$GeneID <- make.unique(as.character(A5SS_S2_GFP_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A5SS_S2_GFP_S2_DC1_GOF) <- A5SS_S2_GFP_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A5SS_S2_GFP_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A5SS_S2_GFP_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5SS_S2_GFP_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A5SS_S2_GFP_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5SS_S2_GFP_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A5SS_S2_GFP_S2_DC1_GOF$GODESCR <- sapply(A5SS_S2_GFP_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A5SS_S2_GFP_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5SS_S2_GFP_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}
#write.table(A5SS_S2_GFP_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS_S2_GFP_S2_DC1_GOF.txt', quote = F, sep = "\t", row.names = F)
```

```         
### MXE S2 Ythdc1-GOF relative to S2 GFP
```

There are a total of 14000 MXE transcripts in S2S YthDC1-GoF relative to S2S GFP after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_S2_GFP_S2_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_S2_DC1_GOF <- MXE_S2_GFP_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(MXE_S2_GFP_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in S2S YthDC1-GoF relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_S2_DC1_GOF$logPValue <- -1 * log10(MXE_S2_GFP_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
MXE_S2_GFP_S2_DC1_GOF$Splicing <- ifelse(MXE_S2_GFP_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Over-expressed")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(MXE_S2_GFP_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
MXE_S2_GFP_S2_DC1_GOF$Color <- ifelse(MXE_S2_GFP_S2_DC1_GOF$FDR < 0.05 & abs(MXE_S2_GFP_S2_DC1_GOF$IncLevelDifference) > 0.2, MXE_S2_GFP_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(MXE_S2_GFP_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts volc_A3SS_S2_Ythdc1_GOF_S2_GFP
ggplot(MXE_S2_GFP_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Over-expressed" = "#ff7f0e",  "Reduced" = "#1f77b4"),
                     labels = c(paste("Not significant -", counts["Not significant"]),
                                paste("Over-expressed -", counts["Over-expressed"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mutually exclusive event in S2 Ythdc1-GOF relative to S2 GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(MXE_S2_GFP_S2_DC1_GOF)) {
# Make GeneID unique
  MXE_S2_GFP_S2_DC1_GOF$GeneID <- make.unique(as.character(MXE_S2_GFP_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(MXE_S2_GFP_S2_DC1_GOF) <- MXE_S2_GFP_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #MXE_S2_GFP_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  MXE_S2_GFP_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   MXE_S2_GFP_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
MXE_S2_GFP_S2_DC1_GOF$GODESCR <- sapply(MXE_S2_GFP_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  MXE_S2_GFP_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}

#write.table(MXE_S2_GFP_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE_S2_GFP_S2_DC1_GOF.txt', quote = F, sep = "\t", row.names = F)
```

```         
   RI S2 Ythdc1-GOF relative to S2 GFP
```

There are a total of 5774 RI events in S2S YthDC1-GoF relative to S2S GFP after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(RI_S2_GFP_S2_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_S2_DC1_GOF <- RI_S2_GFP_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(RI_S2_GFP_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *RI* in S2S YthDC1-GoF relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_S2_DC1_GOF$logPValue <- -1 * log10(RI_S2_GFP_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
RI_S2_GFP_S2_DC1_GOF$Splicing <- ifelse(RI_S2_GFP_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Over-expressed")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(RI_S2_GFP_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
RI_S2_GFP_S2_DC1_GOF$Color <- ifelse(RI_S2_GFP_S2_DC1_GOF$FDR < 0.05 & abs(RI_S2_GFP_S2_DC1_GOF$IncLevelDifference) > 0.2, RI_S2_GFP_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(RI_S2_GFP_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts volc_A3SS_S2_Ythdc1_GOF_S2_GFP
ggplot(RI_S2_GFP_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
  geom_point(alpha = 0.8) + xlim(c(-1,1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Over-expressed" = "#ff7f0e",  "Reduced" = "#1f77b4"),
                     labels = c(paste("Not significant -", counts["Not significant"]),
                                paste("Over-expressed -", counts["Over-expressed"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Intron Retention event in S2 Ythdc1-GOF relative to S2 GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(RI_S2_GFP_S2_DC1_GOF)) {
  
  RI_S2_GFP_S2_DC1_GOF$GeneID <- make.unique(as.character(RI_S2_GFP_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(RI_S2_GFP_S2_DC1_GOF) <- RI_S2_GFP_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #RI_S2_GFP_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  RI_S2_GFP_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   RI_S2_GFP_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
RI_S2_GFP_S2_DC1_GOF$GODESCR <- sapply(RI_S2_GFP_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  RI_S2_GFP_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}

#write.table(RI_S2_GFP_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI_S2_GFP_S2_DC1_GOF.txt', quote = F, sep = "\t", row.names = F)
```

        
    The scatter plot below shows significant the spliced events in S2 Ythdc1-GOF relative to S2 GFP
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to prepare data with read filtering steps
prepare_data <- function(file_path, event_type) {
  # Read the data
  data <- read.table(file_path, header = TRUE, sep = "\t")
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)

  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  # Add necessary columns
  data <- data %>%
    mutate(delta_PSI = IncLevelDifference,
           Significant = ifelse(FDR < 0.05 & abs(delta_PSI) > 0.2, "Significant", "Not Significant"),
           EventType = event_type,
           Event_Types = ifelse(Significant == "Significant", event_type, "Not Significant"))
  
  # Select necessary columns
  data <- data[, c("geneSymbol", "delta_PSI", "FDR", "Significant", "EventType", "Event_Types")]
  
  return(data)
}


file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt"
    )

event_types <- names(file_paths)

# Extract significant genes for each event type
significant_genes_list <- mapply(prepare_data, file_paths, event_types, SIMPLIFY = FALSE)

# Combine all splicing events into a single dataframe
all_genes <- do.call(rbind, significant_genes_list)

# Calculate counts for each event type
event_counts <- as.data.frame(table(all_genes$Event_Types))
colnames(event_counts) <- c("Event_Types", "Count")

# Define colors for splicing event types
type_colors <- c("SE" = "blue", "A3SS" = "black", "A5SS" = "darkgreen", "MXE" = "red", "RI" = "#984EA3", "Not Significant" = "grey")

# Identify top genes (top 100 based on FDR and delta_PSI)
top_genes <- all_genes[order(all_genes$FDR, -abs(all_genes$delta_PSI)),][1:1000,]

# Prepare custom labels with counts for the legend volc_RI_S2_Ythdc1_GOF_S2_GFP
custom_labels <- with(event_counts, setNames(paste(Event_Types, " (", Count, ")", sep=""), Event_Types))

# Create the plot
ggplot(all_genes, aes(x = delta_PSI, y = -log10(FDR), color = Event_Types)) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = type_colors, labels = custom_labels) +
    geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
    geom_vline(xintercept = c(-0.2, 0.2), linetype = "dashed", color = "black") +
    theme_minimal() + xlim(-1.0, 1.0) +
    labs(x = "", y = "Log10 Adj. P-value", title = "Significant spliced genes in S2 YTHDC1-GoF relative to S2 GFP") +
    theme(legend.position = "right", plot.title = element_text(hjust = 0.5)) +
    # Use geom_text_repel to remove boxes and filter for significant genes only
    geom_text_repel(data = top_genes %>% dplyr::filter(Significant == "Significant"), 
                    aes(label = geneSymbol), size = 3, 
                    box.padding = 0.35, point.padding = 0.5, max.overlaps = 10)

```        


below are the genes implicated in S2 Ythdc1-GOF relative to S2 GFP
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}


file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt"
    )

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_S2.GFP_S2.DC1.GOF$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

# Add differential expression information
splicing_diff <- splicing_diff %>%
  left_join(res_S2.GFP_S2.DC1.GOF %>% dplyr::select(current_symbol, log2FoldChange, padj), by = c("geneSymbol" = "current_symbol")) %>%
  mutate(Regulation_DE = ifelse(log2FoldChange > 0.5, "Upregulated", ifelse(log2FoldChange < -0.5, "Downregulated", "Not Significant")))

# Exclude "Not Significant" genes
splicing_diff <- splicing_diff %>% dplyr::filter(Regulation_DE != "Not Significant")

# Create the dot plot
ggplot(splicing_diff, aes(x = log2FoldChange, y = IncLevelDifference, color = Regulation_DE)) +
  geom_point(aes(size = -log10(FDR)), alpha = 0.8) +
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +
  scale_color_manual(values = c("Upregulated" = "red", "Downregulated" = "blue", "Not Significant" = "grey")) +
  labs(x = "log2 Fold Change", y = "Inclusion Level Difference ()", color = "Differential Expression", size = "-log10(FDR)") +
  theme_minimal() +
  geom_text_repel(aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  ggtitle("Significantly Spliced and Differentially Expressed Genes in YTHDC1-GoF relative to S2 GFP")
 #+
 # theme(
 #   text = element_text(size = 20), # Change font size of all text
  #  axis.text = element_text(size = 20), # Change font size of axis text
 #   axis.title = element_text(size = 20), # Change font size of axis titles
 #   plot.title = element_text(size = 20, hjust = 0.5), # Change font size and center plot title
 #   legend.text = element_text(size = 20), # Change font size of legend text
  #  legend.title = element_text(size = 20) # Change font size of legend title
  #)
```
   
   
   
  
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter significant and non-significant genes
significant_genes <- res_S2.GFP_S2.DC1.GOF %>% 
  dplyr::filter(diffexpressed != "Not Significant")
non_significant_genes <- res_S2.GFP_S2.DC1.GOF %>% 
  dplyr::filter(diffexpressed == "Not Significant")

# Function to count the number of genes involved in each splicing event type
count_genes_by_event <- function(data, gene_list) {
  data %>%
    dplyr::filter(geneSymbol %in% gene_list$current_symbol) %>%
    group_by(EventType) %>%
    summarise(
      Count = n(),
      GeneNames = paste(geneSymbol, collapse = ", ")  # Collect gene names
    )
}

# Count for non-significant DEGs
non_sig_event_counts <- count_genes_by_event(all_significant_genes, non_significant_genes)
non_sig_event_counts <- non_sig_event_counts %>% mutate(Significance = "Not Differentially Expressed")

# Count for significant DEGs
sig_event_counts <- count_genes_by_event(all_significant_genes, significant_genes)
sig_event_counts <- sig_event_counts %>% mutate(Significance = "Differentially Expressed")

# Combine the counts into one dataframe
combined_event_counts <- bind_rows(sig_event_counts, non_sig_event_counts)

# Reorder the Significance factor to ensure "Significant" is on top
combined_event_counts$Significance <- factor(combined_event_counts$Significance, 
                                             levels = c("Not Differentially Expressed", "Differentially Expressed"))
# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# Extract the p-value and format it
p_value <- chi_square_result$p.value
p_value_formatted <- sprintf("Chi-Square Test p-value: %.2e", p_value)  # Scientific notation
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)

plot_ly(combined_event_counts, 
                x = ~EventType, 
                y = ~Count, 
                color = ~Significance, 
                type = 'bar', 
                text = ~Count,
                textposition = 'auto',
                textfont = list(color = "black"),
                hoverinfo = 'text') %>%
  layout(
    title = "S2S-YTHDC1-GoF relative to S2S GFP",
    xaxis = list(title = "Splicing Event Type"),
    yaxis = list(title = "Number of Genes"),
    barmode = 'stack',
    legend = list(
      title = list(text = p_value_formatted),
      traceorder = "reversed"))


# Define custom colors
custom_colors <- c("Not Differentially Expressed" = "#a6bddb", "Differentially Expressed" = "#de2d26")

# Create the ggplot
ggplot(combined_event_counts, aes(x = EventType, y = Count, fill = Significance)) +
  geom_bar(stat = "identity", position = "stack", width = 0.7) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 5, fontface = "bold", color = "black") +
  scale_fill_manual(values = custom_colors) +
  labs(
    title = "Number of DEGs S2S-YTHDC1-GoF/S2-GFP splicing types",
    x = "Splicing Event Type",
    y = "Number of Genes",
    fill = sprintf("Gene Expression Status\n(p = %.3g)", p_value)) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank())
```
  
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)


```



```{r}
# Calculate the total number of splicing genes
total_splicing_genes <- length(unique(all_significant_genes$geneSymbol))

# Calculate the number and percentage of DEGs implicated in splicing
num_diff_splicing_genes <- length(unique(splicing_diff$geneSymbol))
percent_diff_splicing_genes <- (num_diff_splicing_genes / total_splicing_genes) * 100

# Print the summary
cat("Total number of genes implicated in splicing:", total_splicing_genes, "\n")
cat("Number of DEGs implicated in splicing:", num_diff_splicing_genes, "\n")
cat("Percentage of DEGs implicated in splicing:", round(percent_diff_splicing_genes, 2), "%\n")

# Optional: Print the number of splicing events per event type
splicing_event_counts <- all_significant_genes %>%
  group_by(EventType) %>%
  summarise(Count = n())

cat("\nNumber of significant splicing events by type:\n")
print(splicing_event_counts)
```
   
        
        METTL3-KO YthDC1-GoF relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
make_single_PSI_box_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- function(event_IDs, Splicing_type) {
    # Define the path to your splicing files
    file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt")
    
    # Check if the specified splicing type is available
    if (!Splicing_type %in% names(file_paths)) {
        stop("Splicing type not found in the list of files.")
    }

    # Read the data file for the specified splicing type
    file_path <- file_paths[[Splicing_type]]
    data <- read.table(file_path, header = TRUE)

    # Filter for specified event IDs
    event_data <- data[data$ID %in% event_IDs, ]

    # Reshape the data
    psi_data <- data.frame(ID = event_data$ID, IncLevel1 = event_data$IncLevel1, IncLevel2 = event_data$IncLevel2)
    psi_data_long <- melt(psi_data, id.vars = "ID", variable.name = "Sample", value.name = "PSI")
    psi_data_long$Condition <- ifelse(psi_data_long$Sample == "IncLevel1", "METTL3-KO Ythdc1-GoF", "METTL3-KO GFP")

    # Reorder the factor levels to put S2S as the first level
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("METTL3-KO Ythdc1-GoF", "METTL3-KO GFP"))

    # Split the concatenated PSI values and convert them to numeric
    psi_data_long$PSI <- as.character(psi_data_long$PSI)
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(PSI, ",")) %>%
        unnest(PSI) %>%
        mutate(PSI = as.numeric(PSI))

    # Create the box plot with statistical comparisons
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) + 
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) + 
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID, scales = "free") +
        theme_bw() + 
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
       labs(fill = "Condition") + 
        scale_fill_manual(values = c("METTL3-KO GFP" = "red", "METTL3-KO Ythdc1-GoF" = "blue")) +
        stat_compare_means(aes(label = ..p.signif..), method = "t.test", comparisons = list(c("METTL3-KO Ythdc1-GoF", "METTL3-KO GFP")))

    print(plot)
}

# Example usage
#make_single_PSI_box_m(event_IDs = c("3972", "3992"), Splicing_type = "SE")

```

     
     Bar chart below show the number of significant splicing difference in METTL3-KO YthDC1-GoF relative to METTL3-KO relative

```{r message=FALSE, warning=FALSE, echo=FALSE}

prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)
  
  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  return(data)
}

# Function to count splicing events after filtering
count_splicing_events <- function(filename) {
  data <- prepare_and_filter_data(filename)
  
  # Filter for significant genes
  significant_genes <- subset(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  n_exclusion <- sum(significant_genes$IncLevelDifference < 0)
  n_inclusion <- sum(significant_genes$IncLevelDifference > 0)
  
  sample_id <- basename(filename)
  sample_id <- gsub(".MATS.JCEC.txt", "", sample_id)
  
  return(data.frame(SampleID = sample_id, Exclusion = n_exclusion, Inclusion = n_inclusion))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt")

# Apply function to each file and combine results
results <- do.call(rbind, lapply(files, count_splicing_events))

# Calculate totals
total_exclusion <- sum(results$Exclusion)
total_inclusion <- sum(results$Inclusion)

# Melt the results data for ggplot
results_melted <- tidyr::pivot_longer(results, cols = c(Exclusion, Inclusion), names_to = "EventType", values_to = "Count")
# Prepare data for t-test
t_test_data <- results %>%
  dplyr::select(SampleID, Exclusion, Inclusion) %>%
  mutate(Exclusion = as.numeric(Exclusion), Inclusion = as.numeric(Inclusion))


t_test_result <- t.test(t_test_data$Inclusion, t_test_data$Exclusion, paired = TRUE)

# Extract p-value from the t-test result
p_value <- t_test_result$p.value
formatted_p_value <- sprintf("P-value: %.3g", p_value)

# Generate the enhanced ggplot with the p-value in the legend Mettl3_ko_ythdc1_gof_mettl3_ko_gfp_AS_Inc_Exc
ggplot(results_melted, aes(x = SampleID, y = Count, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 4, color = "black") +
  labs(title = sprintf("METTL3-KO-YthDC1-GoF/METTL3-KO GFP\nTotal Exclusion: %d, Total Inclusion: %d", 
                       total_exclusion, total_inclusion),
       x = "Splicing Event Type", 
       y = "Number of Significant Splicing Events",
       fill = paste("Splicing Event (", formatted_p_value, ")", sep = "")) +
  scale_fill_manual(values = c("Exclusion" = "#377EB8", "Inclusion" = "#ff7f0e")) +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "top",
        legend.title = element_text(face = "bold"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', color = 'gray')) +
  guides(fill = guide_legend(title = paste("Splicing Event (", formatted_p_value, ")", sep = "")))

```






    Below are the number of spliced events in METTL3-KO YthDC1-GoF relative to METTL3-KO GFP that might be of interest to us and validate some of the genes from PCR
```{r message=FALSE, warning=FALSE, echo=FALSE}

genes_of_interest <- c(
  "Srrm234", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", 
  "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", 
  "Sema1b", "Gbeta13F", "Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", 
  "lncRNA:Hsromega", "CDase", "gce", "Kcmf1", "NKAIN", "CG5445", "Ptp4E", 
  "RpL36", "mrj", "CG13124", "Fer1HCH", "CtBP", "sd", "zfh1", "CG31145", 
  "Imp", "mbl", "fog", "CG11155", "CG4630", "His4r", "Nedd4", "Traf-like", 
  "eIF4G1", "iav", "wdb", "Bsg", "CG11357", "mtgo", "AnxB11", "unc-13", 
  "par-1", "Dscam1", "SP1029", "tmod", "Atpalpha", "Gyc76C", "CG14253"
)


# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and splicing type
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extracting the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  return(gene_event_counts)
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize a list to hold the results from each file
results_list <- lapply(files, count_splicing_events, genes = genes_of_interest)

# Combine results from all files into a single data frame
final_results <- bind_rows(results_list)

# Reshape the data to a wide format for inclusion and exclusion counts
final_results_wide <- final_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final table
print(final_results_wide)

# Write the final results to a file
#write.table(final_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_0_Sig_gene_of_interest.txt", row.names = FALSE, quote = FALSE, sep = "\t")

```



```{r}
# Define vector of gene names
genes_of_interest <- c("Srrm234", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", 
                       "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F", 
                       "Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, ID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and extract high-frequency exons
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extract the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  # Calculate exon frequencies and extract the highest frequency exons
  exon_frequency <- significant_genes %>%
    group_by(geneSymbol) %>%
    summarise(Frequency = n()) %>%
    arrange(desc(Frequency)) %>%
    slice(1:10) %>%  # Extract top 10 highest frequency exons
    mutate(EventType = event_type) # Add event type
  
  return(list(gene_counts = gene_event_counts, high_freq_exons = exon_frequency))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize lists to hold results
gene_results_list <- list()
exon_results_list <- list()

for (file in files) {
  results <- count_splicing_events(file, genes_of_interest)
  gene_results_list <- append(gene_results_list, list(results$gene_counts))
  exon_results_list <- append(exon_results_list, list(results$high_freq_exons))
}

# Combine results from all files into single data frames
final_gene_results <- bind_rows(gene_results_list)
final_exon_results <- bind_rows(exon_results_list)

# Reshape the gene data to a wide format for inclusion and exclusion counts
final_gene_results_wide <- final_gene_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final tables
print(final_gene_results_wide)
print(final_exon_results)



```


There are a total of 22567(26,350) SE transcripts in METTL3-KO YthDC1-GoF relative to METTL3-KO after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
se_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
se_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- se_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *Exon Slipping gene (SE)* in METTL3-KO YthDC1-GoF relative to METTL3-KO

```{r message=FALSE, warning=FALSE, echo=FALSE}
se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & 
                                                     abs(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, 
                                                   se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts
ggplot(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "#0000FF"),
                     labels = c(paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Exon Skiping transcripts/genes in METTL3-KO Ythdc1-GoF relative to METTL3-KO GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF)) {
# Make GeneID unique
  se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF) <- se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  se_mettl3_KO_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}

#write.table(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_SE.txt', quote = F, sep = "\t", row.names = F)
```



```{r message=FALSE, warning=FALSE, echo=FALSE}
mettl3_KO_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt", header = TRUE, stringsAsFactors = FALSE)

# Parse the comma-separated values and calculate the sum for each sample
mettl3_KO_GFP_mettl3_KO_DC1_GOF <- mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  rowwise() %>%
  mutate(IJC_SAMPLE_1 = sum(as.numeric(unlist(strsplit(IJC_SAMPLE_1, ",")))),
    SJC_SAMPLE_1 = sum(as.numeric(unlist(strsplit(SJC_SAMPLE_1, ",")))),
    IJC_SAMPLE_2 = sum(as.numeric(unlist(strsplit(IJC_SAMPLE_2, ",")))),
    SJC_SAMPLE_2 = sum(as.numeric(unlist(strsplit(SJC_SAMPLE_2, ",")))),
    IncLevel1 = mean(as.numeric(unlist(strsplit(IncLevel1, ","))), na.rm = TRUE),
    IncLevel2 = mean(as.numeric(unlist(strsplit(IncLevel2, ","))), na.rm = TRUE)) %>%
  ungroup()

# Calculate combined read counts and log2 fold change
mettl3_KO_GFP_mettl3_KO_DC1_GOF <- mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  mutate(
    read_count_1 = (IJC_SAMPLE_1 + SJC_SAMPLE_1),
    read_count_2 = (IJC_SAMPLE_2 + SJC_SAMPLE_2),
    A = (log2(read_count_1 + 1) + log2(read_count_2 + 1)) / 2,
    M = log2((IncLevel1 + 1) / (IncLevel2 + 1)))

# Replace -Inf and Inf values
mettl3_KO_GFP_mettl3_KO_DC1_GOF$A[is.infinite(mettl3_KO_GFP_mettl3_KO_DC1_GOF$A)] <- NA
mettl3_KO_GFP_mettl3_KO_DC1_GOF$M[is.infinite(mettl3_KO_GFP_mettl3_KO_DC1_GOF$M)] <- NA

# Categorize points based on fold change and FDR
mettl3_KO_GFP_mettl3_KO_DC1_GOF <- mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  mutate(color_group = case_when(
    FDR > 0.05 ~ "Not Significant",
    M > 0.2 ~ "Over-expressed",
    M < -0.2 ~ "Reduced",
    TRUE ~ "Not Significant"  # default case
  ))

# Filter for significant genes
significant_genes <- mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(FDR < 0.05 & abs(M) > 0.2)

# Sort by FDR and select top 100
top_genes <- significant_genes %>%
  arrange(FDR) %>%
  slice(1:100)

# Plot the MA plot with customized colors and labels
#ggplot(na.omit(mettl3_KO_GFP_mettl3_KO_DC1_GOF), aes(x = A, y = M, color = color_group)) +
#  geom_point(alpha = 0.6) +
#  theme_minimal() +
#  labs(
#    x = "Normalized Read Counts",
#    y = "Log2 Fold Change",
#    title = "MA Plot of Log2 Fold Change in METTL3-KO Ythdc1-GoF relative to METTL3-KO GFP"
#  ) +
#  geom_hline(yintercept = 0, linetype = "dashed", color = "black") +
#  scale_color_manual(values = c("Not Significant" = "grey", "Over-expressed" = "red", "Reduced" = "blue")) +
#  theme(legend.title = element_blank()) +
#  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black")
#
```

```         
 ### A3SS mettl3_KO_GFP_mettl3_KO_DC1_GOF
```

There are a total of 11154 A3SS transcripts in METTL3-KO YthDC1-GoF relative to METTL3-KO after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *A3SS* in METTL3-KO YthDC1-GoF relative to METTL3-KO_GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & 
                                                    abs(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, 
                                                   A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones volc_SE_Mettl3_KO_GFP
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts SE_METTL3-KOYthdc1-GoF_METTL3-KOGFP
ggplot(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "#0000FF"),
                     labels = c(paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() + xlim(c(-1, 1)) +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "METTL3-KO Ythdc1-KO/METTL3-KO-GFP(A3'SS)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 



```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF)) {
 # Make GeneID unique
  A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF) <- A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}

#write.table(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF_A3.txt', quote = F, sep = "\t", row.names = F)
```



```         
### A5 mettl3_KO_GFP_mettl3_KO_DC1_GOF
```

There are a total of 18194 A3SS transcripts in METTL3-KO YthDC1-GoF relative to METTL3-KO after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *A5SS* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts volc_A3SS_Mettl3-KO_GFP
ggplot(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "#0000FF"),
                     labels = c(paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() + xlim(c(-1, 1)) +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "A5SS transcripts/genes in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}

if ("GeneID" %in% colnames(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF)) {
 # Make GeneID unique
  A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF) <- A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first") 
   
# Retrieve GO term descriptions
A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}

#write.table(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF.txt', quote = F, sep = "\t", row.names = F)
```


```         
### RI mettl3_KO_GFP_mettl3_KO_DC1_GOF
```

There are a total of 6827 RI transcripts in METTL3-KO YthDC1-GoF relative to METTL3-KO after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)

nrow(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *RI* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR)

RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Excluded", "Retained")


significant_genes <- subset(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)


RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")


counts <- table(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color)


ggplot(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Retained" = "#ff7f0e",  "Excluded" = "#0000FF"),
                     labels = c(
                                paste("Excluded -", counts["Excluded"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Retained -", counts["Retained"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Intron Retention transcripts in METTL3-KO Ythdc1-KO/METTL3-KO GFP") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF)) {
 # Make GeneID unique
  RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF) <- RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
     if (!is.na(go_id)) {
       Term(go_id)
       } else {
         NA}})

  
  RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n")
}
#write.table(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF.txt', quote = F, sep = "\t", row.names = F)
```

```         
### MXE mettl3_KO_GFP_mettl3_KO_DC1_GOF
```

There are a total of 19616 MXE in METTL3-KO YthDC1-GoF relative to METTL3-KO after filtering out transcripts with less than 5 reads

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF <- MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),labels = c(paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "METTL3-KO-YTHDC-GoF/METTL3-KO-GFP(MXE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)) {
# Make GeneID unique
  MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF) <- MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```



    The scatter plot below shows significant the spliced events in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to prepare data with read filtering steps
prepare_data <- function(file_path, event_type) {
  # Read the data
  data <- read.table(file_path, header = TRUE, sep = "\t")
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)

  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  # Add necessary columns
  data <- data %>%
    mutate(delta_PSI = IncLevelDifference,
           Significant = ifelse(FDR < 0.05 & abs(delta_PSI) > 0.2, "Significant", "Not Significant"),
           EventType = event_type,
           Event_Types = ifelse(Significant == "Significant", event_type, "Not Significant"))
  
  # Select necessary columns
  data <- data[, c("geneSymbol", "delta_PSI", "FDR", "Significant", "EventType", "Event_Types")]
  
  return(data)
}


file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt"
    )

event_types <- names(file_paths)

# Extract significant genes for each event type
significant_genes_list <- mapply(prepare_data, file_paths, event_types, SIMPLIFY = FALSE)

# Combine all splicing events into a single dataframe
all_genes <- do.call(rbind, significant_genes_list)

# Calculate counts for each event type
event_counts <- as.data.frame(table(all_genes$Event_Types))
colnames(event_counts) <- c("Event_Types", "Count")

# Define colors for splicing event types
type_colors <- c("SE" = "blue", "A3SS" = "black", "A5SS" = "darkgreen", "MXE" = "red", "RI" = "#ff7f0e", "Not Significant" = "grey")

# Identify top genes (top 100 based on FDR and delta_PSI) volc_RI_Mettl3-KO_GFP
top_genes <- all_genes[order(all_genes$FDR, -abs(all_genes$delta_PSI)),][1:1000,]

# Prepare custom labels with counts for the legend 
custom_labels <- with(event_counts, setNames(paste(Event_Types, " (", Count, ")", sep=""), Event_Types))

# Create the plot
ggplot(all_genes, aes(x = delta_PSI, y = -log10(FDR), color = Event_Types)) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = type_colors, labels = custom_labels) +
    geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
    geom_vline(xintercept = c(-0.2, 0.2), linetype = "dashed", color = "black") +
    theme_minimal() + xlim(-1.0, 1.0) +
    labs(x = "", y = "Log10 Adj. P-value", title = "Significant splied genes in METTL3-KO YTHDC1-GoF relative to METTL3-KO GFP") +
    theme(legend.position = "right", plot.title = element_text(hjust = 0.5)) +
    # Use geom_text_repel to remove boxes and filter for significant genes only
    geom_text_repel(data = top_genes %>% dplyr::filter(Significant == "Significant"), 
                    aes(label = geneSymbol), size = 3, 
                    box.padding = 0.35, point.padding = 0.5, max.overlaps = 10)

```        

    
    
    
    Differential expressed gene in METTL3-KO YTHDC1-KO
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt")

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

# Add differential expression information
splicing_diff <- splicing_diff %>%
  left_join(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>% dplyr::select(current_symbol, log2FoldChange, padj), by = c("geneSymbol" = "current_symbol")) %>%
  mutate(Regulation_DE = ifelse(log2FoldChange > 2, "Upregulated", ifelse(log2FoldChange < -2, "Downregulated", "Not Significant")))

# Exclude "Not Significant" genes
splicing_diff <- splicing_diff %>% dplyr::filter(Regulation_DE != "Not Significant")

```


```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter significant and non-significant genes
significant_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>% 
  dplyr::filter(diffexpressed != "Not Significant")

non_significant_genes <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>% 
  dplyr::filter(diffexpressed == "Not Significant")

# Function to count the number of genes involved in each splicing event type
count_genes_by_event <- function(data, gene_list) {
  data %>%
    dplyr::filter(geneSymbol %in% gene_list$current_symbol) %>%
    group_by(EventType) %>%
    summarise(
      Count = n(),
      GeneNames = paste(geneSymbol, collapse = ", ")  # Collect gene names
    )
}

# Count for significant DEGs
sig_event_counts <- count_genes_by_event(all_significant_genes, significant_genes)
sig_event_counts <- sig_event_counts %>% mutate(Significance = "Differentially Expressed")

# Count for non-significant DEGs
non_sig_event_counts <- count_genes_by_event(all_significant_genes, non_significant_genes)
non_sig_event_counts <- non_sig_event_counts %>% mutate(Significance = "Not Differentially Expressed")

# Combine the counts into one dataframe
combined_event_counts <- bind_rows(sig_event_counts, non_sig_event_counts)

# Reorder the Significance factor to ensure "Significant" is on top
combined_event_counts$Significance <- factor(combined_event_counts$Significance, 
                                             levels = c("Not Differentially Expressed", "Differentially Expressed"))

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# Extract the p-value and format it
p_value <- chi_square_result$p.value
p_value_formatted <- sprintf("Chi-Square Test p-value: %.2e", p_value)  # Scientific notation


plot_ly(combined_event_counts, 
                x = ~EventType, 
                y = ~Count, 
                color = ~Significance, 
                type = 'bar', 
                text = ~Count,  # Display the count number on each bar
                textposition = 'auto',  # Position the text within the bars
                textfont = list(color = "black"),  # Set the color of the text to black
                hoverinfo = 'text') %>%
  layout(title = "METTL3-KO YTHDC1-GoF relative to METTL3-KO GFP",
    xaxis = list(title = "Splicing Event Type"),
    yaxis = list(title = "Number of Genes"),
    barmode = 'stack',
    legend = list(
      title = list(text = p_value_formatted),  # Add the p-value to the legend title
      traceorder = "reversed"))

custom_colors <- c("Not Differentially Expressed" = "#a6bddb", "Differentially Expressed" = "#de2d26")
ggplot(combined_event_counts, aes(x = EventType, y = Count, fill = Significance)) +
  geom_bar(stat = "identity", position = "stack", width = 0.7) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 5, fontface = "bold", color = "black") +
  scale_fill_manual(values = custom_colors) +
  labs(
    title = "Number of DEGs in METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP",
    x = "Splicing Event Type",
    y = "Number of Genes",
    fill = sprintf("Gene Expression Status\n(p = %.2e)", p_value)) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank())

```
  
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)


```
 
```{r}
# Calculate the total number of splicing genes
total_splicing_genes <- length(unique(all_significant_genes$geneSymbol))

# Calculate the number and percentage of DEGs implicated in splicing
num_diff_splicing_genes <- length(unique(splicing_diff$geneSymbol))
percent_diff_splicing_genes <- (num_diff_splicing_genes / total_splicing_genes) * 100

# Print the summary
cat("Total number of genes implicated in splicing:", total_splicing_genes, "\n")
cat("Number of DEGs implicated in splicing:", num_diff_splicing_genes, "\n")
cat("Percentage of DEGs implicated in splicing:", round(percent_diff_splicing_genes, 2), "%\n")

# Optional: Print the number of splicing events per event type
splicing_event_counts <- all_significant_genes %>%
  group_by(EventType) %>%
  summarise(Count = n())

cat("\nNumber of significant splicing events by type:\n")
print(splicing_event_counts)
```
 












    Mettl3-KO_DC1_GOF/S2_GFP
  
  
    Bar chart below show the number of significant splicing difference in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}

prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)
  
  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  return(data)
}

# Function to count splicing events after filtering
count_splicing_events <- function(filename) {
  data <- prepare_and_filter_data(filename)
  
  # Filter for significant genes
  significant_genes <- subset(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  n_exclusion <- sum(significant_genes$IncLevelDifference < 0)
  n_inclusion <- sum(significant_genes$IncLevelDifference > 0)
  
  sample_id <- basename(filename)
  sample_id <- gsub(".MATS.JCEC.txt", "", sample_id)
  
  return(data.frame(SampleID = sample_id, Exclusion = n_exclusion, Inclusion = n_inclusion))
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt")

# Apply function to each file and combine results
results <- do.call(rbind, lapply(files, count_splicing_events))

# Calculate totals
total_exclusion <- sum(results$Exclusion)
total_inclusion <- sum(results$Inclusion)

# Melt the results data for ggplot
results_melted <- tidyr::pivot_longer(results, cols = c(Exclusion, Inclusion), names_to = "EventType", values_to = "Count")
# Perform a paired t-test
t_test_result <- t.test(results$Inclusion, results$Exclusion, paired = TRUE)

# Extract p-value from the t-test result
p_value <- t_test_result$p.value
formatted_p_value <- sprintf("P-value: %.3g", p_value)

# Generate the enhanced ggplot with the p-value in the legend
ggplot(results_melted, aes(x = SampleID, y = Count, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 4, color = "black") +
  labs(title = sprintf("Mettl3-KO-Ythdc1-GoF/S2-GFP: Total Exclusion : %d, Total Inclusion: %d", total_exclusion, total_inclusion),
       x = "Splicing Event Type", 
       y = "Number of Significant Splicing Events",
       fill = paste("Splicing Event (", formatted_p_value, ")", sep = "")) +
  scale_fill_manual(values = c("Exclusion" = "#a6bddb", "Inclusion" = "#ff7f0e")) +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "top",
        legend.title = element_text(face = "bold"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', color = 'gray')) +
  guides(fill = guide_legend(title = paste("Splicing Event (", formatted_p_value, ")", sep = "")))


# Print the t-test results
print(t_test_result)


```



    Below are the number of spliced events in Ythdc1-KO relative to S2S that might be of interest to us and validate some of the genes from PCR
```{r message=FALSE, warning=FALSE, echo=FALSE}


genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega",   "Bsg", "NK7.1", "alpha-Man-Ia", "Vinc", "Ptp4E", "shi", "RpL15", "scrib", "dlg1", "PlexA", "CG13124", "Pka-R2", "Tep2", "mfas", "CG13366", "CG45050", "Fer1HCH", "CG11357", "CG2767", "CaMKII", "ND-B12", "ATPCL", "AnxB11", "Best1", "CDC50", "CG10616", "CG32772", "Cals", "sd", "CG4822", "Dscam1", "l(2)k01209", 
"par-1", "CkIIalpha")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(
      S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and splicing type
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extracting the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  return(gene_event_counts)
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize a list to hold the results from each file
results_list <- lapply(files, count_splicing_events, genes = genes_of_interest)

# Combine results from all files into a single data frame
final_results <- bind_rows(results_list)

# Reshape the data to a wide format for inclusion and exclusion counts
final_results_wide <- final_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final table
print(final_results_wide)


#write.table(final_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/S2_GFP_mettl3_KO_DC1_GOF_Sig_gene_of_interest.txt", row.names = FALSE, quote = FALSE, sep = "\t")

```

frequency of gene

```{r}
# Define vector of gene names
genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, ID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and extract high-frequency exons
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extract the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  # Calculate exon frequencies and extract the highest frequency exons
  exon_frequency <- significant_genes %>%
    group_by(geneSymbol) %>%
    summarise(Frequency = n()) %>%
    arrange(desc(Frequency)) %>%
    slice(1:10) %>%  # Extract top 10 highest frequency exons
    mutate(EventType = event_type) # Add event type
  
  return(list(gene_counts = gene_event_counts, high_freq_exons = exon_frequency))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize lists to hold results
gene_results_list <- list()
exon_results_list <- list()

for (file in files) {
  results <- count_splicing_events(file, genes_of_interest)
  gene_results_list <- append(gene_results_list, list(results$gene_counts))
  exon_results_list <- append(exon_results_list, list(results$high_freq_exons))
}

# Combine results from all files into single data frames
final_gene_results <- bind_rows(gene_results_list)
final_exon_results <- bind_rows(exon_results_list)

# Reshape the gene data to a wide format for inclusion and exclusion counts
final_gene_results_wide <- final_gene_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final tables
print(final_gene_results_wide)
print(final_exon_results)

# Write the final tables to files
#write.table(final_gene_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/Sig_gene_of_interest.txt", row.names = FALSE, quote = F, sep = "\t")
#write.table(final_exon_results, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/High_frequency_exons.txt", row.names = FALSE, quote = F, sep = "\t")

```












```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_mettl3_KO_DC1_GOF <- SE_S2_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(SE_S2_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(SE_S2_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
SE_S2_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(SE_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(SE_S2_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
SE_S2_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(SE_S2_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(SE_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, SE_S2_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(SE_S2_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(SE_S2_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-Ythdc1-GoF/S2-GFP (SE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(SE_S2_GFP_mettl3_KO_DC1_GOF)) {
# Make GeneID unique
  SE_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(SE_S2_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(SE_S2_GFP_mettl3_KO_DC1_GOF) <- SE_S2_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #SE_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  SE_S2_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   SE_S2_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
SE_S2_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(SE_S2_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  SE_S2_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(SE_S2_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```







```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2_GFP_mettl3_KO_DC1_GOF <- A3_S2_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(A3_S2_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(A3_S2_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A3_S2_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(A3_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A3_S2_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A3_S2_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(A3_S2_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(A3_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, A3_S2_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A3_S2_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(A3_S2_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-Ythdc1-GoF/S2-GFP (A3'SS)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A3_S2_GFP_mettl3_KO_DC1_GOF)) {
# Make GeneID unique
  A3_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(A3_S2_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A3_S2_GFP_mettl3_KO_DC1_GOF) <- A3_S2_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A3_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A3_S2_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A3_S2_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A3_S2_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(A3_S2_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A3_S2_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(A3_S2_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```









```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2_GFP_mettl3_KO_DC1_GOF <- A5_S2_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(A5_S2_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(A5_S2_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A5_S2_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(A5_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A5_S2_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A5_S2_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(A5_S2_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(A5_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, A5_S2_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A5_S2_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(A5_S2_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-Ythdc1-GoF/S2-GFP (A5'SS)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A5_S2_GFP_mettl3_KO_DC1_GOF)) {
# Make GeneID unique
  A5_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(A5_S2_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A5_S2_GFP_mettl3_KO_DC1_GOF) <- A5_S2_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A5_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A5_S2_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A5_S2_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A5_S2_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(A5_S2_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A5_S2_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(A5_S2_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```








```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_mettl3_KO_DC1_GOF <- MXE_S2_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(MXE_S2_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(MXE_S2_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
MXE_S2_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(MXE_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(MXE_S2_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
MXE_S2_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(MXE_S2_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(MXE_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, MXE_S2_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(MXE_S2_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(MXE_S2_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-Ythdc1-GoF/S2-GFP (MXE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(MXE_S2_GFP_mettl3_KO_DC1_GOF)) {
# Make GeneID unique
  MXE_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(MXE_S2_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(MXE_S2_GFP_mettl3_KO_DC1_GOF) <- MXE_S2_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #MXE_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  MXE_S2_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   MXE_S2_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
MXE_S2_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(MXE_S2_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  MXE_S2_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(MXE_S2_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```





```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_mettl3_KO_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_mettl3_KO_DC1_GOF <- RI_S2_GFP_mettl3_KO_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(RI_S2_GFP_mettl3_KO_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_mettl3_KO_DC1_GOF$logPValue <- -1 * log10(RI_S2_GFP_mettl3_KO_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
RI_S2_GFP_mettl3_KO_DC1_GOF$Splicing <- ifelse(RI_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(RI_S2_GFP_mettl3_KO_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
RI_S2_GFP_mettl3_KO_DC1_GOF$Color <- ifelse(RI_S2_GFP_mettl3_KO_DC1_GOF$FDR < 0.05 & abs(RI_S2_GFP_mettl3_KO_DC1_GOF$IncLevelDifference) > 0.2, RI_S2_GFP_mettl3_KO_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(RI_S2_GFP_mettl3_KO_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(RI_S2_GFP_mettl3_KO_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-Ythdc1-GoF/S2-GFP (RI)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(RI_S2_GFP_mettl3_KO_DC1_GOF)) {
# Make GeneID unique
  RI_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- make.unique(as.character(RI_S2_GFP_mettl3_KO_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(RI_S2_GFP_mettl3_KO_DC1_GOF) <- RI_S2_GFP_mettl3_KO_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #RI_S2_GFP_mettl3_KO_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  RI_S2_GFP_mettl3_KO_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   RI_S2_GFP_mettl3_KO_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
RI_S2_GFP_mettl3_KO_DC1_GOF$GODESCR <- sapply(RI_S2_GFP_mettl3_KO_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  RI_S2_GFP_mettl3_KO_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_mettl3_KO_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(RI_S2_GFP_mettl3_KO_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```








```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to prepare data with read filtering steps
prepare_data <- function(file_path, event_type) {
  # Read the data
  data <- read.table(file_path, header = TRUE, sep = "\t")
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)

  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  # Add necessary columns
  data <- data %>%
    mutate(delta_PSI = IncLevelDifference,
           Significant = ifelse(FDR < 0.05 & abs(delta_PSI) > 0.2, "Significant", "Not Significant"),
           EventType = event_type,
           Event_Types = ifelse(Significant == "Significant", event_type, "Not Significant"))
  
  # Select necessary columns
  data <- data[, c("geneSymbol", "delta_PSI", "FDR", "Significant", "EventType", "Event_Types")]
  
  return(data)
}


file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt"
    )

event_types <- names(file_paths)

# Extract significant genes for each event type
significant_genes_list <- mapply(prepare_data, file_paths, event_types, SIMPLIFY = FALSE)

# Combine all splicing events into a single dataframe
all_genes <- do.call(rbind, significant_genes_list)

# Calculate counts for each event type
event_counts <- as.data.frame(table(all_genes$Event_Types))
colnames(event_counts) <- c("Event_Types", "Count")

# Define colors for splicing event types
type_colors <- c("SE" = "blue", "A3SS" = "black", "A5SS" = "darkgreen", "MXE" = "red", "RI" = "#984EA3", "Not Significant" = "grey")
# Identify top genes (top 100 based on FDR and delta_PSI) volc_RI_Mettl3-KO_GFP
top_genes <- all_genes[order(all_genes$FDR, -abs(all_genes$delta_PSI)),][1:1000,]

# Prepare custom labels with counts for the legend 
custom_labels <- with(event_counts, setNames(paste(Event_Types, " (", Count, ")", sep=""), Event_Types))

# Create the plot
ggplot(all_genes, aes(x = delta_PSI, y = -log10(FDR), color = Event_Types)) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = type_colors, labels = custom_labels) +
    geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
    geom_vline(xintercept = c(-0.2, 0.2), linetype = "dashed", color = "black") +
    theme_minimal() + xlim(-1.0, 1.0) +
    labs(x = "", y = "Log10 Adj. P-value", title = "METTL3-KO-YTHDC1-GoF/S2-GFP") +
    theme(legend.position = "right", plot.title = element_text(hjust = 0.5)) +
    # Use geom_text_repel to remove boxes and filter for significant genes only
    geom_text_repel(data = top_genes %>% dplyr::filter(Significant == "Significant"), 
                    aes(label = geneSymbol), size = 3, 
                    box.padding = 0.35, point.padding = 0.5, max.overlaps = 10)

```        


    Differential expressed gene 
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt")

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter significant and non-significant genes
significant_genes <- res_S2_GFP_mettl3_KO_DC1_GOF %>% 
  dplyr::filter(diffexpressed != "Not Significant")

non_significant_genes <- res_S2_GFP_mettl3_KO_DC1_GOF %>% 
  dplyr::filter(diffexpressed == "Not Significant")

# Function to count the number of genes involved in each splicing event type
count_genes_by_event <- function(data, gene_list) {
  data %>%
    dplyr::filter(geneSymbol %in% gene_list$current_symbol) %>%
    group_by(EventType) %>%
    summarise(
      Count = n(),
      GeneNames = paste(geneSymbol, collapse = ", "))
}

# Count for significant DEGs
sig_event_counts <- count_genes_by_event(all_significant_genes, significant_genes)
sig_event_counts <- sig_event_counts %>% mutate(Significance = "Differentially Expressed")

# Count for non-significant DEGs
non_sig_event_counts <- count_genes_by_event(all_significant_genes, non_significant_genes)
non_sig_event_counts <- non_sig_event_counts %>% mutate(Significance = "Not Differentially Expressed")

# Combine the counts into one dataframe
combined_event_counts <- bind_rows(sig_event_counts, non_sig_event_counts)

# Reorder the Significance factor to ensure "Significant" is on top
combined_event_counts$Significance <- factor(combined_event_counts$Significance, 
                                             levels = c("Not Differentially Expressed", "Differentially Expressed"))

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# Extract the p-value and format it
p_value <- chi_square_result$p.value
p_value_formatted <- sprintf("Chi-Square Test p-value: %.2e", p_value)  # Scientific notation


plot_ly(combined_event_counts, 
                x = ~EventType, 
                y = ~Count, 
                color = ~Significance, 
                type = 'bar', 
                text = ~Count,  # Display the count number on each bar
                textposition = 'auto',  # Position the text within the bars
                textfont = list(color = "black"),  # Set the color of the text to black
                hoverinfo = 'text') %>%
  layout(title = "METTL3-KO-YTHDC1-GoF/S2-GFP",
    xaxis = list(title = "Splicing Event Type"),
    yaxis = list(title = "Number of Genes"),
    barmode = 'stack',
    legend = list(
      title = list(text = p_value_formatted),  # Add the p-value to the legend title
      traceorder = "reversed"))

custom_colors <- c("Not Differentially Expressed" = "#a6bddb", "Differentially Expressed" = "#de2d26")
# Create the ggplot with p-value in legend
ggplot(combined_event_counts, aes(x = EventType, y = Count, fill = Significance)) +
  geom_bar(stat = "identity", position = "stack", width = 0.7) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 5, fontface = "bold", color = "black") +
  scale_fill_manual(values = custom_colors) +
  labs(
    title = "Number of DEGs METTL3-KO-YTHDC1-GoF/S2-GFP in splicing",
    x = "Splicing Event Type",
    y = "Number of Genes",
    fill = sprintf("Gene Expression Status\n(p = %.2e)", p_value)) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank())

```
  
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)


```






mettl3_KO_DC1_GOF_S2_DC1_GOF
  
  
    Bar chart below show the number of significant splicing difference in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}

prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)
  
  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  return(data)
}

# Function to count splicing events after filtering
count_splicing_events <- function(filename) {
  data <- prepare_and_filter_data(filename)
  
  # Filter for significant genes
  significant_genes <- subset(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  n_exclusion <- sum(significant_genes$IncLevelDifference < 0)
  n_inclusion <- sum(significant_genes$IncLevelDifference > 0)
  
  sample_id <- basename(filename)
  sample_id <- gsub(".MATS.JCEC.txt", "", sample_id)
  
  return(data.frame(SampleID = sample_id, Exclusion = n_exclusion, Inclusion = n_inclusion))
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A5SS.MATS.JCEC.txt")

# Apply function to each file and combine results
results <- do.call(rbind, lapply(files, count_splicing_events))

# Calculate totals
total_exclusion <- sum(results$Exclusion)
total_inclusion <- sum(results$Inclusion)

# Melt the results data for ggplot
results_melted <- tidyr::pivot_longer(results, cols = c(Exclusion, Inclusion), names_to = "EventType", values_to = "Count")
# Perform a paired t-test
t_test_result <- t.test(results$Inclusion, results$Exclusion, paired = TRUE)

# Extract p-value from the t-test result
p_value <- t_test_result$p.value
formatted_p_value <- sprintf("P-value: %.3g", p_value)

# Generate the enhanced ggplot with the p-value in the legend
ggplot(results_melted, aes(x = SampleID, y = Count, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 4, color = "black") +
  labs(title = sprintf("S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF : Total Exclusion : %d, Total Inclusion: %d", total_exclusion, total_inclusion),
       x = "Splicing Event Type", 
       y = "Number of Significant Splicing Events",
       fill = paste("Splicing Event (", formatted_p_value, ")", sep = "")) +
  scale_fill_manual(values = c("Exclusion" = "#a6bddb", "Inclusion" = "#ff7f0e")) +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "top",
        legend.title = element_text(face = "bold"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', color = 'gray')) +
  guides(fill = guide_legend(title = paste("Splicing Event (", formatted_p_value, ")", sep = "")))


# Print the t-test results
print(t_test_result)


```



    Below are the number of spliced events in Ythdc1-KO relative to S2S that might be of interest to us and validate some of the genes from PCR
```{r message=FALSE, warning=FALSE, echo=FALSE}


genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega",   "Bsg", "NK7.1", "alpha-Man-Ia", "Vinc", "Ptp4E", "shi", "RpL15", "scrib", "dlg1", "PlexA", "CG13124", "Pka-R2", "Tep2", "mfas", "CG13366", "CG45050", "Fer1HCH", "CG11357", "CG2767", "CaMKII", "ND-B12", "ATPCL", "AnxB11", "Best1", "CDC50", "CG10616", "CG32772", "Cals", "sd", "CG4822", "Dscam1", "l(2)k01209", 
"par-1", "CkIIalpha")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(
      S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and splicing type
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extracting the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  return(gene_event_counts)
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize a list to hold the results from each file
results_list <- lapply(files, count_splicing_events, genes = genes_of_interest)

# Combine results from all files into a single data frame
final_results <- bind_rows(results_list)

# Reshape the data to a wide format for inclusion and exclusion counts
final_results_wide <- final_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final table
print(final_results_wide)


#write.table(final_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/mettl3_KO_DC1_GOF_S2_DC1_GOF_Sig_gene_of_interest.txt", row.names = FALSE, quote = FALSE, sep = "\t")

```

frequency of gene

```{r}
# Define vector of gene names
genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, ID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and extract high-frequency exons
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extract the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  # Calculate exon frequencies and extract the highest frequency exons
  exon_frequency <- significant_genes %>%
    group_by(geneSymbol) %>%
    summarise(Frequency = n()) %>%
    arrange(desc(Frequency)) %>%
    slice(1:10) %>%  # Extract top 10 highest frequency exons
    mutate(EventType = event_type) # Add event type
  
  return(list(gene_counts = gene_event_counts, high_freq_exons = exon_frequency))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/SE.MATS.JCEC.txt")

# Initialize lists to hold results
gene_results_list <- list()
exon_results_list <- list()

for (file in files) {
  results <- count_splicing_events(file, genes_of_interest)
  gene_results_list <- append(gene_results_list, list(results$gene_counts))
  exon_results_list <- append(exon_results_list, list(results$high_freq_exons))
}

# Combine results from all files into single data frames
final_gene_results <- bind_rows(gene_results_list)
final_exon_results <- bind_rows(exon_results_list)

# Reshape the gene data to a wide format for inclusion and exclusion counts
final_gene_results_wide <- final_gene_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final tables
print(final_gene_results_wide)
print(final_exon_results)


#write.table(final_gene_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/Sig_gene_of_interest.txt", row.names = FALSE, quote = F, sep = "\t")
#write.table(final_exon_results, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/High_frequency_exons.txt", row.names = FALSE, quote = F, sep = "\t")

```












```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO_DC1_GOF_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/SE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO_DC1_GOF_S2_DC1_GOF <- SE_mettl3_KO_DC1_GOF_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$logPValue <- -1 * log10(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing <- ifelse(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color <- ifelse(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR < 0.05 & abs(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference) > 0.2, SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #Mettl3_KO_DC1_GOF_S2_DC1_GOF
ggplot(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF (SE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF)) {
# Make GeneID unique
  SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- make.unique(as.character(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF) <- SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GODESCR <- sapply(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  SE_mettl3_KO_DC1_GOF_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```








```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_mettl3_KO_DC1_GOF_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A3SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_mettl3_KO_DC1_GOF_S2_DC1_GOF <- A3_mettl3_KO_DC1_GOF_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$logPValue <- -1 * log10(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing <- ifelse(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color <- ifelse(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR < 0.05 & abs(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference) > 0.2, A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF (A3'SS)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF)) {
# Make GeneID unique
  A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- make.unique(as.character(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF) <- A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$GODESCR <- sapply(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A3_mettl3_KO_DC1_GOF_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/A3_mettl3_KO_DC1_GOF_S2_DC1_GOF/A3_mettl3_KO_DC1_GOF_S2_DC1_GOF_A3.txt', quote = F, sep = "\t", row.names = F)
```






```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_mettl3_KO_DC1_GOF_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A5SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_mettl3_KO_DC1_GOF_S2_DC1_GOF <- A5_mettl3_KO_DC1_GOF_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$logPValue <- -1 * log10(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing <- ifelse(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color <- ifelse(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR < 0.05 & abs(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference) > 0.2, A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF (A5'SS)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF)) {
# Make GeneID unique
  A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- make.unique(as.character(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF) <- A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$GODESCR <- sapply(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A5_mettl3_KO_DC1_GOF_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/A5_mettl3_KO_DC1_GOF_S2_DC1_GOF_A3.txt', quote = F, sep = "\t", row.names = F)
```






```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_mettl3_KO_DC1_GOF_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/RI.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_mettl3_KO_DC1_GOF_S2_DC1_GOF <- RI_mettl3_KO_DC1_GOF_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$logPValue <- -1 * log10(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing <- ifelse(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color <- ifelse(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR < 0.05 & abs(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference) > 0.2, RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF (RI)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF)) {
# Make GeneID unique
  RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- make.unique(as.character(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF) <- RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$GODESCR <- sapply(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  RI_mettl3_KO_DC1_GOF_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/A5_mettl3_KO_DC1_GOF_S2_DC1_GOF_A3.txt', quote = F, sep = "\t", row.names = F)
```



```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/MXE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF <- MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$logPValue <- -1 * log10(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR)
# Create a binary inclusion/exclusion column
MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing <- ifelse(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color <- ifelse(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$FDR < 0.05 & abs(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$IncLevelDifference) > 0.2, MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF (MXE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF)) {
# Make GeneID unique
  MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- make.unique(as.character(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID))
  
  # Set GeneID as rownames
  rownames(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF) <- MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GODESCR <- sapply(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/A5_mettl3_KO_DC1_GOF_S2_DC1_GOF_A3.txt', quote = F, sep = "\t", row.names = F)
```


```{r}
# Function to prepare data with read filtering steps
prepare_data <- function(file_path, event_type) {
  # Read the data
  data <- read.table(file_path, header = TRUE, sep = "\t")
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)

  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  # Add necessary columns
  data <- data %>%
    mutate(delta_PSI = IncLevelDifference,
           Significant = ifelse(FDR < 0.05 & abs(delta_PSI) > 0.2, "Significant", "Not Significant"),
           EventType = event_type,
           Event_Types = ifelse(Significant == "Significant", event_type, "Not Significant"))
  
  # Select necessary columns
  data <- data[, c("geneSymbol", "delta_PSI", "FDR", "Significant", "EventType", "Event_Types")]
  
  return(data)
}


file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/RI.MATS.JCEC.txt"
    )

event_types <- names(file_paths)

# Extract significant genes for each event type
significant_genes_list <- mapply(prepare_data, file_paths, event_types, SIMPLIFY = FALSE)

# Combine all splicing events into a single dataframe
all_genes <- do.call(rbind, significant_genes_list)

# Calculate counts for each event type
event_counts <- as.data.frame(table(all_genes$Event_Types))
colnames(event_counts) <- c("Event_Types", "Count")

# Define colors for splicing event types
type_colors <- c("SE" = "blue", "A3SS" = "black", "A5SS" = "darkgreen", "MXE" = "red", "RI" = "#984EA3", "Not Significant" = "grey")

# Identify top genes (top 100 based on FDR and delta_PSI) volc_RI_Mettl3-KO_GFP
top_genes <- all_genes[order(all_genes$FDR, -abs(all_genes$delta_PSI)),][1:1000,]

# Prepare custom labels with counts for the legend 
custom_labels <- with(event_counts, setNames(paste(Event_Types, " (", Count, ")", sep=""), Event_Types))

# Create the plot A5_Mettl3_KO_DC1_GOF_S2_DC1_GOF
ggplot(all_genes, aes(x = delta_PSI, y = -log10(FDR), color = Event_Types)) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = type_colors, labels = custom_labels) +
    geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
    geom_vline(xintercept = c(-0.2, 0.2), linetype = "dashed", color = "black") +
    theme_minimal() + xlim(-1.0, 1.0) +
    labs(x = "", y = "Log10 Adj. P-value", title = "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF") +
    theme(legend.position = "right", plot.title = element_text(hjust = 0.5)) +
    # Use geom_text_repel to remove boxes and filter for significant genes only
    geom_text_repel(data = top_genes %>% dplyr::filter(Significant == "Significant"), 
                    aes(label = geneSymbol), size = 3, 
                    box.padding = 0.35, point.padding = 0.5, max.overlaps = 10)
```


    Differential expressed gene 
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/SE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/RI.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/MXE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A5SS.MATS.JCEC.txt")

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

```


```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter significant and non-significant genes
significant_genes <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>% 
  dplyr::filter(diffexpressed != "Not Significant")

non_significant_genes <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>% 
  dplyr::filter(diffexpressed == "Not Significant")

# Function to count the number of genes involved in each splicing event type
count_genes_by_event <- function(data, gene_list) {
  data %>%
    dplyr::filter(geneSymbol %in% gene_list$current_symbol) %>%
    group_by(EventType) %>%
    summarise(
      Count = n(),
      GeneNames = paste(geneSymbol, collapse = ", "))
}

# Count for significant DEGs
sig_event_counts <- count_genes_by_event(all_significant_genes, significant_genes)
sig_event_counts <- sig_event_counts %>% mutate(Significance = "Differentially Expressed")

# Count for non-significant DEGs
non_sig_event_counts <- count_genes_by_event(all_significant_genes, non_significant_genes)
non_sig_event_counts <- non_sig_event_counts %>% mutate(Significance = "Not Differentially Expressed")

# Combine the counts into one dataframe
combined_event_counts <- bind_rows(sig_event_counts, non_sig_event_counts)

# Reorder the Significance factor to ensure "Significant" is on top
combined_event_counts$Significance <- factor(combined_event_counts$Significance, 
                                             levels = c("Not Differentially Expressed", "Differentially Expressed"))

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# Extract the p-value and format it
p_value <- chi_square_result$p.value
p_value_formatted <- sprintf("Chi-Square Test p-value: %.2e", p_value)  # Scientific notation


plot_ly(combined_event_counts, 
                x = ~EventType, 
                y = ~Count, 
                color = ~Significance, 
                type = 'bar', 
                text = ~Count,  # Display the count number on each bar
                textposition = 'auto',  # Position the text within the bars
                textfont = list(color = "black"),  # Set the color of the text to black
                hoverinfo = 'text') %>%
  layout(title = "METTL3-KO-YTHDC1-GoF/S2-GFP",
    xaxis = list(title = "Splicing Event Type"),
    yaxis = list(title = "Number of Genes"),
    barmode = 'stack',
    legend = list(
      title = list(text = p_value_formatted),  # Add the p-value to the legend title
      traceorder = "reversed"))

custom_colors <- c("Not Differentially Expressed" = "#a6bddb", "Differentially Expressed" = "#de2d26")
# Create the ggplot with p-value in legend
ggplot(combined_event_counts, aes(x = EventType, y = Count, fill = Significance)) +
  geom_bar(stat = "identity", position = "stack", width = 0.7) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 5, fontface = "bold", color = "black") +
  scale_fill_manual(values = custom_colors) +
  labs(
    title = "Number of DEGs S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF in splicing",
    x = "Splicing Event Type",
    y = "Number of Genes",
    fill = sprintf("Gene Expression Status\n(p = %.2e)", p_value)) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank())

```
  
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)


```





S2_GFP_mettl3_KO_GFP
  
  
    Bar chart below show the number of significant splicing difference in Ythdc1-KO relative to S2S

```{r message=FALSE, warning=FALSE, echo=FALSE}

prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)
  
  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  return(data)
}

# Function to count splicing events after filtering
count_splicing_events <- function(filename) {
  data <- prepare_and_filter_data(filename)
  
  # Filter for significant genes
  significant_genes <- subset(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  n_exclusion <- sum(significant_genes$IncLevelDifference < 0)
  n_inclusion <- sum(significant_genes$IncLevelDifference > 0)
  
  sample_id <- basename(filename)
  sample_id <- gsub(".MATS.JCEC.txt", "", sample_id)
  
  return(data.frame(SampleID = sample_id, Exclusion = n_exclusion, Inclusion = n_inclusion))
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/SE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A3SS.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/RI.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/MXE.MATS.JCEC.txt",
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A5SS.MATS.JCEC.txt")

# Apply function to each file and combine results
results <- do.call(rbind, lapply(files, count_splicing_events))

# Calculate totals
total_exclusion <- sum(results$Exclusion)
total_inclusion <- sum(results$Inclusion)

# Melt the results data for ggplot
results_melted <- tidyr::pivot_longer(results, cols = c(Exclusion, Inclusion), names_to = "EventType", values_to = "Count")
# Perform a paired t-test
t_test_result <- t.test(results$Inclusion, results$Exclusion, paired = TRUE)

# Extract p-value from the t-test result
p_value <- t_test_result$p.value
formatted_p_value <- sprintf("P-value: %.3g", p_value)

# Generate the enhanced ggplot with the p-value in the legend
ggplot(results_melted, aes(x = SampleID, y = Count, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 4, color = "black") +
  labs(title = sprintf("Mettl3-KO-GFP/S2-GFP : Total Exclusion : %d, Total Inclusion: %d", total_exclusion, total_inclusion),
       x = "Splicing Event Type", 
       y = "Number of Significant Splicing Events",
       fill = paste("Splicing Event (", formatted_p_value, ")", sep = "")) +
  scale_fill_manual(values = c("Exclusion" = "#a6bddb", "Inclusion" = "#ff7f0e")) +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "top",
        legend.title = element_text(face = "bold"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', color = 'gray')) +
  guides(fill = guide_legend(title = paste("Splicing Event (", formatted_p_value, ")", sep = "")))



```



    Below are the number of spliced events in Ythdc1-KO relative to S2S that might be of interest to us and validate some of the genes from PCR
```{r message=FALSE, warning=FALSE, echo=FALSE}


genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega",   "Bsg", "NK7.1", "alpha-Man-Ia", "Vinc", "Ptp4E", "shi", "RpL15", "scrib", "dlg1", "PlexA", "CG13124", "Pka-R2", "Tep2", "mfas", "CG13366", "CG45050", "Fer1HCH", "CG11357", "CG2767", "CaMKII", "ND-B12", "ATPCL", "AnxB11", "Best1", "CDC50", "CG10616", "CG32772", "Cals", "sd", "CG4822", "Dscam1", "l(2)k01209", 
"par-1", "CkIIalpha")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(
      S1R1counts > 0,
      S1R2counts > 0,
      S1R3counts > 0,
      S2R1counts > 0,
      S2R2counts > 0,
      S2R3counts > 0) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and splicing type
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extracting the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  return(gene_event_counts)
}

# File paths
files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/SE.MATS.JCEC.txt")

# Initialize a list to hold the results from each file
results_list <- lapply(files, count_splicing_events, genes = genes_of_interest)

# Combine results from all files into a single data frame
final_results <- bind_rows(results_list)

# Reshape the data to a wide format for inclusion and exclusion counts
final_results_wide <- final_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final table
print(final_results_wide)


#write.table(final_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/mettl3_KO_DC1_GOF_S2_DC1_GOF_Sig_gene_of_interest.txt", row.names = FALSE, quote = FALSE, sep = "\t")

```

frequency of gene

```{r}
# Define vector of gene names
genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", "Sema1b", "Gbeta13F","Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega")

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, ID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to count splicing events per gene and extract high-frequency exons
count_splicing_events <- function(filename, genes) {
  data <- read.table(filename, header = TRUE, stringsAsFactors = FALSE)
  
  # Extract significant genes and filter based on read counts
  significant_genes <- extract_significant_genes(data)
  
  # Filter for significant events and genes of interest
  significant_events <- significant_genes %>%
    dplyr::filter(geneSymbol %in% genes)
  
  # Count the number of inclusion and exclusion events per gene
  gene_event_counts <- significant_events %>%
    group_by(geneSymbol) %>%
    summarise(Exclusion = sum(IncLevelDifference < 0),
              Inclusion = sum(IncLevelDifference > 0))
  
  # Extract the splicing event type from the filename
  event_type <- gsub(pattern = ".*_(.*).MATS.JCEC.txt", replacement = "\\1", x = filename)
  
  # Add the event type to the data frame
  gene_event_counts <- gene_event_counts %>%
    mutate(EventType = event_type)
  
  # Calculate exon frequencies and extract the highest frequency exons
  exon_frequency <- significant_genes %>%
    group_by(geneSymbol) %>%
    summarise(Frequency = n()) %>%
    arrange(desc(Frequency)) %>%
    slice(1:10) %>%  # Extract top 10 highest frequency exons
    mutate(EventType = event_type) # Add event type
  
  return(list(gene_counts = gene_event_counts, high_freq_exons = exon_frequency))
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A5SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A3SS.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/RI.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/MXE.MATS.JCEC.txt", 
           "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/SE.MATS.JCEC.txt")

# Initialize lists to hold results
gene_results_list <- list()
exon_results_list <- list()

for (file in files) {
  results <- count_splicing_events(file, genes_of_interest)
  gene_results_list <- append(gene_results_list, list(results$gene_counts))
  exon_results_list <- append(exon_results_list, list(results$high_freq_exons))
}

# Combine results from all files into single data frames
final_gene_results <- bind_rows(gene_results_list)
final_exon_results <- bind_rows(exon_results_list)

# Reshape the gene data to a wide format for inclusion and exclusion counts
final_gene_results_wide <- final_gene_results %>%
  pivot_wider(names_from = EventType, 
              values_from = c(Exclusion, Inclusion), 
              values_fill = list(Exclusion = 0, Inclusion = 0))

# Print the final tables
print(final_gene_results_wide)
print(final_exon_results)


#write.table(final_gene_results_wide, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/Sig_gene_of_interest.txt", row.names = FALSE, quote = F, sep = "\t")
#write.table(final_exon_results, "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/High_frequency_exons.txt", row.names = FALSE, quote = F, sep = "\t")

```








S2_GFP_mettl3_KO_GFP



```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_mettl3_KO_GFP <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/SE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_mettl3_KO_GFP <- SE_S2_GFP_mettl3_KO_GFP %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(SE_S2_GFP_mettl3_KO_GFP)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_S2_GFP_mettl3_KO_GFP$logPValue <- -1 * log10(SE_S2_GFP_mettl3_KO_GFP$FDR)
# Create a binary inclusion/exclusion column
SE_S2_GFP_mettl3_KO_GFP$Splicing <- ifelse(SE_S2_GFP_mettl3_KO_GFP$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(SE_S2_GFP_mettl3_KO_GFP, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
SE_S2_GFP_mettl3_KO_GFP$Color <- ifelse(SE_S2_GFP_mettl3_KO_GFP$FDR < 0.05 & abs(SE_S2_GFP_mettl3_KO_GFP$IncLevelDifference) > 0.2, SE_S2_GFP_mettl3_KO_GFP$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(SE_S2_GFP_mettl3_KO_GFP$Color)

# Update the scale_color_manual function with new labels including counts 
ggplot(SE_S2_GFP_mettl3_KO_GFP, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-GFP/S2-GFP (SE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(SE_S2_GFP_mettl3_KO_GFP)) {
# Make GeneID unique
  SE_S2_GFP_mettl3_KO_GFP$GeneID <- make.unique(as.character(SE_S2_GFP_mettl3_KO_GFP$GeneID))
  
  # Set GeneID as rownames
  rownames(SE_S2_GFP_mettl3_KO_GFP) <- SE_S2_GFP_mettl3_KO_GFP$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #SE_S2_GFP_mettl3_KO_GFP$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  SE_S2_GFP_mettl3_KO_GFP$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_mettl3_KO_GFP),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   SE_S2_GFP_mettl3_KO_GFP$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_mettl3_KO_GFP),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
SE_S2_GFP_mettl3_KO_GFP$GODESCR <- sapply(SE_S2_GFP_mettl3_KO_GFP$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  SE_S2_GFP_mettl3_KO_GFP$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(SE_S2_GFP_mettl3_KO_GFP),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(SE_S2_GFP_mettl3_KO_GFP, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```





```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2_GFP_mettl3_KO_GFP <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A3SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2_GFP_mettl3_KO_GFP <- A3_S2_GFP_mettl3_KO_GFP %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(A3_S2_GFP_mettl3_KO_GFP)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A3_S2_GFP_mettl3_KO_GFP$logPValue <- -1 * log10(A3_S2_GFP_mettl3_KO_GFP$FDR)
# Create a binary inclusion/exclusion column
A3_S2_GFP_mettl3_KO_GFP$Splicing <- ifelse(A3_S2_GFP_mettl3_KO_GFP$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A3_S2_GFP_mettl3_KO_GFP, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A3_S2_GFP_mettl3_KO_GFP$Color <- ifelse(A3_S2_GFP_mettl3_KO_GFP$FDR < 0.05 & abs(A3_S2_GFP_mettl3_KO_GFP$IncLevelDifference) > 0.2, A3_S2_GFP_mettl3_KO_GFP$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A3_S2_GFP_mettl3_KO_GFP$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(A3_S2_GFP_mettl3_KO_GFP, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-GFP/S2-GFP (A3'SS)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A3_S2_GFP_mettl3_KO_GFP)) {
# Make GeneID unique
  A3_S2_GFP_mettl3_KO_GFP$GeneID <- make.unique(as.character(A3_S2_GFP_mettl3_KO_GFP$GeneID))
  
  # Set GeneID as rownames
  rownames(A3_S2_GFP_mettl3_KO_GFP) <- A3_S2_GFP_mettl3_KO_GFP$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A3_S2_GFP_mettl3_KO_GFP$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A3_S2_GFP_mettl3_KO_GFP$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_S2_GFP_mettl3_KO_GFP),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A3_S2_GFP_mettl3_KO_GFP$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_S2_GFP_mettl3_KO_GFP),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A3_S2_GFP_mettl3_KO_GFP$GODESCR <- sapply(A3_S2_GFP_mettl3_KO_GFP$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A3_S2_GFP_mettl3_KO_GFP$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A3_S2_GFP_mettl3_KO_GFP),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(A3_S2_GFP_mettl3_KO_GFP, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```






```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2_GFP_mettl3_KO_GFP <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A5SS.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2_GFP_mettl3_KO_GFP <- A5_S2_GFP_mettl3_KO_GFP %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(A5_S2_GFP_mettl3_KO_GFP)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
A5_S2_GFP_mettl3_KO_GFP$logPValue <- -1 * log10(A5_S2_GFP_mettl3_KO_GFP$FDR)
# Create a binary inclusion/exclusion column
A5_S2_GFP_mettl3_KO_GFP$Splicing <- ifelse(A5_S2_GFP_mettl3_KO_GFP$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(A5_S2_GFP_mettl3_KO_GFP, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
A5_S2_GFP_mettl3_KO_GFP$Color <- ifelse(A5_S2_GFP_mettl3_KO_GFP$FDR < 0.05 & abs(A5_S2_GFP_mettl3_KO_GFP$IncLevelDifference) > 0.2, A5_S2_GFP_mettl3_KO_GFP$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(A5_S2_GFP_mettl3_KO_GFP$Color)


ggplot(A5_S2_GFP_mettl3_KO_GFP, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-GFP/S2-GFP (A5'SS)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(A5_S2_GFP_mettl3_KO_GFP)) {
# Make GeneID unique
  A5_S2_GFP_mettl3_KO_GFP$GeneID <- make.unique(as.character(A5_S2_GFP_mettl3_KO_GFP$GeneID))
  
  # Set GeneID as rownames
  rownames(A5_S2_GFP_mettl3_KO_GFP) <- A5_S2_GFP_mettl3_KO_GFP$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #A3_S2_GFP_mettl3_KO_GFP$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  A5_S2_GFP_mettl3_KO_GFP$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_S2_GFP_mettl3_KO_GFP),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   A5_S2_GFP_mettl3_KO_GFP$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_S2_GFP_mettl3_KO_GFP),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
A5_S2_GFP_mettl3_KO_GFP$GODESCR <- sapply(A5_S2_GFP_mettl3_KO_GFP$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  A5_S2_GFP_mettl3_KO_GFP$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(A5_S2_GFP_mettl3_KO_GFP),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(A5_S2_GFP_mettl3_KO_GFP, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```





```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_mettl3_KO_GFP <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/RI.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_mettl3_KO_GFP <- RI_S2_GFP_mettl3_KO_GFP %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(RI_S2_GFP_mettl3_KO_GFP)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
RI_S2_GFP_mettl3_KO_GFP$logPValue <- -1 * log10(RI_S2_GFP_mettl3_KO_GFP$FDR)
# Create a binary inclusion/exclusion column
RI_S2_GFP_mettl3_KO_GFP$Splicing <- ifelse(RI_S2_GFP_mettl3_KO_GFP$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(RI_S2_GFP_mettl3_KO_GFP, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
RI_S2_GFP_mettl3_KO_GFP$Color <- ifelse(RI_S2_GFP_mettl3_KO_GFP$FDR < 0.05 & abs(RI_S2_GFP_mettl3_KO_GFP$IncLevelDifference) > 0.2, RI_S2_GFP_mettl3_KO_GFP$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(RI_S2_GFP_mettl3_KO_GFP$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(RI_S2_GFP_mettl3_KO_GFP, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-GFP/S2-GFP (RI)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(RI_S2_GFP_mettl3_KO_GFP)) {
# Make GeneID unique
  RI_S2_GFP_mettl3_KO_GFP$GeneID <- make.unique(as.character(RI_S2_GFP_mettl3_KO_GFP$GeneID))
  
  # Set GeneID as rownames
  rownames(RI_S2_GFP_mettl3_KO_GFP) <- RI_S2_GFP_mettl3_KO_GFP$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #RI_S2_GFP_mettl3_KO_GFP$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  RI_S2_GFP_mettl3_KO_GFP$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_mettl3_KO_GFP),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   RI_S2_GFP_mettl3_KO_GFP$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_mettl3_KO_GFP),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
RI_S2_GFP_mettl3_KO_GFP$GODESCR <- sapply(RI_S2_GFP_mettl3_KO_GFP$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  RI_S2_GFP_mettl3_KO_GFP$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(RI_S2_GFP_mettl3_KO_GFP),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(RI_S2_GFP_mettl3_KO_GFP, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```






```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_mettl3_KO_GFP <- read.table("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/MXE.MATS.JCEC.txt", header = T) #%>%
  #Get rid of columns we aren't really going to use.
  #dplyr::select(., c('ID', 'geneSymbol', 'IJC_SAMPLE_1', 'SJC_SAMPLE_1', 'IJC_SAMPLE_2', 'SJC_SAMPLE_2', "PValue",'FDR', 'IncLevel1', 'IncLevel2', 'IncLevelDifference'))
#MXE_Gof_filt <- MXE_Gof[MXE_Gof$FDR < 0.05, ]
#nrow(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_mettl3_KO_GFP <- MXE_S2_GFP_mettl3_KO_GFP %>%
  tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
  # Calculate combined read counts for each replicate
  mutate(
    S1R1counts = IJC_S1R1 + SJC_S1R1,
    S1R2counts = IJC_S1R2 + SJC_S1R2,
    S1R3counts = IJC_S1R3 + SJC_S1R3,
    S2R1counts = IJC_S2R1 + SJC_S2R1,
    S2R2counts = IJC_S2R2 + SJC_S2R2,
    S2R3counts = IJC_S2R3 + SJC_S2R3
  ) %>%
  # Filter based on read counts
  dplyr::filter(
    S1R1counts > 5,
    S1R2counts > 5,
    S1R3counts > 5,
    S2R1counts > 5,
    S2R2counts > 5,
    S2R3counts > 5)
nrow(MXE_S2_GFP_mettl3_KO_GFP)
```

## Volcano plot shows significantly spliced *MXE* in METTL3-KO Ythdc1-ko relative to METTL3-KO GFP

```{r message=FALSE, warning=FALSE, echo=FALSE}
MXE_S2_GFP_mettl3_KO_GFP$logPValue <- -1 * log10(MXE_S2_GFP_mettl3_KO_GFP$FDR)
# Create a binary inclusion/exclusion column
MXE_S2_GFP_mettl3_KO_GFP$Splicing <- ifelse(MXE_S2_GFP_mettl3_KO_GFP$IncLevelDifference < 0,  "Reduced", "Increased")

# Filter for significant genes based on FDR and IncLevelDifference
significant_genes <- subset(MXE_S2_GFP_mettl3_KO_GFP, FDR < 0.05 & abs(IncLevelDifference) > 0.2)

# Update colors for significant and non-significant genes in the main dataframe
MXE_S2_GFP_mettl3_KO_GFP$Color <- ifelse(MXE_S2_GFP_mettl3_KO_GFP$FDR < 0.05 & abs(MXE_S2_GFP_mettl3_KO_GFP$IncLevelDifference) > 0.2, MXE_S2_GFP_mettl3_KO_GFP$Splicing, "Not significant")

# Sort the significant genes by FDR and select the top ones
# Adjust the number as needed to select the number of top genes you want to label
top_genes <- significant_genes[order(significant_genes$FDR),][1:10000,]
top_genes$Color <- ifelse(top_genes$FDR < 0.05, top_genes$Splicing, "Not significant")

# Calculate counts for each category
counts <- table(MXE_S2_GFP_mettl3_KO_GFP$Color)

# Update the scale_color_manual function with new labels including counts #volc_RI_Mettl3-KO_GFP
ggplot(MXE_S2_GFP_mettl3_KO_GFP, aes(x = IncLevelDifference, y = logPValue, color = Color)) +
geom_point(alpha = 0.8) + xlim(c(-1, 1)) +
  scale_color_manual(values = c("Not significant" = "grey", "Increased" = "#ff7f0e",  "Reduced" = "blue"),
                     labels = c(
                                paste("Increased -", counts["Increased"]),
                                paste("Not significant -", counts["Not significant"]),
                                paste("Reduced -", counts["Reduced"]))) +
  theme_minimal() +
  geom_hline(yintercept = -log10(0.05), linetype="dashed", color="black") +
  geom_vline(xintercept = c(-0.2, 0.2), linetype="dashed", color="black") +
  geom_text_repel(data = top_genes, aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10, color = "black") +
  labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-GFP/S2-GFP (MXE)") +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(alpha = 1))) 


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
if ("GeneID" %in% colnames(MXE_S2_GFP_mettl3_KO_GFP)) {
# Make GeneID unique
  MXE_S2_GFP_mettl3_KO_GFP$GeneID <- make.unique(as.character(MXE_S2_GFP_mettl3_KO_GFP$GeneID))
  
  # Set GeneID as rownames
  rownames(MXE_S2_GFP_mettl3_KO_GFP) <- MXE_S2_GFP_mettl3_KO_GFP$GeneID
  
  # Optionally, remove the GeneID column if it's no longer needed
  #MXE_S2_GFP_mettl3_KO_GFP$GeneID <- NULL
  
  # Now, you can use the mapIds function
    
  MXE_S2_GFP_mettl3_KO_GFP$name <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_mettl3_KO_GFP),
                               column = "GENENAME",
                               keytype = "ENSEMBL",
                               multiVals = "first")
  
   MXE_S2_GFP_mettl3_KO_GFP$GO <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_mettl3_KO_GFP),
                               column = "GO",
                               keytype = "ENSEMBL",
                               multiVals = "first")
   
# Retrieve GO term descriptions
MXE_S2_GFP_mettl3_KO_GFP$GODESCR <- sapply(MXE_S2_GFP_mettl3_KO_GFP$GO, function(go_id) {
  if (!is.na(go_id)) {
    Term(go_id)
  } else {
    NA
  }
})

  
  MXE_S2_GFP_mettl3_KO_GFP$ONTOLOGY <- mapIds(org.Dm.eg.db,
                               keys = rownames(MXE_S2_GFP_mettl3_KO_GFP),
                               column = "ONTOLOGY",
                               keytype = "ENSEMBL",
                               multiVals = "first")
} else {
  cat("The column 'GeneID' does not exist in the data frame.\n") 
}


#write.table(RI_S2_GFP_mettl3_KO_GFP, file = '~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/mettl3_KO_GFP_mettl3_KO_DC1_GOF_MXE.txt', quote = F, sep = "\t", row.names = F)
```


```{r}
# Function to prepare data with read filtering steps
prepare_data <- function(file_path, event_type) {
  # Read the data
  data <- read.table(file_path, header = TRUE, sep = "\t")
  
  # Split the replicate read counts that are separated by commas into different columns
  data <- data %>%
    tidyr::separate(., col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(., col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE)

  # Filter out transcripts based on read counts
  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  # Add necessary columns
  data <- data %>%
    mutate(delta_PSI = IncLevelDifference,
           Significant = ifelse(FDR < 0.05 & abs(delta_PSI) > 0.2, "Significant", "Not Significant"),
           EventType = event_type,
           Event_Types = ifelse(Significant == "Significant", event_type, "Not Significant"))
  
  # Select necessary columns
  data <- data[, c("geneSymbol", "delta_PSI", "FDR", "Significant", "EventType", "Event_Types")]
  
  return(data)
}


file_paths <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/RI.MATS.JCEC.txt")

event_types <- names(file_paths)

# Extract significant genes for each event type
significant_genes_list <- mapply(prepare_data, file_paths, event_types, SIMPLIFY = FALSE)

# Combine all splicing events into a single dataframe
all_genes <- do.call(rbind, significant_genes_list)

# Calculate counts for each event type
event_counts <- as.data.frame(table(all_genes$Event_Types))
colnames(event_counts) <- c("Event_Types", "Count")

# Define colors for splicing event types
type_colors <- c("SE" = "blue", "A3SS" = "black", "A5SS" = "darkgreen", "MXE" = "red", "RI" = "#ff7f0e", "Not Significant" = "grey")

# Identify top genes (top 100 based on FDR and delta_PSI) volc_RI_Mettl3-KO_GFP
top_genes <- all_genes[order(all_genes$FDR, -abs(all_genes$delta_PSI)),][1:1000,]

# Prepare custom labels with counts for the legend 
custom_labels <- with(event_counts, setNames(paste(Event_Types, " (", Count, ")", sep=""), Event_Types))

# Create the plot
ggplot(all_genes, aes(x = delta_PSI, y = -log10(FDR), color = Event_Types)) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = type_colors, labels = custom_labels) +
    geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
    geom_vline(xintercept = c(-0.2, 0.2), linetype = "dashed", color = "black") +
    theme_minimal() + xlim(-1.0, 1.0) +
    labs(x = "", y = "Log10 Adj. P-value", title = "Mettl3-KO-GFP/S2-GFP") +
    theme(legend.position = "right", plot.title = element_text(hjust = 0.5)) +
    # Use geom_text_repel to remove boxes and filter for significant genes only
    geom_text_repel(data = top_genes %>% dplyr::filter(Significant == "Significant"), 
                    aes(label = geneSymbol), size = 3, 
                    box.padding = 0.35, point.padding = 0.5, max.overlaps = 10)
```



   Differential expressed gene 
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Function to process each file
process_file <- function(file_path, event_type) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data) %>%
    dplyr::mutate(
      EventType = event_type,
      Regulation = ifelse(IncLevelDifference > 0, "Inclusion", "Exclusion")
    ) %>%
    dplyr::select(geneSymbol, GeneID, EventType, IncLevelDifference, FDR, IncLevel1, IncLevel2, Regulation)
  return(significant_genes)
}

files <- c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/SE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A3SS.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/RI.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/MXE.MATS.JCEC.txt",
                "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/A5SS.MATS.JCEC.txt")

event_types <- c("SE", "A3SS", "RI", "MXE", "A5SS")

# Extract significant genes for each event type
significant_genes_list <- mapply(process_file, files, event_types, SIMPLIFY = FALSE)

# Combine all significant splicing events into a single dataframe
all_significant_genes <- do.call(rbind, significant_genes_list)

# Find intersection of DEGs and splicing event genes
diff_splicing_genes <- intersect(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF$current_symbol, all_significant_genes$geneSymbol)

# Subset splicing data to keep only differentially expressed genes with splicing events
splicing_diff <- all_significant_genes %>% dplyr::filter(geneSymbol %in% diff_splicing_genes)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Filter significant and non-significant genes
significant_genes <- res_mettl3_KO_GFP_S2_GFP %>% 
  dplyr::filter(diffexpressed != "Not Significant")

non_significant_genes <- res_mettl3_KO_GFP_S2_GFP %>% 
  dplyr::filter(diffexpressed == "Not Significant")

# Function to count the number of genes involved in each splicing event type
count_genes_by_event <- function(data, gene_list) {
  data %>%
    dplyr::filter(geneSymbol %in% gene_list$current_symbol) %>%
    group_by(EventType) %>%
    summarise(
      Count = n(),
      GeneNames = paste(geneSymbol, collapse = ", "))
}

# Count for significant DEGs
sig_event_counts <- count_genes_by_event(all_significant_genes, significant_genes)
sig_event_counts <- sig_event_counts %>% mutate(Significance = "Differentially Expressed")

# Count for non-significant DEGs
non_sig_event_counts <- count_genes_by_event(all_significant_genes, non_significant_genes)
non_sig_event_counts <- non_sig_event_counts %>% mutate(Significance = "Not Differentially Expressed")

# Combine the counts into one dataframe
combined_event_counts <- bind_rows(sig_event_counts, non_sig_event_counts)

# Reorder the Significance factor to ensure "Significant" is on top
combined_event_counts$Significance <- factor(combined_event_counts$Significance, 
                                             levels = c("Not Differentially Expressed", "Differentially Expressed"))

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# Extract the p-value and format it
p_value <- chi_square_result$p.value
p_value_formatted <- sprintf("Chi-Square Test p-value: %.2e", p_value)  # Scientific notation


plot_ly(combined_event_counts, 
                x = ~EventType, 
                y = ~Count, 
                color = ~Significance, 
                type = 'bar', 
                text = ~Count,  # Display the count number on each bar
                textposition = 'auto',  # Position the text within the bars
                textfont = list(color = "black"),  # Set the color of the text to black
                hoverinfo = 'text') %>%
  layout(title = "Mettl3-KO-GFP/S2-GFP",
    xaxis = list(title = "Splicing Event Type"),
    yaxis = list(title = "Number of Genes"),
    barmode = 'stack',
    legend = list(
      title = list(text = p_value_formatted),  # Add the p-value to the legend title
      traceorder = "reversed"))

custom_colors <- c("Not Differentially Expressed" = "#a6bddb", "Differentially Expressed" = "#de2d26")
# Create the ggplot with p-value in legend
ggplot(combined_event_counts, aes(x = EventType, y = Count, fill = Significance)) +
  geom_bar(stat = "identity", position = "stack", width = 0.7) +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 5, fontface = "bold", color = "black") +
  scale_fill_manual(values = custom_colors) +
  labs(
    title = "Number of DEGs Mettl3-KO-GFP/S2-GFP in splicing",
    x = "Splicing Event Type",
    y = "Number of Genes",
    fill = sprintf("Gene Expression Status\n(p = %.2e)", p_value)) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank())

```
  
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Create a contingency table of counts
contingency_table <- combined_event_counts %>%
  dplyr::select(EventType, Significance, Count) %>%
  spread(Significance, Count, fill = 0)

# View the contingency table
print(contingency_table)

# Perform the chi-square test
chi_square_result <- chisq.test(contingency_table[, -1])  # Exclude the EventType column

# View the results
print(chi_square_result)


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Load necessary libraries
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggdist)
library(broom)
library(multcomp)
library(ggsignif)

# Function to extract significant genes and filter based on read counts
extract_significant_genes <- function(data) {
  data <- data %>%
    tidyr::separate(col = IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    tidyr::separate(col = SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', remove = TRUE, convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    dplyr::filter(
      S1R1counts > 5,
      S1R2counts > 5,
      S1R3counts > 5,
      S2R1counts > 5,
      S2R2counts > 5,
      S2R3counts > 5
    ) %>%
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    dplyr::select(geneSymbol, GeneID, IncLevelDifference, FDR, IncLevel1, IncLevel2)
  return(data)
}

# Define file paths and conditions
file_paths <- c(
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt",
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt",
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt",
  "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt"
)

conditions <- c("Mettl3-KO-Ythdc1-GoF", "S2-Ythdc1-GoF", "Ythdc1-KO", "Mettl3-KO", "Mettl3-KO-GFP", "S2-GFP", "S2S")

# Process each file and extract significant genes
process_file <- function(file_path) {
  data <- read.delim(file_path, header = TRUE)
  significant_genes <- extract_significant_genes(data)
  return(significant_genes)
}

# Apply the processing function to all files
significant_genes_list <- lapply(file_paths, process_file)

# Calculate average inclusion levels
calculate_average <- function(comma_separated_string) {
  values <- as.numeric(unlist(strsplit(comma_separated_string, ",")))
  return(mean(values, na.rm = TRUE))
}

process_significant_genes <- function(data) {
  data %>%
    rowwise() %>%
    mutate(Avg_IncLevel2 = calculate_average(IncLevel1)) %>%
    ungroup()
}


significant_genes_list <- lapply(significant_genes_list, process_significant_genes)

# Prepare data for plotting and statistical testing
plot_data_list <- lapply(seq_along(significant_genes_list), function(i) {
  data <- significant_genes_list[[i]]
  condition <- conditions[i]
  data.frame(Condition = condition, Avg_IncLevel2 = data$Avg_IncLevel2)
})

plot_data <- do.call(rbind, plot_data_list)

# Perform ANOVA
anova_result <- aov(Avg_IncLevel2 ~ Condition, data = plot_data)
anova_summary <- summary(anova_result)

# Perform Tukey's HSD test
tukey_result <- TukeyHSD(anova_result)
tukey_summary <- broom::tidy(tukey_result)

# Prepare significance labels for the plot
signif_labels <- tukey_summary %>%
  mutate(p.adj.signif = case_when(
    adj.p.value < 0.001 ~ "***",
    adj.p.value < 0.01 ~ "**",
    adj.p.value < 0.05 ~ "*",
    TRUE ~ "ns"
  ))

# Convert conditions to factors to ensure correct ordering in the plot
plot_data$Condition <- factor(plot_data$Condition, levels = conditions)

# Plot the results
p <- ggplot(plot_data, aes(x = Condition, y = Avg_IncLevel2, fill = Condition, color = Condition)) +
  ggdist::stat_halfeye(adjust = 0.5, width = 0.6, .width = 0, justification = -0.2, point_colour = NA) +
  geom_boxplot(width = 0.15, outlier.shape = NA, position = position_dodge(0.7)) +
  geom_jitter(alpha = 0.6, size = 1.5, position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.7)) +
  labs(title = "Proportion of SE Inclusion Levels(IncLevel 1)", x = "Condition", y = "SE IncLevelDifference ()") +
  theme_minimal(base_size = 16) +
  scale_fill_manual(values = c("Mettl3-KO-DC1-GoF" = "darkblue", 
                               "S2-GFP-S2-DC1-GoF" = "pink", 
                               "DC1-KO" = "darkgreen", 
                               "Mettl3-KO" = "orange")) +
  scale_color_manual(values = c("Mettl3-KO-DC1-GoF" = "darkblue", 
                                "S2-GFP-S2-DC1-GoF" = "pink", 
                                "DC1-KO" = "darkgreen", 
                                "Mettl3-KO" = "orange"))

# Add significance annotations using ggsignif
comparisons <- list(c("Mettl3-KO-DC1-GoF", "S2-GFP-S2-DC1-GoF"), 
                    c("Mettl3-KO-DC1-GoF", "DC1-KO"), 
                    c("Mettl3-KO-DC1-GoF", "Mettl3-KO"),
                    c("S2-GFP-S2-DC1-GoF", "DC1-KO"), 
                    c("S2-GFP-S2-DC1-GoF", "Mettl3-KO"), 
                    c("DC1-KO", "Mettl3-KO"))

# Adjust the positions for significance bars based on the number of comparisons
y_positions <- max(plot_data$Avg_IncLevel2) + seq(0.1, by = 0.15, length.out = length(comparisons))

p <- p +
  geom_signif(comparisons = comparisons,
              annotations = signif_labels$p.adj.signif,
              y_position = y_positions,
              tip_length = 0.01)

print(p)

```


A dot plot showing the Inclusion level of every transcript in each data set. This is similar to the bar plot showing the Inclusion Level Difference between the data set

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Required packages
library(tidyr)
library(dplyr)
library(pheatmap)
library(ggplot2)

# STEP 1: Map object names to experimental groups
group_map_df <- data.frame(
  ObjectName = c(
    # Mettl3-KO-GFP / S2-GFP
    "SE_S2_GFP_mettl3_KO_GFP", "A3_S2_GFP_mettl3_KO_GFP", "A5_S2_GFP_mettl3_KO_GFP", "RI_S2_GFP_mettl3_KO_GFP", "MXE_S2_GFP_mettl3_KO_GFP",
    # METTL3-KO / S2S
    "A3SS_mko", "SE_mettl3_KO", "A5SS_mko", "MXE_S2_MKO", "RI_S2_MKO",
    # Ythdc1-GoF / S2S
    "SE_S2S_DC1_KO", "A3_S2S_DC1_KO", "A5_S2S_DC1_KO", "RI_S2S_DC1_KO", "MXE_S2S_DC1_KO",
    # S2-Ythdc1-GoF / S2-GFP
    "SE_S2_GFP_S2_DC1_GOF", "A3_SS2_GFP_S2_DC1_GOF", "A5SS_S2_GFP_S2_DC1_GOF", "MXE_S2_GFP_S2_DC1_GOF", "RI_S2_GFP_S2_DC1_GOF",
    # Mettl3-KO-Ythdc1-GoF / Mettl3-KO-GFP
    "se_mettl3_KO_GFP_mettl3_KO_DC1_GOF", "A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF", "A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF", 
    "RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF", "MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF",
    # Mettl3-KO-Ythdc1-GoF / S2-GFP
    "SE_S2_GFP_mettl3_KO_DC1_GOF", "A3_S2_GFP_mettl3_KO_DC1_GOF", "A5_S2_GFP_mettl3_KO_DC1_GOF", 
    "MXE_S2_GFP_mettl3_KO_DC1_GOF", "RI_S2_GFP_mettl3_KO_DC1_GOF",
    # S2-Ythdc1-GoF / Mettl3-KO-Ythdc1-GoF
    "SE_mettl3_KO_DC1_GOF_S2_DC1_GOF", "A3_mettl3_KO_DC1_GOF_S2_DC1_GOF", "A5_mettl3_KO_DC1_GOF_S2_DC1_GOF", 
    "RI_mettl3_KO_DC1_GOF_S2_DC1_GOF", "MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF"),
  Group = rep(c("Mettl3-KO-GFP/S2-GFP", "METTL3-KO/S2S", "Ythdc1-GoF/S2S",
                "S2-Ythdc1-GoF/S2-GFP", "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP", 
                "Mettl3-KO-Ythdc1-GoF/S2-GFP", "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF"), each = 5),
  SplicingType = rep(c("SE", "A3SS", "A5SS", "RI", "MXE"), times = 7),
  stringsAsFactors = FALSE)

get_event_counts <- function(ObjectName, Group, SplicingType) {
  data <- get(ObjectName)
  
  # Separate replicate columns and filter
  data <- data %>%
    separate(IJC_SAMPLE_1, into = c("IJC_S1R1", "IJC_S1R2", "IJC_S1R3"), sep = ",", convert = TRUE) %>%
    separate(SJC_SAMPLE_1, into = c("SJC_S1R1", "SJC_S1R2", "SJC_S1R3"), sep = ",", convert = TRUE) %>%
    separate(IJC_SAMPLE_2, into = c("IJC_S2R1", "IJC_S2R2", "IJC_S2R3"), sep = ",", convert = TRUE) %>%
    separate(SJC_SAMPLE_2, into = c("SJC_S2R1", "SJC_S2R2", "SJC_S2R3"), sep = ",", convert = TRUE) %>%
    mutate(
      S1R1counts = IJC_S1R1 + SJC_S1R1,
      S1R2counts = IJC_S1R2 + SJC_S1R2,
      S1R3counts = IJC_S1R3 + SJC_S1R3,
      S2R1counts = IJC_S2R1 + SJC_S2R1,
      S2R2counts = IJC_S2R2 + SJC_S2R2,
      S2R3counts = IJC_S2R3 + SJC_S2R3
    ) %>%
    filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)
  
  sig <- filter(data, FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  
  tibble(
    Sample = ObjectName,
    Group = Group,
    SplicingType = SplicingType,
    Inclusion = sum(sig$IncLevelDifference > 0),
    Exclusion = sum(sig$IncLevelDifference < 0)
  )
}

# STEP 3: Loop over all entries and extract stats
all_counts <- purrr::pmap_dfr(group_map_df, get_event_counts)

# STEP 4: Clustering and plotting per splicing type
splicing_types <- unique(all_counts$SplicingType)

for (stype in splicing_types) {
  message("Processing: ", stype)
  
  sub_data <- filter(all_counts, SplicingType == stype)
  
  mat <- sub_data %>%
    select(Sample, Inclusion, Exclusion) %>%
    column_to_rownames("Sample")
  
  # Z-score normalize
  mat_z <- scale(mat)
  
  # Heatmap
  pheatmap(
    mat_z,
    clustering_distance_rows = "euclidean",
    clustering_method = "complete",
    main = paste("Clustering Heatmap -", stype),
    fontsize_row = 8
  )
  
  # PCA
  pca <- prcomp(mat_z, scale. = FALSE)
  pca_df <- as.data.frame(pca$x)
  pca_df$Group <- sub_data$Group
  pca_df$Sample <- rownames(mat)
  
  # PCA Plot
  print(
    ggplot(pca_df, aes(x = PC1, y = PC2, color = Group, label = Sample)) +
      geom_point(size = 4) +
      geom_text(vjust = 1.2, size = 3) +
      labs(title = paste("PCA of Inclusion/Exclusion -", stype)) +
      theme_minimal()
  )
}

```


# merge all splicing data to make barplot

```{r}
library(dplyr)
library(tidyr)
library(ggplot2)

# --- Prepare and filter data ---
prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

  data <- data %>%
    tidyr::separate(IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', convert = TRUE)

  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)

  return(data)
}

# --- Count inclusion/exclusion per file ---
count_per_direction <- function(file_path, dataset_name, event_type) {
  data <- prepare_and_filter_data(file_path)
  significant <- data %>% dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2)

  incl_count <- nrow(significant %>% dplyr::filter(IncLevelDifference > 0))
  excl_count <- nrow(significant %>% dplyr::filter(IncLevelDifference < 0))

  return(data.frame(
    Dataset = dataset_name,
    Direction = c("Included", "Excluded"),
    EventType = event_type,
    Count = c(incl_count, excl_count)
  ))
}

# --- Define datasets properly ---
dataset_info <- list(
  "Mettl3-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/",
  "Ythdc1-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/",
  "S2-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/",
  "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/",
  "Mettl3-KO-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/",
  "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/",
  "Mettl3-KO-GFP/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/")

event_types <- c("SE", "A5SS", "A3SS", "MXE", "RI")

# --- Process all datasets ---
all_results <- do.call(rbind, lapply(names(dataset_info), function(dataset_name) {
  dir_path <- dataset_info[[dataset_name]]
  do.call(rbind, lapply(event_types, function(event) {
    file_path <- file.path(dir_path, paste0(event, ".MATS.JCEC.txt"))
    count_per_direction(file_path, dataset_name, event)
  }))
}))

# --- Normalize proportions ---
all_results <- all_results %>%
  group_by(Dataset, Direction) %>%
  mutate(Proportion = Count / sum(Count)) %>%
  ungroup()

all_results_level <- c(
  "Mettl3-KO/S2S",
  "Ythdc1-KO/S2S",
  "Mettl3-KO-Ythdc1-GoF/S2-GFP",
  "Mettl3-KO-GFP/S2-GFP",
  "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP",
  "S2-Ythdc1-GoF/S2-GFP",
  "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF")

all_results$Dataset <- factor(all_results$Dataset, levels = all_results_level)
# --- Plot ---
ggplot(all_results, aes(x = Direction, y = Proportion, fill = EventType)) +
  geom_bar(stat = "identity", position = "stack", color = "black") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), size = 3, color = "black") +
  facet_wrap(~Dataset, ncol = 3) +
  labs(title = "Splicing Events (Included vs Excluded)",
       y = "Proportion of Splicing Events", x = "") +
  scale_fill_manual(values = c("SE" = "#d73027", "RI" = "#fee08b", "MXE" = "#8073ac", "A5SS" = "#74add1", "A3SS" = "#1a9850")) +
  theme_minimal(base_size = 13) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5),
        plot.title = element_text(hjust = 0.5, face = "bold"),
        legend.position = "right")


```

```{r}
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggridges)
library(stringr)

# Modified function to add Event Type
collect_all_inc_level_diff_with_event_type <- function(file_path, dataset_name) {
  event_type <- str_extract(basename(file_path), "^(SE|RI|MXE|A3SS|A5SS)")  # Extract from filename
  data <- prepare_and_filter_data(file_path)
  
  filtered_data <- data %>% 
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2) %>%
    mutate(Dataset = dataset_name,
      EventType = event_type,
      Gene = str_remove(GeneID, "\\..*")  # optional cleaning
    )
  
  return(filtered_data)
}

# Load all datasets
all_splicing_data <- purrr::imap_dfr(dataset_info, function(path, dataset_name) {
  files <- list.files(path, pattern = "*.MATS.JCEC.txt$", full.names = TRUE)
  purrr::map_dfr(files, function(f) {
    collect_all_inc_level_diff_with_event_type(f, dataset_name)
  })
})

#all_splicing_data <- all_splicing_data %>%
#  dplyr::filter(EventType == "SE")

# Annotate with m6A status
all_splicing_data <- all_splicing_data %>%
  mutate(m6A_Status = ifelse(geneSymbol %in% peak_data$Gene, "m6A", "Non-m6A"))


# Loop through each dataset and plot separately
unique_datasets <- unique(all_splicing_data$Dataset)

for (ds in unique_datasets) {
  df_subset <- all_splicing_data %>% dplyr::filter(Dataset == ds)

  p <- ggplot(all_splicing_data, aes(x = IncLevelDifference, y = Dataset, fill = m6A_Status)) +
    ggridges::geom_density_ridges(alpha = 0.7, scale = 1.2, rel_min_height = 0.01) +
    scale_fill_manual(values = c("m6A" = "firebrick", "Non-m6A" = "steelblue")) +
    theme_minimal(base_size = 14) +
    labs(
      title = paste("Splicing Changes in", ds),
      x = "PSI (Inclusion Level Difference)",
      y = "Splicing Event Type",
      fill = "m6A Status"
    ) +
    theme(legend.position = "right")

  # Print each plot to screen
  print(p)
}




for (ds in unique_datasets) {
  df_subset <- all_splicing_data %>% dplyr::filter(Dataset == ds)

  v <- ggplot(all_splicing_data, aes(x = EventType, y = IncLevelDifference, fill = m6A_Status)) +
    geom_violin(trim = FALSE) +
    geom_boxplot(width = 0.1, outlier.shape = NA, position = position_dodge(width = 0.9)) +
    scale_fill_manual(values = c("m6A" = "#E41A1C", "Non-m6A" = "#377EB8")) +
    theme_minimal(base_size = 14) +
    labs(
      title = paste("PSI by Event Type and m6A Status:", ds),
      x = "Event Type",
      y = "PSI",
      fill = "m6A Status"
    )

  print(v)
}



```


```{r}
# Filter only SE (Skipped Exon) events
se_data <- all_splicing_data %>% 
  dplyr::filter(EventType == "A3SS")

library(ggridges)


dataset_levels <- c(
  "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF",
  "S2-Ythdc1-GoF/S2-GFP",
  "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP",
  "Mettl3-KO-Ythdc1-GoF/S2-GFP",
  "Mettl3-KO-GFP/S2-GFP",
  "Ythdc1-KO/S2S",
  "Mettl3-KO/S2S")

se_data$Dataset <- factor(se_data$Dataset, levels = dataset_levels)


# Compute Wilcoxon p-values between m6A and Non-m6A for each dataset
stat_results <- se_data %>%
  group_by(Dataset) %>%
  summarise(
    p_value = wilcox.test(IncLevelDifference[m6A_Status == "m6A"],
                          IncLevelDifference[m6A_Status == "Non-m6A"])$p.value,
    .groups = "drop") %>%
  mutate(label = case_when(
    p_value < 0.001 ~ "***",
    p_value < 0.01 ~ "**",
    p_value < 0.05 ~ "*",
    TRUE ~ "ns"))


# Count m6A and Non-m6A per dataset
count_data <- se_data %>%
  group_by(Dataset, m6A_Status) %>%
  summarise(Count = n(), .groups = "drop")

# Create bar plot
# Max height per dataset for positioning
height_data <- count_data %>%
  group_by(Dataset) %>%
  summarise(y_pos = sum(Count) + 5, .groups = "drop") %>%
  left_join(stat_results, by = "Dataset")

# Ensure factor levels in all plotting data frames
count_data$Dataset <- factor(count_data$Dataset, levels = dataset_levels)
height_data$Dataset <- factor(height_data$Dataset, levels = dataset_levels)

ggplot(se_data, aes(x = IncLevelDifference, y = Dataset, fill = m6A_Status)) +
  geom_density_ridges(alpha = 0.7, scale = 1.2, rel_min_height = 0.01) +
  scale_fill_manual(values = c("m6A" = "firebrick", "Non-m6A" = "steelblue")) +
  theme_minimal(base_size = 14) +
  labs(
    title = "PSI Distribution (RI Events) Across Datasets",
    x = "PSI (Inclusion Level Difference)",
    y = "Dataset",
    fill = "m6A Status") +
  theme(legend.position = "right")



ggplot(count_data, aes(x = Dataset, y = Count, fill = m6A_Status)) +
  geom_bar(stat = "identity", position = "stack") +
  geom_text(aes(label = Count), position = position_stack(vjust = 0.5), color = "white", size = 4) +
  geom_text(data = height_data, aes(x = Dataset, y = y_pos, label = label),
            inherit.aes = FALSE, vjust = 0, size = 5) +
  scale_fill_manual(values = c("m6A" = "firebrick", "Non-m6A" = "steelblue")) +
  theme_minimal(base_size = 14) +
  labs(
    title = "Event Counts by Dataset (RI) with Significance",
    x = "Dataset",
    y = "Number of Events",
    fill = "m6A Status") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  coord_flip()





```


```{r}
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggdist)
library(gghalves)


# --- Function to prepare and filter data ---
prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

  data <- data %>%
    tidyr::separate(IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', convert = TRUE)

  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)

  return(data)
}

# --- Function to collect IncLevelDifference for significant events ---
collect_inc_level_diff <- function(file_path, dataset_name) {
  data <- prepare_and_filter_data(file_path)
  significant <- data %>% dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2)

  return(data.frame(
    Dataset = dataset_name,
    IncLevelDifference = significant$IncLevelDifference
  ))
}

# --- Define dataset paths ---
dataset_info <- list(
  "Mettl3-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/",
  "Ythdc1-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/",
  #"S2-YTHDC1-GOF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/",
  #"Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/",
  #"Mettl3-KO-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/",
  #"S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/",
  "Mettl3-KO-GFP/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/")

event_types <- c("SE") #, "A5SS", "A3SS", "MXE", "RI"

# --- Collect IncLevelDifference ---
inc_diff_results <- do.call(rbind, lapply(names(dataset_info), function(dataset_name) {
  dir_path <- dataset_info[[dataset_name]]
  do.call(rbind, lapply(event_types, function(event) {
    file_path <- file.path(dir_path, paste0(event, ".MATS.JCEC.txt"))
    collect_inc_level_diff(file_path, dataset_name)
  }))
}))

# --- Set custom dataset order ---
dataset_levels <- c(
  "Mettl3-KO/S2S",
  "Ythdc1-KO/S2S",
  #"Mettl3-KO-Ythdc1-GoF/S2-GFP",
  "Mettl3-KO-GFP/S2-GFP")#,
  #"Mettl3-KO-YTHDC1-GoF/Mettl3-KO-GFP",
  #"S2-YTHDC1-GOF/S2-GFP",
  #"S2-YTHDC1-GOF/Mettl3-KO-YTHDC1-GoF")

inc_diff_results$Dataset <- factor(inc_diff_results$Dataset, levels = dataset_levels)

# --- Plot Raincloud with Notched Boxplot ---
ggplot(inc_diff_results, aes(x = Dataset, y = IncLevelDifference, fill = Dataset)) +
  ggdist::stat_halfeye(adjust = .5, width = .6, .width = 0, justification = -.3, point_colour = NA) +
  geom_boxplot(width = .15, outlier.shape = NA, notch = TRUE, position = position_nudge(x = 0.15)) +
  gghalves::geom_half_point(side = "l", range_scale = .4, alpha = .5, position = position_nudge(x = -0.15)) +
  coord_flip() + labs(title = "Inclusion Level Differences", x = "Dataset", y = "A5'SS (PSI)") + stat_compare_means(comparisons = combn(unique(inc_diff_results$Dataset), 2, simplify = FALSE), label = "p.signif", method = "wilcox.test", size = 3) +
  theme_minimal(base_size = 14) +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"), axis.text.y = element_text(size = 11), legend.position = "none") +
  scale_fill_brewer(palette = "Set3")


```

```{r}


library(dplyr)
library(tidyr)
library(ggplot2)
library(ggdist)   # for raincloud
library(ggpubr)   # for stat_compare_means()

# --- Prepare and filter data ---
prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

  data <- data %>%
    tidyr::separate(IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', convert = TRUE)

  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
                  S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)

  return(data)
}

# --- Collect ALL IncLevelDifference without filtering ---
#collect_all_inc_level_diff <- function(file_path, dataset_name) {
#  data <- prepare_and_filter_data(file_path)

#  return(data.frame(
#    Dataset = dataset_name,
#    IncLevelDifference = data$IncLevelDifference
#  ))
#}

collect_all_inc_level_diff <- function(file_path, dataset_name) {
  data <- prepare_and_filter_data(file_path)
  
  # Apply filtering for significance: FDR < 0.05 and absolute PSI > 0.2
  filtered_data <- data %>% 
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2)
  
  return(data.frame(
    Dataset = dataset_name,
    IncLevelDifference = filtered_data$IncLevelDifference
  ))
}

# --- Define datasets ---
dataset_info <- list(
  "Mettl3-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/",
  "Ythdc1-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/",
  "S2-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/",
  "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/",
  "Mettl3-KO-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/",
   "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/",
  "Mettl3-KO-GFP/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/"
  )

event_types <- c("A3SS", "SE","A5SS", "A3SS", "MXE", "RI")

# --- Collect all IncLevelDifference values ---
inc_diff_results <- do.call(rbind, lapply(names(dataset_info), function(dataset_name) {
  dir_path <- dataset_info[[dataset_name]]
  do.call(rbind, lapply(event_types, function(event) {
    file_path <- file.path(dir_path, paste0(event, ".MATS.JCEC.txt"))
    collect_all_inc_level_diff(file_path, dataset_name)
  }))
}))

# Split data into Inclusion and Exclusion "#2ca02c" #ff7f0e #1f77b4
inc_diff_inclusion <- inc_diff_results %>% 
  dplyr::filter(IncLevelDifference > 0)

inc_diff_exclusion <- inc_diff_results %>% 
  dplyr::filter(IncLevelDifference < 0)

my_colors <- c(
  "Mettl3-KO/S2S" = "brown",
  "Ythdc1-KO/S2S" = "pink",
  "Mettl3-KO-GFP/S2-GFP" ="green"
  #"Mettl3-KO-Ythdc1-GoF/S2-GFP" = "#2ca02c",
  #"Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP" = " ",
  #"S2-Ythdc1-GoF/S2-GFP" = "red",
  #"S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF" = "#1f77b4"
  )

#Box_inclusion_MXE_MKO
ggplot(inc_diff_inclusion, aes(x = Dataset, y = IncLevelDifference, fill = Dataset, color = Dataset)) +
  #ggdist::stat_halfeye(adjust = 0.5, width = 0.8, .width = 0, justification = -0.3, point_colour = NA) +
  geom_boxplot(width = .3, outlier.shape = NA, notch = TRUE, position = position_nudge(x = 0.15)) +
  geom_jitter(width = 0.15, size = 1, alpha = 0.3) +
  scale_y_continuous(limits = c(-1, 1)) +
  stat_compare_means(comparisons = combn(unique(inc_diff_inclusion$Dataset), 2, simplify = FALSE),label = "p.signif",method = "wilcox.test", size = 3) +
  labs(title = "Distribution of Inclusion Events ( > 0)", x = " ", y = "A3SS (PSI)") +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 360, hjust = 1, size = 11),nplot.title = element_text(hjust = 0.5, face = "bold"), legend.position = "none") +
 scale_fill_manual(values = my_colors) +
  scale_color_manual(values = my_colors) +
  coord_flip()
# Box_exclusion_MXE_MKO
ggplot(inc_diff_exclusion, aes(x = Dataset, y = IncLevelDifference, fill = Dataset, color = Dataset)) +
  #ggdist::stat_halfeye(adjust = 0.5, width = 0.8, .width = 0, justification = -0.3, point_colour = NA) +
  geom_boxplot(width = .3, outlier.shape = NA, notch = TRUE, position = position_nudge(x = 0.15)) +
  geom_jitter(width = 0.15, size = 1, alpha = 0.3) +
  scale_y_continuous(limits = c(-1, 1)) +
  stat_compare_means(comparisons = combn(unique(inc_diff_exclusion$Dataset), 2, simplify = FALSE), label = "p.signif", method = "wilcox.test", size = 3) +
  labs(title = "Distribution of Exclusion Events ( < 0)", x = " ", y = "A3SS (PSI)") +
  theme_minimal(base_size = 14) +
  theme(axis.text.x = element_text(angle = 360, hjust = 1, size = 11), plot.title = element_text(hjust = 0.5, face = "bold"), legend.position = "none") +
 scale_fill_manual(values = my_colors) +
  scale_color_manual(values = my_colors) +
  coord_flip()

```

```{r}
# --- Load libraries ---
library(ggplot2)
library(dplyr)
library(tidyr)
library(ggpubr)
library(gghalves)
library(ggdist)
library(patchwork) # For arranging multiple plots


# --- Prepare reading and filtering ---
prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

  data <- data %>%
    tidyr::separate(IJC_SAMPLE_1, into = c('IJC_S1R1', 'IJC_S1R2', 'IJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_1, into = c('SJC_S1R1', 'SJC_S1R2', 'SJC_S1R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(IJC_SAMPLE_2, into = c('IJC_S2R1', 'IJC_S2R2', 'IJC_S2R3'), sep = ',', convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_2, into = c('SJC_S2R1', 'SJC_S2R2', 'SJC_S2R3'), sep = ',', convert = TRUE)

  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)

  return(data)
}

collect_all_inc_level_diff <- function(file_path, dataset_name) {
  data <- prepare_and_filter_data(file_path)
  return(data.frame(
    Dataset = dataset_name,
    IncLevelDifference = data$IncLevelDifference))
}

dataset_info <- list(
  "Mettl3-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/",
  "Ythdc1-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/",
  "S2-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/",
  "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/",
  "Mettl3-KO-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/",
  "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/",
  "Mettl3-KO-GFP/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/")

event_types <- c("SE", "A5SS", "A3SS", "MXE", "RI")

# --- Collect all IncLevelDifference values ---
inc_diff_results <- do.call(rbind, lapply(names(dataset_info), function(dataset_name) {
  dir_path <- dataset_info[[dataset_name]]
  do.call(rbind, lapply(event_types, function(event) {
    file_path <- file.path(dir_path, paste0(event, ".MATS.JCEC.txt"))
    collect_all_inc_level_diff(file_path, dataset_name)
  }))
}))

# --- Set dataset levels for plotting ---
dataset_levels <- names(dataset_info)
inc_diff_results$Dataset <- factor(inc_diff_results$Dataset, levels = dataset_levels)

# --- Prepare comparisons for stat tests ---
comparisons_list <- combn(unique(inc_diff_results$Dataset), 2, simplify = FALSE)

# --- Manually assign datasets to groups ---
exclusion_like_datasets <- c(
  "Mettl3-KO/S2S",
  "Ythdc1-KO/S2S",
  "Mettl3-KO-GFP/S2-GFP")

inclusion_like_datasets <- c(
  "S2-Ythdc1-GoF/S2-GFP",
  "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP",
  "Mettl3-KO-Ythdc1-GoF/S2-GFP",
  "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF")

# --- Split data ---
exclusion_data <- inc_diff_results %>% 
  dplyr::filter(Dataset %in% exclusion_like_datasets)
inclusion_data <- inc_diff_results %>% 
  dplyr::filter(Dataset %in% inclusion_like_datasets)

# --- Plot function ---
plot_dpsi_box <- function(data, title_text) {
  ggplot(data, aes(x = Dataset, y = IncLevelDifference, fill = Dataset)) +
    geom_boxplot(width = 0.8, outlier.shape = NA, notch = TRUE) +
    gghalves::geom_half_point(aes(color = Dataset), side = "l", range_scale = 0.4, alpha = 0.6, size = 1) +
    stat_compare_means(comparisons = comparisons_list, label = "p.signif", method = "wilcox.test", size = 2.5, tip.length = 0.01) +
    coord_flip() + ylim(-1, 1) + labs(title = title_text, x = NULL,y = expression(Delta*PSI*" (compared to control)")) +
    scale_fill_brewer(palette = "Set3") +
    scale_color_brewer(palette = "Set3") +
    theme_minimal(base_size = 14) +
    theme(plot.title = element_text(hjust = 0.5, face = "bold", size = 14), axis.text.y = element_text(size = 10),
      axis.text.x = element_text(size = 10),
      panel.grid.major.y = element_blank(),
      legend.position = "none")
}

# --- Create plots ---
p_exclusion_final <- plot_dpsi_box(exclusion_data, "Excluded Events (PSI)")
p_inclusion_final <- plot_dpsi_box(inclusion_data, "Included Events (PSI)")


```




```{r}
library(dplyr)
library(tidyr)
library(ggplot2)
library(readr)
library(purrr)
library(stringr)

# --- Function to load and prepare data ---
prepare_and_filter_data <- function(file_path) {
  data <- read.delim(file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

  data <- data %>%
    tidyr::separate(IJC_SAMPLE_1, into = c("IJC_S1R1", "IJC_S1R2", "IJC_S1R3"), sep = ",", convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_1, into = c("SJC_S1R1", "SJC_S1R2", "SJC_S1R3"), sep = ",", convert = TRUE) %>%
    tidyr::separate(IJC_SAMPLE_2, into = c("IJC_S2R1", "IJC_S2R2", "IJC_S2R3"), sep = ",", convert = TRUE) %>%
    tidyr::separate(SJC_SAMPLE_2, into = c("SJC_S2R1", "SJC_S2R2", "SJC_S2R3"), sep = ",", convert = TRUE)

  data <- data %>%
    mutate(S1R1counts = IJC_S1R1 + SJC_S1R1,
           S1R2counts = IJC_S1R2 + SJC_S1R2,
           S1R3counts = IJC_S1R3 + SJC_S1R3,
           S2R1counts = IJC_S2R1 + SJC_S2R1,
           S2R2counts = IJC_S2R2 + SJC_S2R2,
           S2R3counts = IJC_S2R3 + SJC_S2R3) %>%
    dplyr::filter(S1R1counts > 5 & S1R2counts > 5 & S1R3counts > 5 &
           S2R1counts > 5 & S2R2counts > 5 & S2R3counts > 5)

  return(data)
}

# --- Collect inclusion/exclusion IncLevelDifference matrices ---
get_sample_matrix <- function(file_path, sample_id, type = c("inclusion", "exclusion")) {
  type <- match.arg(type)
  data <- prepare_and_filter_data(file_path)
  sig <- data %>% 
    dplyr::filter(FDR < 0.05 & abs(IncLevelDifference) > 0.2)

  if (type == "inclusion") {
    sig <- sig %>% 
      dplyr::filter(IncLevelDifference > 0)
  } else {
    sig <- sig %>% 
      dplyr::filter(IncLevelDifference < 0)
  }

  if (nrow(sig) == 0) return(NULL)

  df <- sig %>%
    dplyr::select(ID, IncLevelDifference) %>%
    mutate(Sample = sample_id) %>%
    pivot_wider(names_from = Sample, values_from = IncLevelDifference)

  return(df)
}

# --- File map ---
dataset_info <- list(
  "Mettl3-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/",
  "Ythdc1-KO/S2S" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/",
  "S2-Ythdc1-GoF/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/",
  "Mettl3-KO-Ythdc1-GoF/Mettl3-KO-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/",
  "Mettl3-KO-Ythdc1/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_DC1_GOF/",
  "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/",
  "Mettl3-KO-GFP/S2-GFP" = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_mettl3_KO_GFP/")

event_types <- c("RI") #, "SE""A5SS", "A3SS", "MXE", "RI

# --- Assemble matrices ---
assemble_matrix <- function(type = c("inclusion", "exclusion")) {
  type <- match.arg(type)
  matrices <- list()

  for (dataset in names(dataset_info)) {
    dir_path <- dataset_info[[dataset]]
    for (event in event_types) {
      file_path <- file.path(dir_path, paste0(event, ".MATS.JCEC.txt"))
      sample_id <- paste(dataset, event, sep = "_")
      mat <- get_sample_matrix(file_path, sample_id, type = type)
      if (!is.null(mat)) {
        matrices[[sample_id]] <- mat
      }
    }
  }

  # Merge all on "ID"
  full_matrix <- reduce(matrices, full_join, by = "ID") %>%
    column_to_rownames("ID") %>%
    as.data.frame()

  # Replace NA with 0 (or use impute with mean/median)
  full_matrix[is.na(full_matrix)] <- 0

  return(full_matrix)
}

# --- Run PCA ---
plot_pca <- function(mat, title) {
  pca <- prcomp(t(mat), scale. = TRUE)
  pca_df <- as.data.frame(pca$x[, 1:2])
  pca_df$Sample <- rownames(pca_df)
  pca_df$Dataset <- sapply(strsplit(rownames(pca_df), "_"), `[`, 1)
  
  # Calculate % variance explained
  var_explained <- pca$sdev^2 / sum(pca$sdev^2)
  
  ggplot(pca_df, aes(x = PC1, y = PC2, color = Dataset, label = Sample)) +
    geom_point(size = 5) +
    geom_text(size = 5, vjust = 1.5) +
    labs(
      title = title,
      x = paste0("PC1 (", round(var_explained[1] * 100, 1), "%)"),
      y = paste0("PC2 (", round(var_explained[2] * 100, 1), "%)")
    ) +
    theme_minimal(base_size = 14) +
    theme(plot.title = element_text(hjust = 0.5))
}

# --- Run everything ---
inclusion_mat <- assemble_matrix("inclusion")
exclusion_mat <- assemble_matrix("exclusion")

# --- Plot ---
plot_pca(inclusion_mat, "PCA of Inclusion Events")
plot_pca(exclusion_mat, "PCA of Exclusion Events")

```


# merge differentially expressed genes res_S2_GFP_mettl3_KO_DC1_GOF_clean
```{r message=FALSE, warning=FALSE, echo=FALSE}

library(dplyr)
library(tidyr)

# Ensure no duplication of 'gene_name'
process_dataframe <- function(df) {
  if ("validated_id" %in% colnames(df)) {
    return(df)
  } else {
    return(df %>% rownames_to_column(var = "validated_id"))
  }
}

# Process each dataframe to ensure 'validated_id' is correctly handled
res_S2_S_DC1_KO.df = process_dataframe(res_S2_S_DC1_KO)

res_mettl3_KO_GFP_S2_GFP_clean.df = process_dataframe(res_mettl3_KO_GFP_S2_GFP)
res_S2_GFP_mettl3_KO_DC1_GOF_clean.df = process_dataframe(res_S2_GFP_mettl3_KO_DC1_GOF)


res_S2_GFP_S2_DC1_GOF.df = process_dataframe(res_S2.GFP_S2.DC1.GOF)

#res_S2S_ythdc_wt.df = process_dataframe(res_S2S_ythdc_wt.df)

res_mettl3_KO_GFP_mettl3_KO_DC1_GOF.df  = process_dataframe(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF)

res_S2_m6A.df = process_dataframe(res_S2_m6A)

res_S2.YTHDC_gof_mettl3.KO.DC1.GOF.df = process_dataframe(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF)

# Select and rename columns for each dataset
res_mettl3_KO_GFP_S2_GFP_selected <- res_mettl3_KO_GFP_S2_GFP_clean.df %>%
  dplyr::select(validated_id, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diffexpressed) %>%
  rename_with(~ paste0(., "_METTL3_KO_GFP_S2_GFP"), -validated_id)


res_S2_GFP_mettl3_KO_DC1_GOF_selected <- res_S2_GFP_mettl3_KO_DC1_GOF_clean.df %>%
  dplyr::select(validated_id, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diffexpressed,GODESCR) %>%
  rename_with(~ paste0(., "_METTL3_KO_Ythdc1_GoF_S2-GFP"), -validated_id)


res_S2_m6A_selected <- res_S2_m6A.df %>%
  dplyr::select(validated_id, current_symbol, GENENAME, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diffexpressed) %>%
  rename_with(~ paste0(., "_METTL3_KO_S2S"), -validated_id, -current_symbol, -GENENAME)

res_S2_S_DC1_KO_selected <- res_S2_S_DC1_KO.df %>%
  dplyr::select(validated_id, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diffexpressed) %>%
  rename_with(~ paste0(., "_Ythdc1_KO_S2S"), -validated_id)

res_S2_GFP_S2_DC1_GOF_selected <- res_S2_GFP_S2_DC1_GOF.df %>%
  dplyr::select(validated_id, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diffexpressed) %>%
  rename_with(~ paste0(., "_S2_Ythdc1_GoF_S2_GFP"), -validated_id)

res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_selected <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF.df %>%
  dplyr::select(validated_id, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diffexpressed) %>%
  rename_with(~ paste0(., "_S2_Ythdc1_GoF_METTL3_KO_Ythdc1_GoF"), -validated_id)

#res_S2S_ythdc_wt_selected <- res_S2S_ythdc_wt.df %>%
 # dplyr::select(validated_id, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, symbol, ENTREZID, GENENAME, GO, GODESCR, diffexpressed) %>%
#  rename_with(~ paste0(., "_S2S_ythdc_wt"), -validated_id)

res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_selected <- res_mettl3_KO_GFP_mettl3_KO_DC1_GOF.df %>%
  dplyr::select(validated_id, baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diffexpressed) %>%
  rename_with(~ paste0(., "_Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP"), -validated_id)





# Merge all datasets
merged_df <- res_S2_m6A_selected %>%
  full_join(res_S2_S_DC1_KO_selected, by = "validated_id") %>%
  full_join(res_S2_GFP_S2_DC1_GOF_selected, by = "validated_id") %>%
  full_join(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_selected, by = "validated_id") %>%
  full_join(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_selected, by = "validated_id") %>%
  full_join(res_mettl3_KO_GFP_S2_GFP_selected, by="validated_id") %>%
  full_join(res_S2_GFP_mettl3_KO_DC1_GOF_selected, by = "validated_id")


#print(merged_df)

merged_df = unique(merged_df)


write.table(merged_df, file = "2025_0415_merged_ALL_DEG.txt", quote = F, row.names = F, sep = "\t")
```

       
       
       Venn diagram shows overlapping genes between METTL3-KO-YTHDC1_GoF vs S2S YTHDC1-GoF  and METTL3-KO vs S2S and Ythdc1-KO vs S2S 
```{r message=FALSE, warning=FALSE, echo=FALSE}


library(ggvenn)

# Extract the `gene_name` column from each dataset
genes_S2_S_METTL3_KO = res_S2_m6A_selected$validated_id
genes_S2_S_DC1_KO = res_S2_S_DC1_KO_selected$validated_id
genes_S2_GFP_S2_DC1_GOF = res_S2_GFP_S2_DC1_GOF_selected$validated_id
genes_METTL3_KO_GFP_METTL3_KO_DC1_GOF = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_selected$validated_id

# Combine the gene lists into a named list for plotting
gene_lists <- list(
  `S2S_METTL3_KO` = genes_S2_S_METTL3_KO,
  `S2S_YTHDC1_KO` = genes_S2_S_DC1_KO,
  `S2S_GFP_S2S_YTHDC1_GOF` = genes_S2_GFP_S2_DC1_GOF,
  `METTL3_KO_GFP_METTL3_KO_YTHDC1_GOF` = genes_METTL3_KO_GFP_METTL3_KO_DC1_GOF)

# Create the Venn diagram using ggvenn
ggvenn(
  data = gene_lists,
  fill_color = c("#FF9999", "#9999FF", "#99FF99", "#FFCC99"),
  stroke_size = 0.5,
  text_size = 5,
  set_name_size = 5)

```
     
    
     Incorporate miCLIP with Differential Expression
```{r message=FALSE, warning=FALSE, echo=FALSE, out.width="100%", out.height="100%"}
library(dplyr)

#df_S2_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE)
#df_S2_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE)


df_s2_cell_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" &  Location != "null", Location != "N/A") #%>% dplyr::filter(fold_enrichment <= 2)Location != "intron" &

df_s2_cell_1ug$Dataset <- "S2_1ug"


df_s2_cell_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "null", Location != "N/A")#%>% dplyr::filter(fold_enrichment <= 2)Location != "intron"


df_s2_cell_5ug$Dataset <- "S2_5ug"

#df_DC_GFP_Input_FR <- read.delim('~/Dropbox/all_files/new_peak_files_java/Ythdc/DC_GFP_Input_FR.tsv', header = T) #%>% dplyr::filter(fold_enrichment <= 2)
#df_DC_GFP_Input_FR$Dataset <- "DC_GFP"

#df_DC_HA_Input_rep1_FR <- read.delim('~/Dropbox/all_files/new_peak_files_java/Ythdc/DC_HA_Input_rep1_FR.tsv', header = T) #%>% dplyr::filter(fold_enrichment <= 2)
#df_DC_HA_Input_rep1_FR$Dataset <- "DC_HA"
 
#wt_head_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond1.tsv', header = TRUE)


#ko_head_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond2.tsv', header = TRUE)



#wt_WF_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond2.tsv', header = TRUE)


#ko_WF_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond1.tsv', header = TRUE)



#wt_WF_peaks$Dataset = "WT-Wholefly_dependent"
#ko_WF_peaks$Dataset = "Mettl3-KO_Wholefly_independent"
#wt_head_peaks$Dataset = "WT-Head_dependent"
#ko_head_peaks$Dataset = "Mettl3-KO-Head_independent"

# Combine data frames
peak_data <- bind_rows(
  df_s2_cell_1ug, df_s2_cell_5ug)#, df_DC_GFP_Input_FR, 
  #df_DC_HA_Input_rep1_FR, ko_WF_peaks, wt_WF_peaks, ko_head_peaks, wt_head_peaks)

# Standardize column names for merging
colnames(peak_data)[colnames(peak_data) == "Gene_ID"] <- "Gene"


peak_summary <- peak_data %>%
  group_by(Gene) %>%
  summarise(
    Peak_Count = n(),
    Locations = paste(unique(Location), collapse = ", "),
    Strands = paste(unique(strand), collapse = ", "),
    Lengths = paste(unique(length), collapse = ", "),
    Fold_Enrichments = paste(unique(fold_enrichment), collapse = ", "),
    Sources = paste(unique(Dataset), collapse = ", "))

merge_with_peaks <- function(deg_df, peak_summary) {
  colnames(deg_df)[colnames(deg_df) == "current_symbol"] <- "Gene"
  return(left_join(deg_df, peak_summary, by = "Gene"))
}


S2_S_DC1_KO_merged_data = merge_with_peaks(res_S2_S_DC1_KO, peak_summary)
S2_m6A_merged_data = merge_with_peaks(res_S2_m6A, peak_summary)
mettl3.KO.GFP_mettl3.KO.DC1.GOF_merged_data = merge_with_peaks(res_mettl3_KO_GFP_mettl3_KO_DC1_GOF, peak_summary)
S2.GFP_S2.DC1.GOF_merged_data = merge_with_peaks(res_S2.GFP_S2.DC1.GOF, peak_summary)
S2.DC1.GOF_mettl3.KO.DC1.GOF_merged_data = merge_with_peaks(res_S2.YTHDC_gof_mettl3.KO.DC1.GOF, peak_summary)

mettl3_KO_GFP_S2_GFP_merged_data = merge_with_peaks(res_mettl3_KO_GFP_S2_GFP, peak_summary)

S2_GFP_mettl3_KO_DC1_GOF_merged_data = merge_with_peaks(res_S2_GFP_mettl3_KO_DC1_GOF, peak_summary)

prepare_rmats <- function(df, prefix, event_type) {
  colnames(df)[colnames(df) == "geneSymbol"] <- "Gene"
  
  df <- df %>%
    dplyr::select(Gene, starts_with(prefix), Splicing_Col = Splicing, PValue, FDR, IncLevel1, IncLevel2, IncLevelDifference) %>% 
    mutate(
      EventType = event_type,
      Splicing_Status = case_when(
        Splicing_Col == "Increased" ~ "Increased",
        Splicing_Col == "Reduced" ~ "Reduced",
        TRUE ~ "Not Significant"),
      
      Splicing_Coordinates = paste(df[[4]], df[[5]], df[[6]], df[[7]], df[[8]], df[[9]], df[[10]], df[[11]], sep = ":")
    )
  return(df)
}


Mettl3_KO_S2S_SE = prepare_rmats(SE_mettl3_KO, "Mettl3_KO_S2S", "SE")
DC1_KO_S2S_SE = prepare_rmats(SE_S2S_DC1_KO, "Ythdc1_KO_S2S", "SE")
YTHDC_GoF_S2_GFP_SE = prepare_rmats(SE_S2_GFP_S2_DC1_GOF, "Ythdc1_S2_GFP", "SE")
METTL3KO_Ythdc1_GoF_Mettl3_KO_SE = prepare_rmats(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP", "SE")
S2_GFP_mettl3_KO_DC1_GOF_SE = prepare_rmats(SE_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "SE")
Mettl3_KO_DC1_GOF_S2_DC1_GOF_SE = prepare_rmats(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "SE")
Mettl3_KO_GFP_S2_GFP_SE = prepare_rmats(SE_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "SE")

Mettl3_KO_S2S_A3 = prepare_rmats(A3SS_mko, "Mettl3_KO_S2S", "A3")
DC1_KO_S2S_A3 = prepare_rmats(A3_S2S_DC1_KO, "Ythdc1_KO_S2S", "A3")
YTHDC_GoF_S2_GFP_A3 = prepare_rmats(A3_SS2_GFP_S2_DC1_GOF, "Ythdc1_GoF_S2_GFP", "A3")
METTL3KO_Ythdc1_GoF_Mettl3_KO_A3 = prepare_rmats(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP", "A3")
METTL3KO_Ythdc1_GoF_S2_GFP_A3 = prepare_rmats(A3_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "A3")
S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_A3 = prepare_rmats(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "A3")
METTL3KO_GFP_S2_GFP_A3 = prepare_rmats(A3_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "A3")


Mettl3_KO_S2S_A5 = prepare_rmats(A5SS_mko, "Mettl3_KO_S2S", "A5")
DC1_KO_S2S_A5 = prepare_rmats(A5_S2S_DC1_KO, "Ythdc1_KO_S2S", "A5")
YTHDC_GoF_S2_GFP_A5 = prepare_rmats(A5SS_S2_GFP_S2_DC1_GOF, "Ythdc1_GoF_S2_GFP", "A5")
METTL3KO_Ythdc1_GoF_Mettl3_KO_A5 = prepare_rmats(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP", "A5")
METTL3KO_Ythdc1_GoF_S2_GFP_A5 = prepare_rmats(A5_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "A5")
S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_A5 = prepare_rmats(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "A5")
METTL3KO_GFP_S2_GFP_A5 = prepare_rmats(A5_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "A5")

Mettl3_KO_S2S_RI = prepare_rmats(RI_S2_MKO, "Mettl3_KO_S2S", "RI")
DC1_KO_S2S_RI = prepare_rmats(RI_S2S_DC1_KO, "Ythdc1_KO_S2S", "RI")
YTHDC_GoF_S2_GFP_RI = prepare_rmats(RI_S2_GFP_S2_DC1_GOF, "Ythdc1_GoF_S2_GFP", "RI")
METTL3KO_Ythdc1_GoF_Mettl3_KO_RI = prepare_rmats(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP", "RI")
METTL3KO_Ythdc1_GoF_S2_GFP_RI = prepare_rmats(RI_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "RI")
S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_RI = prepare_rmats(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "RI")
METTL3KO_GFP_S2_GFP_RI = prepare_rmats(RI_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "RI")



Mettl3_KO_S2S_MXE <- prepare_rmats(MXE_S2_MKO, "Mettl3_KO_S2S", "MXE")
DC1_KO_S2S_MXE <- prepare_rmats(MXE_S2S_DC1_KO, "Ythdc1_KO_S2S", "MXE")
YTHDC_GoF_S2_GFP_MXE <- prepare_rmats(MXE_S2_GFP_S2_DC1_GOF, "Ythdc1_GoF_S2_GFP", "MXE")
METTL3KO_Ythdc1_GoF_Mettl3_KO_MXE <- prepare_rmats(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP", "MXE")
METTL3KO_Ythdc1_GoF_S2_GFP_MXE <- prepare_rmats(MXE_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "MXE")
S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_MXE <- prepare_rmats(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "MXE")
METTL3KO_GFP_S2_GFP_MXE <- prepare_rmats(MXE_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "MXE")


rmats_data <- bind_rows(Mettl3_KO_S2S_SE, DC1_KO_S2S_SE, YTHDC_GoF_S2_GFP_SE, METTL3KO_Ythdc1_GoF_Mettl3_KO_SE,S2_GFP_mettl3_KO_DC1_GOF_SE,Mettl3_KO_DC1_GOF_S2_DC1_GOF_SE,Mettl3_KO_GFP_S2_GFP_SE,
                        Mettl3_KO_S2S_A3,DC1_KO_S2S_A3, YTHDC_GoF_S2_GFP_A3, METTL3KO_Ythdc1_GoF_Mettl3_KO_A3,METTL3KO_Ythdc1_GoF_S2_GFP_A3,S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_A3,METTL3KO_GFP_S2_GFP_A3,
                        Mettl3_KO_S2S_A5, DC1_KO_S2S_A5, YTHDC_GoF_S2_GFP_A5, METTL3KO_Ythdc1_GoF_Mettl3_KO_A5,METTL3KO_Ythdc1_GoF_S2_GFP_A5,S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_A5,METTL3KO_GFP_S2_GFP_A5,
                        Mettl3_KO_S2S_RI,DC1_KO_S2S_RI, YTHDC_GoF_S2_GFP_RI, METTL3KO_Ythdc1_GoF_Mettl3_KO_RI,METTL3KO_Ythdc1_GoF_S2_GFP_RI,S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_RI,METTL3KO_GFP_S2_GFP_RI,
                        Mettl3_KO_S2S_MXE, DC1_KO_S2S_MXE, YTHDC_GoF_S2_GFP_MXE, METTL3KO_Ythdc1_GoF_Mettl3_KO_MXE,METTL3KO_Ythdc1_GoF_S2_GFP_MXE,S2_Ythdc1_GoF_METTL3KO_Ythdc1_GoF_MXE,METTL3KO_GFP_S2_GFP_MXE)


S2S_DC1_KO_DEG_rmats <- S2_S_DC1_KO_merged_data %>%
  full_join(rmats_data, by = "Gene")

S2S_DC1_KO_DEG_rmats <- S2S_DC1_KO_DEG_rmats %>%
  distinct(gene_name, .keep_all = TRUE)

S2_m6A_DEG_rmats <- S2_m6A_merged_data %>%
  full_join(rmats_data, by = "Gene")
S2_m6A_DEG_rmats <- S2_m6A_DEG_rmats %>%
  distinct(gene_name, .keep_all = TRUE)

mettl3.KO.GFP_mettl3.KO.DC1.GOF_DEG_rmats <- mettl3.KO.GFP_mettl3.KO.DC1.GOF_merged_data %>%
  full_join(rmats_data, by = "Gene")
mettl3.KO.GFP_mettl3.KO.DC1.GOF_DEG_rmats <- mettl3.KO.GFP_mettl3.KO.DC1.GOF_DEG_rmats %>%
  distinct(gene_name, .keep_all = TRUE)

S2.GFP_S2.DC1.GOF_DEG_rmats <- S2.GFP_S2.DC1.GOF_merged_data %>%
  full_join(rmats_data, by = "Gene")

S2.GFP_S2.DC1.GOF_DEG_rmats <- S2.GFP_S2.DC1.GOF_DEG_rmats %>%
  distinct(gene_name, .keep_all = TRUE)

S2.DC1.GOF_mettl3.KO.DC1.GOF_DEG_rmats <- S2.DC1.GOF_mettl3.KO.DC1.GOF_merged_data %>%
  full_join(rmats_data, by = "Gene")

S2.DC1.GOF_mettl3.KO.DC1.GOF_DEG_rmats <- S2.DC1.GOF_mettl3.KO.DC1.GOF_DEG_rmats %>%
  distinct(gene_name, .keep_all = TRUE)


mettl3_KO_GFP_S2_GFP_DEG_rmats <- mettl3_KO_GFP_S2_GFP_merged_data %>%
  full_join(rmats_data, by = "Gene")

mettl3_KO_GFP_S2_GFP_DEG_rmats <- mettl3_KO_GFP_S2_GFP_DEG_rmats %>%
  distinct(gene_name, .keep_all = TRUE)

S2_GFP_mettl3_KO_DC1_GOF_DEG_rmats <- S2_GFP_mettl3_KO_DC1_GOF_merged_data %>%
  full_join(rmats_data, by = "Gene")

S2_GFP_mettl3_KO_DC1_GOF_DEG_rmats <- S2_GFP_mettl3_KO_DC1_GOF_DEG_rmats %>%
  distinct(gene_name, .keep_all = TRUE)


#write.table(S2_S_DC1_KO_DEG_rmats, file = "DC1_KO_S2S_DEG_spliced_S2_Peaks.txt", quote = FALSE, row.names = FALSE, sep = "\t")

#write.table(S2_m6A_DEG_rmats, file = "Mettl3_KO_S2S_DEG_spliced_S2_Peaks.txt", quote = FALSE, row.names = FALSE, sep = "\t")

#write.table(S2.GFP_S2.DC1.GOF_DEG_rmats, file = "S2_DC1_GOF_S2_GFP_DEG_spliced_S2_Peaks.txt", quote = FALSE, row.names = FALSE, sep = "\t")

#write.table(mettl3.KO.GFP_mettl3.KO.DC1.GOF_DEG_rmats, file = "Mettl3_KO_DC1_GOF_Mettl3_KO_GFP_DEG_spliced_S2_Peaks.txt", quote = FALSE, row.names = FALSE, sep = "\t")

#write.table(mettl3_KO_GFP_S2_GFP_DEG_rmats, file = "Mettl3_KO_GFP_S2_GFP_DEG_spliced_S2_Peaks.txt", quote = FALSE, row.names = FALSE, sep = "\t")

#write.table(S2_GFP_mettl3_KO_DC1_GOF_DEG_rmats, file = "Mettl3_KO_DC1_GOF_S2_GFP_DEG_spliced_S2_Peaks.txt", quote = FALSE, row.names = FALSE, sep = "\t")

#write.table(S2.DC1.GOF_mettl3.KO.DC1.GOF_DEG_rmats, file = "S2_DC1_GOF_mettl3_KO_DC1_GOF_DEG_spliced_S2_Peaks.txt", quote = FALSE, row.names = FALSE, sep = "\t")

```
     

### Venn diagram of Peaks, AS and DEG
```{r}
df_s2_cell_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A") 
df_s2_cell_1ug$Dataset <- "S2_1ug"

df_s2_cell_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A")
df_s2_cell_5ug$Dataset <- "S2_5ug"


Sig_Mettl3_KO_S2S <- res_S2_m6A %>% 
  dplyr::filter(diffexpressed != "Not Significant")
Sig_DC1_KO_S2S <- res_S2_S_DC1_KO %>% 
  dplyr::filter(diffexpressed != "Not Significant")
Sig_S2.DC1.GOF_S2.GFP <- res_S2.GFP_S2.DC1.GOF %>% 
  dplyr::filter(diffexpressed != "Not Significant")
Sig_mettl3_KO_DC1_GOF_mettl3_KO_GFP <-res_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
    dplyr::filter(diffexpressed != "Not Significant")
Sig_S2.YTHDC_Gof_mettl3.KO.DC1.GOF <- res_S2.YTHDC_gof_mettl3.KO.DC1.GOF %>%
  dplyr::filter(diffexpressed != "Not Significant")
Sig_mettl3_KO_DC1_GOF_S2_GFP <- res_S2_GFP_mettl3_KO_DC1_GOF %>% 
  dplyr::filter(diffexpressed != "Not Significant")
Sig_mettl3_KO_GFP_S2_GFP <- res_mettl3_KO_GFP_S2_GFP %>% 
  dplyr::filter(diffexpressed != "Not Significant")


gene_sets <- list("S2_miCLIP" = peak_data$Gene,
  "Mettl3-KO/S2S" = Sig_Mettl3_KO_S2S$current_symbol,
  "Ythdc1-KO/S2S" = Sig_DC1_KO_S2S$current_symbol,
  #"S2-Ythdc1-GoF/S2-GFP" = Sig_S2.DC1.GOF_S2.GFP$current_symbol,
  #"METTL3-KO-Ythdc-GoF/Mettl3-KO-GFP" = Sig_mettl3_KO_DC1_GOF_mettl3_KO_GFP$current_symbol,
  #"S2-Ythdc1-GoF/METTL3-KO-YTHDC1-GoF" = Sig_S2.YTHDC_Gof_mettl3.KO.DC1.GOF$current_symbol,
  #"METTL3-KO-Ythdc1-GoF/S2-GFP" = Sig_mettl3_KO_DC1_GOF_S2_GFP$current_symbol)
  "METTL3-KO-GFP/S2-GFP" = Sig_mettl3_KO_GFP_S2_GFP$current_symbol)

# Filter out empty gene lists
#gene_lists <- gene_lists[sapply(gene_lists, length) > 0]"#4DAF4A", "#984EA3",  "#377EB8""#0073C2FF", "black", "pink", "orange"




library(ggVennDiagram)
ggVennDiagram(gene_sets,
              edge_size = 1, 
              set_color = c("red", "pink", "purple", "blue")) +
  scale_fill_gradient(low = "white", high = "white") +  # Remove blue color
  ggtitle("Overlap between S2 Peaks and DEGs") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        panel.grid = element_blank(),
        panel.background = element_blank(),
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        show_intersect = TRUE, label_percent_digit = 1, label = "percent") 

library(tidyverse)
library(ComplexUpset)

gene_sets <- list(
  "Mettl3-KO/S2S" = Sig_Mettl3_KO_S2S$gene_name,
  "Ythdc1-KO/S2S" = Sig_DC1_KO_S2S$gene_name,
  "S2-Ythdc1-GoF/S2-GFP" = Sig_S2.DC1.GOF_S2.GFP$gene_name,
  "METTL3-KO-Ythdc-GoF/Mettl3-KO-GFP" = Sig_mettl3_KO_DC1_GOF_mettl3_KO_GFP$gene_name,
  "S2-Ythdc1-GoF/METTL3-KO-YTHDC1-GoF" = Sig_S2.YTHDC_Gof_mettl3.KO.DC1.GOF$gene_name,
  "METTL3-KO-Ythdc1-GoF/S2-GFP" = Sig_mettl3_KO_DC1_GOF_S2_GFP$gene_name,
  "METTL3-KO-GFP/S2-GFP" = Sig_mettl3_KO_GFP_S2_GFP$gene_name)

# Build presence/absence matrix correctly
gene_binary_df <- gene_sets %>%
  purrr::imap(~ tibble(Gene = .x, Set = .y)) %>%
  bind_rows() %>%
  mutate(Present = TRUE) %>%
  pivot_wider(names_from = Set, values_from = Present, values_fill = FALSE)


ComplexUpset::upset(
  gene_binary_df, intersect = names(gene_sets),
  name = "Significant Genes",
  width_ratio = 0.3,
  sort_intersections_by = "degree",
  min_size = 5,
  base_annotations = list(
    'Intersection size' = intersection_size(
      text = list(vjust = -0.5, size = 3))),
  themes = upset_modify_themes(
    list('intersections_matrix' = theme(axis.text.x = element_text(angle = 45, hjust = 1)))))

```
 
 
 
```{r}
library(dplyr)
library(tidyr)
library(ComplexUpset)

# Combine long-form: samples per gene
samples_per_gene <- bind_rows(
  list(
    "Mettl3-KO/S2S" = Sig_Mettl3_KO_S2S %>% dplyr::select(current_symbol),
    "Ythdc1-KO/S2S" = Sig_DC1_KO_S2S %>% dplyr::select(current_symbol),
    "S2-Ythdc1-GoF/S2-GFP" = Sig_S2.DC1.GOF_S2.GFP %>% dplyr::select(current_symbol),
    "METTL3-KO-Ythdc-GoF/Mettl3-KO-GFP" = Sig_mettl3_KO_DC1_GOF_mettl3_KO_GFP %>% dplyr::select(current_symbol),
    "S2-Ythdc1-GoF/METTL3-KO-YTHDC1-GoF" = Sig_S2.YTHDC_Gof_mettl3.KO.DC1.GOF %>% dplyr::select(current_symbol),
    "METTL3-KO-Ythdc1-GoF/S2-GFP" = Sig_mettl3_KO_DC1_GOF_S2_GFP %>% dplyr::select(current_symbol),
    "METTL3-KO-GFP/S2-GFP" = Sig_mettl3_KO_GFP_S2_GFP %>% dplyr::select(current_symbol)),
  .id = "Sample") %>%
  rename(Sample = Sample, Gene = current_symbol)


gene_as_set_matrix <- samples_per_gene %>%
  mutate(Present = TRUE) %>%
  pivot_wider(names_from = Gene, values_from = Present, values_fill = FALSE)


sample_names <- setdiff(colnames(gene_as_set_matrix), "Sample")
sample_colors <- RColorBrewer::brewer.pal(n = length(sample_names), name = "Set3")
names(sample_colors) <- sample_names

ComplexUpset::upset(
  gene_as_set_matrix,
  intersect = sample_names,
  name = "Samples sharing gene",
  width_ratio = 0.3,
  min_size = 5,
  sort_intersections_by = "degree",
  base_annotations = list(
    'Intersection size' = intersection_size(
      text = list(vjust = -0.5, size = 3))),
  themes = upset_modify_themes(
    list('intersections_matrix' = theme(axis.text.x = element_text(angle = 45, hjust = 1)))))

```


# Cumulative distribution 
```{r}
library(ggplot2)
library(dplyr)
library(tidyr)
library(purrr)
df_s2_cell_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A") 

df_s2_cell_1ug$Dataset <- "S2_1ug"


df_s2_cell_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A")


df_s2_cell_5ug$Dataset <- "S2_5ug"

#df_DC_GFP_Input_FR <- read.delim('~/Dropbox/all_files/new_peak_files_java/Ythdc/DC_GFP_Input_FR.tsv', header = T) #%>% dplyr::filter(fold_enrichment <= 2)
#df_DC_GFP_Input_FR$Dataset <- "DC_GFP"

#df_DC_HA_Input_rep1_FR <- read.delim('~/Dropbox/all_files/new_peak_files_java/Ythdc/DC_HA_Input_rep1_FR.tsv', header = T) #%>% dplyr::filter(fold_enrichment <= 2)
#df_DC_HA_Input_rep1_FR$Dataset <- "DC_HA"
 
#wt_head_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond1.tsv', header = TRUE)


#ko_head_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond2.tsv', header = TRUE)



#wt_WF_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond2.tsv', header = TRUE)


#ko_WF_peaks <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond1.tsv', header = TRUE)



#wt_WF_peaks$Dataset = "WT-Wholefly_dependent"
#ko_WF_peaks$Dataset = "Mettl3-KO_Wholefly_independent"
#wt_head_peaks$Dataset = "WT-Head_dependent"
#ko_head_peaks$Dataset = "Mettl3-KO-Head_independent"

# Combine data frames
peak_data <- bind_rows(
  df_s2_cell_1ug, df_s2_cell_5ug)#, df_DC_GFP_Input_FR, 
  #df_DC_HA_Input_rep1_FR, ko_WF_peaks, wt_WF_peaks, ko_head_peaks, wt_head_peaks)

# Standardize column names for merging
colnames(peak_data)[colnames(peak_data) == "Gene_ID"] <- "Gene"

# Adjustable thresholds
padj_threshold <- 0.05
log2fc_threshold <- 0.5


# List of differential expression result files and corresponding labels
deg_files <- list(
  #list(data = res_S2_S_DC1_KO_clean %>% mutate(ENTREZID = as.integer(ENTREZID)), label = "YTHDC1-KO/S2S"),
  #list(data = res_S2_m6A_clean %>% mutate(ENTREZID = as.integer(ENTREZID)), label = "METTL3-KO/S2S"),
  list(data = res_mettl3_KO_GFP_mettl3_KO_DC1_GOF_clean %>% mutate(ENTREZID = as.integer(ENTREZID)), label = "METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP"),
  list(data = res_S2.GFP_S2.DC1.GOF_clean %>% mutate(ENTREZID = as.integer(ENTREZID)), label = "S2-YTHDC1-GOF/S2_GFP"), 
  List(data = res_S2.YTHDC_gof_mettl3.KO.DC1.GOF_clean %>% mutate(ENTREZID = as.integer(ENTREZID)), label = "S2-YTHDC1-GoF/METTL3-KO-YTHDC1-GoF"), 
  list(data = res_S2_GFP_mettl3_KO_DC1_GOF_clean %>% mutate(ENTREZID = as.integer(ENTREZID)), label = "METTL3-KO-YTHDC1-GoF/S2-GFP"))#,
  #list(data = res_mettl3_KO_GFP_S2_GFP_clean %>% mutate(ENTREZID = as.integer(ENTREZID)), label = "METTL3-KO-GFP/S2-GFP"))

# Prepare data for plotting
plot_data <- lapply(deg_files, function(file) {
  file$data %>%
    mutate(
      Significance = ifelse(padj < padj_threshold & abs(log2FoldChange) > log2fc_threshold, "Significant", "Not Significant"),
      m6A_Status = ifelse(current_symbol %in% peak_data$Gene, "m6A", "Non-m6A"),
      Dataset = file$label)
}) %>%
  bind_rows()

# Calculate counts and p-values for each dataset
legend_data <- plot_data %>%
  group_by(Dataset, m6A_Status) %>%
  summarise(Count = n(), .groups = "drop") %>%
  spread(m6A_Status, Count, fill = 0) %>%
  left_join(plot_data %>%
      group_by(Dataset) %>%
      summarise(t_test = list(t.test(log2FoldChange ~ m6A_Status, data = cur_data())),
        .groups = "drop") %>%
      mutate(p_value = map_dbl(t_test, ~ .x$p.value)),
    by = "Dataset") %>%
  mutate(legend_text = paste(
      sprintf("Dataset: %s", Dataset),
      sprintf("m6A: %d", m6A),
      sprintf("Non-m6A: %d", `Non-m6A`),
      sprintf("p-value: %.2e", p_value),
      sep = "\n"))

cdf_plots <- plot_data %>%
  split(.$Dataset) %>%
  purrr::imap(~ {
    dataset_name <- .y
    data <- .x
    
    # Get counts and p-value for the legend
    legend_info <- legend_data %>%
      dplyr::filter(Dataset == dataset_name)
    
    m6A_count <- legend_info$m6A
    non_m6A_count <- legend_info$`Non-m6A`
    p_value <- legend_info$p_value
    
    # Create custom legend labels
    custom_labels <- c(
      sprintf("m6A (n = %d)", m6A_count),
      sprintf("Non-m6A (n = %d)", non_m6A_count))
    
    ggplot(data, aes(x = log2FoldChange, color = m6A_Status, group = m6A_Status)) +
      stat_ecdf(size = 1.2) +
      labs(
        title = paste("S2S peaks implicated in differentially expressed genes in", dataset_name),
        x = "Log2 Fold Change",
        y = "Cumulative Fraction",
        color = sprintf("p-value: %.2e", p_value)) +
      scale_color_manual(
        values = c("m6A" = "red", "Non-m6A" = "blue"),
        labels = custom_labels) +
      theme_minimal(base_size = 14) +
      theme(plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
        legend.position = "right",
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 10),
        plot.caption = element_blank()) +
      xlim(-5, 5)})

# Print all the plots
for (plot in cdf_plots) {
  print(plot)
}



#write.table(plot_data, file ="~/Dropbox/2022_signor_lai_m6A/S2S_rna-seq_results/2024_1013_DEG/CDF_plot_input_file_S2_peaks_DEG.txt", sep = "\t", quote = F, row.name = T, col.names = T)
```




 merged the CDF files
```{r}
library(ggplot2)
library(dplyr)
library(tidyr)
library(purrr)

# Prepare p-values for annotation
legend_data <- plot_data %>%
  group_by(Dataset, m6A_Status) %>%
  summarise(Count = n(), .groups = "drop") %>%
  spread(m6A_Status, Count, fill = 0) %>%
  left_join(
    plot_data %>%
      group_by(Dataset) %>%
      summarise(t_test = list(t.test(log2FoldChange ~ m6A_Status, data = cur_data())),
                .groups = "drop") %>%
      mutate(p_value = map_dbl(t_test, ~ .x$p.value)),
    by = "Dataset"
  ) %>%
  mutate(legend_text = paste(
    sprintf("Dataset: %s", Dataset),
    sprintf("m6A (n = %d)", m6A),
    sprintf("Non-m6A (n = %d)", `Non-m6A`),
    sprintf("p-value: %.2e", p_value),
    sep = "\n"
  ))

# Merge p-values into plot_data
plot_data <- plot_data %>%
  left_join(legend_data %>% dplyr::select(Dataset, p_value), by = "Dataset")

# Generate the combined CDF plot with p-values
cdf_plot <- ggplot(plot_data, aes(x = log2FoldChange, color = m6A_Status, linetype = Dataset, group = interaction(m6A_Status, Dataset))) +
  stat_ecdf(size = 1.2) +
  labs(
    title = "Cumulative Distribution of Log2 Fold Change in GoF Datasets",
    x = "Log2 Fold Change",
    y = "Cumulative Fraction",
    color = "m6A Status",
    linetype = "Dataset"
  ) +
  scale_color_manual(values = c("m6A" = "red", "Non-m6A" = "blue")) +
  theme_minimal(base_size = 14) +
  theme(
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
    legend.position = "right",
    legend.title = element_text(size = 12, face = "bold"),
    legend.text = element_text(size = 10)
  ) +
  xlim(-5, 5) +
  # Add p-values as text annotations
  geom_text(data = legend_data, aes(x = -4, y = 0.2 + as.numeric(factor(Dataset)) * 0.1, label = sprintf("p = %.2e", p_value)), inherit.aes = FALSE, size = 5)

# Print the single combined plot
print(cdf_plot)


```



```{r}

# Step 1: Only retain peaks in 5'UTR, CDS, 3'UTR
peak_filtered <- peak_data %>%
  dplyr::filter(Location %in% c("5'UTR", "CDS", "3'UTR")) %>%
  dplyr::select(Gene, Location)

# Step 2: Assign one location per gene  use the most frequent if multiple
peak_location_per_gene <- peak_filtered %>%
  group_by(Gene, Location) %>%
  summarise(n = n(), .groups = "drop") %>%
  group_by(Gene) %>%
  slice_max(n, with_ties = FALSE) %>%
  ungroup()

# Step 3: Merge with differential expression data
plot_data_with_location <- plot_data %>%
  dplyr::filter(m6A_Status == "m6A") %>%
  left_join(peak_location_per_gene, by = c("current_symbol" = "Gene")) %>%
  dplyr::filter(!is.na(Location)) %>%
  mutate(Location = factor(Location, levels = c("5'UTR", "CDS", "3'UTR")))


# Step 3: Create comparisons
comparisons <- list(
  c("5'UTR", "CDS"),
  c("5'UTR", "3'UTR"),
  c("CDS", "3'UTR"))

# Step 4: Plot with brackets
ggplot(plot_data_with_location, aes(x = Location, y = log2FoldChange, fill = Location)) +
  geom_violin(trim = FALSE, alpha = 0.7) +
  geom_boxplot(width = 0.2, outlier.shape = NA, color = "black", notch = T) +
  stat_summary(fun = median, geom = "point", size = 3, color = "black") +
  stat_compare_means(comparisons = comparisons, method = "wilcox.test", label = "p.signif") +
  facet_wrap(~Dataset) +
  theme_minimal(base_size = 14) +
  labs(
    title = "Differential Expression of m6A Genes by Peak Location",
    subtitle = "Peak location: 5'UTR, CDS, or 3'UTR (only m6A genes with peaks)",
    x = "Peak Location",
    y = "Log2 Fold Change") +
  scale_fill_brewer(palette = "Set2") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 16, face = "bold"),
    plot.subtitle = element_text(hjust = 0.5, size = 12),
    legend.position = "none")


```


 

 
 
## Venn diagram showing the peaks implicated in differential expression
```{r}
# Load required libraries
library(ggvenn)
splicing_files <- list(
  SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
  A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
  RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt",
  MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
  A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt")

df_s2_cell_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A") 
# Optionally, add a new column for labeling if needed
df_s2_cell_1ug$Dataset <- "S2_1ug"


df_s2_cell_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A")

# Optionally, add a new column for labeling if needed
df_s2_cell_5ug$Dataset <- "S2_5ug"

# Assuming peaks have a 'Gene_ID' column
ko_peaks <- unique(df_s2_cell_1ug$Gene_ID)
wt_peaks <- unique(df_s2_cell_5ug$Gene_ID)
all_peaks <- union(ko_peaks, wt_peaks)

# Initialize an empty data frame for results
splicing_gene_stats <- data.frame(SplicingType = character(),
                                  TotalGenes = numeric(),
                                  GenesWithm6A = numeric(),
                                  Percentage = numeric())

# Iterate through each splicing type
for (splicing_type in names(splicing_files)) {
  # Read splicing data
  splicing_data <- read.delim(splicing_files[[splicing_type]], header = TRUE)
  
  # Filter data based on IncLevelDifference and FDR
  filtered_splicing_data <- splicing_data %>%
    dplyr::filter(IncLevelDifference >= 0.2 & FDR <= 0.5)
  
  # Extract unique gene symbols
  splicing_genes <- unique(filtered_splicing_data$geneSymbol)  # Adjust column name if necessary
  
  # Count total genes
  total_genes <- length(splicing_genes)
  
  # Find overlap with m6A peak genes
  genes_with_m6A <- length(intersect(splicing_genes, all_peaks))
  
  # Calculate percentage
  percentage <- (genes_with_m6A / total_genes) * 100
  
  # Append to the data frame
  splicing_gene_stats <- rbind(splicing_gene_stats, 
                               data.frame(SplicingType = splicing_type,
                                          TotalGenes = total_genes,
                                          GenesWithm6A = genes_with_m6A,
                                          Percentage = round(percentage, 2)))
}




# Iterate through each splicing type
for (splicing_type in names(splicing_files)) {
  # Read splicing data for the current type
  splicing_data <- read.delim(splicing_files[[splicing_type]], header = TRUE)
  splicing_genes <- unique(splicing_data$geneSymbol)  # Adjust 'geneSymbol' if needed
  
  # Combine data into a list for the current Venn diagram
  venn_data <- list(
    Splicing = splicing_genes,
    `Mettl3-KO-Head` = ko_peaks,
    `WT-Head` = wt_peaks
  )
  
  # Plot Venn diagram
  venn_plot <- ggvenn(venn_data, fill_color = c("#F8766D", "#00BFC4", "#C77CFF")) +
    ggtitle(paste("Venn Diagram for", splicing_type)) +
    theme(plot.title = element_text(hjust = 0.5))  # Center-align title
  
  # Print the plot
  print(venn_plot)
}


```


```{r}
# Assuming peaks have a 'Gene_ID' column
ko_peaks <- unique(ko_head_peaks$Gene_ID)
wt_peaks <- unique(wt_head_peaks$Gene_ID)
all_peaks <- union(ko_peaks, wt_peaks)

# Initialize an empty data frame for results
splicing_gene_stats <- data.frame(SplicingType = character(),
                                  TotalGenes = numeric(),
                                  GenesWithm6A = numeric(),
                                  Percentage = numeric())

# Iterate through each splicing type
for (splicing_type in names(splicing_files)) {
  # Read splicing data
  splicing_data <- read.delim(splicing_files[[splicing_type]], header = TRUE)
  splicing_genes <- unique(splicing_data$geneSymbol)  # Adjust column name if necessary
  
  # Count total genes
  total_genes <- length(splicing_genes)
  
  # Find overlap with m6A peak genes
  genes_with_m6A <- length(intersect(splicing_genes, all_peaks))
  
  # Calculate percentage
  percentage <- (genes_with_m6A / total_genes) * 100
  
  # Append to the data frame
  splicing_gene_stats <- rbind(splicing_gene_stats, 
                               data.frame(SplicingType = splicing_type,
                                          TotalGenes = total_genes,
                                          GenesWithm6A = genes_with_m6A,
                                          Percentage = round(percentage, 2)))
}

# Display the results
print(splicing_gene_stats)


# Initialize an empty data frame for results
splicing_gene_stats <- data.frame(SplicingType = character(),
                                  GenesWithm6A = numeric())

```



```{r}

splicing_files <- list(
         SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/SE.MATS.JCEC.txt",
         A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
         A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
         MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/MXE.MATS.JCEC.txt",
         RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_DC1_GOF_S2_DC1_GOF/RI.MATS.JCEC.txt")

ko_head_peaks <- read.delim("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv", header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null" & Location != "N/A")

wt_head_peaks <- read.delim("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv", header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null" & Location != "N/A")

# Assuming peaks have a 'Gene_ID' column
ko_peaks <- unique(ko_head_peaks$Gene_ID)
wt_peaks <- unique(wt_head_peaks$Gene_ID)
all_peaks <- union(ko_peaks, wt_peaks)

# Initialize an empty data frame for results
splicing_gene_stats <- data.frame(SplicingType = character(),
                                  TotalGenes = numeric(),
                                  GenesWithm6A = numeric(),
                                  Percentage = numeric())

# Iterate through each splicing type
for (splicing_type in names(splicing_files)) {
  # Read splicing data
  splicing_data <- read.delim(splicing_files[[splicing_type]], header = TRUE)
  
  # Filter data based on IncLevelDifference and FDR
  filtered_splicing_data <- splicing_data %>%
    dplyr::filter(IncLevelDifference >= 0.2 & FDR <= 0.5)
  
  # Extract unique gene symbols
  splicing_genes <- unique(filtered_splicing_data$geneSymbol)  # Adjust column name if necessary
  
  # Count total genes
  total_genes <- length(splicing_genes)
  
  # Find overlap with m6A peak genes
  genes_with_m6A <- length(intersect(splicing_genes, all_peaks))
  
  # Calculate percentage
  percentage <- (genes_with_m6A / total_genes) * 100
  
  # Append to the data frame
  splicing_gene_stats <- rbind(splicing_gene_stats, 
                               data.frame(SplicingType = splicing_type,
                                          TotalGenes = total_genes,
                                          GenesWithm6A = genes_with_m6A,
                                          Percentage = round(percentage, 2)))
}

# Add label column for the legend and percentages for the plot
splicing_gene_stats <- splicing_gene_stats %>%
  mutate(Label = paste0(SplicingType, " (", GenesWithm6A, " genes, ", Percentage, "%)"),
         PercentageLabel = paste0(round(Percentage, 1), "%"))

# Plot the pie chart
pie_chart <- ggplot(splicing_gene_stats, aes(x = "", y = GenesWithm6A, fill = Label)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  geom_text(aes(label = PercentageLabel), position = position_stack(vjust = 0.5)) +  # Add percentage labels
  labs(title = "S2 METTL3-independent peaks associated with S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF", x = NULL, y = NULL) +
  theme_minimal() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        plot.title = element_text(hjust = 0.5),
        legend.title = element_blank()) +
  scale_fill_brewer(palette = "Set3")

# Print the pie chart
print(pie_chart)


```


```{r}



#df_S2_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE)
#df_S2_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE)

#df_Head_FR_WT <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond1.tsv', header = TRUE)
#df_Head_FR_METTL3 <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond2.tsv', header = TRUE)

#df_Wholefly_FR_WT <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond2.tsv', header = TRUE)

#df_Wholefly_FR_METTL3 <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond1.tsv', header = TRUE)


# Initialize an empty data frame for results
all_splicing_stats <- data.frame()

# Iterate through each splicing type
for (splicing_type in names(splicing_files)) {
  # Read splicing data
  splicing_data <- read.delim(splicing_files[[splicing_type]], header = TRUE)
  
  # Debug: Check the first few rows of the splicing data
  print(paste("Processing splicing type:", splicing_type))
  print(head(splicing_data))
  
  # Filter data based on IncLevelDifference and FDR
  filtered_splicing_data <- splicing_data %>%
    dplyr::filter(IncLevelDifference >= 0.2 & FDR <= 0.5)
  
  # Debug: Check if filtering works
  print(paste("Filtered data for", splicing_type, "has rows:", nrow(filtered_splicing_data)))
  
  # Extract unique gene symbols
  splicing_genes <- unique(filtered_splicing_data$geneSymbol)
  
  # If no genes remain after filtering, skip to the next splicing type
  if (length(splicing_genes) == 0) next
  
  # Count total genes
  total_genes <- length(splicing_genes)
  
  # Find overlap with m6A peak genes
  genes_with_m6A <- length(intersect(splicing_genes, all_peaks))
  
  # Calculate percentage
  percentage_with_m6A <- (genes_with_m6A / total_genes) * 100
  
  # Save stats to the combined data frame
  all_splicing_stats <- rbind(all_splicing_stats, 
                              data.frame(SplicingType = splicing_type,
                                         TotalGenes = total_genes,
                                         GenesWithm6A = genes_with_m6A,
                                         PercentageWithm6A = round(percentage_with_m6A, 2)))
  
  # Prepare data for the pie chart
  pie_data <- data.frame(
    Category = c("With m6A", "Without m6A"),
    Count = c(genes_with_m6A, total_genes - genes_with_m6A))
  
  # Debug: Check pie_data
  print(pie_data)
  
  # Perform a chi-squared test for significance
  observed <- c(genes_with_m6A, total_genes - genes_with_m6A)
  expected <- c(total_genes / 2, total_genes / 2)
  p_value <- chisq.test(observed, p = expected / sum(expected))$p.value
  
  # Plot the pie chart for the current splicing type
  pie_chart <- ggplot(pie_data, aes(x = "", y = Count, fill = Category)) +
    geom_bar(stat = "identity", width = 1) +
    coord_polar("y", start = 0) +
    geom_text(aes(label = paste0(Count, " (", round(Count / sum(Count) * 100, 1), "%)")), 
              position = position_stack(vjust = 0.5)) +
    labs(title = paste("S2S m6A Peaks in", splicing_type, "S2-Ythdc1-GoF/Mettl3-KO-Ythdc1-GoF\np-value: ", format(p_value, digits = 3)),
         x = NULL, y = NULL) +
    theme_minimal() +
    theme(
      axis.text = element_blank(),
      axis.ticks = element_blank(),
      panel.grid = element_blank(),
      panel.border = element_blank(),
      plot.title = element_text(hjust = 0.5),
      legend.title = element_blank(),
      legend.position = "right") +
    scale_fill_manual(
      values = c("With m6A" = "red", "Without m6A" = "darkgrey"), # Custom colors
      labels = pie_data$Category)
  
  # Print the pie chart
  print(pie_chart)
}

# Debug: Check final stats
print(all_splicing_stats)

#write.csv(splicing_gene_stats, "splicing_gene_stats.csv", row.names = FALSE)

```

```{r}
# Initialize data frames for summary statistics and gene-level details
# Initialize data frames for overall statistics
all_splicing_stats <- data.frame()
gene_m6A_status <- data.frame()

# Iterate through each splicing type
for (splicing_type in names(splicing_files)) {
  # Read splicing data
  splicing_data <- read.delim(splicing_files[[splicing_type]], header = TRUE)
  
  # Filter data based on IncLevelDifference and FDR
  filtered_splicing_data <- splicing_data %>%
    dplyr::filter(IncLevelDifference >= 0.2 & FDR <= 0.5)
  
  # Extract unique gene symbols
  splicing_genes <- unique(filtered_splicing_data$geneSymbol)
  
  # If no genes remain after filtering, skip to the next splicing type
  if (length(splicing_genes) == 0) next
  
  # Identify genes with m6A peaks
  genes_with_m6A <- intersect(splicing_genes, all_peaks)
  
  # Create a gene-level dataframe for this splicing type
  gene_data <- data.frame(
    GeneSymbol = splicing_genes,
    SplicingType = splicing_type,
    Has_m6A = ifelse(splicing_genes %in% genes_with_m6A, "Yes", "No"))
  
  # Save the event-specific gene-level details
  #write.table(gene_data, file = paste0("gene_m6A_status_", splicing_type, ".txt"), row.names = FALSE, sep = "\t", quote = F)
  
  # Append to combined gene-level dataset
  gene_m6A_status <- rbind(gene_m6A_status, gene_data)
  
  # Summary statistics
  total_genes <- length(splicing_genes)
  m6A_gene_count <- length(genes_with_m6A)
  percentage_with_m6A <- (m6A_gene_count / total_genes) * 100
  
  splicing_stats <- data.frame(
    SplicingType = splicing_type,
    TotalGenes = total_genes,
    GenesWithm6A = m6A_gene_count,
    PercentageWithm6A = round(percentage_with_m6A, 2))
  
  # Save event-specific summary statistics
  write.table(splicing_stats, file = paste0("m6A_splicing_stats_", splicing_type, ".txt"), row.names = FALSE, quote = F, sep = "\t")
  
  # Append to combined statistics dataset
  all_splicing_stats <- rbind(all_splicing_stats, splicing_stats)
}

# Save the combined summary statistics
#write.table(all_splicing_stats, file = "m6A_splicing_stats_all.txt", row.names = FALSE, quote = F, sep = "\t")

# Save the combined gene-level details
#write.table(gene_m6A_status, file = "gene_m6A_status_all.txt", row.names = FALSE, quote = F, sep = "\t")





```


 
    merging the splicing 
```{r}
library(dplyr)

# Function to prepare rMATS splicing data with sample ID in column names
prepare_rmats <- function(df, prefix, event_type, sample_id) {
  colnames(df)[colnames(df) == "geneSymbol"] <- "Gene"
  
  df <- df %>%
    dplyr::select(Gene, starts_with(prefix), Splicing_Col = Splicing, 4:11) %>% 
    mutate(
      EventType = event_type,
      Splicing_Status = case_when(
        Splicing_Col == "Increased" ~ "Increased",
        Splicing_Col == "Reduced" ~ "Reduced",
        TRUE ~ "Not Significant"
      ),
      Splicing_Coordinates = paste(df[[4]], df[[5]], df[[6]], df[[7]], df[[8]], df[[9]], df[[10]], df[[11]], sep = ":"))
  
  colnames(df) <- ifelse(
    colnames(df) %in% c("Gene", "EventType"),
    colnames(df),
    paste(sample_id, colnames(df), sep = "_"))
  
  return(df)
}

# Prepare individual splicing data frames with sample IDs in column names
Mettl3_KO_S2S_SE <- prepare_rmats(SE_mettl3_KO, "Mettl3_KO_S2S", "SE", "Mettl3_KO_S2S")
DC1_KO_S2S_SE <- prepare_rmats(SE_S2S_DC1_KO, "DC1_KO_S2S", "SE", "DC1_KO_S2S")
YTHDF_GoF_S2_GFP_SE <- prepare_rmats(SE_S2_GFP_S2_DC1_GOF, "YTHDF_GoF_S2_GFP", "SE", "YTHDF_GoF_S2_GFP")
METTL3KO_Ythdc1_GoF_Mettl3_KO_SE <- prepare_rmats(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "METTL3KO_Ythdc1_GoF_Mettl3_KO", "SE", "METTL3KO_Ythdc1_GoF_Mettl3_KO")

Mettl3_KO_S2S_A3 <- prepare_rmats(A3SS_mko, "Mettl3_KO_S2S", "A3", "Mettl3_KO_S2S")
DC1_KO_S2S_A3 <- prepare_rmats(A3_S2S_DC1_KO, "DC1_KO_S2S", "A3", "DC1_KO_S2S")
YTHDF_GoF_S2_GFP_A3 <- prepare_rmats(A3_SS2_GFP_S2_DC1_GOF, "YTHDF_GoF_S2_GFP", "A3", "YTHDF_GoF_S2_GFP")
METTL3KO_Ythdc1_GoF_Mettl3_KO_A3 <- prepare_rmats(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "METTL3KO_Ythdc1_GoF_Mettl3_KO", "A3", "METTL3KO_Ythdc1_GoF_Mettl3_KO")

# Combine all splicing event data frames
rmats_data <- bind_rows(
  Mettl3_KO_S2S_SE, DC1_KO_S2S_SE, YTHDF_GoF_S2_GFP_SE, METTL3KO_Ythdc1_GoF_Mettl3_KO_SE,
  Mettl3_KO_S2S_A3, DC1_KO_S2S_A3, YTHDF_GoF_S2_GFP_A3, METTL3KO_Ythdc1_GoF_Mettl3_KO_A3
)

# Save merged splicing data with sample ID in header to file
#write.csv(rmats_data, "merged_splicing_data_with_sample_ids.csv", row.names = FALSE)


```

     Are any transcript involve in the formation of protein!!
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2gof_dc1ko = c("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/S2S_DC1_KO/")
S2gof_dc1ko_data = maser(S2gof_dc1ko,c("YthDC1-KO","S2S") , ftype = "JCEC")

S2gof_dc1gof = c("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/S2_GFP_S2_DC1_GOF/")
S2gof_dc1gof_data = maser(S2gof_dc1gof,c("S2 YthDC1-GoF","S2 GFP") , ftype = "JCEC")

S2gof_mettl3_KO_GFP_mettl3_KO_DC1_GOF = c("~/Dropbox/all_files/S2_m6A/S2_DC1_METTL3_NEW_SAMPLES/mettl3_KO_GFP_mettl3_KO_DC1_GOF/")
S2gof_mettl3_KO_GFP_mettl3_KO_DC1_GOF_data = maser(S2gof_mettl3_KO_GFP_mettl3_KO_DC1_GOF,c("METTL3-KO YthDC1-GoF","METTL3-KO") , ftype = "JCEC")

S2S_mettl3ko = c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/")
S2S_mettl3ko_data = maser(S2S_mettl3ko,c("METTLE-KO","S2S") , ftype = "JCEC")

S2S_dcgof = c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_dc1gof_splicing/")
S2S_dcgof_data = maser(S2S_dcgof,c("DC1_GoF", "S2S") , ftype = "JCEC")
```


```{r message=FALSE, warning=FALSE, echo=FALSE}
S2gof_dc1ko_top = topEvents(S2gof_dc1ko_data, fdr = 0.05, deltaPSI = 0.2)
S2gof_dc1gof_data_top = topEvents(S2gof_dc1gof_data, fdr = 0.05, deltaPSI = 0.2)
S2gof_mettl3_KO_GFP_mettl3_KO_DC1_GOF_top = topEvents(S2gof_mettl3_KO_GFP_mettl3_KO_DC1_GOF_data, fdr = 0.05, deltaPSI = 0.2)
S2S_mettl3ko_top = topEvents(S2S_mettl3ko_data, fdr = 0.05, deltaPSI = 0.2)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2gof_dc1ko_top_filt = filterByCoverage(S2gof_dc1ko_top, avg_reads = 5)
S2gof_dc1gof_data_top_filt = filterByCoverage(S2gof_dc1gof_data_top, avg_reads = 5)
S2gof_mettl3_KO_GFP_mettl3_KO_DC1_GOF_top_filt = filterByCoverage(S2gof_mettl3_KO_GFP_mettl3_KO_DC1_GOF_top, avg_reads = 5)
S2S_mettl3ko_top_filt = filterByCoverage(S2S_mettl3ko_top, avg_reads = 5)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
ens_gtf <- rtracklayer::import.gff("~/Dropbox/all_files/dm6_new.gtf")
# Convert the GTF file to a data frame and filter for exon entries
known_splice_sites <- ens_gtf %>%
  as.data.frame() %>%
  dplyr::filter(type == "exon") %>%
  dplyr::select(seqnames, start, end, strand, gene_id, transcript_id, gene_name)

# Create a standardized bed-like format for known splice sites
known_splice_sites_bed <- known_splice_sites %>%
  mutate(chr = as.character(seqnames),
         exon_start = start,
         exon_end = end) %>%
  dplyr::select(chr, exon_start, exon_end, gene_id, transcript_id, gene_name)

# Convert known splice sites to GRanges object for comparison
known_splice_sites_gr <- makeGRangesFromDataFrame(known_splice_sites_bed, keep.extra.columns = TRUE)

```





    Identify novel transcripts
```{r message=FALSE, warning=FALSE, echo=FALSE}
# Example for SE (Skipped Exon) rMATS output file
S2S_DC1_KO_rmats_output <- read.delim('~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt', header = TRUE)

# Extract relevant columns for exon start and end positions
rmats_se_sites <- S2S_DC1_KO_rmats_output %>%
  dplyr::select(chr = chr, exonStart_0base, exonEnd, strand, geneSymbol) %>%
  mutate(exon_start = exonStart_0base + 1)  # Convert to 1-based coordinates

# Create a GRanges object for rMATS data
rmats_gr <- makeGRangesFromDataFrame(rmats_se_sites, 
                                     seqnames.field = "chr", 
                                     start.field = "exon_start", 
                                     end.field = "exonEnd", 
                                     strand.field = "strand", 
                                     keep.extra.columns = TRUE)

# Step 3: Use findOverlaps to find known vs novel splice sites
overlap <- findOverlaps(rmats_gr, known_splice_sites_gr)

# Extract rMATS splice sites that do not overlap with known splice sites (novel sites)
novel_splice_sites_gr <- rmats_gr[-queryHits(overlap)]

# Convert novel splice sites to a data frame for easier handling
novel_splice_sites_df <- as.data.frame(novel_splice_sites_gr)

# Step 4: Display or save the novel splice junctions
print(unique(novel_splice_sites_df))
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
library(dplyr)
library(GenomicRanges)
library(rtracklayer)

# Step 1: Read the GTF file and extract known splice sites
#ens_gtf <- rtracklayer::import.gff("~/Dropbox/all_files/dm6_new.gtf")

known_splice_sites <- ens_gtf %>%
  as.data.frame() %>%
  dplyr::filter(type == "exon") %>%
  dplyr::select(seqnames, start, end, strand, gene_id, transcript_id, gene_name, type)

known_splice_sites_bed <- known_splice_sites %>%
  mutate(chr = as.character(seqnames),
         exon_start = start,
         exon_end = end) %>%
  dplyr::select(chr, exon_start, exon_end, gene_id, transcript_id, gene_name, type)

known_splice_sites_gr <- makeGRangesFromDataFrame(known_splice_sites_bed, keep.extra.columns = TRUE)

# Function to process rMATS data and extract novel splice sites
process_rmats_data <- function(rmats_data, event_type, suffix) {
  print(paste("Processing:", suffix, "Event Type:", event_type))
  
  # Helper to generate the correct column names based on suffix
  get_col <- function(base) paste0(suffix, "_", base)

  if (event_type == "SE") {
    rmats_sites <- rmats_data %>%
      dplyr::select(
        chr = !!get_col("chr"),
        exonStart_0base = !!get_col("exonStart_0base"),
        exonEnd = !!get_col("exonEnd"),
        strand = !!get_col("strand"),
        Gene,
        Splicing_Status = !!get_col("Splicing_Status"),
        Splicing_Coordinates = !!get_col("Splicing_Coordinates")
      ) %>%
      mutate(exon_start = exonStart_0base + 1)
    start_column <- "exon_start"
    end_column <- "exonEnd"

  } else if (event_type == "A3" || event_type == "A5") {
    rmats_sites <- rmats_data %>%
      dplyr::select(
        chr = !!get_col("chr"),
        longExonStart_0base = !!get_col("longExonStart_0base"),
        longExonEnd = !!get_col("longExonEnd"),
        strand = !!get_col("strand"),
        Gene,
        Splicing_Status = !!get_col("Splicing_Status"),
        Splicing_Coordinates = !!get_col("Splicing_Coordinates")
      ) %>%
      mutate(exon_start = longExonStart_0base + 1)
    start_column <- "exon_start"
    end_column <- "longExonEnd"

  } else if (event_type == "RI") {
    rmats_sites <- rmats_data %>%
      dplyr::select(
        chr = !!get_col("chr"),
        riExonStart_0base = !!get_col("exonStart_0base"),
        riExonEnd = !!get_col("exonEnd"),
        strand = !!get_col("strand"),
        Gene,
        Splicing_Status = !!get_col("Splicing_Status"),
        Splicing_Coordinates = !!get_col("Splicing_Coordinates")
      ) %>%
      mutate(exon_start = riExonStart_0base + 1)
    start_column <- "exon_start"
    end_column <- "riExonEnd"

  } else if (event_type == "MXE") {
    rmats_sites <- rmats_data %>%
      dplyr::select(
        chr = !!get_col("chr"),
        X1stExonStart_0base = !!get_col("upstreamES"),
        X1stExonEnd = !!get_col("upstreamEE"),
        strand = !!get_col("strand"),
        Gene,
        Splicing_Status = !!get_col("Splicing_Status"),
        Splicing_Coordinates = !!get_col("Splicing_Coordinates")
      ) %>%
      mutate(exon_start = X1stExonStart_0base + 1)
    start_column <- "exon_start"
    end_column <- "X1stExonEnd"
  }

  print(paste("Number of extracted sites:", nrow(rmats_sites)))

  rmats_gr <- makeGRangesFromDataFrame(
    rmats_sites,
    seqnames.field = "chr",
    start.field = start_column,
    end.field = end_column,
    strand.field = "strand",
    keep.extra.columns = TRUE
  )

  overlap <- findOverlaps(rmats_gr, known_splice_sites_gr)
  novel_splice_sites_gr <- rmats_gr[-queryHits(overlap)]
  novel_splice_sites_df <- as.data.frame(novel_splice_sites_gr)
  novel_splice_sites_df <- novel_splice_sites_df %>%
    rename_with(~paste0(., "_", suffix), everything())

  return(novel_splice_sites_df)
}

# List of all samples
samples <- list(
  list(data = Mettl3_KO_S2S_SE, event = "SE", suffix = "Mettl3_KO_S2S"),
  list(data = DC1_KO_S2S_SE, event = "SE", suffix = "DC1_KO_S2S"),
  list(data = YTHDF_GoF_S2_GFP_SE, event = "SE", suffix = "YTHDF_GoF_S2_GFP"),
  list(data = METTL3KO_Ythdc1_GoF_Mettl3_KO_SE, event = "SE", suffix = "METTL3KO_Ythdc1_GoF_Mettl3_KO"),
  list(data = Mettl3_KO_S2S_A3, event = "A3", suffix = "Mettl3_KO_S2S"),
  list(data = DC1_KO_S2S_A3, event = "A3", suffix = "DC1_KO_S2S"),
  list(data = YTHDF_GoF_S2_GFP_A3, event = "A3", suffix = "YTHDF_GoF_S2_GFP"),
  list(data = METTL3KO_Ythdc1_GoF_Mettl3_KO_A3, event = "A3", suffix = "METTL3KO_Ythdc1_GoF_Mettl3_KO"),
  list(data = Mettl3_KO_S2S_A5, event = "A5", suffix = "Mettl3_KO_S2S"),
  list(data = DC1_KO_S2S_A5, event = "A5", suffix = "DC1_KO_S2S"),
  list(data = YTHDF_GoF_S2_GFP_A5, event = "A5", suffix = "YTHDF_GoF_S2_GFP"),
  list(data = METTL3KO_Ythdc1_GoF_Mettl3_KO_A5, event = "A5", suffix = "METTL3KO_Ythdc1_GoF_Mettl3_KO"),
  list(data = Mettl3_KO_S2S_RI, event = "RI", suffix = "Mettl3_KO_S2S"),
  list(data = DC1_KO_S2S_RI, event = "RI", suffix = "DC1_KO_S2S"),
  list(data = YTHDF_GoF_S2_GFP_RI, event = "RI", suffix = "YTHDF_GoF_S2_GFP"),
  list(data = METTL3KO_Ythdc1_GoF_Mettl3_KO_RI, event = "RI", suffix = "METTL3KO_Ythdc1_GoF_Mettl3_KO"),
  list(data = Mettl3_KO_S2S_MXE, event = "MXE", suffix = "Mettl3_KO_S2S"),
  list(data = DC1_KO_S2S_MXE, event = "MXE", suffix = "DC1_KO_S2S"),
  list(data = YTHDF_GoF_S2_GFP_MXE, event = "MXE", suffix = "YTHDF_GoF_S2_GFP"),
  list(data = METTL3KO_Ythdc1_GoF_Mettl3_KO_MXE, event = "MXE", suffix = "METTL3KO_Ythdc1_GoF_Mettl3_KO")
)

# Process all samples
novel_splice_sites_list <- lapply(samples, function(sample) {
  result <- process_rmats_data(sample$data, sample$event, sample$suffix)
  print(paste("Processed dataset:", sample$suffix, "with", nrow(result), "rows"))
  return(result)
})

# Ensure start and end are integers for each df
novel_splice_sites_list <- lapply(novel_splice_sites_list, function(df) {
  df$start <- as.integer(df$start)
  df$end <- as.integer(df$end)
  return(df)
})

# Combine all into one dataframe
all_novel_splice_sites <- bind_rows(novel_splice_sites_list)

# Export unique splice sites from each sample
lapply(1:length(novel_splice_sites_list), function(i) {
  result <- novel_splice_sites_list[[i]]
  suffix <- samples[[i]]$suffix
  
  # Build dynamic column names
  start_col <- paste0("start_", suffix)
  end_col <- paste0("end_", suffix)
  chr_col <- paste0("seqnames_", suffix)

  unique_result <- result %>%
    distinct(!!sym(chr_col), !!sym(start_col), !!sym(end_col), .keep_all = TRUE)

  output_file <- paste0("novel_splice_sites_", suffix, ".txt")
  # write.table(unique_result, output_file, row.names = FALSE, sep = "\t", quote = FALSE)
  print(paste("File saved:", output_file, "with", nrow(unique_result), "unique rows"))
})

```


# filter for significant exon coordinates
```{r message=FALSE, warning=FALSE, echo=FALSE}
library(dplyr)


prepare_rmats <- function(df, prefix, event_type, source) {
  colnames(df)[colnames(df) == "geneSymbol"] <- "Gene"
  
  fdr_col <- grep("FDR", colnames(df), value = TRUE)
  inc_level_diff_col <- grep("IncLevelDifference", colnames(df), value = TRUE)

  df <- df %>%
    dplyr::filter(!!sym(fdr_col) < 0.05) %>%
    dplyr::filter(abs(!!sym(inc_level_diff_col)) > 0.2)

  df <- df %>%
    dplyr::select(Gene, ID, PValue, FDR, IncLevel1, IncLevel2, IncLevelDifference, starts_with(prefix), Splicing_Col = Splicing) %>% 
    mutate(EventType = event_type,
      Splicing_Status = case_when(
        Splicing_Col == "Increased" ~ "Increased",
        Splicing_Col == "Reduced" ~ "Reduced",
        TRUE ~ "Not Significant"),
      Splicing_Coordinates = paste(df[[4]], df[[5]], df[[6]], df[[7]], df[[8]], df[[9]], df[[10]], df[[11]], sep = ":"))
  
  colnames(df) <- paste(source, colnames(df), sep = "_")
  colnames(df)[grepl("Splicing_Coordinates", colnames(df))] <- "Splicing_Coordinates"

  return(df)
}

# Prepare individual splicing data frames with sources
Mettl3_KO_S2S_SE <- prepare_rmats(SE_mettl3_KO, "Mettl3_KO_S2S", "SE", "Mettl3_KO_S2S")
DC1_KO_S2S_SE <- prepare_rmats(SE_S2S_DC1_KO, "Ythdc1_KO_S2S", "SE", "Ythdc1_KO_S2S")
YTHDC_GoF_S2_GFP_SE <- prepare_rmats(SE_S2_GFP_S2_DC1_GOF, "S2_Ythdc1_GoF_S2_GFP", "SE", "S2_Ythdc1_GoF_S2_GFP")
METTL3KO_Ythdc1_GoF_Mettl3_KO_SE <- prepare_rmats(se_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP", "SE", "Mettl3_KO_Ythdc1_GoF_Mettl3_KO_GFP")

METTL3KO_Ythdc1_GoF_S2_GFP_SE <- prepare_rmats(SE_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "SE", "Mettl3_KO_Ythdc1_GoF_S2_GFP")

S2_YTHDF_GoF_METTL3KO_Ythdc1_GoF_SE <- prepare_rmats(SE_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "SE", "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF")

METTL3_KO_S2_GFP_SE <- prepare_rmats(SE_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "SE", "Mettl3_KO_GFP_S2_GFP")


# Combine all splicing event data frames by Splicing_Coordinates
merged_splicing_data <- list(
  Mettl3_KO_S2S_SE,
  DC1_KO_S2S_SE,
  YTHDC_GoF_S2_GFP_SE,
  METTL3KO_Ythdc1_GoF_Mettl3_KO_SE,
  METTL3KO_Ythdc1_GoF_S2_GFP_SE,
  S2_YTHDF_GoF_METTL3KO_Ythdc1_GoF_SE,
  METTL3_KO_S2_GFP_SE) %>%
  reduce(full_join, by = "Splicing_Coordinates")


#write.table(merged_splicing_data, "2025_0418_merged_splicing_SE.txt", row.names = FALSE, sep = '\t', quote = F)


```

    Separate the spliced data into separate files after merging
```{r message=FALSE, warning=FALSE, echo=FALSE}
Mettl3_KO_S2S_A3 <- prepare_rmats(A3SS_mko, "Mettl3_KO_S2S", "A3", "Mettl3_KO_S2S")
DC1_KO_S2S_A3 <- prepare_rmats(A3_S2S_DC1_KO, "DC1_KO_S2S", "A3", "DC1_KO_S2S")
YTHDC_GoF_S2_GFP_A3 <- prepare_rmats(A3_SS2_GFP_S2_DC1_GOF, "S2_Ythdc1_GoF_S2_GFP", "A3", "S2_Ythdc1_GoF_S2_GFP")
METTL3KO_Ythdc1_GoF_Mettl3_KO_A3 <- prepare_rmats(A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP", "A3", "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP")


METTL3KO_Ythdc1_GoF_S2_GFP_A3 <- prepare_rmats(A3_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "A3", "Mettl3_KO_Ythdc1_GoF_S2_GFP")

S2_YTHDF_GoF_METTL3KO_Ythdc1_GoF_A3 <- prepare_rmats(A3_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "A3", "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF")

METTL3_KO_S2_GFP_A3 <- prepare_rmats(A3_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "A3", "Mettl3_KO_GFP_S2_GFP")

merged_splicing_data <- list(
  Mettl3_KO_S2S_A3,
  DC1_KO_S2S_A3,
  YTHDC_GoF_S2_GFP_A3,
  METTL3KO_Ythdc1_GoF_Mettl3_KO_A3, 
  METTL3KO_Ythdc1_GoF_S2_GFP_A3,
  S2_YTHDF_GoF_METTL3KO_Ythdc1_GoF_A3,
  METTL3_KO_S2_GFP_A3) %>%
  reduce(full_join, by = "Splicing_Coordinates")


#write.table(merged_splicing_data, "2025_0418_merged_splicing_A3.txt", row.names = FALSE, sep = '\t', quote = F)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
Mettl3_KO_S2S_A5 <- prepare_rmats(A5SS_mko, "Mettl3_KO_S2S", "A5", "Mettl3_KO_S2S")
DC1_KO_S2S_A5 <- prepare_rmats(A5_S2S_DC1_KO, "DC1_KO_S2S", "A5", "DC1_KO_S2S")
YTHDC_GoF_S2_GFP_A5 <- prepare_rmats(A5SS_S2_GFP_S2_DC1_GOF, "S2_Ythdc1_GoF_S2_GFP", "A5", "S2_Ythdc1_GoF_S2_GFP")
METTL3KO_Ythdc1_GoF_Mettl3_KO_A5 <- prepare_rmats(A5_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP", "A5", "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP")



S2_GFP_mettl3_KO_DC1_GOF_A5 <- prepare_rmats(A5_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "A5", "Mettl3_KO_Ythdc1_GoF_S2_GFP")
S2_YTHDF_GoF_METTL3KO_Ythdc1_GoF_A5 <- prepare_rmats(A5_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "A5", "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF")
S2_GFP_mettl3_KO_GFP_A5 <- prepare_rmats(A5_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "A5", "Mettl3_KO_GFP_S2_GFP")

merged_splicing_data <- list(Mettl3_KO_S2S_A5,
  DC1_KO_S2S_A5,
  YTHDC_GoF_S2_GFP_A5,
  METTL3KO_Ythdc1_GoF_Mettl3_KO_A5,
  S2_GFP_mettl3_KO_DC1_GOF_A5,
  S2_YTHDF_GoF_METTL3KO_Ythdc1_GoF_A5,
  S2_GFP_mettl3_KO_GFP_A5) %>%
  reduce(full_join, by = "Splicing_Coordinates")


#write.table(merged_splicing_data, "2025_0418_merged_splicing_A5.txt", row.names = FALSE, sep = '\t', quote = F)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}


Mettl3_KO_S2S_RI <- prepare_rmats(RI_S2_MKO, "Mettl3_KO_S2S", "RI", "Mettl3_KO_S2S")
DC1_KO_S2S_RI <- prepare_rmats(RI_S2S_DC1_KO, "DC1_KO_S2S", "RI", "DC1_KO_S2S")
YTHDC_GoF_S2_GFP_RI <- prepare_rmats(RI_S2_GFP_S2_DC1_GOF, "S2_Ythdc1_GoF_S2_GFP", "RI", "S2_Ythdc1_GoF_S2_GFP")
METTL3KO_Ythdc1_GoF_Mettl3_KO_RI <- prepare_rmats(RI_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP", "RI", "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP")

S2_GFP_mettl3_KO_DC1_GOF_RI <- prepare_rmats(RI_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "RI", "Mettl3_KO_Ythdc1_GoF_S2_GFP")

S2_YTHDC_GoF_METTL3KO_Ythdc1_GoF_RI <- prepare_rmats(RI_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "RI", "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF")

S2_GFP_mettl3_KO_GFP_RI <- prepare_rmats(RI_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "RI", "Mettl3_KO_GFP_S2_GFP")

merged_splicing_data <- list(Mettl3_KO_S2S_RI,
  DC1_KO_S2S_RI,
  YTHDC_GoF_S2_GFP_RI,
  METTL3KO_Ythdc1_GoF_Mettl3_KO_RI, 
  S2_GFP_mettl3_KO_DC1_GOF_RI, 
  S2_YTHDC_GoF_METTL3KO_Ythdc1_GoF_RI,
  S2_GFP_mettl3_KO_GFP_RI) %>%
  reduce(full_join, by = "Splicing_Coordinates")


#write.table(merged_splicing_data, "2025_0418_merged_splicing_RI.txt", row.names = FALSE, sep = '\t', quote = F)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}

Mettl3_KO_S2S_MXE <- prepare_rmats(MXE_S2_MKO, "Mettl3_KO_S2S", "MXE", "Mettl3_KO_S2S")
DC1_KO_S2S_MXE <- prepare_rmats(MXE_S2S_DC1_KO, "DC1_KO_S2S", "MXE", "DC1_KO_S2S")
YTHDC_GoF_S2_GFP_MXE <- prepare_rmats(MXE_S2_GFP_S2_DC1_GOF, "S2_Ythdc1_GoF_S2_GFP", "MXE", "S2_Ythdc1_GoF_S2_GFP")
METTL3KO_Ythdc1_GoF_Mettl3_KO_MXE <- prepare_rmats(MXE_mettl3_KO_GFP_mettl3_KO_DC1_GOF, "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP", "MXE", "METTL3KO_Ythdc1_GoF_Mettl3_KO_GFP")


Mettl3_KO_Ythdc1_GoF_S2_GFP_MXE <- prepare_rmats(MXE_S2_GFP_mettl3_KO_DC1_GOF, "Mettl3_KO_Ythdc1_GoF_S2_GFP", "MXE", "Mettl3_KO_Ythdc1_GoF_S2_GFP")

S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF_MXE <- prepare_rmats(MXE_mettl3_KO_DC1_GOF_S2_DC1_GOF, "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF", "MXE", "S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF")

Mettl3_KO_GFP_S2_GFP_MXE <- prepare_rmats(MXE_S2_GFP_mettl3_KO_GFP, "Mettl3_KO_GFP_S2_GFP", "MXE", "Mettl3_KO_GFP_S2_GFP")


merged_splicing_data <- list(
  Mettl3_KO_S2S_MXE,
  DC1_KO_S2S_MXE,
  YTHDC_GoF_S2_GFP_MXE,
  METTL3KO_Ythdc1_GoF_Mettl3_KO_MXE,
  Mettl3_KO_Ythdc1_GoF_S2_GFP_MXE,
  S2_Ythdc1_GoF_Mettl3_KO_Ythdc1_GoF_MXE,
  Mettl3_KO_GFP_S2_GFP_MXE) %>%
  reduce(full_join, by = "Splicing_Coordinates")


#write.table(merged_splicing_data, "2025_0418_merged_splicing_MXE.txt", row.names = FALSE, sep = '\t', quote = F)
```



    function to make box plot for METTL3, YTHDC1 and METTL3-KO-YTHDC1-GoF
```{r message=FALSE, warning=FALSE, echo=FALSE}
file_paths_config <- list(
    METTL3_KO = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt"),
    Ythdc1_KO = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt"),
    
    Ythdc1_GoF = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt"),

    METTL3_KO_Ythdc1_GoF = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt"))


function_to_make_PSI_box_Mettl3ko_Ythdc1_KO <- function(event_IDs, Splicing_type) {
    # Retrieve file paths from the configuration
    file_path_METTL3_KO <- file_paths_config$METTL3_KO[[Splicing_type]]
    file_path_Ythdc1_KO <- file_paths_config$Ythdc1_KO[[Splicing_type]]
    file_path_Ythdc1_GoF <- file_paths_config$Ythdc1_GoF[[Splicing_type]]
    file_path_METTL3_KO_Ythdc1_GoF <- file_paths_config$METTL3_KO_Ythdc1_GoF[[Splicing_type]]

    # Process METTL3-KO data
    data_METTL3_KO <- read.table(file_path_METTL3_KO, header = TRUE)
    data_METTL3_KO_filtered <- data_METTL3_KO[data_METTL3_KO$ID == event_IDs[1], ]
    data_METTL3_KO_filtered$Dataset <- "METTL3-KO"

    # Process Ythdc1-KO data
    data_Ythdc1_KO <- read.table(file_path_Ythdc1_KO, header = TRUE)
    data_Ythdc1_KO_filtered <- data_Ythdc1_KO[data_Ythdc1_KO$ID == event_IDs[2], ]
    data_Ythdc1_KO_filtered$Dataset <- "Ythdc1-KO"

    # Process Ythdc1-GoF data
    data_Ythdc1_GoF <- read.table(file_path_Ythdc1_GoF, header = TRUE)
    data_Ythdc1_GoF_filtered <- data_Ythdc1_GoF[data_Ythdc1_GoF$ID == event_IDs[3], ]
    data_Ythdc1_GoF_filtered$Dataset <- "Ythdc1-GoF"

    # Process METTL3-KO-Ythdc1-GoF data
    data_METTL3_KO_Ythdc1_GoF <- read.table(file_path_METTL3_KO_Ythdc1_GoF, header = TRUE)
    data_METTL3_KO_Ythdc1_GoF_filtered <- data_METTL3_KO_Ythdc1_GoF[data_METTL3_KO_Ythdc1_GoF$ID == event_IDs[4], ]
    data_METTL3_KO_Ythdc1_GoF_filtered$Dataset <- "METTL3-KO-Ythdc1-GoF"

    # Combine the datasets
    combined_data <- rbind(
        data_METTL3_KO_filtered,
        data_Ythdc1_KO_filtered,
        data_Ythdc1_GoF_filtered,
        data_METTL3_KO_Ythdc1_GoF_filtered)

    # Reshape the combined data for visualization
    psi_data <- data.frame(
        ID = combined_data$ID,
        IncLevel1 = combined_data$IncLevel1,
        IncLevel2 = combined_data$IncLevel2,
        Dataset = combined_data$Dataset,
        FDR = combined_data$FDR  # Include FDR from the input data
    )
    psi_data_long <- melt(psi_data, id.vars = c("ID", "Dataset", "FDR"), variable.name = "Sample", value.name = "PSI")

    # Split the concatenated PSI values into individual rows
    psi_data_long <- psi_data_long %>%
        mutate(PSI = strsplit(as.character(PSI), ",")) %>%
        unnest(PSI)
    psi_data_long$PSI <- as.numeric(psi_data_long$PSI)

    # Add Condition based on Dataset and Sample
    psi_data_long$Condition <- ifelse(
        psi_data_long$Dataset == "METTL3-KO",
        ifelse(psi_data_long$Sample == "IncLevel1", "METTL3-KO", "S2S"),
        ifelse(psi_data_long$Dataset == "Ythdc1-KO",
            ifelse(psi_data_long$Sample == "IncLevel1", "Ythdc1-KO", "S2S"),
            ifelse(psi_data_long$Dataset == "Ythdc1-GoF",
                ifelse(psi_data_long$Sample == "IncLevel1", "Ythdc1-GoF", "S2-GFP"),
                ifelse(psi_data_long$Sample == "IncLevel1", "METTL3-KO-Ythdc1-GoF", "METTL3-KO-GFP"))))

    # Reorder the factor levels to ensure control samples are on the left
    psi_data_long$Condition <- factor(
        psi_data_long$Condition,
        levels = c("S2S", "S2-GFP", "METTL3-KO-GFP", "METTL3-KO", "Ythdc1-KO", "Ythdc1-GoF", "METTL3-KO-Ythdc1-GoF"))

    # Extract FDR values for annotations
    fdr_annotations <- psi_data_long %>%
        group_by(ID, Dataset) %>%
        summarise(FDR = unique(FDR))  # Ensure one FDR per ID and Dataset

    # Create the box plot
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) +
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID + Dataset, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
        scale_fill_manual(values = c(
            "S2S" = "red", "S2-GFP" = "brown", "METTL3-KO-GFP" = "orange",
            "METTL3-KO" = "blue", "Ythdc1-KO" = "green",
            "Ythdc1-GoF" = "purple", "METTL3-KO-Ythdc1-GoF" = "pink")) +
        geom_text(data = fdr_annotations,
                  aes(label = paste0("FDR: ", signif(FDR, 3)), x = 2, y = 0.95),
                  inherit.aes = FALSE,
                  size = 4,
                  hjust = 0)

    # Print the plot
    print(plot)
}

```


function to make box plot for METTL3, YTHDC1 and METTL3-KO-YTHDC1-GoF
```{r message=FALSE, warning=FALSE, echo=FALSE}
file_paths_config_MY <- list(
    METTL3_KO = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI.MATS.JCEC.txt"),
    Ythdc1_KO = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI.MATS.JCEC.txt"))

make_PSI_box_Mettl3ko_Ythdc1_KO <- function(event_IDs, Splicing_type) {
    # Check input length
    if (length(event_IDs) != 2) {
        stop("Please provide exactly two event IDs: one for METTL3-KO and one for Ythdc1-KO.")
    }

    # Retrieve file paths
    file_path_METTL3_KO <- file_paths_config_MY$METTL3_KO[[Splicing_type]]
    file_path_Ythdc1_KO <- file_paths_config_MY$Ythdc1_KO[[Splicing_type]]

    # Check file existence
    if (!file.exists(file_path_METTL3_KO) | !file.exists(file_path_Ythdc1_KO)) {
        stop("One or both input files do not exist!")
    }

    # Load Data
    data_METTL3_KO <- read.table(file_path_METTL3_KO, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
    data_Ythdc1_KO <- read.table(file_path_Ythdc1_KO, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

    # Filter for specified event IDs
    data_METTL3_KO_filtered <- data_METTL3_KO %>% dplyr::filter(ID == event_IDs[1]) %>% mutate(Dataset = "METTL3-KO")
    data_Ythdc1_KO_filtered <- data_Ythdc1_KO %>% dplyr::filter(ID == event_IDs[2]) %>% mutate(Dataset = "Ythdc1-KO")

    # Combine Data
    combined_data <- bind_rows(data_METTL3_KO_filtered, data_Ythdc1_KO_filtered)

    # Reshape for plotting
    psi_data_long <- combined_data %>%
        dplyr::select(ID, Dataset, FDR, IncLevel1, IncLevel2) %>%
        pivot_longer(cols = starts_with("IncLevel"), names_to = "Sample", values_to = "PSI") %>%
        separate_rows(PSI, sep = ",") %>%
        mutate(PSI = as.numeric(PSI)) %>%
        mutate(Condition = case_when(
            Dataset == "METTL3-KO" & Sample == "IncLevel1" ~ "METTL3-KO",
            Dataset == "METTL3-KO" & Sample == "IncLevel2" ~ "S2S",
            Dataset == "Ythdc1-KO" & Sample == "IncLevel1" ~ "Ythdc1-KO",
            Dataset == "Ythdc1-KO" & Sample == "IncLevel2" ~ "S2S"))

    # Ensure ordered factor levels
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("S2S", "METTL3-KO", "Ythdc1-KO"))

    # Extract FDR values
    fdr_annotations <- psi_data_long %>%
        group_by(ID, Dataset) %>%
        summarise(FDR = unique(FDR), .groups = "drop")

    # Boxplot
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) +
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID + Dataset, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
        scale_fill_manual(values = c(
            "S2S" = "red", "METTL3-KO" = "blue", "Ythdc1-KO" = "green")) +
        geom_text(data = fdr_annotations,
                  aes(label = paste0("FDR: ", signif(FDR, 3)), x = 2, y = 0.95),
                  inherit.aes = FALSE,
                  size = 4,
                  hjust = 0)

    # Print the plot
    print(plot)
}
```


```{r message=FALSE, warning=FALSE, echo=FALSE}
file_paths_config_GFP <- list(
    
    Ythdc1_GoF = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI.MATS.JCEC.txt"),

    METTL3_KO_Ythdc1_GoF = list(
        SE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE.MATS.JCEC.txt",
        A3SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS.MATS.JCEC.txt",
        A5SS = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS.MATS.JCEC.txt",
        MXE = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE.MATS.JCEC.txt",
        RI = "~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI.MATS.JCEC.txt"))


make_PSI_box_S2_GFP_METTL3_GFP <- function(event_IDs, Splicing_type) {
    # Check if exactly two event IDs are provided
    if (length(event_IDs) != 2) {
        stop("Please provide exactly two event IDs: one for Ythdc1-GoF and one for METTL3-KO-Ythdc1-GoF.")
    }

    # Retrieve file paths
    file_path_Ythdc1_GoF <- file_paths_config_GFP$Ythdc1_GoF[[Splicing_type]]
    file_path_METTL3_KO_Ythdc1_GoF <- file_paths_config_GFP$METTL3_KO_Ythdc1_GoF[[Splicing_type]]

    # Check if files exist
    if (!file.exists(file_path_Ythdc1_GoF) | !file.exists(file_path_METTL3_KO_Ythdc1_GoF)) {
        stop("One or both input files do not exist!")
    }

    # Load Data
    data_Ythdc1_GoF <- read.table(file_path_Ythdc1_GoF, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
    data_METTL3_KO_Ythdc1_GoF <- read.table(file_path_METTL3_KO_Ythdc1_GoF, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

    # Filter for specified event IDs
    data_Ythdc1_GoF_filtered <- data_Ythdc1_GoF %>% dplyr::filter(ID == event_IDs[1]) %>% mutate(Dataset = "Ythdc1-GoF")
    data_METTL3_KO_Ythdc1_GoF_filtered <- data_METTL3_KO_Ythdc1_GoF %>% dplyr::filter(ID == event_IDs[2]) %>% mutate(Dataset = "METTL3-KO-Ythdc1-GoF")

    # Combine Data
    combined_data <- bind_rows(data_Ythdc1_GoF_filtered, data_METTL3_KO_Ythdc1_GoF_filtered)

    # Reshape for plotting
    psi_data_long <- combined_data %>%
        dplyr::select(ID, Dataset, FDR, IncLevel1, IncLevel2) %>%
        pivot_longer(cols = starts_with("IncLevel"), names_to = "Sample", values_to = "PSI") %>%
        separate_rows(PSI, sep = ",") %>%
        mutate(PSI = as.numeric(PSI)) %>%
        mutate(Condition = case_when(
            Dataset == "Ythdc1-GoF" & Sample == "IncLevel1" ~ "Ythdc1-GoF",
            Dataset == "Ythdc1-GoF" & Sample == "IncLevel2" ~ "S2-GFP",
            Dataset == "METTL3-KO-Ythdc1-GoF" & Sample == "IncLevel1" ~ "METTL3-KO-Ythdc1-GoF",
            Dataset == "METTL3-KO-Ythdc1-GoF" & Sample == "IncLevel2" ~ "METTL3-KO-GFP"
        ))

    # Ensure ordered factor levels
    psi_data_long$Condition <- factor(psi_data_long$Condition, levels = c("S2-GFP", "METTL3-KO-GFP", "Ythdc1-GoF", "METTL3-KO-Ythdc1-GoF"))

    # Extract FDR values
    fdr_annotations <- psi_data_long %>%
        group_by(ID, Dataset) %>%
        summarise(FDR = unique(FDR), .groups = "drop")

    # Boxplot
    plot <- ggplot(psi_data_long, aes(x = Condition, y = PSI, fill = Condition)) +
        geom_boxplot(alpha = 0.6, outlier.shape = NA) +
        geom_jitter(position = position_jitter(width = 0.2), size = 1.5, alpha = 0.6) +
        scale_y_continuous(limits = c(0, 1)) +
        facet_wrap(~ ID + Dataset, scales = "free") +
        theme_bw() +
        theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(face = "plain", colour = "black", size = 12),
              axis.title.y = element_text(face = "plain", colour = "black", size = 12),
              legend.title = element_blank(),
              legend.text = element_text(face = "plain", colour = "black", size = 12)) +
        ylab("PSI") +
        scale_fill_manual(values = c(
          "S2-GFP" = "orange", "METTL3-KO-GFP" = "cyan",
            "Ythdc1-GoF" = "purple", "METTL3-KO-Ythdc1-GoF" = "pink")) +
        geom_text(data = fdr_annotations,
                  aes(label = paste0("FDR: ", signif(FDR, 3)), x = 2, y = 0.95),
                  inherit.aes = FALSE,
                  size = 4,
                  hjust = 0)

    # Print the plot
    print(plot)
}
```




Let's make box plot for METTL3-KO and Ythdc1


```{r message=FALSE, warning=FALSE, echo=FALSE} 
S2S_mettl3ko = c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/")

S2S_ythdc.ko = c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/")

Sgfp_ythdcgof = c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/")

mettl3.ko_ythdc.gof = c("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/")

S2S_mettl3ko_data = maser(S2S_mettl3ko, c("METTLE-KO","S2S") , ftype = "JCEC")

S2S_ythdc1ko_data = maser(S2S_ythdc.ko, c("YTHDC1-KO","S2S") , ftype = "JCEC")


Sgfp_ythdcgof_data = maser(Sgfp_ythdcgof, c("S2-YTHDC1-GoF","S2-GFP") , ftype = "JCEC")

mettl3.ko_ythdc.gof_data = maser(mettl3.ko_ythdc.gof, c("METTL3-KO_YTHDC1-GoF","METTL3-KO-GFP") , ftype = "JCEC")

```


```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_data_filt = filterByCoverage(S2S_mettl3ko_data, avg_reads = 5)
S2S_mettl3ko_data_top = topEvents(S2S_mettl3ko_data_filt, fdr = 0.05, deltaPSI = 0.2)

S2S_ythdc1_data_filt = filterByCoverage(S2S_ythdc1ko_data, avg_reads = 5)
S2S_ythdc1_data_top = topEvents(S2S_ythdc1_data_filt, fdr = 0.05, deltaPSI = 0.2)


Sgfp_ythdcgof_data_filt = filterByCoverage(Sgfp_ythdcgof_data, avg_reads = 5)
Sgfp_ythdcgof_data_top = topEvents(Sgfp_ythdcgof_data_filt, fdr = 0.05, deltaPSI = 0.2)

mettl3.ko_ythdc.gof_data_filt = filterByCoverage(mettl3.ko_ythdc.gof_data, avg_reads = 5)
mettl3.ko_ythdc.gof_data_top = topEvents(mettl3.ko_ythdc.gof_data_filt, fdr = 0.05, deltaPSI = 0.2)

```



```{r message=FALSE, warning=FALSE, echo=FALSE}

S2S_mettl3ko_Sxl = geneEvents(S2S_mettl3ko_data_top, geneS = "Sxl", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Sxl)

S2S_ythdcko_Sxl = geneEvents(S2S_ythdc1_data_top, geneS = "Sxl", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Sxl)

Sgfp_ythdcgof_Sxl = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Sxl", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Sxl)

mettl3.ko_ythdc.gof_Sxl = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Sxl", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Sxl)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Sxl@SE_stats
S2S_ythdcko_Sxl@SE_stats

Sgfp_ythdcgof_Sxl@SE_stats
mettl3.ko_ythdc.gof_Sxl@SE_stats

```
# Sxl: chrX	-	7096814	7096935	7091673	7092170	7096997	7097066

```{r message=FALSE, warning=FALSE, echo=FALSE}

make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(3973, 3783), Splicing_type = "SE")
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(3973, 3783, 3757,4113), Splicing_type = "SE")
```

### fl(2)d
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_fl2 = geneEvents(S2S_mettl3ko_data_top, geneS = "fl(2)d", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_fl2)

S2S_ythdcko_fl2 = geneEvents(S2S_ythdc1_data_top, geneS = "fl(2)d", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_fl2)

Sgfp_ythdcgof_fl2 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "fl(2)d", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_fl2)


mettl3.ko_ythdc.gof_fl2 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "fl(2)d", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_fl2)


```

fl(2)d:chr2R	-	13821765	13822175	13821765	13822125	13824283	13824553

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_fl2@A3SS_stats

S2S_ythdcko_fl2@A3SS_stats





```


```{r message=FALSE, warning=FALSE, echo=FALSE}
make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(216, 214), Splicing_type = "A3SS")
```



### cv-c
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_cvc = geneEvents(S2S_mettl3ko_data_top, geneS = "cv-c", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_cvc)

S2S_ythdcko_cvc = geneEvents(S2S_ythdc1_data_top, geneS = "cv-c", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_cvc)

Sgfp_ythdcgof_cvc = geneEvents(Sgfp_ythdcgof_data_top, geneS = "cv-c", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_cvc)


mettl3.ko_ythdc.gof_cvc = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "cv-c", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_cvc)


```




### nrv3
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_nrv3 = geneEvents(S2S_mettl3ko_data_top, geneS = "nrv3", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_nrv3)

S2S_ythdcko_nrv3 = geneEvents(S2S_ythdc1_data_top, geneS = "nrv3", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_nrv3)

Sgfp_ythdcgof_nrv3 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "nrv3", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_nrv3)


mettl3.ko_ythdc.gof_nrv3 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "nrv3", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_nrv3)


```



```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_mfas = geneEvents(S2S_mettl3ko_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_mfas)

S2S_ythdcko_mfas = geneEvents(S2S_ythdc1_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_mfas)

Sgfp_ythdcgof_mfas = geneEvents(Sgfp_ythdcgof_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_mfas)


mettl3.ko_ythdc.gof_mfas = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_mfas)


```




### chas
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_chas = geneEvents(S2S_mettl3ko_data_top, geneS = "chas", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_chas)

S2S_ythdcko_chas = geneEvents(S2S_ythdc1_data_top, geneS = "chas", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_chas)

Sgfp_ythdcgof_chas = geneEvents(Sgfp_ythdcgof_data_top, geneS = "chas", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_chas)


mettl3.ko_ythdc.gof_chas = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "chas", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_chas)


```

### Mob2
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Mob2 = geneEvents(S2S_mettl3ko_data_top, geneS = "Mob2", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Mob2)

S2S_ythdcko_Mob2 = geneEvents(S2S_ythdc1_data_top, geneS = "Mob2", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Mob2)

Sgfp_ythdcgof_Mob2 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Mob2", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Mob2)


mettl3.ko_ythdc.gof_Mob2 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Mob2", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Mob2)


```

Mob2_chr3L_-_11527886_11527960_11533411_11533643_11527454_11527816_11534389_11534478

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Mob2@MXE_stats
S2S_ythdcko_Mob2@MXE_stats
Sgfp_ythdcgof_Mob2@MXE_stats
mettl3.ko_ythdc.gof_Mob2@MXE_stats
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(8646, 8261, 5926,7365), Splicing_type = "MXE")
```

### slgA
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_slgA = geneEvents(S2S_mettl3ko_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_slgA)

S2S_ythdcko_slgA = geneEvents(S2S_ythdc1_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_slgA)

Sgfp_ythdcgof_slgA = geneEvents(Sgfp_ythdcgof_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_slgA)


mettl3.ko_ythdc.gof_slgA = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_slgA)


```

slgA:slgA_chrX_+_21379587_21379709_21376109_21376666_21379972_21380094

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_slgA@SE_stats
S2S_ythdcko_slgA@SE_stats
make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(6359, 6131), Splicing_type = "SE")
```




```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Aldh= geneEvents(S2S_mettl3ko_data_top, geneS = "Aldh-III", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Aldh)

S2S_ythdcko_Aldh = geneEvents(S2S_ythdc1_data_top, geneS = "Aldh-III", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Aldh)



Sgfp_ythdcgof_Aldh= geneEvents(Sgfp_ythdcgof_data_top, geneS = "Aldh-III", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Aldh)

mettl3.ko_ythdc.gof_Aldh = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Aldh-III", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Aldh)



```


### unc-13
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_unc13 = geneEvents(S2S_mettl3ko_data_top, geneS = "unc-13", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_unc13)

S2S_ythdcko_unc13 = geneEvents(S2S_ythdc1_data_top, geneS = "unc-13", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_unc13)

Sgfp_ythdcgof_unc13 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "unc-13", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_unc13)


mettl3.ko_ythdc.gof_unc13 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "unc-13", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_unc13)


```

unc-13_chr4_-_881842_881949_878850_878992_882960_883081 

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_unc13@SE_stats
S2S_ythdcko_unc13@SE_stats
Sgfp_ythdcgof_unc13@SE_stats
mettl3.ko_ythdc.gof_unc13@SE_stats
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(7955, 160, 181,203), Splicing_type = "SE")
```


### ABCD ABCD_chr4_191308_191386_PSI: ABCD	chr4	-	191308	191386	188424	188523	193544	193829

    
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_ABCD = geneEvents(S2S_mettl3ko_data_top, geneS = "ABCD", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_ABCD)

S2S_ythdcko_ABCD = geneEvents(S2S_ythdc1_data_top, geneS = "ABCD", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_ABCD)

Sgfp_ythdcgof_ABCD = geneEvents(Sgfp_ythdcgof_data_top, geneS = "ABCD", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_ABCD)


mettl3.ko_ythdc.gof_ABCD = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "ABCD", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_ABCD)


```


```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_ABCD@SE_stats
S2S_ythdcko_ABCD@SE_stats
Sgfp_ythdcgof_ABCD@SE_stats
mettl3.ko_ythdc.gof_ABCD@SE_stats
```



```{r}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(271, 222, 259, 272), Splicing_type = "SE")
```

### Liprin-alpha
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_LIP = geneEvents(S2S_mettl3ko_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_LIP)

S2S_ythdcko_LIP = geneEvents(S2S_ythdc1_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_LIP)

Sgfp_ythdcgof_LIP = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_LIP)


mettl3.ko_ythdc.gof_LIP = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_LIP)


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_LIP@A5SS_stats
mettl3.ko_ythdc.gof_LIP@A5SS_stats
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(59, NA, NA, 57), Splicing_type = "A5SS")
```


### Fmr1 chr3R:10,105,059-10,105,369 Eph
    
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Fmr1 = geneEvents(S2S_mettl3ko_data_top, geneS = "Fmr1", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Fmr1)

S2S_ythdcko_Fmr1 = geneEvents(S2S_ythdc1_data_top, geneS = "Fmr1", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Fmr1)

Sgfp_ythdcgof_Fmr1 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Fmr1", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Fmr1)


mettl3.ko_ythdc.gof_Fmr1 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Fmr1", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Fmr1)


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Eph = geneEvents(S2S_mettl3ko_data_top, geneS = "Eph", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Eph)

S2S_ythdcko_Eph = geneEvents(S2S_ythdc1_data_top, geneS = "Eph", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Eph)

Sgfp_ythdcgof_Eph = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Eph", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Eph)


mettl3.ko_ythdc.gof_Eph = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Eph", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Eph)


```




   

### Bsg Bsg	chr2L	+	8089356	8089736	8089490	8089736	8083419	8083521

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Bsg = geneEvents(S2S_mettl3ko_data_top, geneS = "Bsg", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Bsg)

S2S_ythdcko_Bsg = geneEvents(S2S_ythdc1_data_top, geneS = "Bsg", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Bsg)

Sgfp_ythdcgof_Bsg = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Bsg", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Bsg)


mettl3.ko_ythdc.gof_Bsg = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Bsg", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Bsg)


```


```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Bsg@A3SS_stats
S2S_ythdcko_Bsg@A3SS_stats
Sgfp_ythdcgof_Bsg@A3SS_stats
mettl3.ko_ythdc.gof_Bsg@A3SS_stats
```

Bsg	chr2L	+	8089356	8089736	8089490	8089736	8083419	8083521

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(4826, 4700, 4580, 4912), Splicing_type = "A3SS")
```


 
###aqz
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_aqz = geneEvents(S2S_mettl3ko_data_top, geneS = "aqz", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_aqz)

S2S_ythdcko_aqz = geneEvents(S2S_ythdc1_data_top, geneS = "aqz", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_aqz)

Sgfp_ythdcgof_aqz = geneEvents(Sgfp_ythdcgof_data_top, geneS = "aqz", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_aqz)


mettl3.ko_ythdc.gof_aqz = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "aqz", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_aqz)


```
chr3R	-	8819855	8819996	8820087	8820239	8817661	8819263	8820311	8820682

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_aqz@MXE_stats
S2S_ythdcko_aqz@MXE_stats
Sgfp_ythdcgof_aqz@MXE_stats
mettl3.ko_ythdc.gof_aqz@MXE_stats
```


```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(21448, 18937, 14732, 19350), Splicing_type = "MXE")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_sd = geneEvents(S2S_mettl3ko_data_top, geneS = "sd", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_sd)

S2S_ythdcko_sd = geneEvents(S2S_ythdc1_data_top, geneS = "sd", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_sd)

Sgfp_ythdcgof_sd = geneEvents(Sgfp_ythdcgof_data_top, geneS = "sd", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_sd)


mettl3.ko_ythdc.gof_sd = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "sd", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_sd)
```
chrX 15812326	15812438: sd	chrX	+	15813551	15813602	15812326	15812530	15813735	15814295

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_sd@SE_stats
S2S_ythdcko_sd@SE_stats
Sgfp_ythdcgof_sd@SE_stats
mettl3.ko_ythdc.gof_sd@SE_stats
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(3482, 15760, 14638, 17283), Splicing_type = "SE")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_mfas = geneEvents(S2S_mettl3ko_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_mfas)

S2S_ythdcko_mfas = geneEvents(S2S_ythdc1_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_mfas)

Sgfp_ythdcgof_mfas = geneEvents(Sgfp_ythdcgof_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_mfas)


mettl3.ko_ythdc.gof_mfas = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "mfas", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_mfas)
```


```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_mfas@A5SS_stats
S2S_ythdcko_mfas@SE_stats
Sgfp_ythdcgof_mfas@SE_stats
mettl3.ko_ythdc.gof_mfas@SE_stats
```

### slgA
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_slgA = geneEvents(S2S_mettl3ko_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_slgA)

S2S_ythdcko_slgA = geneEvents(S2S_ythdc1_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_slgA)

Sgfp_ythdcgof_slgA = geneEvents(Sgfp_ythdcgof_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_slgA)


mettl3.ko_ythdc.gof_slgA = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "slgA", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_slgA)
```

slgA	chrX	+	21379587	21379709	21376109	21376666	21379972	21380094
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_slgA@SE_stats
S2S_ythdcko_slgA@SE_stats
Sgfp_ythdcgof_slgA@SE_stats
mettl3.ko_ythdc.gof_slgA@SE_stats
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(6362, 6131), Splicing_type = "SE")
```
       
       Liprin-alpha, PlexA
```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_LPA = geneEvents(S2S_mettl3ko_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_LPA)

S2S_ythdcko_LPA = geneEvents(S2S_ythdc1_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_LPA)

Sgfp_ythdcgof_LPA = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_LPA)


mettl3.ko_ythdc.gof_LPA = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Liprin-alpha", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_LPA)
```

### CaMKII	chr4	-	1051258	1051378	1051342	1051378	1050454	1050942 CG4822

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_CaMKII = geneEvents(S2S_mettl3ko_data_top, geneS = "CaMKII", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_CaMKII)

S2S_ythdcko_CaMKII = geneEvents(S2S_ythdc1_data_top, geneS = "CaMKII", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_CaMKII)

Sgfp_ythdcgof_CaMKII = geneEvents(Sgfp_ythdcgof_data_top, geneS = "CaMKII", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_CaMKII)


mettl3.ko_ythdc.gof_CaMKII = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "CaMKII", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_CaMKII)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_CaMKII@A5SS_stats
S2S_ythdcko_CaMKII@A5SS_stats
Sgfp_ythdcgof_CaMKII@A5SS_stats
mettl3.ko_ythdc.gof_CaMKII@A5SS_stats
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(3861, 3643, NA,3683), Splicing_type = "A5SS")
```


### Acsl	chr2R	+	8672055	8672406 Acsl_chr2R_+_8672055_8672406_PSI
```{r}
S2S_mettl3ko_Hsc704 = geneEvents(S2S_mettl3ko_data_top, geneS = "Acsl", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Hsc704)

S2S_ythdcko_Hsc704 = geneEvents(S2S_ythdc1_data_top, geneS = "Acsl", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Hsc704)

Sgfp_ythdcgof_Hsc704 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Acsl", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Hsc704)


mettl3.ko_ythdc.gof_Hsc704 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Acsl", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Hsc704)
```



```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Hsc704@A5SS_stats
S2S_ythdcko_Hsc704@A5SS_stats
Sgfp_ythdcgof_Hsc704@A5SS_stats
mettl3.ko_ythdc.gof_Hsc704@A5SS_stats
```



```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(31360, 2317,2341, NA), Splicing_type = "A5SS")
```



### Pvf2	chr2R	+	8672055	8672406 Acsl_chr2R_+_8672055_8672406_PSI
```{r}
S2S_mettl3ko_Pvf2 = geneEvents(S2S_mettl3ko_data_top, geneS = "Pvf2", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Pvf2)

S2S_ythdcko_Pvf2 = geneEvents(S2S_ythdc1_data_top, geneS = "Pvf2", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Pvf2)

Sgfp_ythdcgof_Pvf2 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Pvf2", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Pvf2)


mettl3.ko_ythdc.gof_Pvf2 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Pvf2", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Pvf2)
```



```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Pvf2@SE_stats
S2S_ythdcko_Pvf2@SE_stats

```


chr2L	-	7083829	7083908
```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(8364, 7991), Splicing_type = "SE")
```



### sky	chr2R	+	8672055	8672406 Acsl_chr2R_+_8672055_8672406_PSI
```{r}
S2S_mettl3ko_sky = geneEvents(S2S_mettl3ko_data_top, geneS = "sky", fdr = 0.05, deltaPSI = 0.2)


S2S_ythdcko_sky = geneEvents(S2S_ythdc1_data_top, geneS = "sky", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_sky)

Sgfp_ythdcgof_sky = geneEvents(Sgfp_ythdcgof_data_top, geneS = "sky", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_sky)


mettl3.ko_ythdc.gof_sky = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "sky", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_sky)
```


### CG4822

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_CG4822 = geneEvents(S2S_mettl3ko_data_top, geneS = "CG4822", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_CG4822)

S2S_ythdcko_CG4822 = geneEvents(S2S_ythdc1_data_top, geneS = "CG4822", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_CG4822)

Sgfp_ythdcgof_CG4822 = geneEvents(Sgfp_ythdcgof_data_top, geneS = "CG4822", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_CG4822)


mettl3.ko_ythdc.gof_CG4822 = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "CG4822", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_CG4822)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_CG4822@SE_stats
S2S_ythdcko_CG4822@SE_stats
Sgfp_ythdcgof_CG4822@SE_stats
mettl3.ko_ythdc.gof_CG4822@SE_stats
```
chr2L	-	120510	120994: chr2L_-_120510_120994_CG4822
```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(NA, 8361, 8014,9148), Splicing_type = "SE")
```


### Nrg: Not significant

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Nrg = geneEvents(S2S_mettl3ko_data_top, geneS = "Nrg", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_Nrg)

S2S_ythdcko_Nrg = geneEvents(S2S_ythdc1_data_top, geneS = "Nrg", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_Nrg)

Sgfp_ythdcgof_Nrg = geneEvents(Sgfp_ythdcgof_data_top, geneS = "Nrg", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_Nrg)


mettl3.ko_ythdc.gof_Nrg = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "Nrg", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_Nrg)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_Nrg@A5SS_stats
S2S_ythdcko_Nrg@A5SS_stats
Sgfp_ythdcgof_Nrg@A5SS_stats
mettl3.ko_ythdc.gof_Nrg@A5SS_stats
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(17339, 15689, 15673,NA), Splicing_type = "A5SS")
```

### spi

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_spi = geneEvents(S2S_mettl3ko_data_top, geneS = "spi", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_spi)

S2S_ythdcko_spi = geneEvents(S2S_ythdc1_data_top, geneS = "spi", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_spi)

Sgfp_ythdcgof_spi = geneEvents(Sgfp_ythdcgof_data_top, geneS = "spi", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_spi)


mettl3.ko_ythdc.gof_spi = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "spi", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_spi)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_spi@MXE_stats
S2S_ythdcko_spi@MXE_stats
Sgfp_ythdcgof_spi@MXE_stats
mettl3.ko_ythdc.gof_spi@MXE_stats
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(5766, 5740, 4424,6113), Splicing_type = "MXE")
```


### CtBP

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_CtBPi = geneEvents(S2S_mettl3ko_data_top, geneS = "CtBP", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_CtBPi)

S2S_ythdcko_CtBP = geneEvents(S2S_ythdc1_data_top, geneS = "CtBP", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_CtBP)

Sgfp_ythdcgof_CtBP = geneEvents(Sgfp_ythdcgof_data_top, geneS = "CtBP", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_CtBP)


mettl3.ko_ythdc.gof_CtBP = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "CtBP", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_CtBP)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_spi@MXE_stats
S2S_ythdcko_spi@MXE_stats
Sgfp_ythdcgof_spi@MXE_stats
mettl3.ko_ythdc.gof_CtBP@MXE_stats
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(5766, 5740, 4424,NA), Splicing_type = "MXE")
```

### capt


```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_capt = geneEvents(S2S_mettl3ko_data_top, geneS = "capt", fdr = 0.05, deltaPSI = 0.2)

print(S2S_mettl3ko_capt)

S2S_ythdcko_capt = geneEvents(S2S_ythdc1_data_top, geneS = "capt", fdr = 0.05, deltaPSI = 0.2)
print(S2S_ythdcko_capt)

Sgfp_ythdcgof_capt = geneEvents(Sgfp_ythdcgof_data_top, geneS = "capt", fdr = 0.05, deltaPSI = 0.2)
print(Sgfp_ythdcgof_capt)


mettl3.ko_ythdc.gof_capt = geneEvents(mettl3.ko_ythdc.gof_data_top, geneS = "capt", fdr = 0.05, deltaPSI = 0.2)
print(mettl3.ko_ythdc.gof_capt)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2S_mettl3ko_capt@A5SS_stats
S2S_ythdcko_capt@A5SS_stats
Sgfp_ythdcgof_capt@A5SS_stats
mettl3.ko_ythdc.gof_capt@A5SS_stats
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
function_to_make_PSI_box_Mettl3ko_Ythdc1_KO(event_IDs = c(4761, 4488, 4378,NA), Splicing_type = "A5SS")
```




# Correlation plot between METTLE-KO and YTHDC1-KO Alternative splicing
```{r message=FALSE, warning=FALSE, echo=FALSE}

A3_S2_GFP_S2_DC1_GOF_filtered <- A3_SS2_GFP_S2_DC1_GOF  %>%
  dplyr::filter(FDR <= 0.05 & abs(IncLevelDifference) > 0.2)

A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF_filtered <- A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF %>%
  dplyr::filter(FDR <= 0.05 & abs(IncLevelDifference) > 0.2)

# Merge datasets by geneSymbol and exonStart_0base
merged_data <- merge(A3_S2_GFP_S2_DC1_GOF_filtered, A3_mettl3_KO_GFP_mettl3_KO_DC1_GOF_filtered, 
                     by.x = c("geneSymbol", "longExonStart_0base"), 
                     by.y = c("geneSymbol", "longExonStart_0base"),
                     suffixes = c("_MKO", "_DC1KO"))

# Add a column to highlight genes of interest
genes_of_interest <- c("Srrm234", "CG7971", "Sxl", "fl(2)d", "Aldh-III", "nrv3", "mob2", "chas", "cv-c", 
                       "mfas", "slgA", "Liprin-alpha", "Hsc70-4", "Pif1A", "Galphaq", "kuz", "C1GalTA", 
                       "Sema1b", "Gbeta13F", "Cka", "gro", "REPTOR", "Hph", "Wbp2", "PlexA", "lncRNA:Hsromega")

merged_data$Highlight <- ifelse(merged_data$geneSymbol %in% genes_of_interest, "interest", "Not interested")

# Calculate the correlation coefficient  & > 0.2
correlation <- cor(merged_data$IncLevelDifference_MKO, 
                   merged_data$IncLevelDifference_DC1KO, 
                   use = "complete.obs")

# Count genes for the legend
counts <- table(merged_data$Highlight)
total_shared_genes <- nrow(merged_data)

# Create the correlation plot &  > 0.2)
ggplot(merged_data, aes(x = IncLevelDifference_MKO, y = IncLevelDifference_DC1KO)) +
  geom_point(aes(color = Highlight), alpha = 0.8) +
  scale_color_manual(
    values = c("interest" = "red", 
               "Not interested" = "grey"), 
    labels = c(
      paste("interest -", counts["interest"]),
      paste("Not interested -", counts["Not interested"]),
      paste("Total Shared Genes:", total_shared_genes)
    )) +
  geom_text_repel(data = subset(merged_data, Highlight == "interest"), 
                  aes(label = geneSymbol), size = 3, box.padding = 0.35, point.padding = 0.5, max.overlaps = 10) +
  geom_smooth(method = "lm", color = "blue", linetype = "dashed") +
  labs(title = "A3SS for METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP and S2-YTHDC1-GoF/S2-GFP (FDR <= 0.05)",
       subtitle = paste("Correlation coefficient:", round(correlation, 2)),
       x = " (S2-YTHDC1-GoF/S2-GFP)", 
       y = " (YMETTL3-KO-YTHDC1-GoF/METTL3-KO GFP)") +
  xlim(-1, 1) + ylim(-1, 1) +
  theme_minimal() +
  theme(legend.title = element_blank(), plot.title = element_text(hjust = 0.5))

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
library(rtracklayer) # for reading in bed file
library(here)
library(dplyr)
library(ggplot2)
S2_1ug_peaks<- import(here("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.narrowPeak"))
S2_5ug_peaks<- import(here("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.narrowPeak"))

# Extend the regions by 50bp on both sides
S2_1ug_extended <- resize(S2_1ug_peaks, width = width(S2_1ug_peaks) + 100, fix = "center")
S2_5ug_extended <- resize(S2_5ug_peaks, width = width(S2_5ug_peaks) + 100, fix = "center")

# Export the extended peak files
export(S2_1ug_extended, here("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_summits_extended.bed"))
export(S2_5ug_extended, here("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_summits_extended.bed"))

S2_1ug_summit_peaks<- import(here("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_summits.bed"))
S2_5ug_summit_peaks<- import(here("~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_summits.bed"))

S2_1ug_overlap_S2_5ug_peaks<- subsetByOverlaps(S2_1ug_peaks, S2_5ug_peaks)
S2_1ug_overlap_S2_5ug_summit_peaks<- subsetByOverlaps(S2_1ug_extended, S2_5ug_extended)
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
S2_1ug_bw<- import(here("~/Dropbox/all_files/new_peak_files_java/new_peakcalls_macs3/S2_cell_1ug_5ug_IP_Input_bamcompare.bw"))
S2_5ug_bw<- import(here("~/Dropbox/all_files/new_peak_files_java/new_peakcalls_macs3/S2_cell_5ug_5ug_IP_Input_bamcompare.bw"))

# it is a GRanges object
S2_1ug_bw
S2_5ug_bw
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#BiocManager::install("EnrichedHeatmap")
library(EnrichedHeatmap)
# extend 1000 bp on each side and use 50bp bin
mat1<- normalizeToMatrix(S2_1ug_bw, S2_5ug_summit_peaks, value_column = "score",
                         extend= 1000, mean_mode = "w0", w=10)

mat2<- normalizeToMatrix(S2_5ug_bw, S2_5ug_summit_peaks, value_column = "score",
                         extend= 1000, mean_mode = "w0", w=10)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
dim(mat1)
dim(mat2)
#mat1[1:5, 1:200 ]
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
quantile(mat1, c(0.1,0.25,0.5,0.9,1))
quantile(mat2, c(0.1,0.25,0.5,0.9,1))
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
col_fun<- circlize::colorRamp2(c(0, 25), c("white", "red"))
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
ht_list =
  EnrichedHeatmap(mat1, pos_line = FALSE, column_title="S2_1ug", name = "S2_1ug", col=col_fun) +
  EnrichedHeatmap(mat2, pos_line = FALSE, column_title="S2_5ug", name = "S2_5ug", col=col_fun)

draw(ht_list, main_heatmap =2)
```



# Peak calling for S2 Head and Whole fly 
```{r message=FALSE, warning=FALSE, echo=FALSE}
df_S2_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE)
df_S2_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE)

#df_Head_FR_WT <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond1.tsv', header = TRUE)
#df_Head_FR_METTL3 <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/wt_vs_ime4_head_minus_cond2.tsv', header = TRUE)

#df_Wholefly_FR_WT <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond2.tsv', header = TRUE)

#df_Wholefly_FR_METTL3 <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/mettl3_vs_wt_minus_cond1.tsv', header = TRUE)

# Put all data frames into a named list
peak_files <- list(
  "S2 1ug" = df_S2_1ug,
  "S2 5ug" = df_S2_5ug)
  #"Head WT" = df_Head_FR_WT, 
  #"Head METTL3-KO" = df_Head_FR_METTL3, 
  #"Whole fly WT" = df_Wholefly_FR_WT, 
  #"Whole fly METTL3-KO" = df_Wholefly_FR_METTL3)


# Filter out rows with fold_enrichment < 2 for each dataframe
for (name in names(peak_files)) {
  df <- peak_files[[name]]
  
  # Check if 'fold_enrichment' column exists
  if ("fold_enrichment" %in% colnames(df)) {
    peak_files[[name]] <- df %>% dplyr::filter(fold_enrichment >= 2)
  } else {
    warning(paste("The column 'fold_enrichment' is not found in", name))
  }
}

# Count peaks for each sample and store them in a new data frame
peak_counts <- sapply(peak_files, nrow)
peak_counts_df <- data.frame(
  Sample = factor(names(peak_counts), levels = names(peak_counts)),
  Peaks = peak_counts)

# Plot the bar plot
ggplot(peak_counts_df, aes(x = Sample, y = Peaks, fill = Sample)) +
  geom_bar(stat = "identity", show.legend = FALSE) +
  geom_text(aes(label = Peaks), vjust = -0.3, color = "black", size = 3.5) +
  scale_fill_manual(values = c(
    'Whole fly METTL3-KO' = '#98df8a', 'Whole fly WT' = '#98df8a',
    'Head METTL3-KO' = '#ff7f0e', 'Head WT' = '#ff7f0e',
    'S2 1ug' = 'darkred', 'S2 5ug' = 'darkred')) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(title = "Number of S2 cells and differential peaks called in Head and Whole body", x = "Sample", y = "Number of Peaks") +
  theme(plot.title = element_text(hjust = 0.5))

```

```{r}
library(ggplot2)
library(dplyr)

# Read and filter the peak files
df_s2_cell_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A") 
df_s2_cell_1ug$Dataset <- "S2_1ug"

df_s2_cell_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "" & Location != "intron" & Location != "null", Location != "N/A")
df_s2_cell_5ug$Dataset <- "S2_5ug"

# Count the number of peaks in each dataset
peak_counts <- data.frame(
  Dataset = c("S2_1ug", "S2_5ug"),
  Peak_Count = c(nrow(df_s2_cell_1ug), nrow(df_s2_cell_5ug)))

# Create the lollipop plot
ggplot(peak_counts, aes(x = Dataset, y = Peak_Count)) +
  geom_segment(aes(x = Dataset, xend = Dataset, y = 0, yend = Peak_Count), color = "blue") +  # Lollipop stems 
  geom_point(aes(x = Dataset, y = Peak_Count), color = "red", size = 4) +  # Lollipop heads
  geom_text(aes(x = Dataset, y = Peak_Count, label = Peak_Count), vjust = -1, color = "black", size = 5) +  # Add peak counts as labels
  labs(title = "Number of Peaks in Each Dataset",
       x = "Dataset",
       y = "Number of Peaks") +
  theme_minimal() + coord_flip() +
  theme(axis.text.x = element_text(size = 12, face = "bold"),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5))

# Count the number of unique genes in each location for each dataset
gene_counts_1ug <- df_s2_cell_1ug %>%
  group_by(Location) %>%
  summarise(Gene_Count = n_distinct(Gene_ID)) %>%
  mutate(Dataset = "S2_1ug")

gene_counts_5ug <- df_s2_cell_5ug %>%
  group_by(Location) %>%
  summarise(Gene_Count = n_distinct(Gene_ID)) %>%
  mutate(Dataset = "S2_5ug")

# Combine the counts into a single data frame
gene_counts <- bind_rows(gene_counts_1ug, gene_counts_5ug)

# Filter for 5'UTR, CDS, and 3'UTR
gene_counts <- gene_counts %>%
  filter(Location %in% c("5'UTR", "CDS", "3'UTR"))

# Create the lollipop plot
ggplot(gene_counts, aes(x = Dataset, y = Gene_Count, color = Location)) +
  geom_segment(aes(x = Dataset, xend = Dataset, y = 0, yend = Gene_Count), color = "gray") +  # Lollipop stems
  geom_point(aes(x = Dataset, y = Gene_Count), size = 4) +  # Lollipop heads
  geom_text(aes(x = Dataset, y = Gene_Count, label = Gene_Count), vjust = -1, color = "black", size = 5) +  # Add gene counts as labels
  labs(title = "Number of Genes in Each Dataset by Location",
       x = "Dataset",
       y = "Number of Genes",
       color = "Location") +
  theme_minimal() + coord_flip() +
  theme(axis.text.x = element_text(size = 12, face = "bold"),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5),
        legend.position = "right") +
  scale_color_manual(values = c("5'UTR" = "lightgreen", "CDS" = "brown", "3'UTR" = "navy")) 
```


```{r}
library(readr)
library(dplyr)
library(ggplot2)

df_s2_cell_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE) %>%
  filter(Location != "" & Location != "intron" & Location != "null" & Location != "N/A") %>%
  mutate(Dataset = "S2_1ug")

df_s2_cell_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE) %>%
  filter(Location != "" & Location != "intron" & Location != "null" & Location != "N/A") %>%
  mutate(Dataset = "S2_5ug")

# Count peaks per chromosome
peak_counts_1ug <- df_s2_cell_1ug %>%
  group_by(chr) %>%
  summarise(Peak_Count = n()) %>%
  mutate(Dataset = "S2_1ug")

peak_counts_5ug <- df_s2_cell_5ug %>%
  group_by(chr) %>%
  summarise(Peak_Count = n()) %>%
  mutate(Dataset = "S2_5ug")

# Count genes per chromosome
gene_counts_1ug <- df_s2_cell_1ug %>%
  group_by(chr) %>%
  summarise(Gene_Count = n_distinct(Gene_ID)) %>%
  mutate(Dataset = "S2_1ug")

gene_counts_5ug <- df_s2_cell_5ug %>%
  group_by(chr) %>%
  summarise(Gene_Count = n_distinct(Gene_ID)) %>%
  mutate(Dataset = "S2_5ug")

# Merge peak and gene counts
peak_counts <- bind_rows(peak_counts_1ug, peak_counts_5ug)
gene_counts <- bind_rows(gene_counts_1ug, gene_counts_5ug)

# Combine both into a single dataset
plot_data <- left_join(peak_counts, gene_counts, by = c("chr", "Dataset"))

# Create grouped lollipop plot with gene count labels
ggplot(plot_data, aes(x = chr, y = Peak_Count, color = Dataset)) +
  geom_segment(aes(x = chr, xend = chr, y = 0, yend = Peak_Count), color = "gray", position = position_dodge(width = 0.5)) +  
  geom_point(aes(fill = Dataset), color = "black", size = 4, shape = 21, position = position_dodge(width = 0.5)) +  
  geom_text(aes(label = paste0(Peak_Count, " (", Gene_Count, " genes)")), 
            vjust = -1, position = position_dodge(width = 0.5), size = 5) +  
  labs(title = "Number of Peaks and Genes per Chromosome in S2 Cells",
       x = "Chromosome",
       y = "Number of Peaks") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +  
  theme_minimal() +
  coord_flip() +
  theme(axis.text.x = element_text(size = 12, face = "bold", angle = 45, hjust = 1),  
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 16, face = "bold", hjust = 0.5),
        legend.position = "right")



```

```{r message=FALSE, warning=FALSE, echo=FALSE}
peak_files <- list(
  'Whole fly METTL3-KO' = df_Wholefly_FR_METTL3,
  'Whole fly WT' = df_Wholefly_FR_WT,
  'Head METTL3-KO' = df_Head_FR_METTL3,
  'Head WT' = df_Head_FR_WT,
  'S2 1ug' = df_S2_1ug,
  'S2 5ug' = df_S2_5ug)

# Colors for pie chart
feature_colors <- c("5'UTR" = "#FF9999", "CDS" = "#66B2FF", "intron" = "#99FF99", "Others" = "#FFD700", "3'UTR" = "red")

# Define a function to create a pie chart for each dataset
plot_pie_chart <- function(df, name) {
  # Replace missing or undefined 'Location' values with 'Others'
  df$Location[is.na(df$Location) | df$Location == "N/A" | df$Location == "null"] <- "Others"
  
  # Calculate the frequency of each location
  location_freq <- table(df$Location)
  location_labels <- names(location_freq)
  
  # Calculate percentages and create labels
  location_pct <- round(location_freq / sum(location_freq) * 100, 1)
  labels <- paste(location_labels, " (", location_freq, ", ", location_pct, "%)", sep = "")
  
  # Set colors for each category
  colors <- feature_colors[location_labels]
  
  # Plot pie chart
  pie(location_freq, labels = labels, main = paste("Distribution of Genomic Features in", name), col = colors)
}

# Generate pie charts for each dataset
for (name in names(peak_files)) {
  plot_pie_chart(peak_files[[name]], name)
}

```


General distribution of peak length 
```{r message=FALSE, warning=FALSE, echo=FALSE}
for (name in names(peak_files)) {
  peak_files[[name]] <- peak_files[[name]] %>%
    mutate(width = end - start) 
}



ggplot(peak_files[['Whole fly WT']], aes(x = width)) +
  geom_histogram(binwidth = 20, color = "black", fill = "darkgreen") +
  labs(title = "Whole fly WT peak calling", x = "Peak Width (bp)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(peak_files[['Whole fly METTL3-KO']], aes(x = width)) +
  geom_histogram(binwidth = 20, color = "black", fill = "darkgreen") +
  labs(title = "Whole fly METTL3-KO peak calling", x = "Peak Width (bp)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(peak_files[['Head WT']], aes(x = width)) +
  geom_histogram(binwidth = 20, color = "black", fill = "darkgreen") +
  labs(title = "Head WT peak calling", x = "Peak Width (bp)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(peak_files[['Head METTL3-KO']], aes(x = width)) +
  geom_histogram(binwidth = 20, color = "black", fill = "darkgreen") +
  labs(title = "Head METTL3-KO peak calling", x = "Peak Width (bp)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(peak_files[['S2 1ug']], aes(x = width)) +
  geom_histogram(binwidth = 20, color = "black", fill = "darkgreen") +
  labs(title = "S2 1ug peak calling", x = "Peak Width (bp)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(peak_files[['S2 5ug']], aes(x = width)) +
  geom_histogram(binwidth = 20, color = "black", fill = "darkgreen") +
  labs(title = "S2 5ug peak calling", x = "Peak Width (bp)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))


```

```{r message=FALSE, warning=FALSE, echo=FALSE}
paired_files <- list(
  c('Whole fly WT', 'Whole fly METTL3-KO'),
  c('Head WT', 'Head METTL3-KO'),
  c('S2 1ug', 'S2 5ug')
)
plot_peak_distribution <- function(data, title) {
  ggplot(data, aes(x = width)) +
    geom_histogram(binwidth = 20, color = "black", fill = "darkgreen") +
    labs(title = title, x = "Peak Width (bp)", y = "Frequency") +
    theme_minimal() +
    theme(plot.title = element_text(hjust = 0.5))
}

compare_peak_lengths <- function(file1, file2) {
  widths1 <- file1$width
  widths2 <- file2$width
  
  t_test_result <- t.test(widths1, widths2)
  return(t_test_result)
}

# Example: Compare Whole fly WT and METTL3-KO
result <- compare_peak_lengths(peak_files[['Whole fly WT']], peak_files[['Whole fly METTL3-KO']])
print(result)


results <- list()
for (pair in paired_files) {
  file1 <- pair[1]
  file2 <- pair[2]
  
  # Plot
  plot1 <- plot_peak_distribution(peak_files[[file1]], paste0(file1, " peak calling"))
  plot2 <- plot_peak_distribution(peak_files[[file2]], paste0(file2, " peak calling"))
  
  print(plot1)
  print(plot2)
  
  # T-test
  test_result <- compare_peak_lengths(peak_files[[file1]], peak_files[[file2]])
  results[[paste0(file1, " vs ", file2)]] <- test_result
}

# Display test results
results
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
library(ggvenn)
peak_files_list <- list(
  "S2 1ug" = df_s2_cell_1ug$Gene_ID,
  "S2 5ug" = df_s2_cell_5ug$Gene_ID#, 
  #"Head WT" = df_Head_FR_WT$Gene_ID, 
  #"Head METTL3-KO" = df_Head_FR_METTL3$Gene_ID, 
  #"Whole fly WT" = df_Wholefly_FR_WT$Gene_ID, 
  #"Whole fly METTL3-KO" = df_Wholefly_FR_METTL3$Gene_ID
)

# Create ggvenn plot
ggvenn(
  peak_files_list, auto_scale = T, show_percentage = F,
  fill_color = c("red", "blue"),#, "green", "purple", "orange", "cyan"),
  stroke_size = 0.5,
  set_name_size = 5) + ggtitle("Venn diagram showing the unique and shared number of peaks called in S2 cells")
```

```{r}
# Load required libraries
library(dplyr)
library(clusterProfiler)
library(org.Dm.eg.db)  # Drosophila genome annotation

# Extract unique genes from each dataset
genes_1ug <- df_s2_cell_1ug %>% 
  dplyr::select(Gene_ID) %>% distinct()
genes_5ug <- df_s2_cell_5ug %>% 
  dplyr::select(Gene_ID) %>% distinct()

# Find shared genes (intersection)
shared_genes <- inner_join(genes_1ug, genes_5ug, by = "Gene_ID")

# Convert gene symbols to Entrez IDs for GO/KEGG analysis
entrez_ids <- bitr(shared_genes$Gene_ID, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Dm.eg.db)

# Perform GO Enrichment (Biological Process)
go_results <- enrichGO(
  gene = entrez_ids$ENTREZID, 
  OrgDb = org.Dm.eg.db, 
  keyType = "ENTREZID", 
  ont = "BP", 
  pAdjustMethod = "BH", 
  qvalueCutoff = 0.05)

# Plot GO results (Top 10)
dotplot(go_results, showCategory = 20, title = "GO for S2 shared peaks")

# View top GO and KEGG terms
head(go_results)



```

KEGG 
```{r}


# Convert SYMBOL to FlyBase Gene ID (FBgn)
flybase_ids <- bitr(
  shared_genes$Gene_ID, 
  fromType = "SYMBOL", 
  toType = "FLYBASE", 
  OrgDb = org.Dm.eg.db)

# Convert FlyBase (FBgn) to KEGG-compatible Dmel_CGXXXX format
kegg_ids <- bitr(
  flybase_ids$FLYBASE, 
  fromType = "FLYBASE", 
  toType = "FLYBASECG",  # KEGG Gene ID (Dmel_CGXXXX)
  OrgDb = org.Dm.eg.db)

# Remove NA values (unmapped genes)
kegg_ids <- na.omit(kegg_ids)

# Add "Dmel_" prefix to KEGG Gene IDs
kegg_ids$KEGG_ID <- paste0("Dmel_", kegg_ids$FLYBASECG)

# Run KEGG pathway enrichment with KEGG-compatible IDs
kegg_results <- enrichKEGG(
  gene = kegg_ids$KEGG_ID,  # Use modified IDs with "Dmel_" prefix
  organism = 'dme', 
  pAdjustMethod = "BH",
  pvalueCutoff = 0.05)

# Plot KEGG results (Top 20 pathways)
dotplot(kegg_results, showCategory = 30, title = "Enriched KEGG pathway for S2 shared peaks") 

# View top KEGG pathways
head(kegg_results)


```


## pheatmap of the genes
```{r}
library(dplyr)
library(tidyr)
library(pheatmap)

# Function to create a heatmap of the top 50 genes by fold enrichment
create_top_50_heatmap <- function(df, dataset_name) {
  
  # Filter for relevant genomic regions
  df_filtered <- df %>%
    dplyr::filter(Location %in% c("5'UTR", "CDS", "3'UTR")) %>%
    group_by(Gene_ID, Location) %>%
    summarize(fold_enrichment = sum(fold_enrichment, na.rm = TRUE), .groups = "drop")
  
  # Identify top 50 genes with highest total fold enrichment
  top_genes <- df_filtered %>%
    group_by(Gene_ID) %>%
    summarize(Total_Pileup = sum(fold_enrichment), .groups = "drop") %>%
    arrange(desc(Total_Pileup)) %>%
    slice_head(n = 50) %>%
    pull(Gene_ID)
  
  # Filter the dataset for only the top 50 genes
  df_top <- df_filtered %>% dplyr::filter(Gene_ID %in% top_genes)
  
  # Convert to wide format with genomic regions as columns
  df_wide <- df_top %>%
    pivot_wider(names_from = Location, values_from = fold_enrichment, values_fill = list(fold_enrichment = 0)) %>%
    dplyr::select(Gene_ID, `5'UTR`, CDS, `3'UTR`)
  
  # Convert to matrix for pheatmap
  gene_matrix <- as.matrix(df_wide[, -1])
  rownames(gene_matrix) <- df_wide$Gene_ID
  
  # Generate the heatmap
  pheatmap(gene_matrix, 
           cluster_rows = FALSE, 
           cluster_cols = FALSE,  
           main = paste("Top 50 Genes Heatmap of Fold Enrichment for", dataset_name),
           color = colorRampPalette(c("navy", "white", "firebrick3"))(100),
           scale = "row", 
           fontsize_row = 8,
           fontsize_col = 12,
           border_color = "grey60", 
           treeheight_row = 50)
}

# Apply the function to each dataset in peak_files
lapply(names(peak_files), function(dataset_name) {
  create_top_50_heatmap(peak_files[[dataset_name]], dataset_name)
})

```



```{r}

# Read and process peak files
df_s2_cell_1ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "", Location != "intron", Location != "null", Location != "N/A") %>%
  mutate(Dataset = "S2_1ug", Peak_Length = end - start)

df_s2_cell_5ug <- read.delim('~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_peaks.tsv', header = TRUE) %>%
  dplyr::filter(Location != "", Location != "intron", Location != "null", Location != "N/A") %>%
  mutate(Dataset = "S2_5ug", Peak_Length = end - start)

# Combine both datasets
peak_lengths <- bind_rows(df_s2_cell_1ug, df_s2_cell_5ug)

# Plot density
# Plot density (without fill)
ggplot(peak_lengths, aes(x = Peak_Length, color = Dataset)) +
  geom_density(size = 1) +  # Density plot with lines only
  scale_x_log10() +  # Log scale for better visualization
  labs(title = "Density Plot of m6A Peak length in S2 cells",
       x = "Peak Length (bp)",
       y = "Density") +
  theme_minimal() +
  theme(plot.title = element_text(size = 16, face = "bold", hjust = 0.5),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14, face = "bold")) +
  scale_color_manual(values = c("S2_1ug" = "blue", "S2_5ug" = "red"))

```


```{r message=FALSE, warning=FALSE, echo=FALSE}

library(Guitar)
stBedFiles <- list(
"~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_1ug_minus_summits.bed",
"~/Dropbox/all_files/new_peak_files_java/all_macs2_call_diff_star_mapped/s2_cell_5ug_minus_summits.bed") 
Txtd = ("~/Dropbox/dm6.sqlite")
txdb <- loadDb(Txtd)
m = GuitarPlot(txTxdb = txdb,
           stBedFiles = stBedFiles,
           headOrtail = F,
           enableCI = FALSE,
           mapFilterTranscript = FALSE,
           pltTxType =  c("mrna"),
           stGroupName = c("S2_1ug", "S2_5ug"))
m
```



# RNA binding motif analyses
```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_RNA_MKO_S2S_dn = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_METLL3-KO_resultMaps/pVal.dn.vs.bg.RNAmap.txt")
SE_RNA_MKO_S2S_up = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_METLL3-KO_resultMaps/pVal.up.vs.bg.RNAmap.txt")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_RNA_MKO_S2S_merged <- rbind(SE_RNA_MKO_S2S_dn, SE_RNA_MKO_S2S_up)
# Convert data from wide to long format
SE_RNA_MKO_S2S_merged_long <- melt(SE_RNA_MKO_S2S_merged, id.vars = "RBP", variable.name = "Region", value.name = "p_value")

# Compute -log10(p-value) for better visualization
SE_RNA_MKO_S2S_merged_long$log_p_value <- -log10(SE_RNA_MKO_S2S_merged_long$p_value)

top_rbps <- SE_RNA_MKO_S2S_merged_long %>%
  group_by(RBP) %>%
  summarise(min_p = min(p_value, na.rm = TRUE)) %>%
  arrange(min_p) %>%
  slice(1:10) %>%
  pull(RBP)


SE_RNA_MKO_S2S_merged_filtered <- SE_RNA_MKO_S2S_merged_long %>% filter(RBP %in% top_rbps)

# Create the dot plot
ggplot(SE_RNA_MKO_S2S_merged_filtered, aes(x = Region, y = RBP)) +
  geom_point(aes(size = log_p_value, color = p_value)) +
  scale_color_gradient(low = "red", high = "blue", name = "p-value") +
  scale_size(range = c(1, 10), name = "-log10(p-value)") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
  labs(title = " ",
       x = "Splicing Region",
       y = "RNA Binding Protein (RBP)")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_RNA_MKO_S2S_dn = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_METLL3-KO_resultMaps/pVal.dn.vs.bg.RNAmap.txt")
SE_RNA_MKO_S2S_up = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_METLL3-KO_resultMaps/pVal.up.vs.bg.RNAmap.txt")
A5_RNA_MKO_S2S_dn = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt")
A5_RNA_MKO_S2S_up = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt")
A3_RNA_MKO_S2S_dn = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt")
A3_RNA_MKO_S2S_up = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt")
MXE_RNA_MKO_S2S_dn = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt")
MXE_RNA_MKO_S2S_up = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt")
RI_RNA_MKO_S2S_dn = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt")
RI_RNA_MKO_S2S_up = read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_RNA_MKO_S2S_merged <- rbind(SE_RNA_MKO_S2S_dn, SE_RNA_MKO_S2S_up)
A5_RNA_MKO_S2S_merged <- rbind(A5_RNA_MKO_S2S_dn, A5_RNA_MKO_S2S_up)
A3_RNA_MKO_S2S_merged <- rbind(A3_RNA_MKO_S2S_dn, A3_RNA_MKO_S2S_up)
MXE_RNA_MKO_S2S_merged <- rbind(MXE_RNA_MKO_S2S_dn, MXE_RNA_MKO_S2S_up)
RI_RNA_MKO_S2S_merged <- rbind(RI_RNA_MKO_S2S_dn, RI_RNA_MKO_S2S_up)

# Extract unique RBPs from each dataset
SE_RBPs <- unique(SE_RNA_MKO_S2S_merged$RBP)
A5_RBPs <- unique(A5_RNA_MKO_S2S_merged$RBP)
A3_RBPs <- unique(A3_RNA_MKO_S2S_merged$RBP)
MXE_RBPs <- unique(MXE_RNA_MKO_S2S_merged$RBP)
RI_RBPs <- unique(RI_RNA_MKO_S2S_merged$RBP)

# Find common RBPs across all five datasets
common_RBPs <- Reduce(intersect, list(SE_RBPs, A5_RBPs, A3_RBPs, MXE_RBPs, RI_RBPs))

# Print common RBPs
print(common_RBPs)

```

```{r message=FALSE, warning=FALSE, echo=FALSE}

plot_dot <- function(data, event_type) {
  # Convert to long format
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Compute -log10(p-value)
  data_long$log_p_value <- -log10(data_long$p_value)
  
  top_rbps <- data_long %>%
  group_by(RBP) %>%
  summarise(min_p = min(p_value, na.rm = TRUE)) %>%
  arrange(min_p) %>%
  slice(1:10) %>%
  pull(RBP)
    
  data_filtered <- data_long %>% dplyr::filter(RBP %in% top_rbps)
  
  
  # Generate dot plot
  ggplot(data_filtered, aes(x = Region, y = RBP)) +
    geom_point(aes(size = log_p_value, color = p_value)) +
    scale_color_gradient(low = "red", high = "blue", name = "p-value") +
    scale_size(range = c(1, 10), name = "-log10(p-value)") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    labs(title = paste("Common RBPs in", event_type, "Splicing Events"),
         x = "Splicing Region",
         y = "RNA Binding Protein (RBP)")
}

# Generate plots for each splicing event
plot_dot(SE_RNA_MKO_S2S_merged, "SE")
plot_dot(A5_RNA_MKO_S2S_merged, "A5SS")
plot_dot(A3_RNA_MKO_S2S_merged, "A3SS")
plot_dot(MXE_RNA_MKO_S2S_merged, "MXE")
plot_dot(RI_RNA_MKO_S2S_merged, "RI")

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
# Step 1: Read and merge datasets by splicing event type
SE_RNA_MKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_METLL3-KO_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/SE_METLL3-KO_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

A5_RNA_MKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A5SS_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

A3_RNA_MKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/A3SS_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

MXE_RNA_MKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/MXE_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

RI_RNA_MKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI_METTL3-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_mettl3KO_splicing/RI_METTL3-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

# Step 2: Filter RBPs with at least one p-value < 0.05
filter_significant_rbps <- function(data) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Group by RBP and Region, taking the lowest p-value per group
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop")

  significant_rbps <- data_long %>%
    group_by(RBP) %>%
    dplyr::filter(any(p_value < 0.05)) %>%  # Keep only RBPs with at least one significant p-value
    distinct(RBP) %>%
    pull(RBP)
  
  return(significant_rbps)
}

# Find RBPs with p-value < 0.05 in at least one region
SE_RBPs <- filter_significant_rbps(SE_RNA_MKO_S2S_merged)
A5_RBPs <- filter_significant_rbps(A5_RNA_MKO_S2S_merged)
A3_RBPs <- filter_significant_rbps(A3_RNA_MKO_S2S_merged)
MXE_RBPs <- filter_significant_rbps(MXE_RNA_MKO_S2S_merged)
RI_RBPs <- filter_significant_rbps(RI_RNA_MKO_S2S_merged)

# Step 3: Identify the 50 shared significant RBPs across all datasets
get_shared_rbps <- function(...) {
  rbp_lists <- list(...)
  common_rbps <- Reduce(intersect, rbp_lists)  # Find common RBPs
  
  if (length(common_rbps) > 50) {
    return(common_rbps[1:50])  # Return the first 50 RBPs
  } else {
    return(common_rbps)  # Return all RBPs if there are fewer than 50
  }
}

shared_50_rbps <- get_shared_rbps(SE_RBPs, A5_RBPs, A3_RBPs, MXE_RBPs, RI_RBPs)

# Step 4: Define function for dot plot
plot_dot <- function(data, event_type, shared_rbps) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Group by RBP and Region, selecting the minimum p-value to avoid duplicates
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop")
  
  # Calculate -log10(p-value)
  data_long$log_p_value <- -log10(data_long$p_value)
  
  # Filter out small dots (e.g., p-value > 0.05 means log_p_value < 1.3)
  data_filtered <- data_long %>%
    dplyr:: filter(RBP %in% shared_rbps & log_p_value >= 1.3)  # Remove low significance
  
  # Plot
  ggplot(data_filtered, aes(x = Region, y = RBP)) +
    geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots
    scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                          name = "log_adjpval") +  
    scale_size(range = c(3, 10), name = "-log10(p-value)") +  
    theme_minimal() +
    theme(
      panel.grid.major = element_line(color = "grey85"),  
      panel.grid.minor = element_line(color = "grey95"),  
      panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
      axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
      axis.text.y = element_text(size = 10)
    ) +
    labs(title = paste("Shared 50 RBPs in METTL3-KO/S2S", event_type, "Splicing Events"),
         x = "Splicing Region",
         y = "RNA Binding Protein (RBP)")
}


# Step 5: Generate plots for each splicing event
plot_dot(SE_RNA_MKO_S2S_merged, "SE", shared_50_rbps)
plot_dot(A5_RNA_MKO_S2S_merged, "A5SS", shared_50_rbps)
plot_dot(A3_RNA_MKO_S2S_merged, "A3SS", shared_50_rbps)
plot_dot(MXE_RNA_MKO_S2S_merged, "MXE", shared_50_rbps)
plot_dot(RI_RNA_MKO_S2S_merged, "RI", shared_50_rbps)


```

merge Mettl-KO/S2S


```{r}
# Step 1: Add Splicing_Type column to each dataset
SE_RNA_MKO_S2S_merged$Splicing_Type <- "SE"
A5_RNA_MKO_S2S_merged$Splicing_Type <- "A5SS"
A3_RNA_MKO_S2S_merged$Splicing_Type <- "A3SS"
MXE_RNA_MKO_S2S_merged$Splicing_Type <- "MXE"
RI_RNA_MKO_S2S_merged$Splicing_Type <- "RI"

# Step 2: Combine all datasets into one
all_data_SGFP_S2S <- bind_rows(
  SE_RNA_MKO_S2S_merged, 
  A5_RNA_MKO_S2S_merged, 
  A3_RNA_MKO_S2S_merged, 
  MXE_RNA_MKO_S2S_merged, 
  RI_RNA_MKO_S2S_merged)

# Step 3: Convert to long format
data_long <- melt(all_data_SGFP_S2S, id.vars = c("RBP", "Splicing_Type"), variable.name = "Region", value.name = "p_value")

# Step 4: Keep only the most significant (smallest) p-value for each RBP-Region-Splicing_Type
data_long <- data_long %>%
  group_by(RBP, Region, Splicing_Type) %>%
  summarize(p_value = min(p_value), .groups = "drop")

# Step 5: Calculate -log10(p-value)
data_long$log_p_value <- -log10(data_long$p_value)

# Step 6: Filter to keep only shared RBPs and significant values
data_filtered <- data_long %>%
  dplyr::filter(RBP %in% shared_50_rbps & log_p_value >= 1.3)  # Remove low significance

# Step 7: Create a merged dot plot
ggplot(data_filtered, aes(x = Region, y = RBP)) +
  geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots
  scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                        name = "log_adjpval") +  
  scale_size(range = c(3, 10), name = "-log10(p-value)") +  
  theme_minimal() +
  theme(
    panel.grid.major = element_line(color = "grey85"),  
    panel.grid.minor = element_line(color = "grey95"),  
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
    axis.text.y = element_text(size = 10)
  ) +
  labs(title = "Shared 50 RBPs in Mettl3-KO/S2S Splicing Events",
       x = "Splicing Region",
       y = "RNA Binding Protein (RBP)") +
  facet_wrap(~ Splicing_Type, scales = "free_x", nrow = 1)  #  One merged plot with separate splicing event panels

```


RNA Binding YTHDC1-KO
```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_RNA_YKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE_YTHDC1-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/SE_YTHDC1-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

A5_RNA_YKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS_YTHDC1-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A5SS_YTHDC1-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

A3_RNA_YKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS_YTHDC1-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/A3SS_YTHDC1-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

MXE_RNA_YKO_S2S_merged <- rbind(
    read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE_YTHDC1-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/MXE_YTHDC1-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

RI_RNA_YKO_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI_YTHDC1-KO_S2S_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2S_DC1_KO/RI_YTHDC1-KO_S2S_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

# Step 2: Filter RBPs with at least one p-value < 0.05
filter_significant_rbps <- function(data) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Group by RBP and Region, taking the lowest p-value per group
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop")

  significant_rbps <- data_long %>%
    group_by(RBP) %>%
    dplyr::filter(any(p_value < 0.05)) %>%  # Keep only RBPs with at least one significant p-value
    distinct(RBP) %>%
    pull(RBP)
  
  return(significant_rbps)
}


# Find RBPs with p-value < 0.05 in at least one region
SE_RBPs <- filter_significant_rbps(SE_RNA_YKO_S2S_merged)
A5_RBPs <- filter_significant_rbps(A5_RNA_YKO_S2S_merged)
A3_RBPs <- filter_significant_rbps(A3_RNA_YKO_S2S_merged)
MXE_RBPs <- filter_significant_rbps(MXE_RNA_YKO_S2S_merged)
RI_RBPs <- filter_significant_rbps(RI_RNA_YKO_S2S_merged)

# Step 3: Identify the 50 shared significant RBPs across all datasets
get_shared_rbps <- function(...) {
  rbp_lists <- list(...)
  common_rbps <- Reduce(intersect, rbp_lists)  # Find common RBPs
  
  if (length(common_rbps) > 50) {
    return(common_rbps[1:50])  # Return the first 50 RBPs
  } else {
    return(common_rbps)  # Return all RBPs if there are fewer than 50
  }
}

shared_50_rbps <- get_shared_rbps(SE_RBPs, A5_RBPs, A3_RBPs, MXE_RBPs, RI_RBPs)

# Step 4: Define function for dot plot
plot_dot <- function(data, event_type, shared_rbps) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Group by RBP and Region, selecting the minimum p-value to avoid duplicates
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop")
  
  # Calculate -log10(p-value)
  data_long$log_p_value <- -log10(data_long$p_value)
  
  # Filter out small dots (e.g., p-value > 0.05 means log_p_value < 1.3)
  data_filtered <- data_long %>%
    filter(RBP %in% shared_rbps & log_p_value >= 1.3)  # Remove low significance
  
  # Plot
  ggplot(data_filtered, aes(x = Region, y = RBP)) +
    geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots
    scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                          name = "log_adjpval") +  
    scale_size(range = c(3, 10), name = "-log10(p-value)") +  
    theme_minimal() +
    theme(
      panel.grid.major = element_line(color = "grey85"),  
      panel.grid.minor = element_line(color = "grey95"),  
      panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
      axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
      axis.text.y = element_text(size = 10)
    ) +
    labs(title = paste("Shared 50 RBPs in Ythdc1-KO/S2S", event_type, "Splicing Events"),
         x = "Splicing Region",
         y = "RNA Binding Protein (RBP)")
}


# Step 5: Generate plots for each splicing event YTHDC1-KO_S2S_SE_RNA RBP_50_Ythdc1_S2S_RI
plot_dot(SE_RNA_YKO_S2S_merged, "SE", shared_50_rbps)
plot_dot(A5_RNA_YKO_S2S_merged, "A5SS", shared_50_rbps)
plot_dot(A3_RNA_YKO_S2S_merged, "A3SS", shared_50_rbps)
plot_dot(MXE_RNA_YKO_S2S_merged, "MXE", shared_50_rbps)
plot_dot(RI_RNA_YKO_S2S_merged, "RI", shared_50_rbps)
```


merge Ythdc1-KO/S2S


```{r}
# Step 1: Add Splicing_Type column to each dataset
SE_RNA_YKO_S2S_merged$Splicing_Type <- "SE"
A5_RNA_YKO_S2S_merged$Splicing_Type <- "A5SS"
A3_RNA_YKO_S2S_merged$Splicing_Type <- "A3SS"
MXE_RNA_YKO_S2S_merged$Splicing_Type <- "MXE"
RI_RNA_YKO_S2S_merged$Splicing_Type <- "RI"

# Step 2: Combine all datasets into one
all_data_SGFP_S2S <- bind_rows(
  SE_RNA_YKO_S2S_merged, 
  A5_RNA_YKO_S2S_merged, 
  A3_RNA_YKO_S2S_merged, 
  MXE_RNA_YKO_S2S_merged, 
  RI_RNA_YKO_S2S_merged)

# Step 3: Convert to long format
data_long <- melt(all_data_SGFP_S2S, id.vars = c("RBP", "Splicing_Type"), variable.name = "Region", value.name = "p_value")

# Step 4: Keep only the most significant (smallest) p-value for each RBP-Region-Splicing_Type
data_long <- data_long %>%
  group_by(RBP, Region, Splicing_Type) %>%
  summarize(p_value = min(p_value), .groups = "drop")

# Step 5: Calculate -log10(p-value)
data_long$log_p_value <- -log10(data_long$p_value)

# Step 6: Filter to keep only shared RBPs and significant values
data_filtered <- data_long %>%
  dplyr::filter(RBP %in% shared_50_rbps & log_p_value >= 1.3)  # Remove low significance

# Step 7: Create a merged dot plot
ggplot(data_filtered, aes(x = Region, y = RBP)) +
  geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots Merge_RBP_Mettl3_KO_S2S
  scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                        name = "log_adjpval") +  
  scale_size(range = c(3, 10), name = "-log10(p-value)") +  
  theme_minimal() +
  theme(
    panel.grid.major = element_line(color = "grey85"),  
    panel.grid.minor = element_line(color = "grey95"),  
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
    axis.text.y = element_text(size = 10)
  ) +
  labs(title = "Shared 50 RBPs in Ythdc1-KO/S2S Splicing Events",
       x = "Splicing Region",
       y = "RNA Binding Protein (RBP)") +
  facet_wrap(~ Splicing_Type, scales = "free_x", nrow = 1)  # One merged plot with separate splicing event panels

```

# RNA bidding proteins for S2_DC1_GOF_mettl3_KO_DC1_GOF
```{r message=FALSE, warning=FALSE, echo=FALSE}
SE_RNA_SGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE_S2S-GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/SE_S2S-GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

A5_RNA_SGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS_S2S-GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A5SS_S2S-GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

A3_RNA_SGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS_S2S-GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/A3SS_S2S-GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

MXE_RNA_SGFP_S2S_merged <- rbind(
    read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE_S2S-GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/MXE_S2S-GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

RI_RNA_SGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI_S2S-GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/S2_GFP_S2_DC1_GOF/RI_S2S-GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt"))

# Step 2: Filter RBPs with at least one p-value < 0.05
filter_significant_rbps <- function(data) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Group by RBP and Region, taking the lowest p-value per group
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop")

  significant_rbps <- data_long %>%
    group_by(RBP) %>%
    dplyr::filter(any(p_value < 0.05)) %>%  # Keep only RBPs with at least one significant p-value RBP_50_Ythdc1_S2S_MXE
    distinct(RBP) %>%
    pull(RBP)
  
  return(significant_rbps)
}

# Find RBPs with p-value < 0.05 in at least one region
SE_RBPs <- filter_significant_rbps(SE_RNA_SGFP_S2S_merged)
A5_RBPs <- filter_significant_rbps(A5_RNA_SGFP_S2S_merged)
A3_RBPs <- filter_significant_rbps(A3_RNA_SGFP_S2S_merged)
MXE_RBPs <- filter_significant_rbps(MXE_RNA_SGFP_S2S_merged)
RI_RBPs <- filter_significant_rbps(RI_RNA_SGFP_S2S_merged)

# Step 3: Identify the 50 shared significant RBPs across all datasets
get_shared_rbps <- function(...) {
  rbp_lists <- list(...)
  common_rbps <- Reduce(intersect, rbp_lists)  # Find common RBPs
  
  if (length(common_rbps) > 50) {
    return(common_rbps[1:50])  # Return the first 50 RBPs
  } else {
    return(common_rbps)  # Return all RBPs if there are fewer than 50
  }
}

shared_50_rbps <- get_shared_rbps(SE_RBPs, A5_RBPs, A3_RBPs, MXE_RBPs, RI_RBPs)

# Step 4: Define function for dot plot
plot_dot <- function(data, event_type, shared_rbps) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Group by RBP and Region, selecting the minimum p-value to avoid duplicates
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop")
  
  # Calculate -log10(p-value)
  data_long$log_p_value <- -log10(data_long$p_value)
  
  # Filter out small dots (e.g., p-value > 0.05 means log_p_value < 1.3)
  data_filtered <- data_long %>%
    dplyr::filter(RBP %in% shared_rbps & log_p_value >= 1.3)  # Remove low significance
  
  # Plot
  ggplot(data_filtered, aes(x = Region, y = RBP)) +
    geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots
    scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                          name = "log_adjpval") +  
    scale_size(range = c(3, 10), name = "-log10(p-value)") +  
    theme_minimal() +
    theme(
      panel.grid.major = element_line(color = "grey85"),  
      panel.grid.minor = element_line(color = "grey95"),  
      panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
      axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
      axis.text.y = element_text(size = 10)
    ) +
    labs(title = paste("Shared 50 RBPs in Shared 50 RBPs in S2-YTHDC1-GoF/S2S", event_type, "Splicing Events"),
         x = "Splicing Region",
         y = "RNA Binding Protein (RBP)")
}


# Step 5: Generate plots for each splicing event 1500 RNA_BP_S2GFP_S2S_A5 merge_MKO_RBP
plot_dot(SE_RNA_SGFP_S2S_merged, "SE", shared_50_rbps)
plot_dot(A5_RNA_SGFP_S2S_merged, "A5SS", shared_50_rbps)
plot_dot(A3_RNA_SGFP_S2S_merged, "A3SS", shared_50_rbps)
plot_dot(MXE_RNA_SGFP_S2S_merged, "MXE", shared_50_rbps)
plot_dot(RI_RNA_SGFP_S2S_merged, "RI", shared_50_rbps)
```


merged

```{r}
# Step 1: Add Splicing_Type column to each dataset
SE_RNA_SGFP_S2S_merged$Splicing_Type <- "SE"
A5_RNA_SGFP_S2S_merged$Splicing_Type <- "A5SS"
A3_RNA_SGFP_S2S_merged$Splicing_Type <- "A3SS"
MXE_RNA_SGFP_S2S_merged$Splicing_Type <- "MXE"
RI_RNA_SGFP_S2S_merged$Splicing_Type <- "RI"

# Step 2: Combine all datasets into one
all_data_SGFP_S2S <- bind_rows(
  SE_RNA_SGFP_S2S_merged, 
  A5_RNA_SGFP_S2S_merged, 
  A3_RNA_SGFP_S2S_merged, 
  MXE_RNA_SGFP_S2S_merged, 
  RI_RNA_SGFP_S2S_merged)

# Step 3: Convert to long format
data_long <- melt(all_data_SGFP_S2S, id.vars = c("RBP", "Splicing_Type"), variable.name = "Region", value.name = "p_value")

# Step 4: Keep only the most significant (smallest) p-value for each RBP-Region-Splicing_Type
data_long <- data_long %>%
  group_by(RBP, Region, Splicing_Type) %>%
  summarize(p_value = min(p_value), .groups = "drop")

# Step 5: Calculate -log10(p-value)
data_long$log_p_value <- -log10(data_long$p_value)

# Step 6: Filter to keep only shared RBPs and significant values
data_filtered <- data_long %>%
  dplyr::filter(RBP %in% shared_50_rbps & log_p_value >= 1.3)  # Remove low significance

# Step 7: Create a merged dot plot
ggplot(data_filtered, aes(x = Region, y = RBP)) +
  geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots merge_MKO_RBP
  scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                        name = "log_adjpval") +  
  scale_size(range = c(3, 10), name = "-log10(p-value)") +  
  theme_minimal() +
  theme(panel.grid.major = element_line(color = "grey85"),  
    panel.grid.minor = element_line(color = "grey95"),  
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
    axis.text.y = element_text(size = 10)) +
  labs(title = "Shared 50 RBPs in S2-YTHDC1-GoF/S2-GFP Splicing Events",
       x = "Splicing Region",
       y = "RNA Binding Protein (RBP)") +
  facet_wrap(~ Splicing_Type, scales = "free_x", nrow = 1)  # One merged plot with separate splicing event panels

```

RNA binding motif analyses for METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP
```{r message=FALSE, warning=FALSE, echo=FALSE}

# Step 1: Load and Merge Data
SE_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
)

A5_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
)

A3_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
)

MXE_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
)

RI_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
)

# Step 2: Filter RBPs with at least one significant p-value and remove non-significant regions
filter_significant_rbps_and_regions <- function(data) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Keep only regions with significant p-values
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop") %>%
    filter(p_value < 0.05)  # Only keep significant regions
  
  # Get the RBPs that have at least one significant region
  significant_rbps <- unique(data_long$RBP)
  
  # Filter original dataset to include only significant RBPs and regions
  filtered_data <- data %>%
    filter(RBP %in% significant_rbps)  # Keep only significant RBPs
  
  return(filtered_data)
}

# Apply filtering to all datasets
SE_RNA_MGFP_S2S_merged <- filter_significant_rbps_and_regions(SE_RNA_MGFP_S2S_merged)
A5_RNA_MGFP_S2S_merged <- filter_significant_rbps_and_regions(A5_RNA_MGFP_S2S_merged)
A3_RNA_MGFP_S2S_merged <- filter_significant_rbps_and_regions(A3_RNA_MGFP_S2S_merged)
MXE_RNA_MGFP_S2S_merged <- filter_significant_rbps_and_regions(MXE_RNA_MGFP_S2S_merged)
RI_RNA_MGFP_S2S_merged <- filter_significant_rbps_and_regions(RI_RNA_MGFP_S2S_merged)

# Step 3: Identify the 50 shared significant RBPs across all datasets
get_shared_rbps <- function(...) {
  rbp_lists <- list(...)
  common_rbps <- Reduce(intersect, rbp_lists)  # Find common RBPs
  
  if (length(common_rbps) > 50) {
    return(common_rbps[1:50])  # Return the first 50 RBPs
  } else {
    return(common_rbps)  # Return all RBPs if there are fewer than 50
  }
}

# Extract RBPs from filtered datasets
SE_RBPs <- unique(SE_RNA_MGFP_S2S_merged$RBP)
A5_RBPs <- unique(A5_RNA_MGFP_S2S_merged$RBP)
A3_RBPs <- unique(A3_RNA_MGFP_S2S_merged$RBP)
MXE_RBPs <- unique(MXE_RNA_MGFP_S2S_merged$RBP)
RI_RBPs <- unique(RI_RNA_MGFP_S2S_merged$RBP)

# Get the 50 most shared RBPs
shared_50_rbps <- get_shared_rbps(SE_RBPs, A5_RBPs, A3_RBPs, MXE_RBPs, RI_RBPs)

# Step 4: Define function for dot plot
plot_dot <- function(data, event_type, shared_rbps) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Group by RBP and Region, selecting the minimum p-value to avoid duplicates
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop")
  
  # Calculate -log10(p-value)
  data_long$log_p_value <- -log10(data_long$p_value)
  
  # Filter out small dots (e.g., p-value > 0.05 means log_p_value < 1.3)
  data_filtered <- data_long %>%
    filter(RBP %in% shared_rbps & log_p_value >= 1.3)  # Remove low significance
  
  # Plot
  ggplot(data_filtered, aes(x = Region, y = RBP)) +
    geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots
    scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                          name = "log_adjpval") +  
    scale_size(range = c(3, 10), name = "-log10(p-value)") +  
    theme_minimal() +
    theme(
      panel.grid.major = element_line(color = "grey85"),  
      panel.grid.minor = element_line(color = "grey95"),  
      panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
      axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
      axis.text.y = element_text(size = 10)
    ) +
    labs(title = paste("Shared 50 RBPs in METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP", event_type, "Splicing Events"),
         x = "Splicing Region",
         y = "RNA Binding Protein (RBP)")
}

# Step 5: Generate plots for each splicing event RNA_BP_S2GFP_S2S_A5
plot_dot(SE_RNA_MGFP_S2S_merged, "SE", shared_50_rbps)
plot_dot(A5_RNA_MGFP_S2S_merged, "A5SS", shared_50_rbps)
plot_dot(A3_RNA_MGFP_S2S_merged, "A3SS", shared_50_rbps)
plot_dot(MXE_RNA_MGFP_S2S_merged, "MXE", shared_50_rbps)
plot_dot(RI_RNA_MGFP_S2S_merged, "RI", shared_50_rbps)


```

```{r message=FALSE, warning=FALSE, echo=FALSE}


# Function to filter RBPs with at least one significant p-value and remove non-significant regions
filter_significant_rbps_and_regions <- function(data) {
  data_long <- melt(data, id.vars = "RBP", variable.name = "Region", value.name = "p_value")
  
  # Keep only regions with significant p-values
  data_long <- data_long %>%
    group_by(RBP, Region) %>%
    summarize(p_value = min(p_value), .groups = "drop") %>%
    dplyr::filter(p_value < 0.05)  # Only keep significant regions
  
  # Get the RBPs that have at least one significant region
  significant_rbps <- unique(data_long$RBP)
  
  # Filter original dataset to include only significant RBPs and regions
  filtered_data <- data %>%
    dplyr::filter(RBP %in% significant_rbps)  # Keep only significant RBPs
  
  return(filtered_data)
}

# Load and merge datasets
SE_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/SE_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
) %>% mutate(Splicing_Type = "SE")

A5_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A5SS_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
) %>% mutate(Splicing_Type = "A5SS")

A3_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/A3SS_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
) %>% mutate(Splicing_Type = "A3SS")

MXE_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/MXE_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
) %>% mutate(Splicing_Type = "MXE")

RI_RNA_MGFP_S2S_merged <- rbind(
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI_METTL3-KO_GFP_resultMaps/pVal.dn.vs.bg.RNAmap.txt"),
  read.delim("~/Dropbox/all_files/S2_m6A/Significant_Splicng_S2S_DC1_METTL3-KO/Alt_Spliced_files/mettl3_KO_GFP_mettl3_KO_DC1_GOF/RI_METTL3-KO_GFP_resultMaps/pVal.up.vs.bg.RNAmap.txt")
) %>% mutate(Splicing_Type = "RI")

# Apply filtering to all datasets
datasets <- list(SE_RNA_MGFP_S2S_merged, A5_RNA_MGFP_S2S_merged, A3_RNA_MGFP_S2S_merged, MXE_RNA_MGFP_S2S_merged, RI_RNA_MGFP_S2S_merged)
filtered_datasets <- lapply(datasets, filter_significant_rbps_and_regions)

# Combine all datasets into one
all_data <- bind_rows(filtered_datasets)

# Rename SE sample from "R1-98" (if necessary)
all_data$RBP <- gsub("R1-R9", "SE", all_data$RBP)

# Identify the 50 shared significant RBPs across all splicing types
shared_rbps <- Reduce(intersect, lapply(filtered_datasets, function(x) unique(x$RBP)))
shared_50_rbps <- if (length(shared_rbps) > 50) shared_rbps[1:50] else shared_rbps


# Convert data to long format for plotting
data_long <- melt(all_data, id.vars = c("RBP", "Splicing_Type"), variable.name = "Region", value.name = "p_value")

# Keep only the most significant (smallest) p-value for each RBP-Region-Splicing Type
data_long <- data_long %>%
  group_by(RBP, Region, Splicing_Type) %>%
  summarize(p_value = min(p_value), .groups = "drop")  # Select smallest p-value

# Calculate -log10(p-value)
data_long$log_p_value <- -log10(data_long$p_value)

# Filter to keep only shared RBPs and significant values
data_filtered <- data_long %>%
  filter(RBP %in% shared_50_rbps & log_p_value >= 1.3)  # Remove low significance

# Plot with separate panels for each splicing type Merge_RBP_S2_GFP
ggplot(data_filtered, aes(x = Region, y = RBP)) +
  geom_point(aes(size = log_p_value, color = log_p_value), shape = 16) +  # Filled dots
  scale_color_gradientn(colors = c("#91bfdb", "#4575b4", "#fc8d59", "#fee08b", "#d73027"), 
                        name = "log_adjpval") +  
  scale_size(range = c(3, 10), name = "-log10(p-value)") +  
  theme_minimal() +
  theme(
    panel.grid.major = element_line(color = "grey85"),  
    panel.grid.minor = element_line(color = "grey95"),  
    panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5), 
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10),
    axis.text.y = element_text(size = 10)
  ) +
  labs(title = "Shared 50 RBPs in METTL3-KO-YTHDC1-GoF/METTL3-KO-GFP Splicing Events",
       x = "Splicing Region",
       y = "RNA Binding Protein (RBP)") +
  facet_wrap(~ Splicing_Type, scales = "free_x", nrow = 1) 

```

```{r message=FALSE, warning=FALSE, echo=FALSE}
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
```
